// tdesign-react/cjs/calendar/CalendarCellComp.js
/**
 * tdesign v1.17.1
 * (c) 2026 tdesign
 * @license MIT
 */

'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var slicedToArray = require('../_chunks/dep-c60a3e4c.js');
var React = require('react');
var hooks_useConfig = require('../hooks/useConfig.js');
var locale_LocalReceiver = require('../locale/LocalReceiver.js');
var calendar__util = require('./_util.js');
var calendar_hooks_usePrefixClass = require('./hooks/usePrefixClass.js');
require('../_chunks/dep-98e6819c.js');
require('../_chunks/dep-7f3fa935.js');
require('../config-provider/ConfigContext.js');
require('../_chunks/dep-c446aa11.js');
require('../_chunks/dep-c290284d.js');
require('../_chunks/dep-2ee09518.js');
require('../_chunks/dep-a6fa90dc.js');
require('dayjs');
require('../_chunks/dep-012d9a7c.js');
require('../_chunks/dep-150a112c.js');
require('../_chunks/dep-e360e17f.js');
require('../_chunks/dep-3d05de97.js');
require('../_chunks/dep-4f1729f0.js');
require('../_chunks/dep-a54f202c.js');
require('../_chunks/dep-51284537.js');
require('../_chunks/dep-8a493834.js');
require('../_chunks/dep-9f5b2891.js');
require('../_chunks/dep-338f013a.js');
require('../_chunks/dep-c37c5e17.js');
require('../_chunks/dep-eee1aab9.js');
require('../_chunks/dep-11347ea1.js');
require('../_chunks/dep-a830ce1e.js');
require('../_chunks/dep-770bc0ae.js');
require('../_chunks/dep-aaf27e39.js');
require('../_chunks/dep-b47d855f.js');
require('../_chunks/dep-81bce4e6.js');
require('../_chunks/dep-863f8f4d.js');
require('../_chunks/dep-84e4d0da.js');
require('../_chunks/dep-f78183d7.js');
require('../_chunks/dep-2f45f6c4.js');
require('../_chunks/dep-427f8bc0.js');
require('../config-provider/ConfigProvider.js');
require('../_chunks/dep-ba90997f.js');
require('../_chunks/dep-22026fea.js');
require('../_chunks/dep-551d5940.js');
require('../_chunks/dep-8005b808.js');
require('../_chunks/dep-2add8824.js');
require('../_chunks/dep-ce83473b.js');
require('../_chunks/dep-8c880a65.js');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

var React__default = /*#__PURE__*/_interopDefaultLegacy(React);

var CalendarCellComp = function CalendarCellComp(props) {
  var mode = props.mode,
    cell = props.cell,
    cellAppend = props.cellAppend,
    theme = props.theme,
    _props$isDisabled = props.isDisabled,
    isDisabled = _props$isDisabled === void 0 ? false : _props$isDisabled,
    cellData = props.cellData,
    isCurrent = props.isCurrent,
    isNow = props.isNow,
    fillWithZero = props.fillWithZero,
    createCalendarCell = props.createCalendarCell,
    onCellClick = props.onCellClick,
    onCellDoubleClick = props.onCellDoubleClick,
    onCellRightClick = props.onCellRightClick;
  var _useLocaleReceiver = locale_LocalReceiver.useLocaleReceiver("calendar"),
    _useLocaleReceiver2 = slicedToArray._slicedToArray(_useLocaleReceiver, 2),
    local = _useLocaleReceiver2[0],
    t = _useLocaleReceiver2[1];
  var monthLabelList = t(local.cellMonth).split(",");
  var getMonthCN = function getMonthCN(month) {
    return monthLabelList[month];
  };
  var _useConfig = hooks_useConfig["default"](),
    calendarConfig = _useConfig.calendar;
  var fix0 = function fix0(num) {
    var _ref;
    var fillZero = num < 10 && ((_ref = fillWithZero !== null && fillWithZero !== void 0 ? fillWithZero : calendarConfig.fillWithZero) !== null && _ref !== void 0 ? _ref : true);
    return fillZero ? "0".concat(num) : num;
  };
  var prefixCls = calendar_hooks_usePrefixClass["default"]();
  return /* @__PURE__ */React__default["default"].createElement("td", {
    className: prefixCls([calendar__util.blockName, "table-body-cell"], isDisabled ? "is-disabled" : null, isCurrent ? "is-checked" : null, isNow ? [calendar__util.blockName, "table-body-cell--now"] : null),
    onClick: onCellClick,
    onDoubleClick: onCellDoubleClick,
    onContextMenu: onCellRightClick
  }, function () {
    if (cell && typeof cell === "function") return cell(mode === "month" ? createCalendarCell(cellData) : cellData);
    if (cell && typeof cell !== "function") return cell;
    var cellCtx;
    if (mode === "year") {
      var mIndex = cellData.date.getMonth();
      cellCtx = theme === "full" ? getMonthCN(mIndex) : t(local.monthSelection, {
        month: "".concat(mIndex + 1)
      });
    } else {
      cellCtx = fix0(cellData.date.getDate());
    }
    return /* @__PURE__ */React__default["default"].createElement("div", {
      className: prefixCls([calendar__util.blockName, "table-body-cell-display"])
    }, cellCtx);
  }(), function () {
    var cellCtx;
    if (cellAppend && typeof cellAppend === "function") {
      cellCtx = cellAppend(mode === "month" ? createCalendarCell(cellData) : cellData);
    }
    if (cellAppend && typeof cellAppend !== "function") {
      cellCtx = cellAppend;
    }
    return cellAppend && /* @__PURE__ */React__default["default"].createElement("div", {
      className: prefixCls([calendar__util.blockName, "table-body-cell-content"])
    }, cellCtx);
  }());
};

exports["default"] = CalendarCellComp;
