// tdesign-react/cjs/tag-input/useTagList.js
/**
 * tdesign v1.17.1
 * (c) 2026 tdesign
 * @license MIT
 */

'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var defineProperty = require('../_chunks/dep-c446aa11.js');
var toConsumableArray = require('../_chunks/dep-05bc49e6.js');
var _typeof = require('../_chunks/dep-c290284d.js');
var slicedToArray = require('../_chunks/dep-c60a3e4c.js');
var React = require('react');
var log = require('../_chunks/dep-3c628d18.js');
var hooks_useConfig = require('../hooks/useConfig.js');
var hooks_useControlled = require('../hooks/useControlled.js');
var tag_index = require('../tag/index.js');
var isFunction = require('../_chunks/dep-11347ea1.js');
require('../_chunks/dep-7f3fa935.js');
require('../_chunks/dep-8c880a65.js');
require('../_chunks/dep-98e6819c.js');
require('../config-provider/ConfigContext.js');
require('../_chunks/dep-2ee09518.js');
require('../_chunks/dep-a6fa90dc.js');
require('dayjs');
require('../_chunks/dep-012d9a7c.js');
require('../_chunks/dep-150a112c.js');
require('../_chunks/dep-e360e17f.js');
require('../_chunks/dep-3d05de97.js');
require('../_chunks/dep-4f1729f0.js');
require('../_chunks/dep-a54f202c.js');
require('../_chunks/dep-51284537.js');
require('../_chunks/dep-8a493834.js');
require('../_chunks/dep-9f5b2891.js');
require('../_chunks/dep-338f013a.js');
require('../_chunks/dep-c37c5e17.js');
require('../_chunks/dep-eee1aab9.js');
require('../_chunks/dep-a830ce1e.js');
require('../_chunks/dep-770bc0ae.js');
require('../_chunks/dep-aaf27e39.js');
require('../_chunks/dep-b47d855f.js');
require('../_chunks/dep-81bce4e6.js');
require('../_chunks/dep-863f8f4d.js');
require('../_chunks/dep-84e4d0da.js');
require('../_chunks/dep-f78183d7.js');
require('../_chunks/dep-2f45f6c4.js');
require('../_util/noop.js');
require('../_chunks/dep-45188d54.js');
require('../_chunks/dep-eccb6a94.js');
require('../_chunks/dep-f5e1d11c.js');
require('../_chunks/dep-b199860d.js');
require('../_chunks/dep-36579bf6.js');
require('../_util/forwardRefWithStatics.js');
require('hoist-non-react-statics');
require('../tag/CheckTag.js');
require('../_chunks/dep-e41137f7.js');
require('classnames');
require('../_chunks/dep-29222115.js');
require('../hooks/useDefaultProps.js');
require('../tag/defaultProps.js');
require('../tag/Tag.js');
require('tdesign-icons-react');
require('../_chunks/dep-183643e1.js');
require('../hooks/useGlobalIcon.js');
require('../tag/CheckTagGroup.js');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

var React__default = /*#__PURE__*/_interopDefaultLegacy(React);

function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { defineProperty._defineProperty(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
function useTagList(props) {
  var _useConfig = hooks_useConfig["default"](),
    prefix = _useConfig.classPrefix;
  var onRemove = props.onRemove,
    max = props.max,
    minCollapsedNum = props.minCollapsedNum,
    size = props.size,
    disabled = props.disabled,
    tagProps = props.tagProps,
    tag = props.tag,
    collapsedItems = props.collapsedItems,
    getDragProps = props.getDragProps;
  var readOnly = props.readOnly || props.readonly;
  var _useControlled = hooks_useControlled["default"](props, "value", props.onChange),
    _useControlled2 = slicedToArray._slicedToArray(_useControlled, 2),
    innerTagValue = _useControlled2[0],
    setTagValue = _useControlled2[1];
  var tagValue;
  if (Array.isArray(innerTagValue)) {
    tagValue = innerTagValue;
  } else {
    if (innerTagValue !== void 0 && innerTagValue !== null) {
      log.log.warnOnce("TagInput", "`value` is expected to be an array, but got ".concat(_typeof._typeof(innerTagValue), ". It will be treated as an empty array."));
    }
    tagValue = [];
  }
  var _useState = React.useState(),
    _useState2 = slicedToArray._slicedToArray(_useState, 2);
    _useState2[0];
    var setOldInputValue = _useState2[1];
  var onClose = function onClose(p) {
    var arr = toConsumableArray._toConsumableArray(tagValue);
    var _arr$splice = arr.splice(p.index, 1),
      _arr$splice2 = slicedToArray._slicedToArray(_arr$splice, 1),
      item = _arr$splice2[0];
    setTagValue(arr, _objectSpread(_objectSpread({
      trigger: "tag-remove"
    }, p), {}, {
      item: item
    }));
    onRemove === null || onRemove === void 0 || onRemove(_objectSpread(_objectSpread({}, p), {}, {
      item: item,
      trigger: "tag-remove",
      value: arr
    }));
  };
  var clearAll = function clearAll(context) {
    setTagValue([], {
      trigger: "clear",
      e: context.e
    });
  };
  var onInnerEnter = function onInnerEnter(value, context) {
    var _tagValue, _props$onEnter;
    var valueStr = value ? String(value).trim() : "";
    var newValue = tagValue;
    var isLimitExceeded = max && ((_tagValue = tagValue) === null || _tagValue === void 0 ? void 0 : _tagValue.length) >= max;
    if (valueStr && !isLimitExceeded) {
      newValue = tagValue instanceof Array ? tagValue.concat(String(valueStr)) : [valueStr];
      setTagValue(newValue, {
        trigger: "enter",
        index: newValue.length - 1,
        item: valueStr,
        e: context.e
      });
    }
    props === null || props === void 0 || (_props$onEnter = props.onEnter) === null || _props$onEnter === void 0 || _props$onEnter.call(props, newValue, _objectSpread(_objectSpread({}, context), {}, {
      inputValue: value
    }));
  };
  var onInputBackspaceKeyUp = function onInputBackspaceKeyUp(value) {
    if (!tagValue || !tagValue.length) return;
    setOldInputValue(value);
  };
  var onInputBackspaceKeyDown = function onInputBackspaceKeyDown(value, context) {
    var e = context.e;
    if (!tagValue || !tagValue.length || readOnly) return;
    if (!value && ["Backspace", "NumpadDelete"].includes(e.key)) {
      var index = tagValue.length - 1;
      var item = tagValue[index];
      var trigger = "backspace";
      var newValue = tagValue.slice(0, -1);
      setTagValue(newValue, {
        e: e,
        index: index,
        item: item,
        trigger: trigger
      });
      onRemove === null || onRemove === void 0 || onRemove({
        e: e,
        index: index,
        item: item,
        trigger: trigger,
        value: newValue
      });
    }
  };
  var renderLabel = function renderLabel(_ref) {
    var displayNode = _ref.displayNode,
      label = _ref.label;
    var newList = minCollapsedNum ? tagValue.slice(0, minCollapsedNum) : tagValue;
    var list = displayNode ? [/* @__PURE__ */React__default["default"].createElement(React.Fragment, {
      key: "display-node"
    }, displayNode)] : newList === null || newList === void 0 ? void 0 : newList.map(function (item, index) {
      var tagContent = isFunction.isFunction(tag) ? tag({
        value: item
      }) : tag;
      var handleClose = function handleClose(context) {
        var _tagProps$onClose;
        tagProps === null || tagProps === void 0 || (_tagProps$onClose = tagProps.onClose) === null || _tagProps$onClose === void 0 || _tagProps$onClose.call(tagProps, context);
        onClose({
          e: context === null || context === void 0 ? void 0 : context.e,
          index: index
        });
      };
      return /* @__PURE__ */React__default["default"].createElement(tag_index.Tag, _objectSpread(_objectSpread(_objectSpread({
        key: index,
        size: size,
        disabled: disabled,
        closable: !readOnly && !disabled
      }, getDragProps === null || getDragProps === void 0 ? void 0 : getDragProps(index, item)), tagProps), {}, {
        onClose: handleClose
      }), tagContent !== null && tagContent !== void 0 ? tagContent : item);
    });
    if (label) {
      list === null || list === void 0 || list.unshift(/* @__PURE__ */React__default["default"].createElement("div", {
        className: "".concat(prefix, "-tag-input__prefix"),
        key: "label"
      }, label));
    }
    if (newList.length !== tagValue.length) {
      var len = tagValue.length - newList.length;
      var options = Array.isArray(props === null || props === void 0 ? void 0 : props.options) ? props.options : tagValue;
      var params = {
        value: tagValue,
        count: tagValue.length - minCollapsedNum,
        collapsedTags: tagValue.slice(minCollapsedNum, tagValue.length),
        collapsedSelectedItems: options.slice(minCollapsedNum, tagValue.length),
        onClose: onClose
      };
      var more = isFunction.isFunction(collapsedItems) ? collapsedItems(params) : collapsedItems;
      list.push(/* @__PURE__ */React__default["default"].createElement(React.Fragment, {
        key: "more"
      }, more !== null && more !== void 0 ? more : /* @__PURE__ */React__default["default"].createElement(tag_index.Tag, _objectSpread({
        size: size
      }, tagProps), "+", len)));
    }
    return list;
  };
  return {
    tagValue: tagValue,
    clearAll: clearAll,
    onClose: onClose,
    onInnerEnter: onInnerEnter,
    onInputBackspaceKeyDown: onInputBackspaceKeyDown,
    onInputBackspaceKeyUp: onInputBackspaceKeyUp,
    renderLabel: renderLabel
  };
}

exports["default"] = useTagList;
