// tdesign-react/cjs/_chunks/dep-147ad148.js
/**
 * tdesign v1.17.1
 * (c) 2026 tdesign
 * @license MIT
 */

'use strict';

var DISTANCE = "32px";
var DEFAULT_Z_INDEX = 6e3;
var DEFAULT_DURATION = 3e3;
var PLACEMENT_OFFSET = {
  center: {
    left: "50%",
    top: "50%",
    transform: "translateX(-50%) translateY(-50%)"
  },
  left: {
    left: DISTANCE,
    top: "50%",
    transform: "translateY(-50%)"
  },
  bottom: {
    bottom: DISTANCE,
    left: "50%",
    transform: "translateX(-50%)"
  },
  right: {
    right: DISTANCE,
    top: "50%",
    transform: "translateY(-50%)"
  },
  top: {
    top: DISTANCE,
    left: "50%",
    transform: "translateX(-50%)"
  },
  "top-left": {
    left: DISTANCE,
    top: DISTANCE
  },
  "top-right": {
    right: DISTANCE,
    top: DISTANCE
  },
  "bottom-left": {
    left: DISTANCE,
    bottom: DISTANCE
  },
  "bottom-right": {
    right: DISTANCE,
    bottom: DISTANCE
  }
};
var PLACEMENT_LIST = Object.keys(PLACEMENT_OFFSET);

exports.DEFAULT_DURATION = DEFAULT_DURATION;
exports.DEFAULT_Z_INDEX = DEFAULT_Z_INDEX;
exports.PLACEMENT_LIST = PLACEMENT_LIST;
exports.PLACEMENT_OFFSET = PLACEMENT_OFFSET;
