// tdesign-react/lib/qrcode/QRCodeStatus.js
/**
 * tdesign v1.17.1
 * (c) 2026 tdesign
 * @license MIT
 */

import React, { useMemo, useCallback } from 'react';
import { RefreshIcon, CheckCircleFilledIcon } from 'tdesign-icons-react';
import { Loading } from '../loading/index.js';
import '../loading/Loading.js';
import '../_chunks/dep-55b77e26.js';
import '../_chunks/dep-067cb22b.js';
import '../_chunks/dep-d1eeb95c.js';
import '../_chunks/dep-673e8854.js';
import '../_chunks/dep-570a5adb.js';
import 'classnames';
import '../_chunks/dep-68899e89.js';
import 'lodash-es';
import '../common/Portal.js';
import 'react-dom';
import '../hooks/useConfig.js';
import '../config-provider/ConfigContext.js';
import '../_chunks/dep-c3e5e988.js';
import '../_chunks/dep-65b05485.js';
import 'dayjs';
import '../hooks/useLayoutEffect.js';
import '../hooks/useDefaultProps.js';
import '../loading/defaultProps.js';
import '../loading/gradient.js';
import '../_chunks/dep-e17ec8df.js';
import '../_chunks/dep-b0cbea79.js';
import '../_chunks/dep-154fcb93.js';
import '../_chunks/dep-e9c0bb63.js';
import '../_chunks/dep-84926167.js';
import '../hooks/useDomRefCallback.js';
import '../loading/plugin.js';
import '../_util/react-render.js';
import '../_chunks/dep-e29eee31.js';
import '../common/PluginContainer.js';
import '../config-provider/ConfigProvider.js';

var defaultSpin = /* @__PURE__ */React.createElement(Loading, {
  size: "32px"
});
function QRcodeStatus(_ref) {
  var locale = _ref.locale,
    classPrefix = _ref.classPrefix,
    onRefresh = _ref.onRefresh,
    statusRender = _ref.statusRender,
    status = _ref.status;
  var defaultExpiredNode = useMemo(function () {
    return /* @__PURE__ */React.createElement(React.Fragment, null, /* @__PURE__ */React.createElement("p", {
      className: "".concat(classPrefix, "-expired__text")
    }, locale === null || locale === void 0 ? void 0 : locale.expiredText), onRefresh && /* @__PURE__ */React.createElement("p", {
      className: "".concat(classPrefix, "-expired__button"),
      onClick: onRefresh
    }, /* @__PURE__ */React.createElement(RefreshIcon, {
      size: 16
    }), locale === null || locale === void 0 ? void 0 : locale.refreshText));
  }, [classPrefix, locale === null || locale === void 0 ? void 0 : locale.expiredText, locale === null || locale === void 0 ? void 0 : locale.refreshText, onRefresh]);
  var defaultScannedNode = useMemo(function () {
    return /* @__PURE__ */React.createElement("p", {
      className: "".concat(classPrefix, "-scanned")
    }, /* @__PURE__ */React.createElement(CheckCircleFilledIcon, {
      size: 16,
      className: "".concat(classPrefix, "-scanned__icon")
    }), locale === null || locale === void 0 ? void 0 : locale.scannedText);
  }, [classPrefix, locale === null || locale === void 0 ? void 0 : locale.scannedText]);
  var defaultNodes = useMemo(function () {
    return {
      expired: defaultExpiredNode,
      loading: defaultSpin,
      scanned: defaultScannedNode
    };
  }, [defaultExpiredNode, defaultScannedNode]);
  var defaultStatusRender = useCallback(function (info) {
    return defaultNodes[info.status];
  }, [defaultNodes]);
  var mergedStatusRender = statusRender !== null && statusRender !== void 0 ? statusRender : defaultStatusRender;
  return /* @__PURE__ */React.createElement(React.Fragment, null, mergedStatusRender({
    status: status,
    onRefresh: onRefresh
  }) || null);
}

export { QRcodeStatus as default };
