// tdesign-react/esm/pagination/hooks/useBoundaryJumper.js
/**
 * tdesign v1.17.1
 * (c) 2026 tdesign
 * @license MIT
 */

import { _ as _defineProperty } from '../../_chunks/dep-55b77e26.js';
import React from 'react';
import classNames from 'classnames';
import { PageLastIcon, PageFirstIcon } from 'tdesign-icons-react';
import useConfig from '../../hooks/useConfig.js';
import useGlobalIcon from '../../hooks/useGlobalIcon.js';
import '../../_chunks/dep-067cb22b.js';
import '../../config-provider/ConfigContext.js';
import 'lodash-es';
import '../../locale/zh_CN.js';
import '../../_chunks/dep-e56ffacc.js';
import 'dayjs';
import '../../_chunks/dep-84036ea0.js';

function useBoundaryJumper(props) {
  var _useConfig = useConfig(),
    classPrefix = _useConfig.classPrefix;
  var _useGlobalIcon = useGlobalIcon({
      PageLastIcon: PageLastIcon,
      PageFirstIcon: PageFirstIcon
    }),
    PageFirstIcon$1 = _useGlobalIcon.PageFirstIcon,
    PageLastIcon$1 = _useGlobalIcon.PageLastIcon;
  var name = "".concat(classPrefix, "-pagination");
  var showFirstAndLastPageBtn = props.showFirstAndLastPageBtn,
    disabled = props.disabled,
    current = props.current,
    pageCount = props.pageCount,
    changeCurrent = props.changeCurrent;
  var firstPageJumper = showFirstAndLastPageBtn && /* @__PURE__ */React.createElement("div", {
    className: classNames("".concat(name, "__btn"), "".concat(name, "__btn-first-page"), _defineProperty({}, "".concat(classPrefix, "-is-disabled"), disabled || current === 1)),
    onClick: function onClick() {
      return changeCurrent(1);
    }
  }, /* @__PURE__ */React.createElement(PageFirstIcon$1, null));
  var lastPageJumper = showFirstAndLastPageBtn && /* @__PURE__ */React.createElement("div", {
    className: classNames("".concat(name, "__btn"), "".concat(name, "__btn-first-page"), _defineProperty({}, "".concat(classPrefix, "-is-disabled"), disabled || current === pageCount)),
    onClick: function onClick() {
      return changeCurrent(pageCount);
    }
  }, /* @__PURE__ */React.createElement(PageLastIcon$1, null));
  return {
    firstPageJumper: firstPageJumper,
    lastPageJumper: lastPageJumper
  };
}

export { useBoundaryJumper as default };
