// tdesign-react/cjs/dialog/hooks/useDialogPosition.js
/**
 * tdesign v1.17.1
 * (c) 2026 tdesign
 * @license MIT
 */

'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var React = require('react');
var _util_style = require('../../_chunks/dep-ee71ddca.js');
require('../../_chunks/dep-427f8bc0.js');
require('../../_chunks/dep-3d05de97.js');
require('../../_chunks/dep-c290284d.js');
require('../../_chunks/dep-a54f202c.js');
require('../../_chunks/dep-4f1729f0.js');

var mousePosition = null;
var getClickPosition = function getClickPosition(e) {
  mousePosition = {
    x: e.clientX,
    y: e.clientY
  };
  setTimeout(function () {
    mousePosition = null;
  }, 100);
};
if (_util_style.canUseDocument) {
  document.addEventListener("click", getClickPosition, true);
}
function useDialogPosition(dialogCardRef) {
  var applyTransform = React.useCallback(function () {
    var _el$offsetParent;
    var el = dialogCardRef.current;
    if (!el || !mousePosition) return;
    var _mousePosition = mousePosition,
      x = _mousePosition.x,
      y = _mousePosition.y;
    var parentRect = ((_el$offsetParent = el.offsetParent) === null || _el$offsetParent === void 0 ? void 0 : _el$offsetParent.getBoundingClientRect()) || {
      left: 0,
      top: 0
    };
    var top = parentRect.top + el.offsetTop;
    var left = parentRect.left + el.offsetLeft;
    var offsetX = x - left;
    var offsetY = y - top;
    el.style.transformOrigin = "".concat(offsetX, "px ").concat(offsetY, "px");
  }, [dialogCardRef]);
  return {
    applyTransform: applyTransform
  };
}

exports["default"] = useDialogPosition;
