// tdesign-react/cjs/sticky-tool/StickyTool.js
/**
 * tdesign v1.17.1
 * (c) 2026 tdesign
 * @license MIT
 */

'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var defineProperty = require('../_chunks/dep-c446aa11.js');
var React = require('react');
var classNames = require('classnames');
var _util_forwardRefWithStatics = require('../_util/forwardRefWithStatics.js');
var hooks_useConfig = require('../hooks/useConfig.js');
var hooks_useDefaultProps = require('../hooks/useDefaultProps.js');
var stickyTool_defaultProps = require('./defaultProps.js');
var stickyTool_StickyItem = require('./StickyItem.js');
require('../_chunks/dep-c290284d.js');
require('hoist-non-react-statics');
require('../config-provider/ConfigContext.js');
require('../_chunks/dep-2ee09518.js');
require('../_chunks/dep-a6fa90dc.js');
require('dayjs');
require('../_chunks/dep-012d9a7c.js');
require('../_chunks/dep-150a112c.js');
require('../_chunks/dep-e360e17f.js');
require('../_chunks/dep-3d05de97.js');
require('../_chunks/dep-4f1729f0.js');
require('../_chunks/dep-a54f202c.js');
require('../_chunks/dep-51284537.js');
require('../_chunks/dep-8a493834.js');
require('../_chunks/dep-9f5b2891.js');
require('../_chunks/dep-338f013a.js');
require('../_chunks/dep-c37c5e17.js');
require('../_chunks/dep-eee1aab9.js');
require('../_chunks/dep-11347ea1.js');
require('../_chunks/dep-a830ce1e.js');
require('../_chunks/dep-770bc0ae.js');
require('../_chunks/dep-aaf27e39.js');
require('../_chunks/dep-b47d855f.js');
require('../_chunks/dep-81bce4e6.js');
require('../_chunks/dep-863f8f4d.js');
require('../_chunks/dep-84e4d0da.js');
require('../_chunks/dep-f78183d7.js');
require('../_chunks/dep-2f45f6c4.js');
require('../popup/index.js');
require('../popup/Popup.js');
require('../_chunks/dep-c60a3e4c.js');
require('../_chunks/dep-98e6819c.js');
require('../_chunks/dep-7f3fa935.js');
require('react-transition-group');
require('../_chunks/dep-ee71ddca.js');
require('../_chunks/dep-427f8bc0.js');
require('../common/Portal.js');
require('react-dom');
require('../hooks/useLayoutEffect.js');
require('../hooks/useAnimation.js');
require('../hooks/useAttach.js');
require('../hooks/useControlled.js');
require('../_util/noop.js');
require('../_chunks/dep-45188d54.js');
require('../_chunks/dep-eccb6a94.js');
require('../_chunks/dep-f5e1d11c.js');
require('../_chunks/dep-b199860d.js');
require('../_chunks/dep-36579bf6.js');
require('../hooks/useMutationObserver.js');
require('../hooks/useLatest.js');
require('../_chunks/dep-aa9a2c54.js');
require('../_chunks/dep-77ef3011.js');
require('../_chunks/dep-b36779a4.js');
require('../_chunks/dep-551d5940.js');
require('../_chunks/dep-8005b808.js');
require('../_chunks/dep-2add8824.js');
require('../_chunks/dep-f2809376.js');
require('../_chunks/dep-3be56e1e.js');
require('../hooks/usePopper.js');
require('../_chunks/dep-05bc49e6.js');
require('../_chunks/dep-8c880a65.js');
require('react-fast-compare');
require('@popperjs/core');
require('../hooks/useWindowSize.js');
require('../popup/defaultProps.js');
require('../popup/hooks/useTrigger.js');
require('../_util/listener.js');
require('../_util/ref.js');
require('react-is');
require('../_util/isFragment.js');
require('../popup/utils/transition.js');
require('../popup/PopupPlugin.js');
require('../_chunks/dep-7bf14076.js');
require('../_util/react-render.js');
require('../common/PluginContainer.js');
require('../config-provider/ConfigProvider.js');
require('../_chunks/dep-ba90997f.js');
require('../_chunks/dep-22026fea.js');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

var React__default = /*#__PURE__*/_interopDefaultLegacy(React);
var classNames__default = /*#__PURE__*/_interopDefaultLegacy(classNames);

function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { defineProperty._defineProperty(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
var StickyTool = _util_forwardRefWithStatics["default"](function (props, ref) {
  var _useDefaultProps = hooks_useDefaultProps["default"](props, stickyTool_defaultProps.stickyToolDefaultProps),
    style = _useDefaultProps.style,
    className = _useDefaultProps.className,
    children = _useDefaultProps.children,
    width = _useDefaultProps.width,
    type = _useDefaultProps.type,
    shape = _useDefaultProps.shape,
    placement = _useDefaultProps.placement,
    offset = _useDefaultProps.offset,
    popupProps = _useDefaultProps.popupProps,
    list = _useDefaultProps.list,
    onClick = _useDefaultProps.onClick,
    onHover = _useDefaultProps.onHover;
  var _useConfig = hooks_useConfig["default"](),
    classPrefix = _useConfig.classPrefix;
  var styles = React.useMemo(function () {
    var position = offset ? [80, 24] : ["80px", "24px"];
    offset === null || offset === void 0 || offset.forEach(function (item, index) {
      position[index] = isNaN(Number(item)) ? "calc( ".concat(position[index], "px + ").concat(item, ")") : "".concat(position[index] + item, "px");
    });
    var styles2 = _objectSpread({}, style);
    placement.split("-").forEach(function (item, index) {
      if (item !== "center") {
        styles2[item] = position[index];
      } else {
        styles2.top = "50%";
        styles2.transform = "translate(0, -50%)";
      }
    });
    if (width) {
      styles2.width = typeof width === "number" ? "".concat(width, "px") : width;
    }
    return styles2;
  }, [offset, placement, width, style]);
  var handleClick = React.useCallback(function (context) {
    onClick === null || onClick === void 0 || onClick(context);
  }, [onClick]);
  var handleHover = React.useCallback(function (context) {
    onHover === null || onHover === void 0 || onHover(context);
  }, [onHover]);
  var stickyItemList = React.useMemo(function () {
    if (list !== null && list !== void 0 && list.length) {
      return list.map(function (item, index) {
        var itemProps = _objectSpread(_objectSpread({}, item), {}, {
          type: type,
          shape: shape,
          placement: placement,
          basePopupProps: popupProps,
          baseWidth: styles.width,
          onClick: handleClick,
          onHover: handleHover
        });
        return /* @__PURE__ */React__default["default"].createElement(stickyTool_StickyItem["default"], _objectSpread({
          key: index
        }, itemProps));
      });
    }
    var childrenList = React__default["default"].Children.toArray(children);
    return childrenList.map(function (child) {
      var itemProps = _objectSpread(_objectSpread({}, child.props), {}, {
        type: type,
        shape: shape,
        placement: placement,
        basePopupProps: popupProps,
        baseWidth: styles.width,
        onClick: handleClick,
        onHover: handleHover
      });
      return /*#__PURE__*/React__default["default"].cloneElement(child, _objectSpread({}, itemProps));
    });
  }, [list, children, type, shape, placement, popupProps, styles, handleClick, handleHover]);
  return /* @__PURE__ */React__default["default"].createElement("div", {
    ref: ref,
    style: styles,
    className: classNames__default["default"]("".concat(classPrefix, "-sticky-tool"), "".concat(classPrefix, "-sticky-tool--").concat(shape), className)
  }, stickyItemList);
}, {
  StickyItem: stickyTool_StickyItem["default"]
});
StickyTool.displayName = "StickyTool";

exports["default"] = StickyTool;
