// tdesign-react/cjs/timeline/Timeline.js
/**
 * tdesign v1.17.1
 * (c) 2026 tdesign
 * @license MIT
 */

'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var defineProperty = require('../_chunks/dep-c446aa11.js');
var React = require('react');
var classNames = require('classnames');
var _util_forwardRefWithStatics = require('../_util/forwardRefWithStatics.js');
var hooks_useConfig = require('../hooks/useConfig.js');
var timeline_TimelineContext = require('./TimelineContext.js');
var timeline_TimelineItem = require('./TimelineItem.js');
var timeline_useAlign = require('./useAlign.js');
require('../_chunks/dep-c290284d.js');
require('hoist-non-react-statics');
require('../config-provider/ConfigContext.js');
require('../_chunks/dep-2ee09518.js');
require('../_chunks/dep-a6fa90dc.js');
require('dayjs');
require('../_chunks/dep-012d9a7c.js');
require('../_chunks/dep-150a112c.js');
require('../_chunks/dep-e360e17f.js');
require('../_chunks/dep-3d05de97.js');
require('../_chunks/dep-4f1729f0.js');
require('../_chunks/dep-a54f202c.js');
require('../_chunks/dep-51284537.js');
require('../_chunks/dep-8a493834.js');
require('../_chunks/dep-9f5b2891.js');
require('../_chunks/dep-338f013a.js');
require('../_chunks/dep-c37c5e17.js');
require('../_chunks/dep-eee1aab9.js');
require('../_chunks/dep-11347ea1.js');
require('../_chunks/dep-a830ce1e.js');
require('../_chunks/dep-770bc0ae.js');
require('../_chunks/dep-aaf27e39.js');
require('../_chunks/dep-b47d855f.js');
require('../_chunks/dep-81bce4e6.js');
require('../_chunks/dep-863f8f4d.js');
require('../_chunks/dep-84e4d0da.js');
require('../_chunks/dep-f78183d7.js');
require('../_chunks/dep-2f45f6c4.js');
require('../_util/parseTNode.js');
require('../_chunks/dep-3c628d18.js');
require('../loading/index.js');
require('../loading/Loading.js');
require('../_chunks/dep-c60a3e4c.js');
require('../_chunks/dep-98e6819c.js');
require('../_chunks/dep-7f3fa935.js');
require('../_chunks/dep-ee71ddca.js');
require('../_chunks/dep-427f8bc0.js');
require('../common/Portal.js');
require('react-dom');
require('../hooks/useLayoutEffect.js');
require('../hooks/useDefaultProps.js');
require('../loading/defaultProps.js');
require('../loading/gradient.js');
require('../_chunks/dep-062bb1b1.js');
require('../_chunks/dep-6e0da644.js');
require('../_chunks/dep-05bc49e6.js');
require('../_chunks/dep-8c880a65.js');
require('../_chunks/dep-e41137f7.js');
require('../_chunks/dep-b8d0b96c.js');
require('../hooks/useDomRefCallback.js');
require('../loading/plugin.js');
require('../_util/react-render.js');
require('../_chunks/dep-7bf14076.js');
require('../common/PluginContainer.js');
require('../config-provider/ConfigProvider.js');
require('../_chunks/dep-ba90997f.js');
require('../_chunks/dep-22026fea.js');
require('../_chunks/dep-551d5940.js');
require('../_chunks/dep-8005b808.js');
require('../_chunks/dep-2add8824.js');
require('../_chunks/dep-8de6ca9d.js');
require('../_chunks/dep-b199860d.js');
require('../_chunks/dep-36579bf6.js');
require('../_chunks/dep-6e5e0470.js');
require('../_chunks/dep-d4a6a9c1.js');
require('../_chunks/dep-cdef7593.js');
require('../_chunks/dep-f5e1d11c.js');
require('../_chunks/dep-204188de.js');
require('../_chunks/dep-7a4e0117.js');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

var React__default = /*#__PURE__*/_interopDefaultLegacy(React);
var classNames__default = /*#__PURE__*/_interopDefaultLegacy(classNames);

var Timeline = _util_forwardRefWithStatics["default"](function (props, ref) {
  var _props$theme = props.theme,
    theme = _props$theme === void 0 ? "default" : _props$theme,
    labelAlign = props.labelAlign,
    children = props.children,
    className = props.className,
    style = props.style,
    _props$reverse = props.reverse,
    reverse = _props$reverse === void 0 ? false : _props$reverse,
    _props$layout = props.layout,
    layout = _props$layout === void 0 ? "vertical" : _props$layout,
    _props$mode = props.mode,
    mode = _props$mode === void 0 ? "alternate" : _props$mode;
  var _useConfig = hooks_useConfig["default"](),
    classPrefix = _useConfig.classPrefix;
  var renderAlign = timeline_useAlign.useAlign(labelAlign, layout);
  var timelineItems = React__default["default"].Children.toArray(children).filter(function (child) {
    var _child$type;
    return ((_child$type = child.type) === null || _child$type === void 0 ? void 0 : _child$type.displayName) === timeline_TimelineItem["default"].displayName;
  });
  var itemsStatus = React__default["default"].Children.map(timelineItems, function (child) {
    var _child$props;
    return ((_child$props = child.props) === null || _child$props === void 0 ? void 0 : _child$props.dotColor) || "primary";
  });
  var hasLabelItem = timelineItems.some(function (item) {
    var _item$props;
    return !!(item !== null && item !== void 0 && (_item$props = item.props) !== null && _item$props !== void 0 && _item$props.label);
  });
  if (reverse) {
    timelineItems.reverse();
  }
  var itemsCounts = React__default["default"].Children.count(timelineItems);
  var timelineClassName = classNames__default["default"]("".concat(classPrefix, "-timeline"), defineProperty._defineProperty(defineProperty._defineProperty(defineProperty._defineProperty(defineProperty._defineProperty(defineProperty._defineProperty({}, "".concat(classPrefix, "-timeline-").concat(renderAlign), true), "".concat(classPrefix, "-timeline-reverse"), reverse), "".concat(classPrefix, "-timeline-").concat(layout), true), "".concat(classPrefix, "-timeline-label"), hasLabelItem), "".concat(classPrefix, "-timeline-label--").concat(mode), true), className);
  return /* @__PURE__ */React__default["default"].createElement(timeline_TimelineContext["default"].Provider, {
    value: {
      theme: theme,
      reverse: reverse,
      itemsStatus: itemsStatus,
      layout: layout,
      globalAlign: labelAlign,
      mode: mode
    }
  }, /* @__PURE__ */React__default["default"].createElement("ul", {
    className: timelineClassName,
    style: style,
    ref: ref
  }, React__default["default"].Children.map(timelineItems, function (ele, index) {
    var _ele$props;
    return /*#__PURE__*/React__default["default"].cloneElement(ele, {
      index: index,
      className: classNames__default["default"]([ele === null || ele === void 0 || (_ele$props = ele.props) === null || _ele$props === void 0 ? void 0 : _ele$props.className], defineProperty._defineProperty({}, "".concat(classPrefix, "-timeline-item--last"), index === itemsCounts - 1))
    });
  })));
}, {
  Item: timeline_TimelineItem["default"]
});
Timeline.displayName = "Timeline";

exports["default"] = Timeline;
