// tdesign-react/cjs/timeline/TimelineItem.js
/**
 * tdesign v1.17.1
 * (c) 2026 tdesign
 * @license MIT
 */

'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var defineProperty = require('../_chunks/dep-c446aa11.js');
var React = require('react');
var classNames = require('classnames');
var _util_parseTNode = require('../_util/parseTNode.js');
var hooks_useConfig = require('../hooks/useConfig.js');
var loading_index = require('../loading/index.js');
var timeline_TimelineContext = require('./TimelineContext.js');
var timeline_useAlign = require('./useAlign.js');
var omit = require('../_chunks/dep-8de6ca9d.js');
require('../_chunks/dep-c290284d.js');
require('../_chunks/dep-3c628d18.js');
require('../_chunks/dep-11347ea1.js');
require('../_chunks/dep-3d05de97.js');
require('../_chunks/dep-a830ce1e.js');
require('../config-provider/ConfigContext.js');
require('../_chunks/dep-2ee09518.js');
require('../_chunks/dep-a6fa90dc.js');
require('dayjs');
require('../_chunks/dep-012d9a7c.js');
require('../_chunks/dep-150a112c.js');
require('../_chunks/dep-e360e17f.js');
require('../_chunks/dep-4f1729f0.js');
require('../_chunks/dep-a54f202c.js');
require('../_chunks/dep-51284537.js');
require('../_chunks/dep-8a493834.js');
require('../_chunks/dep-9f5b2891.js');
require('../_chunks/dep-338f013a.js');
require('../_chunks/dep-c37c5e17.js');
require('../_chunks/dep-eee1aab9.js');
require('../_chunks/dep-770bc0ae.js');
require('../_chunks/dep-aaf27e39.js');
require('../_chunks/dep-b47d855f.js');
require('../_chunks/dep-81bce4e6.js');
require('../_chunks/dep-863f8f4d.js');
require('../_chunks/dep-84e4d0da.js');
require('../_chunks/dep-f78183d7.js');
require('../_chunks/dep-2f45f6c4.js');
require('../loading/Loading.js');
require('../_chunks/dep-c60a3e4c.js');
require('../_chunks/dep-98e6819c.js');
require('../_chunks/dep-7f3fa935.js');
require('../_chunks/dep-ee71ddca.js');
require('../_chunks/dep-427f8bc0.js');
require('../common/Portal.js');
require('react-dom');
require('../hooks/useLayoutEffect.js');
require('../hooks/useDefaultProps.js');
require('../loading/defaultProps.js');
require('../loading/gradient.js');
require('../_chunks/dep-062bb1b1.js');
require('../_chunks/dep-6e0da644.js');
require('../_chunks/dep-05bc49e6.js');
require('../_chunks/dep-8c880a65.js');
require('../_chunks/dep-e41137f7.js');
require('../_chunks/dep-b8d0b96c.js');
require('../hooks/useDomRefCallback.js');
require('../loading/plugin.js');
require('../_util/react-render.js');
require('../_chunks/dep-7bf14076.js');
require('../common/PluginContainer.js');
require('../config-provider/ConfigProvider.js');
require('../_chunks/dep-ba90997f.js');
require('../_chunks/dep-22026fea.js');
require('../_chunks/dep-551d5940.js');
require('../_chunks/dep-8005b808.js');
require('../_chunks/dep-2add8824.js');
require('../_chunks/dep-b199860d.js');
require('../_chunks/dep-36579bf6.js');
require('../_chunks/dep-6e5e0470.js');
require('../_chunks/dep-d4a6a9c1.js');
require('../_chunks/dep-cdef7593.js');
require('../_chunks/dep-f5e1d11c.js');
require('../_chunks/dep-204188de.js');
require('../_chunks/dep-7a4e0117.js');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

var React__default = /*#__PURE__*/_interopDefaultLegacy(React);
var classNames__default = /*#__PURE__*/_interopDefaultLegacy(classNames);

var DefaultTheme = ["default", "primary", "success", "warning", "error"];
var TimelineItem = function TimelineItem(props) {
  var className = props.className,
    _props$style = props.style,
    style = _props$style === void 0 ? {} : _props$style,
    dot = props.dot,
    _props$dotColor = props.dotColor,
    dotColor = _props$dotColor === void 0 ? "primary" : _props$dotColor,
    labelAlign = props.labelAlign,
    children = props.children,
    index = props.index,
    content = props.content,
    label = props.label,
    _props$loading = props.loading,
    loading = _props$loading === void 0 ? false : _props$loading,
    onClick = props.onClick;
  var _useContext = React.useContext(timeline_TimelineContext["default"]),
    theme = _useContext.theme,
    reverse = _useContext.reverse,
    itemsStatus = _useContext.itemsStatus,
    layout = _useContext.layout,
    globalAlign = _useContext.globalAlign,
    mode = _useContext.mode;
  var _useConfig = hooks_useConfig["default"](),
    classPrefix = _useConfig.classPrefix;
  var renderAlign = timeline_useAlign.useAlign(globalAlign, layout);
  var getPositionClassName = function getPositionClassName(index2) {
    var left = layout === "horizontal" ? "top" : "left";
    var right = layout === "horizontal" ? "bottom" : "right";
    if (renderAlign === "alternate") {
      return labelAlign || index2 % 2 === 0 ? "".concat(classPrefix, "-timeline-item-").concat(left) : "".concat(classPrefix, "-timeline-item-").concat(right);
    }
    if (renderAlign === "left" || renderAlign === "top") {
      return "".concat(classPrefix, "-timeline-item-").concat(left);
    }
    if (renderAlign === "right" || renderAlign === "bottom") {
      return "".concat(classPrefix, "-timeline-item-").concat(right);
    }
    return "";
  };
  var dotElement = React.useMemo(function () {
    var ele = _util_parseTNode["default"](dot);
    if (/*#__PURE__*/React__default["default"].isValidElement(ele)) {
      var _ele$props;
      return /*#__PURE__*/React__default["default"].cloneElement(ele, {
        className: classNames__default["default"](ele === null || ele === void 0 || (_ele$props = ele.props) === null || _ele$props === void 0 ? void 0 : _ele$props.className, "".concat(classPrefix, "-timeline-item__dot-content"))
      });
    }
    return ele;
  }, [dot, classPrefix]);
  var handleClick = function handleClick(e) {
    onClick === null || onClick === void 0 || onClick({
      e: e,
      item: omit.omit(props, ["children", "index", "onClick"])
    });
  };
  var itemClassName = classNames__default["default"](defineProperty._defineProperty(defineProperty._defineProperty({}, "".concat(classPrefix, "-timeline-item"), true), "".concat(getPositionClassName(index)), true), className);
  var tailClassName = classNames__default["default"](defineProperty._defineProperty(defineProperty._defineProperty(defineProperty._defineProperty({}, "".concat(classPrefix, "-timeline-item__tail"), true), "".concat(classPrefix, "-timeline-item__tail--theme-").concat(theme), true), "".concat(classPrefix, "-timeline-item__tail--status-").concat(itemsStatus[index]), reverse));
  var dotClassName = classNames__default["default"](defineProperty._defineProperty(defineProperty._defineProperty(defineProperty._defineProperty({}, "".concat(classPrefix, "-timeline-item__dot"), true), "".concat(classPrefix, "-timeline-item__dot--custom"), !!dotElement || !dotElement && loading), "".concat(classPrefix, "-timeline-item__dot--").concat(dotColor), DefaultTheme.includes(dotColor)));
  var labelClassName = classNames__default["default"]("".concat(classPrefix, "-timeline-item__label"), defineProperty._defineProperty({}, "".concat(classPrefix, "-timeline-item__label--").concat(mode), true));
  return /* @__PURE__ */React__default["default"].createElement("li", {
    className: itemClassName,
    style: style,
    onClick: handleClick
  }, mode === "alternate" && label && /* @__PURE__ */React__default["default"].createElement("div", {
    className: labelClassName
  }, label), /* @__PURE__ */React__default["default"].createElement("div", {
    className: "".concat(classPrefix, "-timeline-item__wrapper")
  }, /* @__PURE__ */React__default["default"].createElement("div", {
    className: dotClassName,
    style: {
      borderColor: !DefaultTheme.includes(dotColor) && dotColor
    }
  }, !dotElement && loading && /* @__PURE__ */React__default["default"].createElement(loading_index.Loading, {
    size: "12px",
    className: "".concat(classPrefix, "-timeline-item__dot-content")
  }), dotElement), /* @__PURE__ */React__default["default"].createElement("div", {
    className: tailClassName
  })), /* @__PURE__ */React__default["default"].createElement("div", {
    className: "".concat(classPrefix, "-timeline-item__content")
  }, content || children, mode === "same" && label && /* @__PURE__ */React__default["default"].createElement("div", {
    className: labelClassName
  }, label)));
};
TimelineItem.displayName = "TimelineItem";

exports["default"] = TimelineItem;
