// tdesign-react/es/slider/SliderHandleButton.js
/**
 * tdesign v1.17.1
 * (c) 2026 tdesign
 * @license MIT
 */

import { _ as _defineProperty } from '../_chunks/dep-55b77e26.js';
import { _ as _slicedToArray } from '../_chunks/dep-d1eeb95c.js';
import React, { useRef, useState } from 'react';
import classNames from 'classnames';
import useMouseEvent from '../hooks/useMouseEvent.js';
import Tooltip from '../tooltip/Tooltip.js';
import '../_chunks/dep-067cb22b.js';
import '../_chunks/dep-673e8854.js';
import '../_chunks/dep-570a5adb.js';
import '../hooks/useLatest.js';
import '../_chunks/dep-84926167.js';
import '../hooks/useConfig.js';
import '../config-provider/ConfigContext.js';
import 'lodash-es';
import '../locale/zh_CN.js';
import '../_chunks/dep-e56ffacc.js';
import 'dayjs';
import '../_chunks/dep-84036ea0.js';
import '../hooks/useDefaultProps.js';
import '../popup/index.js';
import '../popup/style/css.js';
import '../popup/Popup.js';
import 'react-transition-group';
import '../_chunks/dep-8c226881.js';
import '../common/Portal.js';
import 'react-dom';
import '../hooks/useLayoutEffect.js';
import '../hooks/useAnimation.js';
import '../hooks/useAttach.js';
import '../hooks/useControlled.js';
import '../_util/noop.js';
import '../hooks/useMutationObserver.js';
import '../hooks/usePopper.js';
import '../_chunks/dep-154fcb93.js';
import '../_chunks/dep-e9c0bb63.js';
import 'react-fast-compare';
import '@popperjs/core';
import '../hooks/useWindowSize.js';
import '../popup/defaultProps.js';
import '../popup/hooks/useTrigger.js';
import '../_util/listener.js';
import '../_util/ref.js';
import 'react-is';
import '../_util/isFragment.js';
import '../popup/utils/transition.js';
import '../popup/PopupPlugin.js';
import '../_chunks/dep-55f620a2.js';
import '../_util/react-render.js';
import '../common/PluginContainer.js';
import '../config-provider/index.js';
import '../config-provider/ConfigProvider.js';
import '../config-provider/type.js';
import '../popup/type.js';
import '../tooltip/defaultProps.js';

function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { _defineProperty(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
var SliderHandleButton = function SliderHandleButton(_ref) {
  var onChange = _ref.onChange,
    onChangeEnd = _ref.onChangeEnd,
    style = _ref.style,
    classPrefix = _ref.classPrefix,
    toolTipProps = _ref.toolTipProps,
    hideTips = _ref.hideTips,
    layout = _ref.layout;
  var sliderNodeRef = useRef(null);
  var _useState = useState(false),
    _useState2 = _slicedToArray(_useState, 2),
    popupVisible = _useState2[0],
    setPopupVisible = _useState2[1];
  var _useMouseEvent = useMouseEvent(sliderNodeRef, {
      onEnter: function onEnter() {
        setPopupVisible(true);
      },
      onDown: function onDown() {
        setPopupVisible(true);
      },
      onMove: function onMove(e) {
        setPopupVisible(true);
        onChange(e);
      },
      onLeave: function onLeave() {
        setPopupVisible(false);
      },
      onUp: function onUp(e) {
        setPopupVisible(false);
        onChange(e);
        onChangeEnd === null || onChangeEnd === void 0 || onChangeEnd(e);
      }
    }),
    isMoving = _useMouseEvent.isMoving;
  var handleNode = /* @__PURE__ */React.createElement("div", {
    ref: sliderNodeRef,
    style: style,
    className: "".concat(classPrefix, "-slider__button-wrapper"),
    onClick: function onClick(e) {
      return e.stopPropagation();
    }
  }, /* @__PURE__ */React.createElement("div", {
    className: classNames("".concat(classPrefix, "-slider__button"), _defineProperty({}, "".concat(classPrefix, "-slider__button--dragging"), isMoving))
  }));
  return hideTips ? handleNode : /* @__PURE__ */React.createElement(Tooltip, _objectSpread({
    visible: popupVisible,
    placement: layout === "horizontal" ? "top" : "right"
  }, toolTipProps), handleNode);
};

export { SliderHandleButton as default };
