// tdesign-react/es/slider/Slider.js
/**
 * tdesign v1.17.1
 * (c) 2026 tdesign
 * @license MIT
 */

import { _ as _defineProperty } from '../_chunks/dep-55b77e26.js';
import { _ as _typeof } from '../_chunks/dep-067cb22b.js';
import { _ as _slicedToArray } from '../_chunks/dep-d1eeb95c.js';
import React, { useRef, useMemo } from 'react';
import classNames from 'classnames';
import { isEqual, isFunction, isString, isNumber } from 'lodash-es';
import { l as largeNumberToFixed } from '../_chunks/dep-adc7334a.js';
import { accAdd, numberToPercent } from '../_util/number.js';
import useConfig from '../hooks/useConfig.js';
import useControlled from '../hooks/useControlled.js';
import useDefaultProps from '../hooks/useDefaultProps.js';
import InputNumber from '../input-number/InputNumber.js';
import { sliderDefaultProps } from './defaultProps.js';
import SliderHandleButton from './SliderHandleButton.js';
import '../_chunks/dep-673e8854.js';
import '../_chunks/dep-570a5adb.js';
import '../_chunks/dep-65b369a5.js';
import '../_chunks/dep-0f787ecd.js';
import '../config-provider/ConfigContext.js';
import '../locale/zh_CN.js';
import '../_chunks/dep-e56ffacc.js';
import 'dayjs';
import '../_chunks/dep-84036ea0.js';
import '../_util/noop.js';
import 'tdesign-icons-react';
import '../button/index.js';
import '../button/style/css.js';
import '../button/Button.js';
import '../_chunks/dep-84926167.js';
import '../_util/parseTNode.js';
import '../hooks/useDomRefCallback.js';
import '../hooks/useRipple.js';
import '../_chunks/dep-e17ec8df.js';
import '../_chunks/dep-8c226881.js';
import '../hooks/useAnimation.js';
import '../loading/index.js';
import '../loading/style/css.js';
import '../loading/Loading.js';
import '../common/Portal.js';
import 'react-dom';
import '../hooks/useLayoutEffect.js';
import '../loading/defaultProps.js';
import '../loading/gradient.js';
import '../_chunks/dep-89bc2167.js';
import '../_chunks/dep-154fcb93.js';
import '../_chunks/dep-e9c0bb63.js';
import '../loading/plugin.js';
import '../_util/react-render.js';
import '../_chunks/dep-55f620a2.js';
import '../common/PluginContainer.js';
import '../config-provider/index.js';
import '../config-provider/ConfigProvider.js';
import '../config-provider/type.js';
import '../loading/type.js';
import '../button/defaultProps.js';
import '../button/type.js';
import '../hooks/useGlobalIcon.js';
import '../input/index.js';
import '../input/style/css.js';
import '../input/Input.js';
import '../_util/forwardRefWithStatics.js';
import 'hoist-non-react-statics';
import '../locale/LocalReceiver.js';
import '../input/defaultProps.js';
import '../input/InputGroup.js';
import '../input/useLengthLimit.js';
import '../input/type.js';
import '../input-number/defaultProps.js';
import '../input-number/useInputNumber.js';
import '../hooks/useCommonClassName.js';
import '../hooks/useMouseEvent.js';
import '../hooks/useLatest.js';
import '../tooltip/Tooltip.js';
import '../popup/index.js';
import '../popup/style/css.js';
import '../popup/Popup.js';
import 'react-transition-group';
import '../hooks/useAttach.js';
import '../hooks/useMutationObserver.js';
import '../hooks/usePopper.js';
import 'react-fast-compare';
import '@popperjs/core';
import '../hooks/useWindowSize.js';
import '../popup/defaultProps.js';
import '../popup/hooks/useTrigger.js';
import '../_util/listener.js';
import '../_util/ref.js';
import 'react-is';
import '../_util/isFragment.js';
import '../popup/utils/transition.js';
import '../popup/PopupPlugin.js';
import '../popup/type.js';
import '../tooltip/defaultProps.js';

function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { _defineProperty(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
var LEFT_NODE = 0;
var RIGHT_NODE = 1;
var Slider = /*#__PURE__*/React.forwardRef(function (originalProps, ref) {
  var _useConfig = useConfig(),
    classPrefix = _useConfig.classPrefix;
  var props = useDefaultProps(originalProps, sliderDefaultProps);
  var disabled = props.disabled,
    inputNumberProps = props.inputNumberProps,
    label = props.label,
    layout = props.layout,
    marks = props.marks,
    max = props.max,
    min = props.min,
    range = props.range,
    showStep = props.showStep,
    step = props.step,
    tooltipProps = props.tooltipProps,
    className = props.className,
    style = props.style,
    onChange = props.onChange,
    onChangeEnd = props.onChangeEnd;
  var sliderRef = useRef(null);
  var _useControlled = useControlled(props, "value", onChange),
    _useControlled2 = _slicedToArray(_useControlled, 2),
    value = _useControlled2[0],
    internalOnChange = _useControlled2[1];
  var isVertical = layout === "vertical";
  var renderValue = Array.isArray(value) ? value : [min, value];
  var start = (renderValue[LEFT_NODE] - min) / (max - min);
  var width = (renderValue[RIGHT_NODE] - renderValue[LEFT_NODE]) / (max - min);
  var end = start + width;
  var precision = useMemo(function () {
    if (!Number.isInteger(step)) return step.toString().split(".")[1].length;
    return void 0;
  }, [step]);
  var dots = useMemo(function () {
    if (Array.isArray(marks)) {
      if (marks.some(function (mark) {
        return typeof mark !== "number";
      })) {
        console.warn('The props "marks" only support number!');
        return [];
      }
      return marks.map(function (mark) {
        return {
          value: mark,
          position: (mark - min) / (max - min),
          label: mark
        };
      });
    }
    if (marks && _typeof(marks) === "object") {
      var result = [];
      Object.keys(marks).forEach(function (key) {
        var numberKey = Number(key);
        if (typeof numberKey !== "number") {
          console.warn('The props "marks" key only support number!');
        } else {
          result.push({
            value: numberKey,
            label: marks[numberKey],
            position: (numberKey - min) / (max - min)
          });
        }
      });
      return result;
    }
    return [];
  }, [max, min, marks]);
  var allDots = useMemo(function () {
    var result = [];
    for (var i = min; i <= max; i = accAdd(i, step)) {
      result.push({
        value: i,
        position: (i - min) / (max - min)
      });
    }
    return result;
  }, [max, min, step]);
  var startDirection = isVertical ? "bottom" : "left";
  var stepDirection = isVertical ? "top" : "left";
  var sizeKey = isVertical ? "height" : "width";
  var renderDots = isVertical ? dots.map(function (item) {
    return _objectSpread(_objectSpread({}, item), {}, {
      position: 1 - item.position
    });
  }) : dots;
  var handleInputChange = function handleInputChange(newValue, nodeIndex, isEnd) {
    var safeValue = Number(newValue.toFixed(32));
    var resultValue = Math.max(Math.min(max, safeValue), min);
    if (precision) resultValue = Number(largeNumberToFixed(String(resultValue), precision));
    if (nodeIndex === LEFT_NODE && value && safeValue > value[RIGHT_NODE]) resultValue = value[RIGHT_NODE];
    if (nodeIndex === RIGHT_NODE && value && safeValue < value[LEFT_NODE]) resultValue = value[LEFT_NODE];
    var finalValue;
    if (Array.isArray(value)) {
      var arrValue = value.slice();
      arrValue[nodeIndex] = resultValue;
      finalValue = arrValue;
    } else {
      finalValue = resultValue;
    }
    if (isEqual(value, finalValue)) {
      if (isEnd) {
        onChangeEnd === null || onChangeEnd === void 0 || onChangeEnd(finalValue);
      }
      return;
    }
    internalOnChange(finalValue);
  };
  var createInput = function createInput(nodeIndex) {
    var inputProps = _typeof(inputNumberProps) === "object" ? inputNumberProps : {};
    var currentValue = renderValue[nodeIndex];
    return /* @__PURE__ */React.createElement(InputNumber, _objectSpread({
      value: currentValue,
      onChange: function onChange(v) {
        if (typeof v !== "undefined") {
          handleInputChange(Number(v), nodeIndex);
        }
      },
      className: classNames("".concat(classPrefix, "-slider-input"), {
        "is-vertical": isVertical
      }),
      disabled: disabled,
      theme: "column",
      min: min,
      max: max
    }, inputProps));
  };
  var nearbyValueChange = function nearbyValueChange(value2, isEnd) {
    var buttonBias = Math.abs(value2 - renderValue[LEFT_NODE]) > Math.abs(value2 - renderValue[RIGHT_NODE]) ? RIGHT_NODE : LEFT_NODE;
    handleInputChange(value2, buttonBias, isEnd);
  };
  var setPosition = function setPosition(position, nodeIndex, isEnd) {
    var index = 0;
    var minDistance = 1;
    for (var i = 0; i < allDots.length; i++) {
      var diff = Math.abs(allDots[i].position - position);
      if (minDistance > diff) {
        index = i;
        minDistance = diff;
      }
    }
    var value2 = allDots[index].value;
    if (nodeIndex === void 0 && range) {
      nearbyValueChange(value2, isEnd);
    } else {
      handleInputChange(value2, nodeIndex, isEnd);
    }
  };
  var onSliderChange = function onSliderChange(event, nodeIndex, isEnd) {
    if (disabled || !sliderRef.current) return;
    var clientKey = isVertical ? "clientY" : "clientX";
    var sliderPositionInfo = sliderRef.current.getBoundingClientRect();
    var sliderOffset = sliderPositionInfo[startDirection];
    var position = (event[clientKey] - sliderOffset) / sliderPositionInfo[sizeKey] * (isVertical ? -1 : 1);
    setPosition(position, nodeIndex, isEnd);
  };
  var handleClickMarks = function handleClickMarks(event, value2) {
    event.stopPropagation();
    nearbyValueChange(value2, true);
  };
  var createHandleButton = function createHandleButton(nodeIndex, style2) {
    var currentValue = renderValue[nodeIndex];
    var tipLabel = currentValue;
    if (isFunction(label)) {
      tipLabel = label({
        value: currentValue,
        position: nodeIndex === LEFT_NODE ? "start" : "end"
      });
    }
    if (isString(label)) {
      tipLabel = label.replace(/\$\{value\}/g, currentValue.toString());
    }
    if (isNumber(tipLabel) && precision) tipLabel = largeNumberToFixed(String(tipLabel), precision);
    return /* @__PURE__ */React.createElement(SliderHandleButton, {
      toolTipProps: _objectSpread({
        content: tipLabel
      }, tooltipProps),
      hideTips: label === false,
      classPrefix: classPrefix,
      layout: layout,
      style: style2,
      onChange: function onChange(e) {
        return onSliderChange(e, nodeIndex);
      },
      onChangeEnd: function onChangeEnd(e) {
        return onSliderChange(e, nodeIndex, true);
      }
    });
  };
  return /* @__PURE__ */React.createElement("div", {
    style: _objectSpread({}, style),
    className: classNames("".concat(classPrefix, "-slider__container"), {
      "is-vertical": isVertical
    }),
    ref: ref
  }, /* @__PURE__ */React.createElement("div", {
    ref: sliderRef,
    className: classNames("".concat(classPrefix, "-slider"), className, _defineProperty(_defineProperty(_defineProperty({}, "".concat(classPrefix, "-is-disabled"), disabled), "".concat(classPrefix, "-slider--vertical"), isVertical), "".concat(classPrefix, "-slider--with-input"), inputNumberProps)),
    onClick: function onClick(e) {
      return onSliderChange(e, void 0, true);
    }
  }, /* @__PURE__ */React.createElement("div", {
    className: classNames("".concat(classPrefix, "-slider__rail"))
  }, /* @__PURE__ */React.createElement("div", {
    style: _defineProperty(_defineProperty({}, startDirection, numberToPercent(start)), sizeKey, numberToPercent(width)),
    className: classNames("".concat(classPrefix, "-slider__track"))
  }), range ? createHandleButton(LEFT_NODE, _defineProperty({}, startDirection, numberToPercent(start))) : null, createHandleButton(RIGHT_NODE, _defineProperty({}, startDirection, numberToPercent(end))), /* @__PURE__ */React.createElement("div", {
    className: "".concat(classPrefix, "-slider__stops")
  }, (showStep ? allDots : renderDots).filter(function (_ref) {
    var position = _ref.position;
    return position !== 0 && position !== 1;
  }).map(function (_ref2) {
    var dotValue = _ref2.value,
      position = _ref2.position;
    return /* @__PURE__ */React.createElement("div", {
      key: dotValue,
      style: _defineProperty({}, stepDirection, numberToPercent(position)),
      className: classNames("".concat(classPrefix, "-slider__stop"))
    });
  })), /* @__PURE__ */React.createElement("div", {
    className: classNames("".concat(classPrefix, "-slider__mark"))
  }, renderDots.map(function (_ref3) {
    var position = _ref3.position,
      value2 = _ref3.value,
      label2 = _ref3.label;
    return /* @__PURE__ */React.createElement("div", {
      key: value2,
      onClick: function onClick(event) {
        return handleClickMarks(event, value2);
      },
      style: _defineProperty({}, stepDirection, numberToPercent(position)),
      className: classNames("".concat(classPrefix, "-slider__mark-text"))
    }, label2);
  })))), inputNumberProps ? /* @__PURE__ */React.createElement("div", {
    className: classNames("".concat(classPrefix, "-slider__input-container"), {
      "is-vertical": isVertical
    })
  }, range && createInput(LEFT_NODE), range && /* @__PURE__ */React.createElement("div", {
    className: "".concat(classPrefix, "-slider__center-line")
  }), createInput(RIGHT_NODE)) : null);
});
Slider.displayName = "Slider";

export { Slider as default };
