// tdesign-react/cjs/qrcode/QRCode.js
/**
 * tdesign v1.17.1
 * (c) 2026 tdesign
 * @license MIT
 */

'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var defineProperty = require('../_chunks/dep-c446aa11.js');
var slicedToArray = require('../_chunks/dep-c60a3e4c.js');
var objectWithoutProperties = require('../_chunks/dep-e41137f7.js');
var React = require('react');
var classNames = require('classnames');
var hooks_useConfig = require('../hooks/useConfig.js');
var hooks_useDefaultProps = require('../hooks/useDefaultProps.js');
var hooks_useVariables = require('../hooks/useVariables.js');
var locale_LocalReceiver = require('../locale/LocalReceiver.js');
var qrcode_defaultProps = require('./defaultProps.js');
var qrcode_QRCodeCanvas = require('./QRCodeCanvas.js');
var qrcode_QRCodeStatus = require('./QRCodeStatus.js');
var qrcode_QRCodeSVG = require('./QRCodeSVG.js');
require('../_chunks/dep-c290284d.js');
require('../_chunks/dep-98e6819c.js');
require('../_chunks/dep-7f3fa935.js');
require('../config-provider/ConfigContext.js');
require('../_chunks/dep-2ee09518.js');
require('../_chunks/dep-a6fa90dc.js');
require('dayjs');
require('../_chunks/dep-012d9a7c.js');
require('../_chunks/dep-150a112c.js');
require('../_chunks/dep-e360e17f.js');
require('../_chunks/dep-3d05de97.js');
require('../_chunks/dep-4f1729f0.js');
require('../_chunks/dep-a54f202c.js');
require('../_chunks/dep-51284537.js');
require('../_chunks/dep-8a493834.js');
require('../_chunks/dep-9f5b2891.js');
require('../_chunks/dep-338f013a.js');
require('../_chunks/dep-c37c5e17.js');
require('../_chunks/dep-eee1aab9.js');
require('../_chunks/dep-11347ea1.js');
require('../_chunks/dep-a830ce1e.js');
require('../_chunks/dep-770bc0ae.js');
require('../_chunks/dep-aaf27e39.js');
require('../_chunks/dep-b47d855f.js');
require('../_chunks/dep-81bce4e6.js');
require('../_chunks/dep-863f8f4d.js');
require('../_chunks/dep-84e4d0da.js');
require('../_chunks/dep-f78183d7.js');
require('../_chunks/dep-2f45f6c4.js');
require('../_chunks/dep-29222115.js');
require('../_chunks/dep-ee71ddca.js');
require('../_chunks/dep-427f8bc0.js');
require('../hooks/useMutationObserver.js');
require('../hooks/useLatest.js');
require('../_chunks/dep-aa9a2c54.js');
require('../_chunks/dep-77ef3011.js');
require('../_chunks/dep-b36779a4.js');
require('../_chunks/dep-551d5940.js');
require('../_chunks/dep-8005b808.js');
require('../_chunks/dep-2add8824.js');
require('../_chunks/dep-f2809376.js');
require('../_chunks/dep-3be56e1e.js');
require('../_chunks/dep-36579bf6.js');
require('../config-provider/ConfigProvider.js');
require('../_chunks/dep-ba90997f.js');
require('../_chunks/dep-22026fea.js');
require('../_chunks/dep-457df323.js');
require('../_chunks/dep-fb453fc0.js');
require('tdesign-icons-react');
require('../loading/index.js');
require('../loading/Loading.js');
require('../common/Portal.js');
require('react-dom');
require('../hooks/useLayoutEffect.js');
require('../loading/defaultProps.js');
require('../loading/gradient.js');
require('../_chunks/dep-062bb1b1.js');
require('../_chunks/dep-6e0da644.js');
require('../_chunks/dep-05bc49e6.js');
require('../_chunks/dep-8c880a65.js');
require('../_chunks/dep-b8d0b96c.js');
require('../hooks/useDomRefCallback.js');
require('../loading/plugin.js');
require('../_util/react-render.js');
require('../_chunks/dep-7bf14076.js');
require('../common/PluginContainer.js');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

var React__default = /*#__PURE__*/_interopDefaultLegacy(React);
var classNames__default = /*#__PURE__*/_interopDefaultLegacy(classNames);

var _excluded = ["className", "value", "borderless", "iconSize", "color", "bgColor", "style", "icon", "size", "type", "status", "onRefresh", "statusRender"];
function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { defineProperty._defineProperty(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
var QRCode = function QRCode(props) {
  var _iconSize$height, _iconSize$width, _style$width, _style$height;
  var _useDefaultProps = hooks_useDefaultProps["default"](props, qrcode_defaultProps.qRCodeDefaultProps),
    className = _useDefaultProps.className,
    value = _useDefaultProps.value,
    borderless = _useDefaultProps.borderless,
    iconSize = _useDefaultProps.iconSize,
    color = _useDefaultProps.color,
    bgColor = _useDefaultProps.bgColor,
    style = _useDefaultProps.style,
    icon = _useDefaultProps.icon,
    size = _useDefaultProps.size,
    type = _useDefaultProps.type,
    status = _useDefaultProps.status,
    onRefresh = _useDefaultProps.onRefresh,
    statusRender = _useDefaultProps.statusRender,
    rest = objectWithoutProperties._objectWithoutProperties(_useDefaultProps, _excluded);
  var _useConfig = hooks_useConfig["default"](),
    classPrefix = _useConfig.classPrefix;
  var _useLocaleReceiver = locale_LocalReceiver.useLocaleReceiver("qrcode"),
    _useLocaleReceiver2 = slicedToArray._slicedToArray(_useLocaleReceiver, 1),
    locale = _useLocaleReceiver2[0];
  var _useVariables = hooks_useVariables["default"]({
      themeFgColor: "--td-text-color-primary",
      themeBgColor: "--td-bg-color-specialcomponent"
    }),
    themeFgColor = _useVariables.themeFgColor,
    themeBgColor = _useVariables.themeBgColor;
  var finalBgColor = React.useMemo(function () {
    return bgColor || themeBgColor || "transparent";
  }, [bgColor, themeBgColor]);
  if (!value) {
    return null;
  }
  var imageSettings = {
    src: icon,
    x: void 0,
    y: void 0,
    height: typeof iconSize === "number" ? iconSize : (_iconSize$height = iconSize === null || iconSize === void 0 ? void 0 : iconSize.height) !== null && _iconSize$height !== void 0 ? _iconSize$height : 40,
    width: typeof iconSize === "number" ? iconSize : (_iconSize$width = iconSize === null || iconSize === void 0 ? void 0 : iconSize.width) !== null && _iconSize$width !== void 0 ? _iconSize$width : 40,
    excavate: true,
    crossOrigin: "anonymous"
  };
  var qrCodeProps = _objectSpread({
    value: value,
    size: size,
    bgColor: finalBgColor,
    fgColor: color || themeFgColor,
    imageSettings: icon ? imageSettings : void 0
  }, rest);
  var cls = classNames__default["default"]("".concat(classPrefix, "-qrcode"), defineProperty._defineProperty(defineProperty._defineProperty({}, "".concat(classPrefix, "-borderless"), borderless), "".concat(classPrefix, "-qrcode-svg"), type === "svg"), className);
  var mergedStyle = _objectSpread(_objectSpread({
    backgroundColor: finalBgColor
  }, style), {}, {
    width: (_style$width = style === null || style === void 0 ? void 0 : style.width) !== null && _style$width !== void 0 ? _style$width : size,
    height: (_style$height = style === null || style === void 0 ? void 0 : style.height) !== null && _style$height !== void 0 ? _style$height : size
  });
  return /* @__PURE__ */React__default["default"].createElement("div", _objectSpread(_objectSpread({}, rest), {}, {
    className: cls,
    style: mergedStyle
  }), status !== "active" && /* @__PURE__ */React__default["default"].createElement("div", {
    className: "".concat(classPrefix, "-mask")
  }, /* @__PURE__ */React__default["default"].createElement(qrcode_QRCodeStatus["default"], {
    classPrefix: classPrefix,
    locale: locale,
    status: status,
    onRefresh: onRefresh,
    statusRender: statusRender
  })), type === "canvas" ? /* @__PURE__ */React__default["default"].createElement(qrcode_QRCodeCanvas.QRCodeCanvas, _objectSpread(_objectSpread({}, qrCodeProps), {}, {
    size: size
  })) : /* @__PURE__ */React__default["default"].createElement(qrcode_QRCodeSVG.QRCodeSVG, _objectSpread(_objectSpread({}, qrCodeProps), {}, {
    size: size
  })));
};
QRCode.displayName = "QRCode";

exports["default"] = QRCode;
