// tdesign-react/cjs/qrcode/QRCodeStatus.js
/**
 * tdesign v1.17.1
 * (c) 2026 tdesign
 * @license MIT
 */

'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var React = require('react');
var tdesignIconsReact = require('tdesign-icons-react');
var loading_index = require('../loading/index.js');
require('../loading/Loading.js');
require('../_chunks/dep-c446aa11.js');
require('../_chunks/dep-c290284d.js');
require('../_chunks/dep-c60a3e4c.js');
require('../_chunks/dep-98e6819c.js');
require('../_chunks/dep-7f3fa935.js');
require('classnames');
require('../_chunks/dep-ee71ddca.js');
require('../_chunks/dep-427f8bc0.js');
require('../_chunks/dep-3d05de97.js');
require('../_chunks/dep-a54f202c.js');
require('../_chunks/dep-4f1729f0.js');
require('../common/Portal.js');
require('react-dom');
require('../hooks/useConfig.js');
require('../config-provider/ConfigContext.js');
require('../_chunks/dep-2ee09518.js');
require('../_chunks/dep-a6fa90dc.js');
require('dayjs');
require('../_chunks/dep-012d9a7c.js');
require('../_chunks/dep-150a112c.js');
require('../_chunks/dep-e360e17f.js');
require('../_chunks/dep-51284537.js');
require('../_chunks/dep-8a493834.js');
require('../_chunks/dep-9f5b2891.js');
require('../_chunks/dep-338f013a.js');
require('../_chunks/dep-c37c5e17.js');
require('../_chunks/dep-eee1aab9.js');
require('../_chunks/dep-11347ea1.js');
require('../_chunks/dep-a830ce1e.js');
require('../_chunks/dep-770bc0ae.js');
require('../_chunks/dep-aaf27e39.js');
require('../_chunks/dep-b47d855f.js');
require('../_chunks/dep-81bce4e6.js');
require('../_chunks/dep-863f8f4d.js');
require('../_chunks/dep-84e4d0da.js');
require('../_chunks/dep-f78183d7.js');
require('../_chunks/dep-2f45f6c4.js');
require('../hooks/useLayoutEffect.js');
require('../hooks/useDefaultProps.js');
require('../loading/defaultProps.js');
require('../loading/gradient.js');
require('../_chunks/dep-062bb1b1.js');
require('../_chunks/dep-6e0da644.js');
require('../_chunks/dep-05bc49e6.js');
require('../_chunks/dep-8c880a65.js');
require('../_chunks/dep-e41137f7.js');
require('../_chunks/dep-b8d0b96c.js');
require('../hooks/useDomRefCallback.js');
require('../loading/plugin.js');
require('../_util/react-render.js');
require('../_chunks/dep-7bf14076.js');
require('../common/PluginContainer.js');
require('../config-provider/ConfigProvider.js');
require('../_chunks/dep-ba90997f.js');
require('../_chunks/dep-22026fea.js');
require('../_chunks/dep-551d5940.js');
require('../_chunks/dep-8005b808.js');
require('../_chunks/dep-2add8824.js');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

var React__default = /*#__PURE__*/_interopDefaultLegacy(React);

var defaultSpin = /* @__PURE__ */React__default["default"].createElement(loading_index.Loading, {
  size: "32px"
});
function QRcodeStatus(_ref) {
  var locale = _ref.locale,
    classPrefix = _ref.classPrefix,
    onRefresh = _ref.onRefresh,
    statusRender = _ref.statusRender,
    status = _ref.status;
  var defaultExpiredNode = React.useMemo(function () {
    return /* @__PURE__ */React__default["default"].createElement(React__default["default"].Fragment, null, /* @__PURE__ */React__default["default"].createElement("p", {
      className: "".concat(classPrefix, "-expired__text")
    }, locale === null || locale === void 0 ? void 0 : locale.expiredText), onRefresh && /* @__PURE__ */React__default["default"].createElement("p", {
      className: "".concat(classPrefix, "-expired__button"),
      onClick: onRefresh
    }, /* @__PURE__ */React__default["default"].createElement(tdesignIconsReact.RefreshIcon, {
      size: 16
    }), locale === null || locale === void 0 ? void 0 : locale.refreshText));
  }, [classPrefix, locale === null || locale === void 0 ? void 0 : locale.expiredText, locale === null || locale === void 0 ? void 0 : locale.refreshText, onRefresh]);
  var defaultScannedNode = React.useMemo(function () {
    return /* @__PURE__ */React__default["default"].createElement("p", {
      className: "".concat(classPrefix, "-scanned")
    }, /* @__PURE__ */React__default["default"].createElement(tdesignIconsReact.CheckCircleFilledIcon, {
      size: 16,
      className: "".concat(classPrefix, "-scanned__icon")
    }), locale === null || locale === void 0 ? void 0 : locale.scannedText);
  }, [classPrefix, locale === null || locale === void 0 ? void 0 : locale.scannedText]);
  var defaultNodes = React.useMemo(function () {
    return {
      expired: defaultExpiredNode,
      loading: defaultSpin,
      scanned: defaultScannedNode
    };
  }, [defaultExpiredNode, defaultScannedNode]);
  var defaultStatusRender = React.useCallback(function (info) {
    return defaultNodes[info.status];
  }, [defaultNodes]);
  var mergedStatusRender = statusRender !== null && statusRender !== void 0 ? statusRender : defaultStatusRender;
  return /* @__PURE__ */React__default["default"].createElement(React__default["default"].Fragment, null, mergedStatusRender({
    status: status,
    onRefresh: onRefresh
  }) || null);
}

exports["default"] = QRcodeStatus;
