// tdesign-react/cjs/time-picker/TimePicker.js
/**
 * tdesign v1.17.1
 * (c) 2026 tdesign
 * @license MIT
 */

'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var defineProperty = require('../_chunks/dep-c446aa11.js');
var slicedToArray = require('../_chunks/dep-c60a3e4c.js');
var React = require('react');
var classNames = require('classnames');
var dayjs = require('dayjs');
var customParseFormat = require('../_chunks/dep-34b7c4f7.js');
var tdesignIconsReact = require('tdesign-icons-react');
var timePicker_panel_SinglePanel = require('../_chunks/dep-7592c92b.js');
var _util_forwardRefWithStatics = require('../_util/forwardRefWithStatics.js');
var _util_noop = require('../_util/noop.js');
var hooks_useConfig = require('../hooks/useConfig.js');
var hooks_useControlled = require('../hooks/useControlled.js');
var hooks_useDefaultProps = require('../hooks/useDefaultProps.js');
var hooks_useGlobalIcon = require('../hooks/useGlobalIcon.js');
var selectInput_index = require('../select-input/index.js');
var timePicker_defaultProps = require('./defaultProps.js');
var timePicker_hooks_useTimePickerTextConfig = require('./hooks/useTimePickerTextConfig.js');
var timePicker_panel_TimePickerPanel = require('./panel/TimePickerPanel.js');
var timePicker_TimeRangePicker = require('./TimeRangePicker.js');
require('../_chunks/dep-c290284d.js');
require('../_chunks/dep-98e6819c.js');
require('../_chunks/dep-7f3fa935.js');
require('../_chunks/dep-a6fa90dc.js');
require('../_chunks/dep-c1575286.js');
require('../_chunks/dep-3be56e1e.js');
require('../_chunks/dep-a830ce1e.js');
require('../_chunks/dep-36579bf6.js');
require('../_chunks/dep-3d05de97.js');
require('../_chunks/dep-4f1729f0.js');
require('../hooks/useDebounce.js');
require('../hooks/usePersistFn.js');
require('../_chunks/dep-f2809376.js');
require('../hooks/usePropsRef.js');
require('../_chunks/dep-2f45f6c4.js');
require('../_chunks/dep-770bc0ae.js');
require('../_chunks/dep-863f8f4d.js');
require('../_chunks/dep-11347ea1.js');
require('../_chunks/dep-8a493834.js');
require('../_chunks/dep-338f013a.js');
require('../_chunks/dep-b199860d.js');
require('../_chunks/dep-a54f202c.js');
require('../_chunks/dep-eccb6a94.js');
require('../_chunks/dep-f5e1d11c.js');
require('hoist-non-react-statics');
require('../config-provider/ConfigContext.js');
require('../_chunks/dep-2ee09518.js');
require('../_chunks/dep-012d9a7c.js');
require('../_chunks/dep-150a112c.js');
require('../_chunks/dep-e360e17f.js');
require('../_chunks/dep-51284537.js');
require('../_chunks/dep-9f5b2891.js');
require('../_chunks/dep-c37c5e17.js');
require('../_chunks/dep-eee1aab9.js');
require('../_chunks/dep-aaf27e39.js');
require('../_chunks/dep-b47d855f.js');
require('../_chunks/dep-81bce4e6.js');
require('../_chunks/dep-84e4d0da.js');
require('../_chunks/dep-f78183d7.js');
require('../_chunks/dep-45188d54.js');
require('../select-input/SelectInput.js');
require('../popup/index.js');
require('../popup/Popup.js');
require('react-transition-group');
require('../_chunks/dep-ee71ddca.js');
require('../_chunks/dep-427f8bc0.js');
require('../common/Portal.js');
require('react-dom');
require('../hooks/useLayoutEffect.js');
require('../hooks/useAnimation.js');
require('../hooks/useAttach.js');
require('../hooks/useMutationObserver.js');
require('../hooks/useLatest.js');
require('../_chunks/dep-aa9a2c54.js');
require('../_chunks/dep-77ef3011.js');
require('../_chunks/dep-b36779a4.js');
require('../_chunks/dep-551d5940.js');
require('../_chunks/dep-8005b808.js');
require('../_chunks/dep-2add8824.js');
require('../hooks/usePopper.js');
require('../_chunks/dep-05bc49e6.js');
require('../_chunks/dep-8c880a65.js');
require('react-fast-compare');
require('@popperjs/core');
require('../hooks/useWindowSize.js');
require('../popup/defaultProps.js');
require('../popup/hooks/useTrigger.js');
require('../_util/listener.js');
require('../_util/ref.js');
require('react-is');
require('../_util/isFragment.js');
require('../popup/utils/transition.js');
require('../popup/PopupPlugin.js');
require('../_chunks/dep-7bf14076.js');
require('../_util/react-render.js');
require('../common/PluginContainer.js');
require('../config-provider/ConfigProvider.js');
require('../_chunks/dep-ba90997f.js');
require('../_chunks/dep-22026fea.js');
require('../select-input/defaultProps.js');
require('../select-input/useMultiple.js');
require('../tag-input/index.js');
require('../tag-input/TagInput.js');
require('../hooks/useDragSorter.js');
require('../hooks/useEventCallback.js');
require('../input/index.js');
require('../input/Input.js');
require('../_chunks/dep-e41137f7.js');
require('../_util/parseTNode.js');
require('../_chunks/dep-3c628d18.js');
require('../locale/LocalReceiver.js');
require('../input/defaultProps.js');
require('../input/InputGroup.js');
require('../input/useLengthLimit.js');
require('../_chunks/dep-6e0da644.js');
require('../_chunks/dep-b8d0b96c.js');
require('../tag-input/defaultProps.js');
require('../tag-input/useHover.js');
require('../tag-input/useTagList.js');
require('../tag/index.js');
require('../tag/CheckTag.js');
require('../_chunks/dep-29222115.js');
require('../tag/defaultProps.js');
require('../tag/Tag.js');
require('../_chunks/dep-183643e1.js');
require('../tag/CheckTagGroup.js');
require('../tag-input/useTagScroll.js');
require('../select-input/useOverlayInnerStyle.js');
require('../hooks/useInnerPopupVisible.js');
require('../select-input/useSingle.js');
require('../loading/index.js');
require('../loading/Loading.js');
require('../loading/defaultProps.js');
require('../loading/gradient.js');
require('../_chunks/dep-062bb1b1.js');
require('../hooks/useDomRefCallback.js');
require('../loading/plugin.js');
require('../_chunks/dep-3784bbea.js');
require('../_chunks/dep-d4a6a9c1.js');
require('../_chunks/dep-bc160f5b.js');
require('../_chunks/dep-363864ae.js');
require('../_chunks/dep-204188de.js');
require('../_chunks/dep-7a4e0117.js');
require('../button/index.js');
require('../button/Button.js');
require('../hooks/useRipple.js');
require('../button/defaultProps.js');
require('../range-input/index.js');
require('../range-input/RangeInput.js');
require('../range-input/defaultProps.js');
require('../range-input/RangeInputPopup.js');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

var React__default = /*#__PURE__*/_interopDefaultLegacy(React);
var classNames__default = /*#__PURE__*/_interopDefaultLegacy(classNames);
var dayjs__default = /*#__PURE__*/_interopDefaultLegacy(dayjs);

function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { defineProperty._defineProperty(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
dayjs__default["default"].extend(customParseFormat.customParseFormat);
var TimePicker = _util_forwardRefWithStatics["default"](function (originalProps, ref) {
  var props = hooks_useDefaultProps["default"](originalProps, timePicker_defaultProps.timePickerDefaultProps);
  var TEXT_CONFIG = timePicker_hooks_useTimePickerTextConfig.useTimePickerTextConfig();
  var allowInput = props.allowInput,
    borderless = props.borderless,
    className = props.className,
    clearable = props.clearable,
    disabled = props.disabled,
    style = props.style,
    _props$format = props.format,
    format = _props$format === void 0 ? timePicker_panel_SinglePanel.DEFAULT_FORMAT : _props$format,
    _props$hideDisabledTi = props.hideDisabledTime,
    hideDisabledTime = _props$hideDisabledTi === void 0 ? true : _props$hideDisabledTi,
    _props$steps = props.steps,
    steps = _props$steps === void 0 ? timePicker_panel_SinglePanel.DEFAULT_STEPS : _props$steps,
    _props$placeholder = props.placeholder,
    placeholder = _props$placeholder === void 0 ? TEXT_CONFIG.placeholder : _props$placeholder,
    disableTime = props.disableTime,
    _props$onBlur = props.onBlur,
    onBlur = _props$onBlur === void 0 ? _util_noop["default"] : _props$onBlur,
    _props$onClose = props.onClose,
    onClose = _props$onClose === void 0 ? _util_noop["default"] : _props$onClose,
    _props$onFocus = props.onFocus,
    onFocus = _props$onFocus === void 0 ? _util_noop["default"] : _props$onFocus,
    _props$onOpen = props.onOpen,
    onOpen = _props$onOpen === void 0 ? _util_noop["default"] : _props$onOpen,
    _props$onInput = props.onInput,
    onInput = _props$onInput === void 0 ? _util_noop["default"] : _props$onInput,
    _props$onPick = props.onPick,
    onPick = _props$onPick === void 0 ? _util_noop["default"] : _props$onPick,
    _props$onClear = props.onClear,
    onClear = _props$onClear === void 0 ? _util_noop["default"] : _props$onClear;
  var _useControlled = hooks_useControlled["default"](props, "value", props.onChange),
    _useControlled2 = slicedToArray._slicedToArray(_useControlled, 2),
    value = _useControlled2[0],
    onChange = _useControlled2[1];
  var _useState = React.useState(false),
    _useState2 = slicedToArray._slicedToArray(_useState, 2),
    isPanelShowed = _useState2[0],
    setPanelShow = _useState2[1];
  var _useState3 = React.useState(""),
    _useState4 = slicedToArray._slicedToArray(_useState3, 2),
    currentValue = _useState4[0],
    setCurrentValue = _useState4[1];
  var _useConfig = hooks_useConfig["default"](),
    classPrefix = _useConfig.classPrefix;
  var _useGlobalIcon = hooks_useGlobalIcon["default"]({
      TimeIcon: tdesignIconsReact.TimeIcon
    }),
    TimeIcon = _useGlobalIcon.TimeIcon;
  var name = "".concat(classPrefix, "-time-picker");
  var inputClasses = classNames__default["default"]("".concat(name, "__group"), defineProperty._defineProperty({}, "".concat(classPrefix, "-is-focused"), isPanelShowed));
  var effectVisibleCurrentValue = function effectVisibleCurrentValue(visible) {
    setPanelShow(visible);
    setCurrentValue(visible ? value !== null && value !== void 0 ? value : "" : "");
  };
  var handleShowPopup = function handleShowPopup(visible, context) {
    effectVisibleCurrentValue(visible);
    visible ? onOpen(context) : onClose(context);
  };
  var handleClear = function handleClear(context) {
    var e = context.e;
    e.stopPropagation();
    setCurrentValue("");
    onChange(null);
    onClear(context);
  };
  var handleInputChange = function handleInputChange(value2, context) {
    setCurrentValue(value2);
    if (allowInput) {
      onInput({
        value: value2,
        e: context.e
      });
    }
  };
  var handleInputBlur = function handleInputBlur(value2, ctx) {
    if (allowInput) {
      var isValidTime = timePicker_panel_SinglePanel.validateInputValue(currentValue, format);
      if (isValidTime) {
        onChange(timePicker_panel_SinglePanel.formatInputValue(currentValue, format));
      }
    }
    onBlur(_objectSpread({
      value: String(value2)
    }, ctx));
  };
  var handleClickConfirm = function handleClickConfirm() {
    var isValidTime = timePicker_panel_SinglePanel.validateInputValue(currentValue, format);
    if (isValidTime) onChange(currentValue);
    effectVisibleCurrentValue(false);
  };
  var handlePanelChange = function handlePanelChange(v, ctx) {
    setCurrentValue(v);
    onPick === null || onPick === void 0 || onPick(v, ctx);
  };
  return /* @__PURE__ */React__default["default"].createElement("div", {
    className: classNames__default["default"](name, className),
    ref: ref,
    style: style
  }, /* @__PURE__ */React__default["default"].createElement(selectInput_index.SelectInput, {
    onFocus: onFocus,
    onClear: handleClear,
    disabled: disabled,
    clearable: clearable,
    allowInput: allowInput,
    className: inputClasses,
    borderless: borderless,
    suffixIcon: /* @__PURE__ */React__default["default"].createElement(TimeIcon, null),
    popupVisible: isPanelShowed,
    onInputChange: handleInputChange,
    onBlur: handleInputBlur,
    onPopupVisibleChange: handleShowPopup,
    placeholder: !value ? placeholder : void 0,
    value: isPanelShowed ? currentValue : value !== null && value !== void 0 ? value : void 0,
    inputValue: isPanelShowed ? currentValue : value !== null && value !== void 0 ? value : void 0,
    inputProps: _objectSpread(_objectSpread({}, props.inputProps), {}, {
      size: props.size
    }),
    popupProps: _objectSpread({
      overlayInnerStyle: {
        width: "auto",
        padding: 0
      }
    }, props.popupProps),
    tips: props.tips,
    status: props.status,
    label: props.label,
    panel: /* @__PURE__ */React__default["default"].createElement(timePicker_panel_TimePickerPanel["default"], {
      steps: steps,
      format: format,
      value: currentValue,
      isFooterDisplay: true,
      isShowPanel: isPanelShowed,
      disableTime: disableTime,
      onChange: handlePanelChange,
      onPick: props.onPick,
      hideDisabledTime: hideDisabledTime,
      handleConfirmClick: handleClickConfirm,
      presets: props.presets
    })
  }));
}, {
  TimeRangePicker: timePicker_TimeRangePicker["default"],
  TimePickerPanel: timePicker_panel_TimePickerPanel["default"]
});
TimePicker.displayName = "TimePicker";

exports["default"] = TimePicker;
