// tdesign-react/cjs/time-picker/TimeRangePicker.js
/**
 * tdesign v1.17.1
 * (c) 2026 tdesign
 * @license MIT
 */

'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var defineProperty = require('../_chunks/dep-c446aa11.js');
var slicedToArray = require('../_chunks/dep-c60a3e4c.js');
var React = require('react');
var classNames = require('classnames');
var dayjs = require('dayjs');
var tdesignIconsReact = require('tdesign-icons-react');
var timePicker_panel_SinglePanel = require('../_chunks/dep-7592c92b.js');
var _util_noop = require('../_util/noop.js');
var hooks_useConfig = require('../hooks/useConfig.js');
var hooks_useControlled = require('../hooks/useControlled.js');
var hooks_useDefaultProps = require('../hooks/useDefaultProps.js');
var hooks_useGlobalIcon = require('../hooks/useGlobalIcon.js');
var rangeInput_index = require('../range-input/index.js');
var timePicker_defaultProps = require('./defaultProps.js');
var timePicker_hooks_useTimePickerTextConfig = require('./hooks/useTimePickerTextConfig.js');
var timePicker_panel_TimePickerPanel = require('./panel/TimePickerPanel.js');
var isArray = require('../_chunks/dep-a54f202c.js');
require('../_chunks/dep-c290284d.js');
require('../_chunks/dep-98e6819c.js');
require('../_chunks/dep-7f3fa935.js');
require('../_chunks/dep-34b7c4f7.js');
require('../_chunks/dep-a6fa90dc.js');
require('../_chunks/dep-c1575286.js');
require('../_chunks/dep-3be56e1e.js');
require('../_chunks/dep-a830ce1e.js');
require('../_chunks/dep-36579bf6.js');
require('../_chunks/dep-3d05de97.js');
require('../_chunks/dep-4f1729f0.js');
require('../hooks/useDebounce.js');
require('../hooks/usePersistFn.js');
require('../_chunks/dep-f2809376.js');
require('../hooks/usePropsRef.js');
require('../_chunks/dep-2f45f6c4.js');
require('../_chunks/dep-770bc0ae.js');
require('../_chunks/dep-863f8f4d.js');
require('../_chunks/dep-11347ea1.js');
require('../_chunks/dep-8a493834.js');
require('../_chunks/dep-338f013a.js');
require('../_chunks/dep-b199860d.js');
require('../_chunks/dep-eccb6a94.js');
require('../_chunks/dep-f5e1d11c.js');
require('../config-provider/ConfigContext.js');
require('../_chunks/dep-2ee09518.js');
require('../_chunks/dep-012d9a7c.js');
require('../_chunks/dep-150a112c.js');
require('../_chunks/dep-e360e17f.js');
require('../_chunks/dep-51284537.js');
require('../_chunks/dep-9f5b2891.js');
require('../_chunks/dep-c37c5e17.js');
require('../_chunks/dep-eee1aab9.js');
require('../_chunks/dep-aaf27e39.js');
require('../_chunks/dep-b47d855f.js');
require('../_chunks/dep-81bce4e6.js');
require('../_chunks/dep-84e4d0da.js');
require('../_chunks/dep-f78183d7.js');
require('../_chunks/dep-45188d54.js');
require('../range-input/RangeInput.js');
require('../_chunks/dep-e41137f7.js');
require('../_util/parseTNode.js');
require('../_chunks/dep-3c628d18.js');
require('../input/index.js');
require('../input/Input.js');
require('../_util/forwardRefWithStatics.js');
require('hoist-non-react-statics');
require('../hooks/useLayoutEffect.js');
require('../_chunks/dep-ee71ddca.js');
require('../_chunks/dep-427f8bc0.js');
require('../locale/LocalReceiver.js');
require('../config-provider/ConfigProvider.js');
require('../_chunks/dep-ba90997f.js');
require('../_chunks/dep-22026fea.js');
require('../_chunks/dep-551d5940.js');
require('../_chunks/dep-8005b808.js');
require('../_chunks/dep-2add8824.js');
require('../input/defaultProps.js');
require('../input/InputGroup.js');
require('../input/useLengthLimit.js');
require('../_chunks/dep-6e0da644.js');
require('../_chunks/dep-05bc49e6.js');
require('../_chunks/dep-8c880a65.js');
require('../_chunks/dep-b8d0b96c.js');
require('../range-input/defaultProps.js');
require('../range-input/RangeInputPopup.js');
require('../popup/index.js');
require('../popup/Popup.js');
require('react-transition-group');
require('../common/Portal.js');
require('react-dom');
require('../hooks/useAnimation.js');
require('../hooks/useAttach.js');
require('../hooks/useMutationObserver.js');
require('../hooks/useLatest.js');
require('../_chunks/dep-aa9a2c54.js');
require('../_chunks/dep-77ef3011.js');
require('../_chunks/dep-b36779a4.js');
require('../hooks/usePopper.js');
require('react-fast-compare');
require('@popperjs/core');
require('../hooks/useWindowSize.js');
require('../popup/defaultProps.js');
require('../popup/hooks/useTrigger.js');
require('../_util/listener.js');
require('../_util/ref.js');
require('react-is');
require('../_util/isFragment.js');
require('../popup/utils/transition.js');
require('../popup/PopupPlugin.js');
require('../_chunks/dep-7bf14076.js');
require('../_util/react-render.js');
require('../common/PluginContainer.js');
require('../select-input/useOverlayInnerStyle.js');
require('../hooks/useInnerPopupVisible.js');
require('../button/index.js');
require('../button/Button.js');
require('../hooks/useDomRefCallback.js');
require('../hooks/useRipple.js');
require('../_chunks/dep-062bb1b1.js');
require('../loading/index.js');
require('../loading/Loading.js');
require('../loading/defaultProps.js');
require('../loading/gradient.js');
require('../loading/plugin.js');
require('../button/defaultProps.js');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

var React__default = /*#__PURE__*/_interopDefaultLegacy(React);
var classNames__default = /*#__PURE__*/_interopDefaultLegacy(classNames);
var dayjs__default = /*#__PURE__*/_interopDefaultLegacy(dayjs);

function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { defineProperty._defineProperty(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
function handlePositionTrans(income) {
  return income === "first" ? "start" : "end";
}
var TimeRangePicker = function TimeRangePicker(originalProps) {
  var props = hooks_useDefaultProps["default"](originalProps, timePicker_defaultProps.timeRangePickerDefaultProps);
  var TEXT_CONFIG = timePicker_hooks_useTimePickerTextConfig.useTimePickerTextConfig();
  var allowInput = props.allowInput,
    borderless = props.borderless,
    clearable = props.clearable,
    disabled = props.disabled,
    format = props.format,
    hideDisabledTime = props.hideDisabledTime,
    _props$placeholder = props.placeholder,
    placeholder = _props$placeholder === void 0 ? TEXT_CONFIG.rangePlaceholder : _props$placeholder,
    size = props.size,
    steps = props.steps,
    disableTime = props.disableTime,
    _props$onBlur = props.onBlur,
    onBlur = _props$onBlur === void 0 ? _util_noop["default"] : _props$onBlur,
    _props$onFocus = props.onFocus,
    onFocus = _props$onFocus === void 0 ? _util_noop["default"] : _props$onFocus,
    _props$onInput = props.onInput,
    onInput = _props$onInput === void 0 ? _util_noop["default"] : _props$onInput,
    style = props.style,
    className = props.className,
    presets = props.presets,
    label = props.label;
  var _useControlled = hooks_useControlled["default"](props, "value", props.onChange),
    _useControlled2 = slicedToArray._slicedToArray(_useControlled, 2),
    value = _useControlled2[0],
    onChange = _useControlled2[1];
  var _useConfig = hooks_useConfig["default"](),
    classPrefix = _useConfig.classPrefix;
  var _useGlobalIcon = hooks_useGlobalIcon["default"]({
      TimeIcon: tdesignIconsReact.TimeIcon
    }),
    TimeIcon = _useGlobalIcon.TimeIcon;
  var _useState = React.useState(false),
    _useState2 = slicedToArray._slicedToArray(_useState, 2),
    isPanelShowed = _useState2[0],
    setPanelShow = _useState2[1];
  var _useState3 = React.useState(void 0),
    _useState4 = slicedToArray._slicedToArray(_useState3, 2),
    currentPanelIdx = _useState4[0],
    setCurrentPanelIdx = _useState4[1];
  var _useState5 = React.useState(["", ""]),
    _useState6 = slicedToArray._slicedToArray(_useState5, 2),
    currentValue = _useState6[0],
    setCurrentValue = _useState6[1];
  var name = "".concat(classPrefix, "-time-picker");
  var inputClasses = classNames__default["default"]("".concat(name, "__group"), defineProperty._defineProperty({}, "".concat(classPrefix, "-is-focused"), isPanelShowed));
  var handleShowPopup = function handleShowPopup(visible, _ref) {
    var trigger = _ref.trigger;
    if (trigger === "trigger-element-click") {
      setPanelShow(true);
      return;
    }
    setPanelShow(visible);
  };
  function handlePickerValue(pickValue, currentValue2) {
    var _currentValue2$, _currentValue2$2;
    if (Array.isArray(pickValue)) return pickValue;
    return currentPanelIdx === 0 ? [pickValue, (_currentValue2$ = currentValue2[1]) !== null && _currentValue2$ !== void 0 ? _currentValue2$ : pickValue] : [(_currentValue2$2 = currentValue2[0]) !== null && _currentValue2$2 !== void 0 ? _currentValue2$2 : pickValue, pickValue];
  }
  var handleOnPick = function handleOnPick(pickValue, e) {
    var _props$onPick;
    var context;
    if (isArray.isArray(pickValue)) {
      context = {
        e: e
      };
    } else if (currentPanelIdx.value === 0) {
      context = {
        e: e,
        position: "start"
      };
    } else {
      context = {
        e: e,
        position: "end"
      };
    }
    (_props$onPick = props.onPick) === null || _props$onPick === void 0 || _props$onPick.call(props, pickValue, context);
  };
  var handleClear = function handleClear(context) {
    var e = context.e;
    e.stopPropagation();
    onChange(void 0);
    setCurrentValue(timePicker_panel_SinglePanel.TIME_PICKER_EMPTY);
  };
  var handleClick = function handleClick(_ref2) {
    var position = _ref2.position;
    setCurrentPanelIdx(position === "first" ? 0 : 1);
  };
  var handleTimeChange = function handleTimeChange(newValue, context) {
    var nextCurrentValue = handlePickerValue(newValue, currentValue);
    setCurrentValue(nextCurrentValue);
    handleOnPick(nextCurrentValue, context);
  };
  var autoSwapTime = function autoSwapTime(valueBeforeConfirm) {
    var _valueBeforeConfirm = slicedToArray._slicedToArray(valueBeforeConfirm, 2),
      startTime = _valueBeforeConfirm[0],
      endTime = _valueBeforeConfirm[1];
    var startDayjs = dayjs__default["default"](startTime, props.format);
    var endDayjs = dayjs__default["default"](endTime, props.format);
    if (startDayjs.isAfter(endDayjs, "second")) return [endTime, startTime];
    return [startTime, endTime];
  };
  var handleInputBlur = function handleInputBlur(value2, _ref3) {
    var e = _ref3.e;
    if (allowInput) {
      var isValidTime = timePicker_panel_SinglePanel.validateInputValue(currentValue[currentPanelIdx], format);
      if (isValidTime) {
        var _currentValue$, _currentValue$2;
        var formattedVal = timePicker_panel_SinglePanel.formatInputValue(currentValue[currentPanelIdx], format);
        currentPanelIdx === 0 ? setCurrentValue([formattedVal, (_currentValue$ = currentValue[1]) !== null && _currentValue$ !== void 0 ? _currentValue$ : formattedVal]) : setCurrentValue([(_currentValue$2 = currentValue[0]) !== null && _currentValue$2 !== void 0 ? _currentValue$2 : formattedVal, formattedVal]);
      }
    }
    onBlur({
      value: value2,
      e: e
    });
  };
  var handleInputChange = function handleInputChange(inputVal, _ref4) {
    var e = _ref4.e,
      position = _ref4.position;
    setCurrentValue(inputVal);
    onInput({
      value: value,
      e: e,
      position: handlePositionTrans(position)
    });
  };
  var handleClickConfirm = function handleClickConfirm() {
    var isValidTime = !currentValue.find(function (v) {
      return !timePicker_panel_SinglePanel.validateInputValue(v, format);
    });
    if (isValidTime) onChange(props.autoSwap ? autoSwapTime(currentValue) : currentValue);
    setPanelShow(false);
  };
  var handleFocus = function handleFocus(value2, _ref5) {
    var e = _ref5.e,
      position = _ref5.position;
    onFocus({
      value: value2,
      e: e,
      position: handlePositionTrans(position)
    });
  };
  React.useEffect(function () {
    setCurrentValue(isPanelShowed ? value !== null && value !== void 0 ? value : timePicker_panel_SinglePanel.TIME_PICKER_EMPTY : timePicker_panel_SinglePanel.TIME_PICKER_EMPTY);
    if (!isPanelShowed) setCurrentPanelIdx(void 0);
  }, [isPanelShowed]);
  return /* @__PURE__ */React__default["default"].createElement("div", {
    className: classNames__default["default"](name, className),
    style: style
  }, /* @__PURE__ */React__default["default"].createElement(rangeInput_index.RangeInputPopup, {
    style: style,
    disabled: disabled,
    popupVisible: isPanelShowed,
    onPopupVisibleChange: handleShowPopup,
    popupProps: _objectSpread({
      overlayInnerStyle: {
        width: "auto",
        padding: 0
      }
    }, props.popupProps),
    onInputChange: handleInputChange,
    inputValue: isPanelShowed ? currentValue : value !== null && value !== void 0 ? value : timePicker_panel_SinglePanel.TIME_PICKER_EMPTY,
    rangeInputProps: _objectSpread({
      size: size,
      borderless: borderless,
      clearable: clearable,
      className: inputClasses,
      value: isPanelShowed ? currentValue : value !== null && value !== void 0 ? value : void 0,
      placeholder: placeholder,
      suffixIcon: /* @__PURE__ */React__default["default"].createElement(TimeIcon, null),
      onClear: handleClear,
      onClick: handleClick,
      onFocus: handleFocus,
      onBlur: handleInputBlur,
      readOnly: !allowInput,
      activeIndex: currentPanelIdx,
      label: label
    }, props.rangeInputProps),
    tips: props.tips,
    status: props.status,
    panel: /* @__PURE__ */React__default["default"].createElement(timePicker_panel_TimePickerPanel["default"], {
      steps: steps,
      format: format,
      disableTime: disableTime,
      isShowPanel: isPanelShowed,
      hideDisabledTime: hideDisabledTime,
      isFooterDisplay: true,
      value: currentValue[currentPanelIdx || 0],
      onChange: handleTimeChange,
      handleConfirmClick: handleClickConfirm,
      position: currentPanelIdx === 0 ? "start" : "end",
      activeIndex: currentPanelIdx,
      presets: presets
    })
  }));
};
TimeRangePicker.displayName = "TimeRangePicker";

exports["default"] = TimeRangePicker;
