// tdesign-react/lib/date-picker/hooks/useDisableDate.js
/**
 * tdesign v1.17.1
 * (c) 2026 tdesign
 * @license MIT
 */

import dayjs from 'dayjs';
import { isObject } from 'lodash-es';
import { b as isEnabledDate } from '../../_chunks/dep-1784ea94.js';
import '../../_chunks/dep-d1eeb95c.js';
import '../../_chunks/dep-673e8854.js';
import '../../_chunks/dep-570a5adb.js';
import '../../_chunks/dep-55b77e26.js';
import '../../_chunks/dep-067cb22b.js';
import '../../_chunks/dep-65b05485.js';
import '../../_chunks/dep-f5983b06.js';
import '../../_chunks/dep-6367027a.js';
import '../../_chunks/dep-8a02427e.js';
import '../../_chunks/dep-65b369a5.js';

function useDisableDate(props) {
  var _disableDate = props.disableDate,
    format = props.format,
    mode = props.mode,
    start = props.start,
    end = props.end;
  return {
    disableDate: function disableDate(value) {
      return !isEnabledDate({
        disableDate: _disableDate,
        format: format,
        mode: mode,
        value: value
      });
    },
    minDate: isObject(_disableDate) && "before" in _disableDate ? new Date(dayjs(_disableDate.before).startOf("day").format()) : start,
    maxDate: isObject(_disableDate) && "after" in _disableDate ? new Date(dayjs(_disableDate.after).endOf("day").format()) : end
  };
}

export { useDisableDate as default };
