// tdesign-react/lib/dialog/hooks/useDialogPosition.js
/**
 * tdesign v1.17.1
 * (c) 2026 tdesign
 * @license MIT
 */

import { useCallback } from 'react';
import { c as canUseDocument } from '../../_chunks/dep-68899e89.js';
import 'lodash-es';

var mousePosition = null;
var getClickPosition = function getClickPosition(e) {
  mousePosition = {
    x: e.clientX,
    y: e.clientY
  };
  setTimeout(function () {
    mousePosition = null;
  }, 100);
};
if (canUseDocument) {
  document.addEventListener("click", getClickPosition, true);
}
function useDialogPosition(dialogCardRef) {
  var applyTransform = useCallback(function () {
    var _el$offsetParent;
    var el = dialogCardRef.current;
    if (!el || !mousePosition) return;
    var _mousePosition = mousePosition,
      x = _mousePosition.x,
      y = _mousePosition.y;
    var parentRect = ((_el$offsetParent = el.offsetParent) === null || _el$offsetParent === void 0 ? void 0 : _el$offsetParent.getBoundingClientRect()) || {
      left: 0,
      top: 0
    };
    var top = parentRect.top + el.offsetTop;
    var left = parentRect.left + el.offsetLeft;
    var offsetX = x - left;
    var offsetY = y - top;
    el.style.transformOrigin = "".concat(offsetX, "px ").concat(offsetY, "px");
  }, [dialogCardRef]);
  return {
    applyTransform: applyTransform
  };
}

export { useDialogPosition as default };
