// tdesign-react/lib/anchor/AnchorTarget.js
/**
 * tdesign v1.17.1
 * (c) 2026 tdesign
 * @license MIT
 */

import React from 'react';
import classNames from 'classnames';
import { FileCopyIcon } from 'tdesign-icons-react';
import copyText from '../_util/copyText.js';
import useConfig from '../hooks/useConfig.js';
import useDefaultProps from '../hooks/useDefaultProps.js';
import useGlobalIcon from '../hooks/useGlobalIcon.js';
import { MessagePlugin } from '../message/index.js';
import { Popup } from '../popup/index.js';
import { anchorTargetDefaultProps } from './defaultProps.js';
import '../_chunks/dep-68899e89.js';
import 'lodash-es';
import '../config-provider/ConfigContext.js';
import '../_chunks/dep-55b77e26.js';
import '../_chunks/dep-067cb22b.js';
import '../_chunks/dep-c3e5e988.js';
import '../_chunks/dep-65b05485.js';
import 'dayjs';
import '../message/Message.js';
import '../_chunks/dep-d1eeb95c.js';
import '../_chunks/dep-673e8854.js';
import '../_chunks/dep-570a5adb.js';
import '../_chunks/dep-9c5d9036.js';
import '../_chunks/dep-84926167.js';
import '../_chunks/dep-af05de5f.js';
import '../_util/noop.js';
import '../hooks/usePersistFn.js';
import '../message/MessageClose.js';
import '../message/useMessageClass.js';
import '../message/MessageIcon.js';
import '../loading/index.js';
import '../loading/Loading.js';
import '../common/Portal.js';
import 'react-dom';
import '../hooks/useLayoutEffect.js';
import '../loading/defaultProps.js';
import '../loading/gradient.js';
import '../_chunks/dep-e17ec8df.js';
import '../_chunks/dep-b0cbea79.js';
import '../_chunks/dep-154fcb93.js';
import '../_chunks/dep-e9c0bb63.js';
import '../hooks/useDomRefCallback.js';
import '../loading/plugin.js';
import '../_util/react-render.js';
import '../_chunks/dep-e29eee31.js';
import '../common/PluginContainer.js';
import '../config-provider/ConfigProvider.js';
import '../message/config.js';
import '../message/defaultProps.js';
import '../popup/Popup.js';
import 'react-transition-group';
import '../hooks/useAnimation.js';
import '../hooks/useAttach.js';
import '../hooks/useControlled.js';
import '../hooks/useMutationObserver.js';
import '../hooks/useLatest.js';
import '../hooks/usePopper.js';
import 'react-fast-compare';
import '@popperjs/core';
import '../hooks/useWindowSize.js';
import '../popup/defaultProps.js';
import '../popup/hooks/useTrigger.js';
import '../_util/listener.js';
import '../_util/ref.js';
import 'react-is';
import '../_util/isFragment.js';
import '../popup/utils/transition.js';
import '../popup/PopupPlugin.js';

var AnchorTarget = function AnchorTarget(props) {
  var _useConfig = useConfig(),
    classPrefix = _useConfig.classPrefix;
  var _useGlobalIcon = useGlobalIcon({
      FileCopyIcon: FileCopyIcon
    }),
    FileCopyIcon$1 = _useGlobalIcon.FileCopyIcon;
  var _useDefaultProps = useDefaultProps(props, anchorTargetDefaultProps),
    id = _useDefaultProps.id,
    tag = _useDefaultProps.tag,
    children = _useDefaultProps.children,
    className = _useDefaultProps.className,
    style = _useDefaultProps.style;
  var tagClassName = classNames("".concat(classPrefix, "-anchor__target"), className);
  var iconClassName = "".concat(classPrefix, "-anchor__copy");
  var handleCopyText = function handleCopyText() {
    var a = document.createElement("a");
    a.href = "#".concat(id);
    copyText(a.href);
    MessagePlugin.success("\u94FE\u63A5\u590D\u5236\u6210\u529F", 1e3);
  };
  var Content = function Content() {
    return /* @__PURE__ */React.createElement(React.Fragment, null, children, /* @__PURE__ */React.createElement(Popup, {
      content: "\u590D\u5236\u94FE\u63A5",
      placement: "top",
      showArrow: true
    }, /* @__PURE__ */React.createElement(FileCopyIcon$1, {
      className: iconClassName,
      onClick: handleCopyText
    })));
  };
  return /*#__PURE__*/React.createElement(tag, {
    id: id,
    className: tagClassName,
    style: style
  }, /*#__PURE__*/React.createElement(Content));
};
AnchorTarget.displayName = "AnchorTarget";

export { AnchorTarget as default };
