// tdesign-react/cjs/cascader/Cascader.js
/**
 * tdesign v1.17.1
 * (c) 2026 tdesign
 * @license MIT
 */

'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var defineProperty = require('../_chunks/dep-c446aa11.js');
var slicedToArray = require('../_chunks/dep-c60a3e4c.js');
var React = require('react');
var classNames = require('classnames');
var _util_parseTNode = require('../_util/parseTNode.js');
var common_FakeArrow = require('../common/FakeArrow.js');
var hooks_useCommonClassName = require('../hooks/useCommonClassName.js');
var hooks_useConfig = require('../hooks/useConfig.js');
var hooks_useDefaultProps = require('../hooks/useDefaultProps.js');
var locale_LocalReceiver = require('../locale/LocalReceiver.js');
var selectInput_index = require('../select-input/index.js');
var cascader_components_Panel = require('./components/Panel.js');
var cascader_core_className = require('./core/className.js');
var cascader_core_effect = require('./core/effect.js');
var cascader_core_helper = require('./core/helper.js');
var cascader_defaultProps = require('./defaultProps.js');
var cascader_hooks = require('./hooks.js');
var pick = require('../_chunks/dep-3784bbea.js');
var omit = require('../_chunks/dep-8de6ca9d.js');
require('../_chunks/dep-c290284d.js');
require('../_chunks/dep-98e6819c.js');
require('../_chunks/dep-7f3fa935.js');
require('../_chunks/dep-3c628d18.js');
require('../_chunks/dep-11347ea1.js');
require('../_chunks/dep-3d05de97.js');
require('../_chunks/dep-a830ce1e.js');
require('../_chunks/dep-6e0da644.js');
require('../_chunks/dep-05bc49e6.js');
require('../_chunks/dep-8c880a65.js');
require('../_chunks/dep-e41137f7.js');
require('../_chunks/dep-b8d0b96c.js');
require('../_chunks/dep-4f1729f0.js');
require('../_chunks/dep-427f8bc0.js');
require('../_chunks/dep-a54f202c.js');
require('../config-provider/ConfigContext.js');
require('../_chunks/dep-2ee09518.js');
require('../_chunks/dep-a6fa90dc.js');
require('dayjs');
require('../_chunks/dep-012d9a7c.js');
require('../_chunks/dep-150a112c.js');
require('../_chunks/dep-e360e17f.js');
require('../_chunks/dep-51284537.js');
require('../_chunks/dep-8a493834.js');
require('../_chunks/dep-9f5b2891.js');
require('../_chunks/dep-338f013a.js');
require('../_chunks/dep-c37c5e17.js');
require('../_chunks/dep-eee1aab9.js');
require('../_chunks/dep-770bc0ae.js');
require('../_chunks/dep-aaf27e39.js');
require('../_chunks/dep-b47d855f.js');
require('../_chunks/dep-81bce4e6.js');
require('../_chunks/dep-863f8f4d.js');
require('../_chunks/dep-84e4d0da.js');
require('../_chunks/dep-f78183d7.js');
require('../_chunks/dep-2f45f6c4.js');
require('../config-provider/ConfigProvider.js');
require('../_chunks/dep-ba90997f.js');
require('../_chunks/dep-22026fea.js');
require('../_chunks/dep-551d5940.js');
require('../_chunks/dep-8005b808.js');
require('../_chunks/dep-2add8824.js');
require('../select-input/SelectInput.js');
require('../popup/index.js');
require('../popup/Popup.js');
require('react-transition-group');
require('../_chunks/dep-ee71ddca.js');
require('../common/Portal.js');
require('react-dom');
require('../hooks/useLayoutEffect.js');
require('../hooks/useAnimation.js');
require('../hooks/useAttach.js');
require('../hooks/useControlled.js');
require('../_util/noop.js');
require('../_chunks/dep-45188d54.js');
require('../_chunks/dep-eccb6a94.js');
require('../_chunks/dep-f5e1d11c.js');
require('../_chunks/dep-b199860d.js');
require('../_chunks/dep-36579bf6.js');
require('../hooks/useMutationObserver.js');
require('../hooks/useLatest.js');
require('../_chunks/dep-aa9a2c54.js');
require('../_chunks/dep-77ef3011.js');
require('../_chunks/dep-b36779a4.js');
require('../_chunks/dep-f2809376.js');
require('../_chunks/dep-3be56e1e.js');
require('../hooks/usePopper.js');
require('react-fast-compare');
require('@popperjs/core');
require('../hooks/useWindowSize.js');
require('../popup/defaultProps.js');
require('../popup/hooks/useTrigger.js');
require('../_util/listener.js');
require('../_util/ref.js');
require('react-is');
require('../_util/isFragment.js');
require('../popup/utils/transition.js');
require('../popup/PopupPlugin.js');
require('../_chunks/dep-7bf14076.js');
require('../_util/react-render.js');
require('../common/PluginContainer.js');
require('../select-input/defaultProps.js');
require('../select-input/useMultiple.js');
require('../tag-input/index.js');
require('../tag-input/TagInput.js');
require('tdesign-icons-react');
require('../hooks/useDragSorter.js');
require('../hooks/useEventCallback.js');
require('../hooks/useGlobalIcon.js');
require('../input/index.js');
require('../input/Input.js');
require('../_util/forwardRefWithStatics.js');
require('hoist-non-react-statics');
require('../input/defaultProps.js');
require('../input/InputGroup.js');
require('../input/useLengthLimit.js');
require('../tag-input/defaultProps.js');
require('../tag-input/useHover.js');
require('../tag-input/useTagList.js');
require('../tag/index.js');
require('../tag/CheckTag.js');
require('../_chunks/dep-29222115.js');
require('../tag/defaultProps.js');
require('../tag/Tag.js');
require('../_chunks/dep-183643e1.js');
require('../tag/CheckTagGroup.js');
require('../tag-input/useTagScroll.js');
require('../select-input/useOverlayInnerStyle.js');
require('../hooks/useInnerPopupVisible.js');
require('../select-input/useSingle.js');
require('../loading/index.js');
require('../loading/Loading.js');
require('../loading/defaultProps.js');
require('../loading/gradient.js');
require('../_chunks/dep-062bb1b1.js');
require('../hooks/useDomRefCallback.js');
require('../loading/plugin.js');
require('../_chunks/dep-d4a6a9c1.js');
require('../_chunks/dep-bc160f5b.js');
require('../_chunks/dep-363864ae.js');
require('../_chunks/dep-204188de.js');
require('../_chunks/dep-7a4e0117.js');
require('./components/Item.js');
require('../checkbox/index.js');
require('../_chunks/dep-ab5baf4d.js');
require('../common/Check.js');
require('../_util/helper.js');
require('../_chunks/dep-b422817e.js');
require('../_chunks/dep-fb263e64.js');
require('../checkbox/defaultProps.js');
require('../_chunks/dep-77f065c8.js');
require('../_chunks/dep-bfa39549.js');
require('../_chunks/dep-2be6b7d2.js');
require('../_chunks/dep-e4edf583.js');
require('../_chunks/dep-fb453fc0.js');
require('mitt');
require('../_chunks/dep-ac66dc4e.js');
require('../_chunks/dep-f8c20fd8.js');
require('../_chunks/dep-84536fdc.js');
require('../_chunks/dep-f4f0baa3.js');
require('../_chunks/dep-d3176201.js');
require('../_chunks/dep-5fd08885.js');
require('../_chunks/dep-f98232bb.js');
require('../_chunks/dep-6e5e0470.js');
require('../_chunks/dep-cdef7593.js');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

var React__default = /*#__PURE__*/_interopDefaultLegacy(React);
var classNames__default = /*#__PURE__*/_interopDefaultLegacy(classNames);

function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { defineProperty._defineProperty(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
var Cascader = function Cascader(originalProps) {
  var _props$popupProps;
  var props = hooks_useDefaultProps["default"](originalProps, cascader_defaultProps.cascaderDefaultProps);
  var _useConfig = hooks_useConfig["default"](),
    classPrefix = _useConfig.classPrefix;
  var _useCommonClassName = hooks_useCommonClassName["default"](),
    STATUS = _useCommonClassName.STATUS;
  var _useLocaleReceiver = locale_LocalReceiver.useLocaleReceiver("cascader"),
    _useLocaleReceiver2 = slicedToArray._slicedToArray(_useLocaleReceiver, 1),
    global = _useLocaleReceiver2[0];
  var COMPONENT_NAME = "".concat(classPrefix, "-cascader");
  var _useCascaderContext = cascader_hooks.useCascaderContext(props),
    cascaderContext = _useCascaderContext.cascaderContext,
    isFilterable = _useCascaderContext.isFilterable,
    innerValue = _useCascaderContext.innerValue,
    getCascaderItems = _useCascaderContext.getCascaderItems;
  var displayValue = React.useMemo(function () {
    return props.multiple ? cascader_core_helper.getMultipleContent(cascaderContext) : cascader_core_helper.getSingleContent(cascaderContext);
  }, [props.multiple, cascaderContext]);
  var panels = React.useMemo(function () {
    return cascader_core_helper.getPanels(cascaderContext.treeNodes);
  }, [cascaderContext]);
  var inputPlaceholder = React.useMemo(function () {
    var _props$placeholder;
    return cascaderContext.visible && !props.multiple && cascader_core_helper.getSingleContent(cascaderContext) || ((_props$placeholder = props.placeholder) !== null && _props$placeholder !== void 0 ? _props$placeholder : global.placeholder);
  }, [props.placeholder, cascaderContext, props.multiple, global.placeholder]);
  var renderSuffixIcon = function renderSuffixIcon() {
    if (props.suffixIcon) {
      return props.suffixIcon;
    }
    var visible2 = cascaderContext.visible,
      disabled = cascaderContext.disabled;
    return /* @__PURE__ */React__default["default"].createElement(common_FakeArrow["default"], {
      className: cascader_core_className.getFakeArrowIconClass(classPrefix, STATUS, cascaderContext),
      isActive: visible2,
      disabled: disabled
    });
  };
  var renderLabel = function renderLabel() {
    var label = _util_parseTNode["default"](props.label);
    if (props.multiple) return label;
    if (!label) return null;
    return /* @__PURE__ */React__default["default"].createElement("div", {
      className: "".concat(classPrefix, "-tag-input__prefix")
    }, label);
  };
  var valueDisplayParams = React.useMemo(function () {
    var arrayValue = innerValue instanceof Array ? innerValue : [innerValue];
    var displayValue2 = props.multiple && props.minCollapsedNum ? arrayValue.slice(0, props.minCollapsedNum) : innerValue;
    var options = getCascaderItems(arrayValue, props.valueType, props.multiple);
    return {
      value: innerValue,
      selectedOptions: options,
      onClose: function onClose(index) {
        cascader_core_effect.handleRemoveTagEffect(cascaderContext, index, props.onRemove);
      },
      displayValue: displayValue2
    };
  }, [cascaderContext, innerValue, props.multiple, props.minCollapsedNum, props.onRemove, getCascaderItems, props.valueType]);
  var renderValueDisplay = function renderValueDisplay() {
    return _util_parseTNode.parseContentTNode(props.valueDisplay, valueDisplayParams);
  };
  var setVisible = cascaderContext.setVisible,
    visible = cascaderContext.visible,
    inputVal = cascaderContext.inputVal,
    setInputVal = cascaderContext.setInputVal;
  var updateScrollTop = function updateScrollTop(content) {
    var cascaderMenuList = content === null || content === void 0 ? void 0 : content.querySelectorAll(".".concat(COMPONENT_NAME, "__menu"));
    requestAnimationFrame(function () {
      cascaderMenuList === null || cascaderMenuList === void 0 || cascaderMenuList.forEach(function (menu) {
        var firstSelectedNode = (menu === null || menu === void 0 ? void 0 : menu.querySelector(".".concat(classPrefix, "-is-selected"))) || (menu === null || menu === void 0 ? void 0 : menu.querySelector(".".concat(classPrefix, "-is-expanded")));
        if (!firstSelectedNode || !menu) return;
        var nodeTop = firstSelectedNode.offsetTop;
        var nodeHeight = firstSelectedNode.offsetHeight;
        var menuHeight = menu.clientHeight;
        var currentScrollTop = menu.scrollTop;
        var nodeVisibleTop = nodeTop - currentScrollTop;
        var nodeVisibleBottom = nodeVisibleTop + nodeHeight;
        var isNodeFullyVisible = nodeVisibleTop >= 0 && nodeVisibleBottom <= menuHeight;
        if (isNodeFullyVisible) return;
        var targetScrollTop = currentScrollTop;
        if (nodeVisibleTop < 0) {
          targetScrollTop = nodeTop;
        } else if (nodeVisibleBottom > menuHeight) {
          targetScrollTop = nodeTop - menuHeight + nodeHeight;
        }
        var maxScrollTop = menu.scrollHeight - menuHeight;
        targetScrollTop = Math.max(0, Math.min(targetScrollTop, maxScrollTop));
        menu.scrollTop = targetScrollTop;
      });
    });
  };
  return /* @__PURE__ */React__default["default"].createElement(selectInput_index.SelectInput, _objectSpread(_objectSpread({
    className: classNames__default["default"](COMPONENT_NAME, props.className),
    style: props.style,
    value: displayValue,
    borderless: props.borderless,
    inputValue: visible ? inputVal : "",
    popupVisible: visible,
    allowInput: isFilterable,
    minCollapsedNum: props.minCollapsedNum,
    collapsedItems: props.collapsedItems,
    readOnly: props.readOnly || props.readonly,
    clearable: props.clearable,
    placeholder: inputPlaceholder,
    multiple: props.multiple,
    loading: props.loading,
    disabled: props.disabled,
    status: props.status,
    tips: props.tips,
    label: renderLabel(),
    valueDisplay: renderValueDisplay(),
    suffix: props.suffix,
    suffixIcon: renderSuffixIcon(),
    updateScrollTop: updateScrollTop,
    popupProps: _objectSpread(_objectSpread({}, props.popupProps), {}, {
      overlayInnerStyle: panels.length && !props.loading ? {
        width: "auto"
      } : {},
      overlayClassName: ["".concat(classPrefix, "-cascader__popup"), (_props$popupProps = props.popupProps) === null || _props$popupProps === void 0 ? void 0 : _props$popupProps.overlayClassName]
    }),
    inputProps: _objectSpread({
      size: props.size
    }, props.inputProps),
    tagInputProps: _objectSpread({
      size: props.size
    }, props.tagInputProps),
    tagProps: _objectSpread({}, props.tagProps),
    onInputChange: function onInputChange(value, ctx) {
      var _props$selectInputPro, _props$selectInputPro2;
      if (!visible || (ctx === null || ctx === void 0 ? void 0 : ctx.trigger) === "clear") {
        return;
      }
      setInputVal("".concat(value));
      props === null || props === void 0 || (_props$selectInputPro = props.selectInputProps) === null || _props$selectInputPro === void 0 || (_props$selectInputPro2 = _props$selectInputPro.onInputChange) === null || _props$selectInputPro2 === void 0 || _props$selectInputPro2.call(_props$selectInputPro, value, ctx);
    },
    onTagChange: function onTagChange(val, ctx) {
      var _props$selectInputPro3, _props$selectInputPro4;
      if (ctx.trigger === "enter" || ctx.trigger === "clear") {
        return;
      }
      cascader_core_effect.handleRemoveTagEffect(cascaderContext, ctx.index, props.onRemove);
      props === null || props === void 0 || (_props$selectInputPro3 = props.selectInputProps) === null || _props$selectInputPro3 === void 0 || (_props$selectInputPro4 = _props$selectInputPro3.onTagChange) === null || _props$selectInputPro4 === void 0 || _props$selectInputPro4.call(_props$selectInputPro3, val, ctx);
    },
    onPopupVisibleChange: function onPopupVisibleChange(val, context) {
      var _props$selectInputPro5, _props$selectInputPro6;
      if (props.disabled) {
        return;
      }
      setVisible(val, context);
      props === null || props === void 0 || (_props$selectInputPro5 = props.selectInputProps) === null || _props$selectInputPro5 === void 0 || (_props$selectInputPro6 = _props$selectInputPro5.onPopupVisibleChange) === null || _props$selectInputPro6 === void 0 || _props$selectInputPro6.call(_props$selectInputPro5, val, context);
    },
    onBlur: function onBlur(val, context) {
      var _props$onBlur, _props$selectInputPro7, _props$selectInputPro8;
      (_props$onBlur = props.onBlur) === null || _props$onBlur === void 0 || _props$onBlur.call(props, _objectSpread({
        value: cascaderContext.value,
        e: context.e,
        inputValue: inputVal
      }, props.multiple ? {
        tagInputValue: context.tagInputValue
      } : {}));
      props === null || props === void 0 || (_props$selectInputPro7 = props.selectInputProps) === null || _props$selectInputPro7 === void 0 || (_props$selectInputPro8 = _props$selectInputPro7.onBlur) === null || _props$selectInputPro8 === void 0 || _props$selectInputPro8.call(_props$selectInputPro7, val, context);
    },
    onFocus: function onFocus(val, context) {
      var _props$onFocus, _props$selectInputPro9, _props$selectInputPro0;
      (_props$onFocus = props.onFocus) === null || _props$onFocus === void 0 || _props$onFocus.call(props, {
        value: cascaderContext.value,
        e: context.e
      });
      props === null || props === void 0 || (_props$selectInputPro9 = props.selectInputProps) === null || _props$selectInputPro9 === void 0 || (_props$selectInputPro0 = _props$selectInputPro9.onFocus) === null || _props$selectInputPro0 === void 0 || _props$selectInputPro0.call(_props$selectInputPro9, val, context);
    },
    onClear: function onClear(context) {
      var _props$selectInputPro1, _props$selectInputPro10, _props$onClear;
      cascader_core_effect.closeIconClickEffect(cascaderContext);
      props === null || props === void 0 || (_props$selectInputPro1 = props.selectInputProps) === null || _props$selectInputPro1 === void 0 || (_props$selectInputPro10 = _props$selectInputPro1.onClear) === null || _props$selectInputPro10 === void 0 || _props$selectInputPro10.call(_props$selectInputPro1, context);
      (_props$onClear = props.onClear) === null || _props$onClear === void 0 || _props$onClear.call(props, context);
    }
  }, omit.omit(props.selectInputProps, ["onTagChange", "onInputChange", "onPopupVisibleChange", "onBlur", "onFocus", "onClear"])), {}, {
    panel: /* @__PURE__ */React__default["default"].createElement(React__default["default"].Fragment, null, props.panelTopContent && _util_parseTNode["default"](props.panelTopContent), /* @__PURE__ */React__default["default"].createElement(cascader_components_Panel["default"], _objectSpread({
      cascaderContext: cascaderContext
    }, pick.pick(props, ["trigger", "onChange", "empty", "loading", "loadingText", "option"]))), props.panelBottomContent && _util_parseTNode["default"](props.panelBottomContent))
  }));
};
Cascader.displayName = "Cascader";

exports["default"] = Cascader;
