// tdesign-react/cjs/cascader/hooks.js
/**
 * tdesign v1.17.1
 * (c) 2026 tdesign
 * @license MIT
 */

'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var defineProperty = require('../_chunks/dep-c446aa11.js');
var slicedToArray = require('../_chunks/dep-c60a3e4c.js');
var React = require('react');
var treeNodeModel = require('../_chunks/dep-77f065c8.js');
var treeStore = require('../_chunks/dep-e4edf583.js');
var hooks_useControlled = require('../hooks/useControlled.js');
var hooks_useDefaultProps = require('../hooks/useDefaultProps.js');
var cascader_core_effect = require('./core/effect.js');
var cascader_core_helper = require('./core/helper.js');
var cascader_defaultProps = require('./defaultProps.js');
var isEqual = require('../_chunks/dep-aa9a2c54.js');
var isFunction = require('../_chunks/dep-11347ea1.js');
var isArray = require('../_chunks/dep-a54f202c.js');
require('../_chunks/dep-c290284d.js');
require('../_chunks/dep-98e6819c.js');
require('../_chunks/dep-7f3fa935.js');
require('../_chunks/dep-3c628d18.js');
require('../_chunks/dep-3784bbea.js');
require('../_chunks/dep-d4a6a9c1.js');
require('../_chunks/dep-36579bf6.js');
require('../_chunks/dep-3d05de97.js');
require('../_chunks/dep-4f1729f0.js');
require('../_chunks/dep-c37c5e17.js');
require('../_chunks/dep-eee1aab9.js');
require('../_chunks/dep-a830ce1e.js');
require('../_chunks/dep-770bc0ae.js');
require('../_chunks/dep-b199860d.js');
require('../_chunks/dep-bc160f5b.js');
require('../_chunks/dep-aaf27e39.js');
require('../_chunks/dep-b47d855f.js');
require('../_chunks/dep-338f013a.js');
require('../_chunks/dep-363864ae.js');
require('../_chunks/dep-e360e17f.js');
require('../_chunks/dep-8a493834.js');
require('../_chunks/dep-204188de.js');
require('../_chunks/dep-7a4e0117.js');
require('../_chunks/dep-8005b808.js');
require('../_chunks/dep-f78183d7.js');
require('../_chunks/dep-fb263e64.js');
require('../_chunks/dep-8de6ca9d.js');
require('../_chunks/dep-22026fea.js');
require('../_chunks/dep-150a112c.js');
require('../_chunks/dep-51284537.js');
require('../_chunks/dep-9f5b2891.js');
require('../_chunks/dep-81bce4e6.js');
require('../_chunks/dep-863f8f4d.js');
require('../_chunks/dep-551d5940.js');
require('../_chunks/dep-2add8824.js');
require('../_chunks/dep-6e5e0470.js');
require('../_chunks/dep-cdef7593.js');
require('../_chunks/dep-f5e1d11c.js');
require('../_chunks/dep-bfa39549.js');
require('../_chunks/dep-fb453fc0.js');
require('mitt');
require('../_chunks/dep-ac66dc4e.js');
require('../_chunks/dep-7bf14076.js');
require('../_chunks/dep-05bc49e6.js');
require('../_chunks/dep-8c880a65.js');
require('../_chunks/dep-b8d0b96c.js');
require('../_chunks/dep-f8c20fd8.js');
require('../_chunks/dep-84536fdc.js');
require('../_chunks/dep-f4f0baa3.js');
require('../_chunks/dep-427f8bc0.js');
require('../_chunks/dep-d3176201.js');
require('../_chunks/dep-5fd08885.js');
require('../_chunks/dep-b36779a4.js');
require('../_chunks/dep-f98232bb.js');
require('../_chunks/dep-84e4d0da.js');
require('../_chunks/dep-b422817e.js');
require('../_chunks/dep-45188d54.js');
require('../_chunks/dep-eccb6a94.js');
require('../_util/noop.js');
require('../_chunks/dep-ba90997f.js');
require('../_chunks/dep-2be6b7d2.js');
require('../_chunks/dep-77ef3011.js');

function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { defineProperty._defineProperty(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
var useCascaderContext = function useCascaderContext(originalProps) {
  var props = hooks_useDefaultProps["default"](originalProps, cascader_defaultProps.cascaderDefaultProps);
  var disabled = props.disabled,
    options = props.options,
    _props$keys = props.keys,
    keys = _props$keys === void 0 ? {} : _props$keys,
    checkStrictly = props.checkStrictly,
    lazy = props.lazy,
    multiple = props.multiple,
    reserveKeyword = props.reserveKeyword,
    valueMode = props.valueMode,
    load = props.load;
  var _useControlled = hooks_useControlled["default"](props, "value", props.onChange),
    _useControlled2 = slicedToArray._slicedToArray(_useControlled, 2),
    innerValue = _useControlled2[0],
    setInnerValue = _useControlled2[1];
  var _useControlled3 = hooks_useControlled["default"](props, "popupVisible", props.onPopupVisibleChange),
    _useControlled4 = slicedToArray._slicedToArray(_useControlled3, 2),
    innerPopupVisible = _useControlled4[0],
    setPopupVisible = _useControlled4[1];
  var _useState = React.useState(""),
    _useState2 = slicedToArray._slicedToArray(_useState, 2),
    inputVal = _useState2[0],
    setInputVal = _useState2[1];
  var _useState3 = React.useState(null),
    _useState4 = slicedToArray._slicedToArray(_useState3, 2),
    treeStore$1 = _useState4[0],
    setTreeStore = _useState4[1];
  var _useState5 = React.useState([]),
    _useState6 = slicedToArray._slicedToArray(_useState5, 2),
    treeNodes = _useState6[0],
    setTreeNodes = _useState6[1];
  var _useState7 = React.useState([]),
    _useState8 = slicedToArray._slicedToArray(_useState7, 2),
    expend = _useState8[0],
    setExpend = _useState8[1];
  var _useState9 = React.useState(void 0),
    _useState0 = slicedToArray._slicedToArray(_useState9, 2),
    scopeVal = _useState0[0],
    setScopeVal = _useState0[1];
  var isParentFilterable = React.useMemo(function () {
    return !!((["parentFirst", "all"].includes(props.valueMode) || props.checkStrictly) && inputVal);
  }, [props.valueMode, props.checkStrictly, inputVal]);
  var cascaderContext = React.useMemo(function () {
    var size = props.size,
      checkStrictly2 = props.checkStrictly,
      lazy2 = props.lazy,
      multiple2 = props.multiple,
      filterable = props.filterable,
      clearable = props.clearable,
      checkProps = props.checkProps,
      max = props.max,
      disabled2 = props.disabled,
      showAllLevels = props.showAllLevels,
      minCollapsedNum = props.minCollapsedNum,
      valueType = props.valueType;
    return {
      value: scopeVal,
      size: size,
      checkStrictly: checkStrictly2,
      lazy: lazy2,
      multiple: multiple2,
      filterable: filterable,
      clearable: clearable,
      checkProps: checkProps,
      max: max,
      disabled: disabled2,
      showAllLevels: showAllLevels,
      minCollapsedNum: minCollapsedNum,
      valueType: valueType,
      treeStore: treeStore$1,
      setValue: function setValue(val, source, node) {
        if (isEqual.isEqual(val, scopeVal)) return;
        setInnerValue(val, {
          source: source,
          node: node
        });
      },
      visible: innerPopupVisible,
      setVisible: setPopupVisible,
      treeNodes: treeNodes,
      setTreeNodes: setTreeNodes,
      inputVal: inputVal,
      setInputVal: setInputVal,
      setExpend: setExpend,
      isParentFilterable: isParentFilterable
    };
  }, [props, scopeVal, innerPopupVisible, treeStore$1, treeNodes, inputVal, setInnerValue, setPopupVisible, isParentFilterable]);
  var isFilterable = React.useMemo(function () {
    return Boolean(props.filterable || isFunction.isFunction(props.filter));
  }, [props.filterable, props.filter]);
  var optionCurrent = React.useRef([]);
  React.useEffect(function () {
    if (!isEqual.isEqual(optionCurrent.current, options)) {
      optionCurrent.current = options;
      handleTreeStore();
    }
  }, [options]);
  var handleTreeStore = function handleTreeStore() {
    if (!treeStore$1) {
      if (!options.length) return;
      var store = new treeStore.TreeStore({
        keys: _objectSpread(_objectSpread({}, keys), {}, {
          children: typeof keys.children === "string" ? keys.children : "children"
        }),
        allowDuplicateValue: props.valueType === "full",
        onLoad: function onLoad() {
          setTimeout(function () {
            store.refreshNodes();
            cascader_core_effect.treeNodesEffect(inputVal, store, setTreeNodes, props.filter, isParentFilterable);
          });
        }
      });
      store.append(options);
      setTreeStore(store);
    } else {
      treeStore$1.reload(options);
      treeStore$1.refreshNodes();
      cascader_core_effect.treeStoreExpendEffect(treeStore$1, scopeVal, [], props.valueType);
      cascader_core_effect.treeNodesEffect(inputVal, treeStore$1, setTreeNodes, props.filter, isParentFilterable);
    }
  };
  React.useEffect(function () {
    if (!treeStore$1) return;
    var treeProps = {
      keys: _objectSpread(_objectSpread({}, keys), {}, {
        children: typeof keys.children === "string" ? keys.children : "children"
      }),
      checkable: true,
      expandMutex: true,
      expandParent: true,
      checkStrictly: checkStrictly,
      disabled: disabled,
      load: load,
      lazy: lazy,
      valueMode: valueMode
    };
    treeStore$1.setConfig(treeProps);
  }, [checkStrictly, disabled, keys, lazy, load, valueMode, treeStore$1]);
  React.useEffect(function () {
    var setValue = cascaderContext.setValue,
      multiple2 = cascaderContext.multiple;
    if (cascader_core_helper.isValueInvalid(innerValue, cascaderContext)) {
      setValue(multiple2 ? [] : "", "invalid-value");
    }
    if (!cascader_core_helper.isEmptyValues(innerValue)) {
      setScopeVal(innerValue);
    } else {
      setScopeVal(multiple2 ? [] : "");
    }
    if (multiple2 && !reserveKeyword) {
      setInputVal("");
    }
  }, [innerValue, multiple, reserveKeyword]);
  React.useEffect(function () {
    if (!treeStore$1) return;
    cascader_core_effect.treeStoreExpendEffect(treeStore$1, scopeVal, expend, props.valueType);
  }, [treeStore$1, scopeVal, expend, props.valueType]);
  React.useEffect(function () {
    if (!treeStore$1) return;
    cascader_core_effect.treeNodesEffect(inputVal, treeStore$1, setTreeNodes, props.filter, isParentFilterable);
  }, [inputVal, treeStore$1, props.filter, isParentFilterable]);
  React.useEffect(function () {
    if (!treeStore$1) return;
    var valueType = props.valueType;
    var getCheckedKeys = function getCheckedKeys() {
      if (valueType !== "full") {
        return cascader_core_helper.getTreeValue(scopeVal);
      }
      var isValidPath = function isValidPath(path) {
        return Array.isArray(path) && path.length > 0;
      };
      if (multiple && Array.isArray(scopeVal)) {
        return scopeVal.filter(isValidPath).map(treeNodeModel.pathToKey);
      }
      return isValidPath(scopeVal) ? [treeNodeModel.pathToKey(scopeVal)] : [];
    };
    treeStore$1.replaceChecked(getCheckedKeys());
  }, [options, scopeVal, treeStore$1, multiple, props]);
  React.useEffect(function () {
    if (!innerPopupVisible && isFilterable) {
      setInputVal("");
    }
  }, [innerPopupVisible, isFilterable]);
  React.useEffect(function () {
    var inputVal2 = cascaderContext.inputVal,
      treeStore2 = cascaderContext.treeStore,
      setTreeNodes2 = cascaderContext.setTreeNodes;
    cascader_core_effect.treeNodesEffect(inputVal2, treeStore2, setTreeNodes2, props.filter, isParentFilterable);
  }, [inputVal, scopeVal, isParentFilterable]);
  var getCascaderItems = function getCascaderItems(arrValue, valueType, multiple2) {
    var treeStore2 = cascaderContext.treeStore;
    if (!treeStore2) return [];
    var optionsData = [];
    var pushNodeData = function pushNodeData(node) {
      if (node !== null && node !== void 0 && node.data) optionsData.push(node.data);
    };
    if (valueType === "full") {
      var values = multiple2 ? arrValue : [arrValue];
      values.forEach(function (v) {
        if (isArray.isArray(v) && v.length) {
          pushNodeData(treeStore2.getNode(v));
        }
      });
      return optionsData;
    }
    arrValue.forEach(function (v) {
      var _treeStore2$getNodes;
      var node = (_treeStore2$getNodes = treeStore2.getNodes(v)) === null || _treeStore2$getNodes === void 0 ? void 0 : _treeStore2$getNodes[0];
      pushNodeData(node);
    });
    return optionsData;
  };
  return {
    cascaderContext: cascaderContext,
    isFilterable: isFilterable,
    innerValue: innerValue,
    getCascaderItems: getCascaderItems
  };
};

exports.useCascaderContext = useCascaderContext;
