// tdesign-react/cjs/input-adornment/InputAdornment.js
/**
 * tdesign v1.17.1
 * (c) 2026 tdesign
 * @license MIT
 */

'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var defineProperty = require('../_chunks/dep-c446aa11.js');
var objectWithoutProperties = require('../_chunks/dep-e41137f7.js');
var React = require('react');
var classNames = require('classnames');
var hooks_useConfig = require('../hooks/useConfig.js');
var isString = require('../_chunks/dep-427f8bc0.js');
var isNumber = require('../_chunks/dep-b8d0b96c.js');
require('../_chunks/dep-c290284d.js');
require('../config-provider/ConfigContext.js');
require('../_chunks/dep-2ee09518.js');
require('../_chunks/dep-a6fa90dc.js');
require('dayjs');
require('../_chunks/dep-012d9a7c.js');
require('../_chunks/dep-150a112c.js');
require('../_chunks/dep-e360e17f.js');
require('../_chunks/dep-3d05de97.js');
require('../_chunks/dep-4f1729f0.js');
require('../_chunks/dep-a54f202c.js');
require('../_chunks/dep-51284537.js');
require('../_chunks/dep-8a493834.js');
require('../_chunks/dep-9f5b2891.js');
require('../_chunks/dep-338f013a.js');
require('../_chunks/dep-c37c5e17.js');
require('../_chunks/dep-eee1aab9.js');
require('../_chunks/dep-11347ea1.js');
require('../_chunks/dep-a830ce1e.js');
require('../_chunks/dep-770bc0ae.js');
require('../_chunks/dep-aaf27e39.js');
require('../_chunks/dep-b47d855f.js');
require('../_chunks/dep-81bce4e6.js');
require('../_chunks/dep-863f8f4d.js');
require('../_chunks/dep-84e4d0da.js');
require('../_chunks/dep-f78183d7.js');
require('../_chunks/dep-2f45f6c4.js');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

var React__default = /*#__PURE__*/_interopDefaultLegacy(React);
var classNames__default = /*#__PURE__*/_interopDefaultLegacy(classNames);

var _excluded = ["prepend", "append", "children", "className", "style"];
function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { defineProperty._defineProperty(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
var InputAdornment = /*#__PURE__*/React.forwardRef(function (props, ref) {
  var _useConfig = hooks_useConfig["default"](),
    classPrefix = _useConfig.classPrefix;
  var prepend = props.prepend,
    append = props.append,
    children = props.children,
    className = props.className,
    style = props.style,
    wrapperProps = objectWithoutProperties._objectWithoutProperties(props, _excluded);
  var renderAddon = function renderAddon(type, classPrefix2, Content) {
    if (typeof Content === "undefined") return null;
    var addonNode;
    var isContentNode = isString.isString(Content) || isNumber.isNumber(Content);
    if (typeof Content === "function") {
      addonNode = /* @__PURE__ */React__default["default"].createElement(Content, null);
    } else {
      addonNode = isContentNode ? /* @__PURE__ */React__default["default"].createElement("span", {
        className: "".concat(classPrefix2, "-input-adornment__text")
      }, Content) : Content;
    }
    return addonNode ? /* @__PURE__ */React__default["default"].createElement("span", {
      className: "".concat(classPrefix2, "-input-adornment__").concat(type)
    }, addonNode) : addonNode;
  };
  var renderChildren = function renderChildren() {
    return React__default["default"].Children.map(children, function (child) {
      if (!child) return null;
      if (/*#__PURE__*/React__default["default"].isValidElement(child)) {
        var childProps = child.props;
        return /*#__PURE__*/React__default["default"].cloneElement(child, _objectSpread(_objectSpread(_objectSpread({}, wrapperProps), childProps), {}, {
          onChange: function onChange() {
            var _wrapperProps$onChang, _wrapperProps$onChang2, _childProps$onChange, _childProps$onChange$;
            for (var _len = arguments.length, args = new Array(_len), _key = 0; _key < _len; _key++) {
              args[_key] = arguments[_key];
            }
            wrapperProps === null || wrapperProps === void 0 || (_wrapperProps$onChang = wrapperProps.onChange) === null || _wrapperProps$onChang === void 0 || (_wrapperProps$onChang2 = _wrapperProps$onChang.call) === null || _wrapperProps$onChang2 === void 0 || _wrapperProps$onChang2.call.apply(_wrapperProps$onChang2, [_wrapperProps$onChang, null].concat(args));
            childProps === null || childProps === void 0 || (_childProps$onChange = childProps.onChange) === null || _childProps$onChange === void 0 || (_childProps$onChange$ = _childProps$onChange.call) === null || _childProps$onChange$ === void 0 || _childProps$onChange$.call.apply(_childProps$onChange$, [_childProps$onChange, null].concat(args));
          }
        }));
      }
      return child;
    });
  };
  return /* @__PURE__ */React__default["default"].createElement("div", {
    ref: ref,
    style: style,
    className: classNames__default["default"]("".concat(classPrefix, "-input-adornment"), className)
  }, renderAddon("prepend", classPrefix, prepend), renderChildren(), renderAddon("append", classPrefix, append));
});
InputAdornment.displayName = "InputAdornment";

exports["default"] = InputAdornment;
