// tdesign-react/cjs/typography/Text.js
/**
 * tdesign v1.17.1
 * (c) 2026 tdesign
 * @license MIT
 */

'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var defineProperty = require('../_chunks/dep-c446aa11.js');
var objectWithoutProperties = require('../_chunks/dep-e41137f7.js');
var slicedToArray = require('../_chunks/dep-c60a3e4c.js');
var React = require('react');
var classNames = require('classnames');
var tdesignIconsReact = require('tdesign-icons-react');
var _util_copyText = require('../_util/copyText.js');
var button_Button = require('../button/Button.js');
var hooks_useConfig = require('../hooks/useConfig.js');
var hooks_useDefaultProps = require('../hooks/useDefaultProps.js');
var locale_LocalReceiver = require('../locale/LocalReceiver.js');
var tooltip_index = require('../tooltip/index.js');
var typography_defaultProps = require('./defaultProps.js');
var typography_ellipsis_Ellipsis = require('./ellipsis/Ellipsis.js');
var typography_ellipsis_useEllipsis = require('./ellipsis/useEllipsis.js');
require('../_chunks/dep-c290284d.js');
require('../_chunks/dep-98e6819c.js');
require('../_chunks/dep-7f3fa935.js');
require('../_chunks/dep-ee71ddca.js');
require('../_chunks/dep-427f8bc0.js');
require('../_chunks/dep-3d05de97.js');
require('../_chunks/dep-a54f202c.js');
require('../_chunks/dep-4f1729f0.js');
require('../_util/parseTNode.js');
require('../_chunks/dep-3c628d18.js');
require('../_chunks/dep-11347ea1.js');
require('../_chunks/dep-a830ce1e.js');
require('../hooks/useDomRefCallback.js');
require('../hooks/useRipple.js');
require('../_chunks/dep-062bb1b1.js');
require('../hooks/useAnimation.js');
require('../config-provider/ConfigContext.js');
require('../_chunks/dep-2ee09518.js');
require('../_chunks/dep-a6fa90dc.js');
require('dayjs');
require('../_chunks/dep-012d9a7c.js');
require('../_chunks/dep-150a112c.js');
require('../_chunks/dep-e360e17f.js');
require('../_chunks/dep-51284537.js');
require('../_chunks/dep-8a493834.js');
require('../_chunks/dep-9f5b2891.js');
require('../_chunks/dep-338f013a.js');
require('../_chunks/dep-c37c5e17.js');
require('../_chunks/dep-eee1aab9.js');
require('../_chunks/dep-770bc0ae.js');
require('../_chunks/dep-aaf27e39.js');
require('../_chunks/dep-b47d855f.js');
require('../_chunks/dep-81bce4e6.js');
require('../_chunks/dep-863f8f4d.js');
require('../_chunks/dep-84e4d0da.js');
require('../_chunks/dep-f78183d7.js');
require('../_chunks/dep-2f45f6c4.js');
require('../loading/index.js');
require('../loading/Loading.js');
require('../common/Portal.js');
require('react-dom');
require('../hooks/useLayoutEffect.js');
require('../loading/defaultProps.js');
require('../loading/gradient.js');
require('../_chunks/dep-6e0da644.js');
require('../_chunks/dep-05bc49e6.js');
require('../_chunks/dep-8c880a65.js');
require('../_chunks/dep-b8d0b96c.js');
require('../loading/plugin.js');
require('../_util/react-render.js');
require('../_chunks/dep-7bf14076.js');
require('../common/PluginContainer.js');
require('../config-provider/ConfigProvider.js');
require('../_chunks/dep-ba90997f.js');
require('../_chunks/dep-22026fea.js');
require('../_chunks/dep-551d5940.js');
require('../_chunks/dep-8005b808.js');
require('../_chunks/dep-2add8824.js');
require('../button/defaultProps.js');
require('../tooltip/Tooltip.js');
require('../popup/index.js');
require('../popup/Popup.js');
require('react-transition-group');
require('../hooks/useAttach.js');
require('../hooks/useControlled.js');
require('../_util/noop.js');
require('../_chunks/dep-45188d54.js');
require('../_chunks/dep-eccb6a94.js');
require('../_chunks/dep-f5e1d11c.js');
require('../_chunks/dep-b199860d.js');
require('../_chunks/dep-36579bf6.js');
require('../hooks/useMutationObserver.js');
require('../hooks/useLatest.js');
require('../_chunks/dep-aa9a2c54.js');
require('../_chunks/dep-77ef3011.js');
require('../_chunks/dep-b36779a4.js');
require('../_chunks/dep-f2809376.js');
require('../_chunks/dep-3be56e1e.js');
require('../hooks/usePopper.js');
require('react-fast-compare');
require('@popperjs/core');
require('../hooks/useWindowSize.js');
require('../popup/defaultProps.js');
require('../popup/hooks/useTrigger.js');
require('../_util/listener.js');
require('../_util/ref.js');
require('react-is');
require('../_util/isFragment.js');
require('../popup/utils/transition.js');
require('../popup/PopupPlugin.js');
require('../tooltip/defaultProps.js');
require('../tooltip/TooltipLite.js');
require('../hooks/useSwitch.js');
require('../hooks/usePersistFn.js');
require('../_chunks/dep-d07eb641.js');
require('./ellipsis/Truncate.js');
require('../_chunks/dep-fb453fc0.js');
require('../_chunks/dep-8de6ca9d.js');
require('../_chunks/dep-6e5e0470.js');
require('../_chunks/dep-d4a6a9c1.js');
require('../_chunks/dep-cdef7593.js');
require('../_chunks/dep-204188de.js');
require('../_chunks/dep-7a4e0117.js');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

var React__default = /*#__PURE__*/_interopDefaultLegacy(React);
var classNames__default = /*#__PURE__*/_interopDefaultLegacy(classNames);

var _excluded = ["theme", "disabled", "className", "copyable", "strong", "mark", "code", "keyboard", "underline", "delete", "italic", "children", "ellipsis"];
function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { defineProperty._defineProperty(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
var Text = /*#__PURE__*/React.forwardRef(function (originalProps, ref) {
  var _copyable$tooltipProp;
  var _useConfig = hooks_useConfig["default"](),
    classPrefix = _useConfig.classPrefix;
  var props = hooks_useDefaultProps["default"](originalProps, typography_defaultProps.textDefaultProps);
  var prefixCls = "".concat(classPrefix, "-typography");
  var _useLocaleReceiver = locale_LocalReceiver.useLocaleReceiver("typography"),
    _useLocaleReceiver2 = slicedToArray._slicedToArray(_useLocaleReceiver, 2),
    local = _useLocaleReceiver2[0],
    t = _useLocaleReceiver2[1];
  var copiedText = t(local.copiedText);
  var theme = props.theme,
    disabled = props.disabled,
    className = props.className,
    copyable = props.copyable,
    strong = props.strong,
    mark = props.mark,
    code = props.code,
    keyboard = props.keyboard,
    underline = props.underline,
    deleteProp = props["delete"],
    italic = props.italic,
    children = props.children,
    ellipsis = props.ellipsis,
    rest = objectWithoutProperties._objectWithoutProperties(props, _excluded);
  var getComponent = function getComponent() {
    var _Object$entries$find;
    var componentMap = {
      strong: !!strong,
      mark: !!mark,
      code: !!code,
      kbd: !!keyboard,
      u: !!underline,
      del: !!deleteProp,
      i: !!italic
    };
    return (_Object$entries$find = Object.entries(componentMap).find(function (_ref) {
      var _ref2 = slicedToArray._slicedToArray(_ref, 2),
        condition = _ref2[1];
      return !!condition;
    })) === null || _Object$entries$find === void 0 ? void 0 : _Object$entries$find[0];
  };
  var currentRef = React.useRef(null);
  var _useEllipsis = typography_ellipsis_useEllipsis["default"](ellipsis),
    ellipsisProps = _useEllipsis.ellipsisProps;
  var Component = getComponent();
  var textEllipsisProps = _objectSpread({}, ellipsisProps);
  var _useState = React.useState(false),
    _useState2 = slicedToArray._slicedToArray(_useState, 2),
    isCopied = _useState2[0],
    setIsCopied = _useState2[1];
  var copyProps = typeof copyable === "boolean" ? {
    text: children === null || children === void 0 ? void 0 : children.toString(),
    onCopy: Function.prototype,
    tooltipProps: isCopied ? {
      content: copiedText
    } : null
  } : {
    text: (copyable === null || copyable === void 0 ? void 0 : copyable.text) || (children === null || children === void 0 ? void 0 : children.toString()),
    onCopy: copyable === null || copyable === void 0 ? void 0 : copyable.onCopy,
    tooltipProps: _objectSpread(_objectSpread({}, copyable === null || copyable === void 0 ? void 0 : copyable.tooltipProps), {}, {
      content: isCopied ? copiedText : copyable === null || copyable === void 0 || (_copyable$tooltipProp = copyable.tooltipProps) === null || _copyable$tooltipProp === void 0 ? void 0 : _copyable$tooltipProp.content
    }),
    suffix: copyable === null || copyable === void 0 ? void 0 : copyable.suffix
  };
  var handleCopy = function handleCopy() {
    _util_copyText["default"](copyProps === null || copyProps === void 0 ? void 0 : copyProps.text);
    setIsCopied(true);
    setTimeout(function () {
      setIsCopied(false);
    }, 1500);
    if (typeof copyProps.onCopy === "function") copyProps.onCopy();
  };
  var renderContent = function renderContent(withChildren) {
    var tooltipProps = copyProps.tooltipProps;
    var wrapWithTooltip = function wrapWithTooltip(wrapContent) {
      return tooltipProps ? /* @__PURE__ */React__default["default"].createElement(tooltip_index.Tooltip, _objectSpread({
        hideEmptyPopup: true
      }, tooltipProps), wrapContent) : wrapContent;
    };
    var getSuffix = function getSuffix() {
      if (typeof (copyProps === null || copyProps === void 0 ? void 0 : copyProps.suffix) === "function") {
        return copyProps.suffix({
          copied: isCopied
        });
      }
      return isCopied ? /* @__PURE__ */React__default["default"].createElement(tdesignIconsReact.CheckIcon, null) : /* @__PURE__ */React__default["default"].createElement(tdesignIconsReact.CopyIcon, null);
    };
    return /* @__PURE__ */React__default["default"].createElement(React__default["default"].Fragment, null, withChildren ? children : null, copyable ? wrapWithTooltip(/* @__PURE__ */React__default["default"].createElement(button_Button["default"], {
      shape: "square",
      theme: "primary",
      variant: "text",
      icon: getSuffix(),
      onClick: handleCopy
    })) : null);
  };
  if (!ellipsis) {
    return Component ? /* @__PURE__ */React__default["default"].createElement("span", _objectSpread({
      className: classNames__default["default"](className, prefixCls),
      ref: ref || currentRef
    }, rest), /* @__PURE__ */React__default["default"].createElement(Component, null, renderContent(true))) : /* @__PURE__ */React__default["default"].createElement("span", _objectSpread({
      className: classNames__default["default"](className, defineProperty._defineProperty(defineProperty._defineProperty({}, "".concat(prefixCls, "--").concat(theme), theme), "".concat(prefixCls, "--disabled"), disabled))
    }, rest), renderContent(true));
  }
  return /* @__PURE__ */React__default["default"].createElement(React__default["default"].Fragment, null, Component ? /* @__PURE__ */React__default["default"].createElement("span", _objectSpread({
    className: classNames__default["default"](className, prefixCls),
    ref: ref || currentRef
  }, rest), /* @__PURE__ */React__default["default"].createElement(typography_ellipsis_Ellipsis["default"], _objectSpread(_objectSpread({}, textEllipsisProps), {}, {
    component: Component
  }), children)) : /* @__PURE__ */React__default["default"].createElement(typography_ellipsis_Ellipsis["default"], _objectSpread(_objectSpread({}, textEllipsisProps), {}, {
    className: classNames__default["default"](className, defineProperty._defineProperty(defineProperty._defineProperty({}, "".concat(prefixCls, "--").concat(theme), theme), "".concat(prefixCls, "--disabled"), disabled))
  }, rest), children), renderContent(false));
});

exports["default"] = Text;
