// tdesign-react/cjs/tree-select/TreeSelect.js
/**
 * tdesign v1.17.1
 * (c) 2026 tdesign
 * @license MIT
 */

'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var defineProperty = require('../_chunks/dep-c446aa11.js');
var slicedToArray = require('../_chunks/dep-c60a3e4c.js');
var React = require('react');
var classNames = require('classnames');
var _util_noop = require('../_util/noop.js');
var _util_parseTNode = require('../_util/parseTNode.js');
var hooks_useConfig = require('../hooks/useConfig.js');
var hooks_useControlled = require('../hooks/useControlled.js');
var hooks_useDefaultProps = require('../hooks/useDefaultProps.js');
var hooks_usePersistFn = require('../hooks/usePersistFn.js');
var hooks_useSwitch = require('../hooks/useSwitch.js');
var selectInput_SelectInput = require('../select-input/SelectInput.js');
var tree_index = require('../tree/index.js');
var treeSelect_defaultProps = require('./defaultProps.js');
var treeSelect_hooks_useTreeSelectLocale = require('./hooks/useTreeSelectLocale.js');
var treeSelect_hooks_useTreeSelectPassthroughProps = require('./hooks/useTreeSelectPassthroughProps.js');
var treeSelect_hooks_useTreeSelectUtils = require('./hooks/useTreeSelectUtils.js');
var treeSelect_SelectArrow = require('./SelectArrow.js');
var isFunction = require('../_chunks/dep-11347ea1.js');
require('../_chunks/dep-c290284d.js');
require('../_chunks/dep-98e6819c.js');
require('../_chunks/dep-7f3fa935.js');
require('../_chunks/dep-3c628d18.js');
require('../config-provider/ConfigContext.js');
require('../_chunks/dep-2ee09518.js');
require('../_chunks/dep-a6fa90dc.js');
require('dayjs');
require('../_chunks/dep-012d9a7c.js');
require('../_chunks/dep-150a112c.js');
require('../_chunks/dep-e360e17f.js');
require('../_chunks/dep-3d05de97.js');
require('../_chunks/dep-4f1729f0.js');
require('../_chunks/dep-a54f202c.js');
require('../_chunks/dep-51284537.js');
require('../_chunks/dep-8a493834.js');
require('../_chunks/dep-9f5b2891.js');
require('../_chunks/dep-338f013a.js');
require('../_chunks/dep-c37c5e17.js');
require('../_chunks/dep-eee1aab9.js');
require('../_chunks/dep-a830ce1e.js');
require('../_chunks/dep-770bc0ae.js');
require('../_chunks/dep-aaf27e39.js');
require('../_chunks/dep-b47d855f.js');
require('../_chunks/dep-81bce4e6.js');
require('../_chunks/dep-863f8f4d.js');
require('../_chunks/dep-84e4d0da.js');
require('../_chunks/dep-f78183d7.js');
require('../_chunks/dep-2f45f6c4.js');
require('../_chunks/dep-45188d54.js');
require('../_chunks/dep-eccb6a94.js');
require('../_chunks/dep-f5e1d11c.js');
require('../_chunks/dep-b199860d.js');
require('../_chunks/dep-36579bf6.js');
require('../popup/index.js');
require('../popup/Popup.js');
require('react-transition-group');
require('../_chunks/dep-ee71ddca.js');
require('../_chunks/dep-427f8bc0.js');
require('../common/Portal.js');
require('react-dom');
require('../hooks/useLayoutEffect.js');
require('../hooks/useAnimation.js');
require('../hooks/useAttach.js');
require('../hooks/useMutationObserver.js');
require('../hooks/useLatest.js');
require('../_chunks/dep-aa9a2c54.js');
require('../_chunks/dep-77ef3011.js');
require('../_chunks/dep-b36779a4.js');
require('../_chunks/dep-551d5940.js');
require('../_chunks/dep-8005b808.js');
require('../_chunks/dep-2add8824.js');
require('../_chunks/dep-f2809376.js');
require('../_chunks/dep-3be56e1e.js');
require('../hooks/usePopper.js');
require('../_chunks/dep-05bc49e6.js');
require('../_chunks/dep-8c880a65.js');
require('react-fast-compare');
require('@popperjs/core');
require('../hooks/useWindowSize.js');
require('../popup/defaultProps.js');
require('../popup/hooks/useTrigger.js');
require('../_util/listener.js');
require('../_util/ref.js');
require('react-is');
require('../_util/isFragment.js');
require('../popup/utils/transition.js');
require('../popup/PopupPlugin.js');
require('../_chunks/dep-7bf14076.js');
require('../_util/react-render.js');
require('../common/PluginContainer.js');
require('../config-provider/ConfigProvider.js');
require('../_chunks/dep-ba90997f.js');
require('../_chunks/dep-22026fea.js');
require('../select-input/defaultProps.js');
require('../select-input/useMultiple.js');
require('../tag-input/index.js');
require('../tag-input/TagInput.js');
require('tdesign-icons-react');
require('../hooks/useDragSorter.js');
require('../hooks/useEventCallback.js');
require('../hooks/useGlobalIcon.js');
require('../input/index.js');
require('../input/Input.js');
require('../_chunks/dep-e41137f7.js');
require('../_util/forwardRefWithStatics.js');
require('hoist-non-react-statics');
require('../locale/LocalReceiver.js');
require('../input/defaultProps.js');
require('../input/InputGroup.js');
require('../input/useLengthLimit.js');
require('../_chunks/dep-6e0da644.js');
require('../_chunks/dep-b8d0b96c.js');
require('../tag-input/defaultProps.js');
require('../tag-input/useHover.js');
require('../tag-input/useTagList.js');
require('../tag/index.js');
require('../tag/CheckTag.js');
require('../_chunks/dep-29222115.js');
require('../tag/defaultProps.js');
require('../tag/Tag.js');
require('../_chunks/dep-183643e1.js');
require('../tag/CheckTagGroup.js');
require('../tag-input/useTagScroll.js');
require('../select-input/useOverlayInnerStyle.js');
require('../hooks/useInnerPopupVisible.js');
require('../select-input/useSingle.js');
require('../loading/index.js');
require('../loading/Loading.js');
require('../loading/defaultProps.js');
require('../loading/gradient.js');
require('../_chunks/dep-062bb1b1.js');
require('../hooks/useDomRefCallback.js');
require('../loading/plugin.js');
require('../_chunks/dep-3784bbea.js');
require('../_chunks/dep-d4a6a9c1.js');
require('../_chunks/dep-bc160f5b.js');
require('../_chunks/dep-363864ae.js');
require('../_chunks/dep-204188de.js');
require('../_chunks/dep-7a4e0117.js');
require('../tree/Tree.js');
require('../tree/defaultProps.js');
require('../tree/hooks/TreeDraggableContext.js');
require('../_util/createHookContext.js');
require('../tree/hooks/useControllable.js');
require('../tree/hooks/useStore.js');
require('../_chunks/dep-e4edf583.js');
require('../_chunks/dep-fb453fc0.js');
require('mitt');
require('../_chunks/dep-ac66dc4e.js');
require('../_chunks/dep-77f065c8.js');
require('../_chunks/dep-fb263e64.js');
require('../_chunks/dep-8de6ca9d.js');
require('../_chunks/dep-6e5e0470.js');
require('../_chunks/dep-cdef7593.js');
require('../_chunks/dep-bfa39549.js');
require('../_chunks/dep-f8c20fd8.js');
require('../_chunks/dep-84536fdc.js');
require('../_chunks/dep-f4f0baa3.js');
require('../_chunks/dep-d3176201.js');
require('../_chunks/dep-5fd08885.js');
require('../_chunks/dep-f98232bb.js');
require('../_chunks/dep-b422817e.js');
require('../hooks/usePrevious.js');
require('../hooks/useUpdateLayoutEffect.js');
require('../hooks/useIsFirstRender.js');
require('../tree/hooks/useTreeConfig.js');
require('../tree/hooks/useTreeVirtualScroll.js');
require('../hooks/useVirtualScroll.js');
require('../tree/TreeItem.js');
require('../checkbox/index.js');
require('../_chunks/dep-ab5baf4d.js');
require('../common/Check.js');
require('../_util/helper.js');
require('../checkbox/defaultProps.js');
require('../hooks/useRipple.js');
require('../tree/hooks/useDraggable.js');
require('../_chunks/dep-d07eb641.js');
require('../common/FakeArrow.js');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

var React__default = /*#__PURE__*/_interopDefaultLegacy(React);
var classNames__default = /*#__PURE__*/_interopDefaultLegacy(classNames);

function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { defineProperty._defineProperty(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
var useMergeFn = function useMergeFn() {
  for (var _len = arguments.length, fns = new Array(_len), _key = 0; _key < _len; _key++) {
    fns[_key] = arguments[_key];
  }
  return hooks_usePersistFn.usePersistFn(function () {
    for (var _len2 = arguments.length, args = new Array(_len2), _key2 = 0; _key2 < _len2; _key2++) {
      args[_key2] = arguments[_key2];
    }
    return fns.forEach(function (fn) {
      return fn === null || fn === void 0 ? void 0 : fn.apply(void 0, args);
    });
  });
};
var TreeSelect = /*#__PURE__*/React.forwardRef(function (originalProps, ref) {
  var props = hooks_useDefaultProps["default"](originalProps, treeSelect_defaultProps.treeSelectDefaultProps);
  var _useTreeSelectLocale = treeSelect_hooks_useTreeSelectLocale.useTreeSelectLocale(props),
    placeholder = _useTreeSelectLocale.placeholder,
    empty = _useTreeSelectLocale.empty,
    loadingItem = _useTreeSelectLocale.loadingItem;
  var _useConfig = hooks_useConfig["default"](),
    classPrefix = _useConfig.classPrefix;
  var className = props.className,
    onInputChange = props.onInputChange,
    disabled = props.disabled,
    multiple = props.multiple,
    prefixIcon = props.prefixIcon,
    label = props.label,
    loading = props.loading,
    size = props.size,
    max = props.max,
    data = props.data,
    panelTopContent = props.panelTopContent,
    panelBottomContent = props.panelBottomContent,
    rawFilter = props.filter,
    rawFilterable = props.filterable,
    onClear = props.onClear,
    valueDisplay = props.valueDisplay,
    treeProps = props.treeProps,
    inputProps = props.inputProps,
    valueType = props.valueType,
    collapsedItems = props.collapsedItems,
    reserveKeyword = props.reserveKeyword,
    onBlur = props.onBlur,
    onFocus = props.onFocus,
    onSearch = props.onSearch,
    onRemove = props.onRemove,
    onEnter = props.onEnter;
  var readOnly = props.readOnly || props.readonly;
  var selectInputProps = treeSelect_hooks_useTreeSelectPassthroughProps.useTreeSelectPassThroughProps(props);
  var _useControlled = hooks_useControlled["default"](props, "value", props.onChange),
    _useControlled2 = slicedToArray._slicedToArray(_useControlled, 2),
    value = _useControlled2[0],
    onChange = _useControlled2[1];
  var _useControlled3 = hooks_useControlled["default"](props, "popupVisible", props.onPopupVisibleChange),
    _useControlled4 = slicedToArray._slicedToArray(_useControlled3, 2),
    popupVisible = _useControlled4[0],
    setPopupVisible = _useControlled4[1];
  var _useSwitch = hooks_useSwitch["default"](),
    _useSwitch2 = slicedToArray._slicedToArray(_useSwitch, 2),
    hover = _useSwitch2[0],
    hoverAction = _useSwitch2[1];
  var _useControlled5 = hooks_useControlled["default"](props, "inputValue", onInputChange),
    _useControlled6 = slicedToArray._slicedToArray(_useControlled5, 2),
    filterInput = _useControlled6[0],
    setFilterInput = _useControlled6[1];
  var treeRef = React.useRef(null);
  var selectInputRef = React.useRef(null);
  var tKeys = React.useMemo(function () {
    return _objectSpread({
      value: "value",
      label: "label",
      children: "children"
    }, props.keys);
  }, [props.keys]);
  var passThroughDefaultStore = React.useMemo(function () {
    return {
      data: data,
      treeProps: _objectSpread({
        keys: tKeys
      }, treeProps),
      valueType: valueType
    };
  }, [tKeys, data, treeProps, valueType]);
  var _useTreeSelectUtils = treeSelect_hooks_useTreeSelectUtils.useTreeSelectUtils(passThroughDefaultStore, treeRef),
    normalizeValue = _useTreeSelectUtils.normalizeValue,
    formatValue = _useTreeSelectUtils.formatValue,
    getNodeItem = _useTreeSelectUtils.getNodeItem;
  React.useImperativeHandle(ref, function () {
    return _objectSpread(_objectSpread({}, selectInputRef.current || {}), treeRef.current || {});
  });
  var defaultFilter = function defaultFilter(text, option) {
    if (!text) return true;
    if (!option.label && !option.value) return false;
    if (option.label && typeof option.label === "string") {
      return option.label.includes(text);
    }
    if (option.data.text && typeof option.data.text === "string") {
      return option.data.text.includes(text);
    }
    return true;
  };
  var filter = onSearch ? void 0 : rawFilter || defaultFilter;
  var filterable = rawFilterable || !!props.filter;
  var normalizedValue = React.useMemo(function () {
    var calcValue = Array.isArray(value) ? value : [value];
    return calcValue.reduce(function (result, value2) {
      var normalized = normalizeValue(value2);
      typeof normalized.value !== "undefined" && result.push(normalized);
      return result;
    }, []);
  }, [normalizeValue, value, data]);
  var internalInputValue = React.useMemo(function () {
    if (multiple) return normalizedValue;
    return filterable && popupVisible ? filterInput : normalizedValue[0] || "";
  }, [multiple, normalizedValue, filterable, popupVisible, filterInput]);
  var normalizedValueDisplay = React.useMemo(function () {
    if (!valueDisplay) {
      return;
    }
    if (multiple) {
      return function (_ref) {
        var onClose = _ref.onClose;
        return isFunction.isFunction(valueDisplay) ? valueDisplay({
          value: normalizedValue,
          onClose: onClose
        }) : valueDisplay;
      };
    }
    var displayNode = isFunction.isFunction(valueDisplay) ? valueDisplay({
      value: normalizedValue[0],
      onClose: _util_noop["default"]
    }) : valueDisplay;
    return normalizedValue.length ? displayNode : "";
  }, [valueDisplay, multiple, normalizedValue]);
  var internalInputValueDisplay = React.useMemo(function () {
    if (filterable && !multiple && popupVisible) {
      return void 0;
    }
    return normalizedValueDisplay;
  }, [filterable, popupVisible, multiple, normalizedValueDisplay]);
  var inputPlaceholder = React.useMemo(function () {
    if (filterable && !multiple && popupVisible && normalizedValue.length) {
      var valueDisplayPlaceholder = normalizedValueDisplay;
      if (typeof valueDisplayPlaceholder === "string") {
        return valueDisplayPlaceholder;
      }
      return typeof normalizedValue[0].label === "string" ? normalizedValue[0].label : String(normalizedValue[0].value);
    }
    return placeholder;
  }, [filterable, multiple, popupVisible, normalizedValue, placeholder, normalizedValueDisplay]);
  var showLoading = !disabled && loading;
  var handleFilter = React.useCallback(function (node) {
    return filterable && filter ? filter(filterInput, node) : true;
  }, [filter, filterInput, filterable]);
  var handleSingleChange = hooks_usePersistFn.usePersistFn(function (value2, context) {
    var $value = Array.isArray(value2) && value2.length ? value2[0] : void 0;
    if ($value) {
      onChange(formatValue($value, context.node.label), _objectSpread(_objectSpread({}, context), {}, {
        data: context.node.data,
        trigger: "check"
      }));
    }
    setPopupVisible(false, _objectSpread(_objectSpread({}, context), {}, {
      trigger: "trigger-element-click"
    }));
  });
  var handleMultiChange = hooks_usePersistFn.usePersistFn(function (value2, context) {
    if (max === 0 || value2.length <= max) {
      onChange(value2.map(function (value3) {
        var _getNodeItem;
        return formatValue(value3, (_getNodeItem = getNodeItem(value3)) === null || _getNodeItem === void 0 ? void 0 : _getNodeItem.label);
      }), _objectSpread(_objectSpread({}, context), {}, {
        data: context.node.data,
        trigger: value2.length > normalizedValue.length ? "check" : "uncheck"
      }));
    }
    if (!reserveKeyword) {
      setFilterInput("", {
        trigger: "change"
      });
    }
  });
  var onInnerPopupVisibleChange = function onInnerPopupVisibleChange(visible, ctx) {
    setPopupVisible(visible, {
      e: ctx.e
    });
    if (!visible) {
      setFilterInput("", {
        trigger: "blur"
      });
    }
  };
  var handleClear = hooks_usePersistFn.usePersistFn(function (ctx) {
    ctx.e.stopPropagation();
    onChange(multiple ? [] : formatValue(void 0), {
      node: null,
      data: null,
      trigger: "clear",
      e: ctx.e
    });
    onClear === null || onClear === void 0 || onClear(ctx);
    setPopupVisible(false, {
      trigger: "clear"
    });
  });
  var handleTagChange = hooks_usePersistFn.usePersistFn(function (tags, ctx) {
    if (ctx.trigger === "tag-remove" || ctx.trigger === "backspace") {
      var index = ctx.index,
        e = ctx.e,
        trigger = ctx.trigger;
      var node = getNodeItem(normalizedValue[index].value);
      onChange(normalizedValue.filter(function (value2, i) {
        return i !== index;
      }).map(function (_ref2) {
        var value2 = _ref2.value,
          label2 = _ref2.label;
        return formatValue(value2, label2);
      }), {
        node: node,
        data: node === null || node === void 0 ? void 0 : node.data,
        trigger: trigger,
        e: e
      });
      onRemove === null || onRemove === void 0 || onRemove({
        value: node.value,
        node: node,
        index: index,
        data: _objectSpread({
          value: node === null || node === void 0 ? void 0 : node.value,
          label: node === null || node === void 0 ? void 0 : node.label
        }, node === null || node === void 0 ? void 0 : node.data),
        e: e,
        trigger: trigger
      });
    }
  });
  var getTreeSelectEventValue = function getTreeSelectEventValue() {
    var selectedOptions = Array.isArray(normalizedValue) ? normalizedValue : [normalizedValue];
    var value2 = selectedOptions.map(function (item) {
      return valueType === "object" ? item : item[tKeys.value];
    });
    return multiple ? value2 : value2[0];
  };
  var handleBlur = hooks_usePersistFn.usePersistFn(function (_, ctx) {
    onBlur === null || onBlur === void 0 || onBlur(_objectSpread({
      value: getTreeSelectEventValue()
    }, ctx));
  });
  var handleFocus = hooks_usePersistFn.usePersistFn(function (_, ctx) {
    onFocus === null || onFocus === void 0 || onFocus({
      value: getTreeSelectEventValue(),
      e: ctx.e
    });
  });
  var handleEnter = hooks_usePersistFn.usePersistFn(function (_, ctx) {
    onSearch === null || onSearch === void 0 || onSearch(ctx.inputValue, {
      e: ctx.e
    });
    onEnter === null || onEnter === void 0 || onEnter({
      inputValue: ctx.inputValue,
      e: ctx.e,
      value: getTreeSelectEventValue()
    });
  });
  var handleFilterChange = hooks_usePersistFn.usePersistFn(function (value2, ctx) {
    if (ctx.trigger === "clear") return;
    setFilterInput(value2, ctx);
    onSearch === null || onSearch === void 0 || onSearch(value2, {
      e: ctx.e
    });
  });
  var renderTree = function renderTree() {
    if (readOnly) return empty;
    if (showLoading) return loadingItem;
    return /* @__PURE__ */React__default["default"].createElement(React__default["default"].Fragment, null, panelTopContent, /* @__PURE__ */React__default["default"].createElement(tree_index.Tree, _objectSpread(_objectSpread({
      ref: treeRef,
      hover: true,
      transition: true,
      filter: filterInput ? handleFilter : null,
      data: data,
      disabled: disabled,
      empty: empty,
      expandOnClickNode: false,
      keys: tKeys
    }, multiple ? {
      checkable: true,
      onChange: handleMultiChange,
      value: normalizedValue.map(function (_ref3) {
        var value2 = _ref3.value;
        return value2;
      })
    } : {
      activable: true,
      actived: normalizedValue.map(function (_ref4) {
        var value2 = _ref4.value;
        return value2;
      }),
      onActive: handleSingleChange
    }), treeProps)), panelBottomContent);
  };
  return /* @__PURE__ */React__default["default"].createElement(selectInput_SelectInput["default"], _objectSpread(_objectSpread(_objectSpread({
    status: props.status,
    tips: props.tips
  }, props.selectInputProps), selectInputProps), {}, {
    ref: selectInputRef,
    className: classNames__default["default"]("".concat(classPrefix, "-tree-select"), className),
    value: internalInputValue,
    inputValue: filterInput,
    panel: renderTree(),
    allowInput: filterable,
    inputProps: _objectSpread(_objectSpread({}, inputProps), {}, {
      size: size
    }),
    tagInputProps: {
      size: size,
      excessTagsDisplayType: "break-line",
      inputProps: inputProps,
      tagProps: props.tagProps
    },
    placeholder: inputPlaceholder,
    popupVisible: popupVisible && !disabled,
    onInputChange: handleFilterChange,
    onPopupVisibleChange: onInnerPopupVisibleChange,
    onFocus: useMergeFn(handleFocus),
    onBlur: useMergeFn(handleBlur),
    onClear: handleClear,
    onTagChange: handleTagChange,
    onEnter: handleEnter,
    onMouseenter: hoverAction.on,
    onMouseleave: hoverAction.off,
    suffixIcon: props.suffixIcon || (readOnly ? null : /* @__PURE__ */React__default["default"].createElement(treeSelect_SelectArrow.SelectArrow, {
      isActive: popupVisible,
      isHighlight: hover || popupVisible,
      disabled: disabled
    })),
    collapsedItems: collapsedItems,
    label: _util_parseTNode["default"](label || prefixIcon),
    valueDisplay: internalInputValueDisplay
  }));
});
TreeSelect.displayName = "TreeSelect";

exports["default"] = TreeSelect;
