// tdesign-react/cjs/time-picker/panel/TimePickerPanel.js
/**
 * tdesign v1.17.1
 * (c) 2026 tdesign
 * @license MIT
 */

'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var defineProperty = require('../../_chunks/dep-c446aa11.js');
var slicedToArray = require('../../_chunks/dep-c60a3e4c.js');
var React = require('react');
var dayjs = require('dayjs');
var log = require('../../_chunks/dep-3c628d18.js');
var timePicker_panel_SinglePanel = require('../../_chunks/dep-7592c92b.js');
var button_index = require('../../button/index.js');
var hooks_useConfig = require('../../hooks/useConfig.js');
var timePicker_hooks_useTimePickerTextConfig = require('../hooks/useTimePickerTextConfig.js');
require('../../_chunks/dep-c290284d.js');
require('../../_chunks/dep-98e6819c.js');
require('../../_chunks/dep-7f3fa935.js');
require('classnames');
require('../../_chunks/dep-34b7c4f7.js');
require('../../_chunks/dep-a6fa90dc.js');
require('../../_chunks/dep-c1575286.js');
require('../../_chunks/dep-3be56e1e.js');
require('../../_chunks/dep-a830ce1e.js');
require('../../_chunks/dep-36579bf6.js');
require('../../_chunks/dep-3d05de97.js');
require('../../_chunks/dep-4f1729f0.js');
require('../../_util/noop.js');
require('../../hooks/useDebounce.js');
require('../../hooks/usePersistFn.js');
require('../../_chunks/dep-f2809376.js');
require('../../hooks/usePropsRef.js');
require('../../_chunks/dep-2f45f6c4.js');
require('../../_chunks/dep-770bc0ae.js');
require('../../_chunks/dep-863f8f4d.js');
require('../../_chunks/dep-11347ea1.js');
require('../../_chunks/dep-8a493834.js');
require('../../_chunks/dep-338f013a.js');
require('../../_chunks/dep-b199860d.js');
require('../../_chunks/dep-a54f202c.js');
require('../../_chunks/dep-eccb6a94.js');
require('../../_chunks/dep-f5e1d11c.js');
require('../../button/Button.js');
require('../../_chunks/dep-e41137f7.js');
require('../../_util/parseTNode.js');
require('../../hooks/useDefaultProps.js');
require('../../hooks/useDomRefCallback.js');
require('../../hooks/useRipple.js');
require('../../_chunks/dep-062bb1b1.js');
require('../../_chunks/dep-ee71ddca.js');
require('../../_chunks/dep-427f8bc0.js');
require('../../hooks/useAnimation.js');
require('../../config-provider/ConfigContext.js');
require('../../_chunks/dep-2ee09518.js');
require('../../_chunks/dep-012d9a7c.js');
require('../../_chunks/dep-150a112c.js');
require('../../_chunks/dep-e360e17f.js');
require('../../_chunks/dep-51284537.js');
require('../../_chunks/dep-9f5b2891.js');
require('../../_chunks/dep-c37c5e17.js');
require('../../_chunks/dep-eee1aab9.js');
require('../../_chunks/dep-aaf27e39.js');
require('../../_chunks/dep-b47d855f.js');
require('../../_chunks/dep-81bce4e6.js');
require('../../_chunks/dep-84e4d0da.js');
require('../../_chunks/dep-f78183d7.js');
require('../../loading/index.js');
require('../../loading/Loading.js');
require('../../common/Portal.js');
require('react-dom');
require('../../hooks/useLayoutEffect.js');
require('../../loading/defaultProps.js');
require('../../loading/gradient.js');
require('../../_chunks/dep-6e0da644.js');
require('../../_chunks/dep-05bc49e6.js');
require('../../_chunks/dep-8c880a65.js');
require('../../_chunks/dep-b8d0b96c.js');
require('../../loading/plugin.js');
require('../../_util/react-render.js');
require('../../_chunks/dep-7bf14076.js');
require('../../common/PluginContainer.js');
require('../../config-provider/ConfigProvider.js');
require('../../_chunks/dep-ba90997f.js');
require('../../_chunks/dep-22026fea.js');
require('../../_chunks/dep-551d5940.js');
require('../../_chunks/dep-8005b808.js');
require('../../_chunks/dep-2add8824.js');
require('../../button/defaultProps.js');
require('../../locale/LocalReceiver.js');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

var React__default = /*#__PURE__*/_interopDefaultLegacy(React);
var dayjs__default = /*#__PURE__*/_interopDefaultLegacy(dayjs);

function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { defineProperty._defineProperty(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
var TimePickerPanel = function TimePickerPanel(props) {
  var _props$format = props.format,
    format = _props$format === void 0 ? timePicker_panel_SinglePanel.DEFAULT_FORMAT : _props$format,
    _props$steps = props.steps,
    steps = _props$steps === void 0 ? timePicker_panel_SinglePanel.DEFAULT_STEPS : _props$steps,
    handleConfirmClick = props.handleConfirmClick,
    isFooterDisplay = props.isFooterDisplay,
    onChange = props.onChange,
    value = props.value,
    _props$isShowPanel = props.isShowPanel,
    isShowPanel = _props$isShowPanel === void 0 ? true : _props$isShowPanel,
    _props$presets = props.presets,
    presets = _props$presets === void 0 ? null : _props$presets;
  var _useState = React.useState(false),
    _useState2 = slicedToArray._slicedToArray(_useState, 2),
    triggerScroll = _useState2[0],
    toggleTriggerScroll = _useState2[1];
  var _useConfig = hooks_useConfig["default"](),
    classPrefix = _useConfig.classPrefix;
  var TEXT_CONFIG = timePicker_hooks_useTimePickerTextConfig.useTimePickerTextConfig();
  var panelClassName = "".concat(classPrefix, "-time-picker__panel");
  var showNowTimeBtn = !!steps.filter(function (v) {
    return Number(v) > 1;
  }).length;
  var defaultValue = React.useMemo(function () {
    var formattedValue = dayjs__default["default"](value, format);
    if (value && formattedValue.isValid()) {
      return formattedValue.format(format);
    }
    return dayjs__default["default"]().hour(0).minute(0).second(0).format(format);
  }, [value, format]);
  React.useEffect(function () {
    if (isShowPanel) toggleTriggerScroll(true);
  }, [isShowPanel]);
  var resetTriggerScroll = React.useCallback(function () {
    toggleTriggerScroll(false);
  }, [toggleTriggerScroll]);
  var handlePresetClick = function handlePresetClick(presetValue) {
    var presetVal = typeof presetValue === "function" ? presetValue() : presetValue;
    if (typeof props.activeIndex === "number") {
      if (Array.isArray(presetVal)) {
        var _props$onChange;
        (_props$onChange = props.onChange) === null || _props$onChange === void 0 || _props$onChange.call(props, presetVal[props.activeIndex]);
      } else {
        log.log.error("TimePicker", "preset: ".concat(presets, " \u9884\u8BBE\u503C\u5FC5\u987B\u662F\u6570\u7EC4!"));
      }
    } else {
      var _props$onChange2;
      (_props$onChange2 = props.onChange) === null || _props$onChange2 === void 0 || _props$onChange2.call(props, presetVal);
    }
  };
  var renderFooter = function renderFooter() {
    if (presets) {
      return Object.keys(presets).map(function (preset) {
        return /* @__PURE__ */React__default["default"].createElement(button_index.Button, {
          key: preset,
          theme: "primary",
          size: "small",
          variant: "text",
          onClick: function onClick() {
            handlePresetClick(presets[preset]);
          }
        }, preset);
      });
    }
    return !showNowTimeBtn ? /* @__PURE__ */React__default["default"].createElement(button_index.Button, {
      theme: "primary",
      variant: "text",
      size: "small",
      onClick: function onClick() {
        onChange === null || onChange === void 0 || onChange(dayjs__default["default"]().format(format));
      }
    }, TEXT_CONFIG.nowTime) : null;
  };
  return /* @__PURE__ */React__default["default"].createElement("div", {
    className: panelClassName
  }, /* @__PURE__ */React__default["default"].createElement("div", {
    className: "".concat(panelClassName, "-section-body")
  }, /* @__PURE__ */React__default["default"].createElement(timePicker_panel_SinglePanel.SinglePanel, _objectSpread(_objectSpread({}, props), {}, {
    onChange: onChange,
    format: format,
    steps: steps,
    value: dayjs__default["default"](value, format).isValid() ? value : defaultValue,
    triggerScroll: triggerScroll,
    isVisible: isShowPanel,
    resetTriggerScroll: resetTriggerScroll
  }))), isFooterDisplay ? /* @__PURE__ */React__default["default"].createElement("div", {
    className: "".concat(panelClassName, "-section-footer")
  }, /* @__PURE__ */React__default["default"].createElement(button_index.Button, {
    size: "small",
    theme: "primary",
    variant: "base",
    disabled: !props.value,
    onClick: function onClick() {
      handleConfirmClick === null || handleConfirmClick === void 0 || handleConfirmClick(defaultValue);
    }
  }, TEXT_CONFIG.confirm), renderFooter()) : null);
};

exports["default"] = TimePickerPanel;
