// tdesign-react/cjs/table/Cell.js
/**
 * tdesign v1.17.1
 * (c) 2026 tdesign
 * @license MIT
 */

'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var _typeof = require('../_chunks/dep-c290284d.js');
var defineProperty = require('../_chunks/dep-c446aa11.js');
var React = require('react');
var classNames = require('classnames');
var table_Ellipsis = require('./Ellipsis.js');
var table_hooks_useFixed = require('./hooks/useFixed.js');
var table_utils = require('./utils.js');
var isFunction = require('../_chunks/dep-11347ea1.js');
var get = require('../_chunks/dep-f8c20fd8.js');
require('../_chunks/dep-c60a3e4c.js');
require('../_chunks/dep-98e6819c.js');
require('../_chunks/dep-7f3fa935.js');
require('../_util/scroll.js');
require('raf');
require('../_chunks/dep-ee71ddca.js');
require('../_chunks/dep-427f8bc0.js');
require('../_chunks/dep-3d05de97.js');
require('../_chunks/dep-a54f202c.js');
require('../_chunks/dep-4f1729f0.js');
require('../_util/easing.js');
require('../hooks/useDebounce.js');
require('../hooks/usePersistFn.js');
require('../_chunks/dep-f2809376.js');
require('../_chunks/dep-a830ce1e.js');
require('../_chunks/dep-3be56e1e.js');
require('../_chunks/dep-36579bf6.js');
require('../tooltip/index.js');
require('../tooltip/Tooltip.js');
require('../_chunks/dep-e41137f7.js');
require('../hooks/useConfig.js');
require('../config-provider/ConfigContext.js');
require('../_chunks/dep-2ee09518.js');
require('../_chunks/dep-a6fa90dc.js');
require('dayjs');
require('../_chunks/dep-012d9a7c.js');
require('../_chunks/dep-150a112c.js');
require('../_chunks/dep-e360e17f.js');
require('../_chunks/dep-51284537.js');
require('../_chunks/dep-8a493834.js');
require('../_chunks/dep-9f5b2891.js');
require('../_chunks/dep-338f013a.js');
require('../_chunks/dep-c37c5e17.js');
require('../_chunks/dep-eee1aab9.js');
require('../_chunks/dep-770bc0ae.js');
require('../_chunks/dep-aaf27e39.js');
require('../_chunks/dep-b47d855f.js');
require('../_chunks/dep-81bce4e6.js');
require('../_chunks/dep-863f8f4d.js');
require('../_chunks/dep-84e4d0da.js');
require('../_chunks/dep-f78183d7.js');
require('../_chunks/dep-2f45f6c4.js');
require('../hooks/useDefaultProps.js');
require('../popup/index.js');
require('../popup/Popup.js');
require('react-transition-group');
require('../common/Portal.js');
require('react-dom');
require('../hooks/useLayoutEffect.js');
require('../hooks/useAnimation.js');
require('../hooks/useAttach.js');
require('../hooks/useControlled.js');
require('../_util/noop.js');
require('../_chunks/dep-45188d54.js');
require('../_chunks/dep-eccb6a94.js');
require('../_chunks/dep-f5e1d11c.js');
require('../_chunks/dep-b199860d.js');
require('../hooks/useMutationObserver.js');
require('../hooks/useLatest.js');
require('../_chunks/dep-aa9a2c54.js');
require('../_chunks/dep-77ef3011.js');
require('../_chunks/dep-b36779a4.js');
require('../_chunks/dep-551d5940.js');
require('../_chunks/dep-8005b808.js');
require('../_chunks/dep-2add8824.js');
require('../hooks/usePopper.js');
require('../_chunks/dep-05bc49e6.js');
require('../_chunks/dep-8c880a65.js');
require('react-fast-compare');
require('@popperjs/core');
require('../hooks/useWindowSize.js');
require('../popup/defaultProps.js');
require('../popup/hooks/useTrigger.js');
require('../_util/listener.js');
require('../_util/ref.js');
require('react-is');
require('../_util/isFragment.js');
require('../popup/utils/transition.js');
require('../popup/PopupPlugin.js');
require('../_chunks/dep-7bf14076.js');
require('../_util/react-render.js');
require('../common/PluginContainer.js');
require('../config-provider/ConfigProvider.js');
require('../_chunks/dep-ba90997f.js');
require('../_chunks/dep-22026fea.js');
require('../tooltip/defaultProps.js');
require('../tooltip/TooltipLite.js');
require('../hooks/useSwitch.js');
require('../_chunks/dep-d07eb641.js');
require('../_chunks/dep-3c628d18.js');
require('../_chunks/dep-ea91dbfd.js');
require('../_chunks/dep-6e0da644.js');
require('../_chunks/dep-b8d0b96c.js');
require('../hooks/useDeepEffect.js');
require('../hooks/usePrevious.js');
require('../_chunks/dep-677950af.js');
require('../_chunks/dep-5fd08885.js');
require('../_chunks/dep-f98232bb.js');
require('../_chunks/dep-7a4e0117.js');
require('../_chunks/dep-5ee57a9e.js');
require('../_chunks/dep-cdef7593.js');
require('../_chunks/dep-3784bbea.js');
require('../_chunks/dep-d4a6a9c1.js');
require('../_chunks/dep-bc160f5b.js');
require('../_chunks/dep-363864ae.js');
require('../_chunks/dep-204188de.js');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

var React__default = /*#__PURE__*/_interopDefaultLegacy(React);
var classNames__default = /*#__PURE__*/_interopDefaultLegacy(classNames);

function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { defineProperty._defineProperty(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
function renderCell(params, extra) {
  var col = params.col,
    row = params.row,
    rowIndex = params.rowIndex;
  if (col.colKey === "serial-number") {
    var _ref = (extra === null || extra === void 0 ? void 0 : extra.pagination) || {},
      current = _ref.current,
      pageSize = _ref.pageSize,
      defaultCurrent = _ref.defaultCurrent,
      defaultPageSize = _ref.defaultPageSize;
    var tCurrent = current || defaultCurrent;
    var tPageSize = pageSize || defaultPageSize;
    if (tPageSize && tCurrent) {
      return tPageSize * (tCurrent - 1) + rowIndex + 1;
    }
    return rowIndex + 1;
  }
  if (isFunction.isFunction(col.cell)) {
    return col.cell(params);
  }
  if (isFunction.isFunction(col.render)) {
    return col.render(_objectSpread(_objectSpread({}, params), {}, {
      type: "cell"
    }));
  }
  var r = col.cell || col.render || get.get(row, col.colKey);
  if (![void 0, "", null].includes(r)) return r;
  if (extra !== null && extra !== void 0 && extra.cellEmptyContent) return extra.cellEmptyContent;
  return r;
}
function renderEllipsisCell(cellParams, params) {
  var cellNode = params.cellNode,
    tableElm = params.tableElm,
    columnLength = params.columnLength,
    classPrefix = params.classPrefix,
    overlayClassName = params.overlayClassName;
  var col = cellParams.col,
    colIndex = cellParams.colIndex;
  var content = isFunction.isFunction(col.ellipsis) ? col.ellipsis(cellParams) : void 0;
  if (_typeof._typeof(col.ellipsis) === "object" && "content" in col.ellipsis && isFunction.isFunction(col.ellipsis.content)) {
    content = col.ellipsis.content(cellParams);
  }
  var tooltipProps = {};
  if (_typeof._typeof(col.ellipsis) === "object") {
    tooltipProps = "props" in col.ellipsis ? col.ellipsis.props : col.ellipsis || void 0;
  }
  var tableElement = tableElm;
  var placement = colIndex === 0 ? "top-left" : "top";
  placement = colIndex === columnLength - 1 ? "top-right" : placement;
  return /* @__PURE__ */React__default["default"].createElement(table_Ellipsis["default"], {
    placement: placement,
    attach: tableElement ? function () {
      return tableElement;
    } : void 0,
    popupContent: content,
    tooltipProps: tooltipProps,
    overlayClassName: overlayClassName,
    classPrefix: classPrefix
  }, cellNode);
}
var Cell = function Cell(props) {
  var cellParams = props.cellParams,
    tableClassNames = props.tableClassNames,
    tableRef = props.tableRef,
    columnLength = props.columnLength,
    classPrefix = props.classPrefix,
    overlayClassName = props.overlayClassName,
    pagination = props.pagination;
  var col = cellParams.col,
    colIndex = cellParams.colIndex,
    rowIndex = cellParams.rowIndex;
  var cellSpans = props.cellSpans,
    dataLength = props.dataLength,
    rowAndColFixedPosition = props.rowAndColFixedPosition,
    cellEmptyContent = props.cellEmptyContent,
    rowspanAndColspan = props.rowspanAndColspan,
    onClick = props.onClick;
  var tableColFixedClasses = tableClassNames.tableColFixedClasses,
    tdEllipsisClass = tableClassNames.tdEllipsisClass,
    tableBaseClass = tableClassNames.tableBaseClass,
    tdAlignClasses = tableClassNames.tdAlignClasses,
    tableDraggableClasses = tableClassNames.tableDraggableClasses;
  var cellNode = renderCell(cellParams, {
    cellEmptyContent: cellEmptyContent,
    pagination: pagination
  });
  var tdStyles = table_hooks_useFixed.getColumnFixedStyles(col, colIndex, rowAndColFixedPosition, tableColFixedClasses);
  var customClasses = table_utils.formatClassNames(col.className, _objectSpread(_objectSpread({}, cellParams), {}, {
    type: "td"
  }));
  var classes = [tdStyles.classes, customClasses, defineProperty._defineProperty(defineProperty._defineProperty(defineProperty._defineProperty(defineProperty._defineProperty(defineProperty._defineProperty({}, tdEllipsisClass, col.ellipsis), tableBaseClass.tdLastRow, rowIndex + cellSpans.rowspan === dataLength), tableBaseClass.tdFirstCol, colIndex === 0 && rowspanAndColspan), tdAlignClasses[col.align], col.align && col.align !== "left"), tableDraggableClasses.handle, col.colKey === "drag")];
  var normalAttrs = isFunction.isFunction(col.attrs) ? col.attrs(_objectSpread(_objectSpread({}, cellParams), {}, {
    type: "td"
  })) : col.attrs;
  var attrs = _objectSpread(_objectSpread({}, normalAttrs), {}, {
    rowSpan: cellSpans.rowspan,
    colSpan: cellSpans.colspan
  });
  return /* @__PURE__ */React__default["default"].createElement("td", _objectSpread(_objectSpread({
    key: col.colKey || colIndex,
    className: classNames__default["default"](classes) || void 0,
    style: tdStyles.style
  }, attrs), {}, {
    onClick: onClick
  }), col.ellipsis ? renderEllipsisCell(cellParams, {
    cellNode: cellNode,
    tableElm: tableRef.current,
    columnLength: columnLength,
    classPrefix: classPrefix,
    overlayClassName: overlayClassName
  }) : cellNode);
};
Cell.displayName = "Cell";

exports["default"] = Cell;
exports.renderCell = renderCell;
