// tdesign-react/cjs/table/FilterController.js
/**
 * tdesign v1.17.1
 * (c) 2026 tdesign
 * @license MIT
 */

'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var _typeof = require('../_chunks/dep-c290284d.js');
var defineProperty = require('../_chunks/dep-c446aa11.js');
var slicedToArray = require('../_chunks/dep-c60a3e4c.js');
var React = require('react');
var classNames = require('classnames');
var tdesignIconsReact = require('tdesign-icons-react');
var log = require('../_chunks/dep-3c628d18.js');
var _util_parseTNode = require('../_util/parseTNode.js');
var button_index = require('../button/index.js');
var checkbox_index = require('../checkbox/index.js');
var hooks_useGlobalIcon = require('../hooks/useGlobalIcon.js');
var input_index = require('../input/index.js');
var locale_LocalReceiver = require('../locale/LocalReceiver.js');
var popup_index = require('../popup/index.js');
var radio_index = require('../radio/index.js');
var isEmpty = require('../_chunks/dep-2be6b7d2.js');
require('../_chunks/dep-98e6819c.js');
require('../_chunks/dep-7f3fa935.js');
require('../_chunks/dep-11347ea1.js');
require('../_chunks/dep-3d05de97.js');
require('../_chunks/dep-a830ce1e.js');
require('../button/Button.js');
require('../_chunks/dep-e41137f7.js');
require('../hooks/useConfig.js');
require('../config-provider/ConfigContext.js');
require('../_chunks/dep-2ee09518.js');
require('../_chunks/dep-a6fa90dc.js');
require('dayjs');
require('../_chunks/dep-012d9a7c.js');
require('../_chunks/dep-150a112c.js');
require('../_chunks/dep-e360e17f.js');
require('../_chunks/dep-4f1729f0.js');
require('../_chunks/dep-a54f202c.js');
require('../_chunks/dep-51284537.js');
require('../_chunks/dep-8a493834.js');
require('../_chunks/dep-9f5b2891.js');
require('../_chunks/dep-338f013a.js');
require('../_chunks/dep-c37c5e17.js');
require('../_chunks/dep-eee1aab9.js');
require('../_chunks/dep-770bc0ae.js');
require('../_chunks/dep-aaf27e39.js');
require('../_chunks/dep-b47d855f.js');
require('../_chunks/dep-81bce4e6.js');
require('../_chunks/dep-863f8f4d.js');
require('../_chunks/dep-84e4d0da.js');
require('../_chunks/dep-f78183d7.js');
require('../_chunks/dep-2f45f6c4.js');
require('../hooks/useDefaultProps.js');
require('../hooks/useDomRefCallback.js');
require('../hooks/useRipple.js');
require('../_chunks/dep-062bb1b1.js');
require('../_chunks/dep-ee71ddca.js');
require('../_chunks/dep-427f8bc0.js');
require('../hooks/useAnimation.js');
require('../loading/index.js');
require('../loading/Loading.js');
require('../common/Portal.js');
require('react-dom');
require('../hooks/useLayoutEffect.js');
require('../loading/defaultProps.js');
require('../loading/gradient.js');
require('../_chunks/dep-6e0da644.js');
require('../_chunks/dep-05bc49e6.js');
require('../_chunks/dep-8c880a65.js');
require('../_chunks/dep-b8d0b96c.js');
require('../loading/plugin.js');
require('../_util/react-render.js');
require('../_chunks/dep-7bf14076.js');
require('../common/PluginContainer.js');
require('../config-provider/ConfigProvider.js');
require('../_chunks/dep-ba90997f.js');
require('../_chunks/dep-22026fea.js');
require('../_chunks/dep-551d5940.js');
require('../_chunks/dep-8005b808.js');
require('../_chunks/dep-2add8824.js');
require('../button/defaultProps.js');
require('../_chunks/dep-ab5baf4d.js');
require('../_util/forwardRefWithStatics.js');
require('hoist-non-react-statics');
require('../common/Check.js');
require('../_util/helper.js');
require('../_chunks/dep-b422817e.js');
require('../_chunks/dep-b199860d.js');
require('../_chunks/dep-36579bf6.js');
require('../_chunks/dep-45188d54.js');
require('../_chunks/dep-eccb6a94.js');
require('../_chunks/dep-f5e1d11c.js');
require('../hooks/useControlled.js');
require('../_util/noop.js');
require('../_chunks/dep-fb263e64.js');
require('../checkbox/defaultProps.js');
require('../input/Input.js');
require('../input/defaultProps.js');
require('../input/InputGroup.js');
require('../input/useLengthLimit.js');
require('../popup/Popup.js');
require('react-transition-group');
require('../hooks/useAttach.js');
require('../hooks/useMutationObserver.js');
require('../hooks/useLatest.js');
require('../_chunks/dep-aa9a2c54.js');
require('../_chunks/dep-77ef3011.js');
require('../_chunks/dep-b36779a4.js');
require('../_chunks/dep-f2809376.js');
require('../_chunks/dep-3be56e1e.js');
require('../hooks/usePopper.js');
require('react-fast-compare');
require('@popperjs/core');
require('../hooks/useWindowSize.js');
require('../popup/defaultProps.js');
require('../popup/hooks/useTrigger.js');
require('../_util/listener.js');
require('../_util/ref.js');
require('react-is');
require('../_util/isFragment.js');
require('../popup/utils/transition.js');
require('../popup/PopupPlugin.js');
require('../_chunks/dep-3ddd7377.js');
require('../radio/defaultProps.js');
require('../_chunks/dep-2bc69864.js');
require('../hooks/useCommonClassName.js');
require('../hooks/useDeepEffect.js');
require('../radio/useKeyboard.js');
require('../_chunks/dep-29222115.js');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

var React__default = /*#__PURE__*/_interopDefaultLegacy(React);
var classNames__default = /*#__PURE__*/_interopDefaultLegacy(classNames);

function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { defineProperty._defineProperty(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
var CheckboxGroup = checkbox_index.Checkbox.Group;
var RadioGroup = radio_index.Radio.Group;
function TableFilterController(props) {
  var tFilterValue = props.tFilterValue,
    innerFilterValue = props.innerFilterValue,
    tableFilterClasses = props.tableFilterClasses,
    isFocusClass = props.isFocusClass,
    column = props.column;
  var _useGlobalIcon = hooks_useGlobalIcon["default"]({
      FilterIcon: tdesignIconsReact.FilterIcon
    }),
    FilterIcon = _useGlobalIcon.FilterIcon;
  var triggerElementRef = React.useRef(null);
  var _useLocaleReceiver = locale_LocalReceiver.useLocaleReceiver("table"),
    _useLocaleReceiver2 = slicedToArray._slicedToArray(_useLocaleReceiver, 2),
    locale = _useLocaleReceiver2[0],
    t = _useLocaleReceiver2[1];
  var _useState = React.useState(false),
    _useState2 = slicedToArray._slicedToArray(_useState, 2),
    filterPopupVisible = _useState2[0],
    setFilterPopupVisible = _useState2[1];
  var onFilterPopupVisibleChange = function onFilterPopupVisibleChange(visible) {
    var _props$onVisibleChang;
    setFilterPopupVisible(visible);
    (_props$onVisibleChang = props.onVisibleChange) === null || _props$onVisibleChang === void 0 || _props$onVisibleChang.call(props, visible);
  };
  var getFilterContent = function getFilterContent(column2) {
    var _column2$filter, _column2$filter2, _column2$filter3, _column2$filter4, _column2$filter5;
    var types = ["single", "multiple", "input"];
    if (column2.type && !types.includes(column2.filter.type)) {
      log.log.error("Table", "TDesign Table Error: column.filter.type must be the following: ".concat(JSON.stringify(types)));
      return;
    }
    var Component = {
      single: RadioGroup,
      multiple: CheckboxGroup,
      input: input_index.Input
    }[column2.filter.type];
    if (!Component && !(column2 !== null && column2 !== void 0 && (_column2$filter = column2.filter) !== null && _column2$filter !== void 0 && _column2$filter.component)) return;
    var filterComponentProps = _objectSpread(_objectSpread({
      options: ["single", "multiple"].includes(column2.filter.type) ? (_column2$filter2 = column2.filter) === null || _column2$filter2 === void 0 ? void 0 : _column2$filter2.list : void 0
    }, ((_column2$filter3 = column2.filter) === null || _column2$filter3 === void 0 ? void 0 : _column2$filter3.props) || {}), {}, {
      onChange: function onChange(val) {
        var _props$onInnerFilterC;
        (_props$onInnerFilterC = props.onInnerFilterChange) === null || _props$onInnerFilterC === void 0 || _props$onInnerFilterC.call(props, val, column2);
      }
    });
    if (column2.colKey && innerFilterValue && column2.colKey in innerFilterValue) {
      filterComponentProps.value = innerFilterValue[column2.colKey];
    }
    if ((_column2$filter4 = column2.filter) !== null && _column2$filter4 !== void 0 && _column2$filter4.confirmEvents) {
      column2.filter.confirmEvents.forEach(function (event) {
        filterComponentProps[event] = function () {
          var _props$onConfirm;
          setFilterPopupVisible(false);
          (_props$onConfirm = props.onConfirm) === null || _props$onConfirm === void 0 || _props$onConfirm.call(props, column2);
        };
      });
    }
    var FilterComponent = (column2 === null || column2 === void 0 || (_column2$filter5 = column2.filter) === null || _column2$filter5 === void 0 ? void 0 : _column2$filter5.component) || Component;
    var filter = column2.filter || {};
    return /* @__PURE__ */React__default["default"].createElement("div", {
      className: tableFilterClasses.contentInner
    }, /* @__PURE__ */React__default["default"].createElement(FilterComponent, _objectSpread(_objectSpread({
      className: filter.classNames,
      style: filter.style
    }, filter.attrs), filterComponentProps)));
  };
  var getBottomButtons = function getBottomButtons(column2) {
    if (!column2.filter.showConfirmAndReset) return;
    return /* @__PURE__ */React__default["default"].createElement("div", {
      className: tableFilterClasses.bottomButtons
    }, /* @__PURE__ */React__default["default"].createElement(button_index.Button, {
      theme: "default",
      size: "small",
      onClick: function onClick() {
        var _props$onReset;
        setFilterPopupVisible(false);
        (_props$onReset = props.onReset) === null || _props$onReset === void 0 || _props$onReset.call(props, column2);
      }
    }, locale.resetText), /* @__PURE__ */React__default["default"].createElement(button_index.Button, {
      theme: "primary",
      size: "small",
      onClick: function onClick() {
        var _props$onConfirm2;
        setFilterPopupVisible(false);
        (_props$onConfirm2 = props.onConfirm) === null || _props$onConfirm2 === void 0 || _props$onConfirm2.call(props, column2);
      }
    }, locale.confirmText));
  };
  if (!column.filter || column.filter && !Object.keys(column.filter).length) return null;
  var defaultFilterIcon = t(locale.filterIcon) || /* @__PURE__ */React__default["default"].createElement(FilterIcon, null);
  var filterValue = tFilterValue === null || tFilterValue === void 0 ? void 0 : tFilterValue[column.colKey];
  var isObjectTrue = _typeof._typeof(filterValue) === "object" && !isEmpty.isEmpty(filterValue);
  var isValueExist = ![null, void 0, ""].includes(filterValue) && _typeof._typeof(filterValue) !== "object";
  return /* @__PURE__ */React__default["default"].createElement("div", {
    className: classNames__default["default"]([tableFilterClasses.icon, defineProperty._defineProperty({}, isFocusClass, isObjectTrue || isValueExist)])
  }, /* @__PURE__ */React__default["default"].createElement(popup_index.Popup, _objectSpread({
    visible: filterPopupVisible,
    destroyOnClose: true,
    trigger: "click",
    placement: "bottom-right",
    showArrow: true,
    overlayClassName: tableFilterClasses.popup,
    onVisibleChange: function onVisibleChange(val) {
      return onFilterPopupVisibleChange(val);
    },
    content: /* @__PURE__ */React__default["default"].createElement("div", {
      className: tableFilterClasses.popupContent
    }, getFilterContent(column), getBottomButtons(column))
  }, props.popupProps), /* @__PURE__ */React__default["default"].createElement("div", {
    ref: triggerElementRef
  }, _util_parseTNode.parseContentTNode(props.filterIcon, {
    col: column,
    colIndex: props.colIndex
  }) || defaultFilterIcon)));
}

exports["default"] = TableFilterController;
