// tdesign-react/cjs/table/TFoot.js
/**
 * tdesign v1.17.1
 * (c) 2026 tdesign
 * @license MIT
 */

'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var defineProperty = require('../_chunks/dep-c446aa11.js');
var React = require('react');
var classNames = require('classnames');
var table_hooks_useClassName = require('./hooks/useClassName.js');
var table_hooks_useFixed = require('./hooks/useFixed.js');
var table_hooks_useRowspanAndColspan = require('./hooks/useRowspanAndColspan.js');
var table_utils = require('./utils.js');
var isFunction = require('../_chunks/dep-11347ea1.js');
var get = require('../_chunks/dep-f8c20fd8.js');
require('../_chunks/dep-c290284d.js');
require('../hooks/useConfig.js');
require('../config-provider/ConfigContext.js');
require('../_chunks/dep-2ee09518.js');
require('../_chunks/dep-a6fa90dc.js');
require('dayjs');
require('../_chunks/dep-012d9a7c.js');
require('../_chunks/dep-150a112c.js');
require('../_chunks/dep-e360e17f.js');
require('../_chunks/dep-3d05de97.js');
require('../_chunks/dep-4f1729f0.js');
require('../_chunks/dep-a54f202c.js');
require('../_chunks/dep-51284537.js');
require('../_chunks/dep-8a493834.js');
require('../_chunks/dep-9f5b2891.js');
require('../_chunks/dep-338f013a.js');
require('../_chunks/dep-c37c5e17.js');
require('../_chunks/dep-eee1aab9.js');
require('../_chunks/dep-a830ce1e.js');
require('../_chunks/dep-770bc0ae.js');
require('../_chunks/dep-aaf27e39.js');
require('../_chunks/dep-b47d855f.js');
require('../_chunks/dep-81bce4e6.js');
require('../_chunks/dep-863f8f4d.js');
require('../_chunks/dep-84e4d0da.js');
require('../_chunks/dep-f78183d7.js');
require('../_chunks/dep-2f45f6c4.js');
require('../_chunks/dep-c60a3e4c.js');
require('../_chunks/dep-98e6819c.js');
require('../_chunks/dep-7f3fa935.js');
require('../_chunks/dep-3c628d18.js');
require('../_chunks/dep-ea91dbfd.js');
require('../_chunks/dep-6e0da644.js');
require('../_chunks/dep-05bc49e6.js');
require('../_chunks/dep-8c880a65.js');
require('../_chunks/dep-e41137f7.js');
require('../_chunks/dep-b8d0b96c.js');
require('../_chunks/dep-427f8bc0.js');
require('../_util/listener.js');
require('../_chunks/dep-ee71ddca.js');
require('../hooks/useDeepEffect.js');
require('../_chunks/dep-77ef3011.js');
require('../_chunks/dep-b36779a4.js');
require('../_chunks/dep-551d5940.js');
require('../_chunks/dep-8005b808.js');
require('../_chunks/dep-2add8824.js');
require('../hooks/usePrevious.js');
require('../_chunks/dep-677950af.js');
require('../_chunks/dep-5fd08885.js');
require('../_chunks/dep-f98232bb.js');
require('../_chunks/dep-b199860d.js');
require('../_chunks/dep-36579bf6.js');
require('../_chunks/dep-7a4e0117.js');
require('../_chunks/dep-5ee57a9e.js');
require('../_chunks/dep-cdef7593.js');
require('../_chunks/dep-3784bbea.js');
require('../_chunks/dep-d4a6a9c1.js');
require('../_chunks/dep-bc160f5b.js');
require('../_chunks/dep-363864ae.js');
require('../_chunks/dep-204188de.js');
require('../hooks/useLayoutEffect.js');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

var React__default = /*#__PURE__*/_interopDefaultLegacy(React);
var classNames__default = /*#__PURE__*/_interopDefaultLegacy(classNames);

function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { defineProperty._defineProperty(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
function TFoot(props) {
  var footData = props.footData,
    columns = props.columns,
    rowKey = props.rowKey,
    footerSummary = props.footerSummary,
    virtualScroll = props.virtualScroll;
  var tfooterRef = React.useRef(null);
  var classnames = table_hooks_useClassName["default"]();
  var _useRowspanAndColspan = table_hooks_useRowspanAndColspan["default"](footData, columns, rowKey, props.rowspanAndColspanInFooter),
    skipSpansMap = _useRowspanAndColspan.skipSpansMap;
  var renderTFootCell = function renderTFootCell(p) {
    var col = p.col,
      row = p.row;
    if (isFunction.isFunction(col.foot)) {
      return col.foot(p);
    }
    return col.foot || get.get(row, col.colKey);
  };
  var theadClasses = [classnames.tableFooterClasses.footer, defineProperty._defineProperty({}, classnames.tableFooterClasses.fixed, props.isFixedHeader)];
  if (!columns) return null;
  var footerDomList = footData === null || footData === void 0 ? void 0 : footData.map(function (row, rowIndex) {
    var trAttributes = table_utils.formatRowAttributes(props.rowAttributes, {
      row: row,
      rowIndex: rowIndex,
      type: "foot"
    });
    var customClasses = table_utils.formatRowClassNames(props.rowClassName, {
      row: row,
      rowIndex: rowIndex,
      type: "foot"
    }, props.rowKey || "id");
    return /* @__PURE__ */React__default["default"].createElement("tr", _objectSpread(_objectSpread({
      key: rowIndex
    }, trAttributes), {}, {
      className: classNames__default["default"](customClasses) || void 0
    }), columns.map(function (col, colIndex) {
      var _props$thWidthList;
      var cellSpans = {};
      var spanState = null;
      if (skipSpansMap.size) {
        var _spanState, _spanState2;
        var cellKey = table_hooks_useRowspanAndColspan.getCellKey(row, rowKey, col.colKey, colIndex);
        spanState = skipSpansMap.get(cellKey) || {};
        ((_spanState = spanState) === null || _spanState === void 0 ? void 0 : _spanState.rowspan) > 1 && (cellSpans.rowspan = spanState.rowspan);
        ((_spanState2 = spanState) === null || _spanState2 === void 0 ? void 0 : _spanState2.colspan) > 1 && (cellSpans.colspan = spanState.colspan);
        if (spanState.skipped) return null;
      }
      var tdStyles = table_hooks_useFixed.getColumnFixedStyles(col, colIndex, props.rowAndColFixedPosition, classnames.tableColFixedClasses);
      var style = _objectSpread({}, tdStyles.style);
      if ((_props$thWidthList = props.thWidthList) !== null && _props$thWidthList !== void 0 && _props$thWidthList[col.colKey]) {
        style.width = "".concat(props.thWidthList[col.colKey] || 0, "px");
      }
      return /* @__PURE__ */React__default["default"].createElement("td", {
        key: col.colKey,
        rowSpan: cellSpans.rowspan,
        colSpan: cellSpans.colspan,
        className: classNames__default["default"](tdStyles.classes) || void 0,
        style: style
      }, renderTFootCell({
        row: row,
        rowIndex: rowIndex,
        col: col,
        colIndex: colIndex
      }));
    }));
  });
  if (!footerSummary && (!props.footData || !props.footData.length)) return null;
  return /* @__PURE__ */React__default["default"].createElement("tfoot", {
    ref: tfooterRef,
    className: classNames__default["default"](theadClasses) || void 0,
    style: {
      visibility: virtualScroll ? "hidden" : "visible"
    }
  }, footerSummary && /* @__PURE__ */React__default["default"].createElement("tr", {
    className: classnames.tableFullRowClasses.base
  }, /* @__PURE__ */React__default["default"].createElement("td", {
    colSpan: columns.length
  }, /* @__PURE__ */React__default["default"].createElement("div", {
    className: classnames.tableFullRowClasses.innerFullElement
  }, footerSummary))), footerDomList);
}

exports["default"] = TFoot;
