// tdesign-react/cjs/table/THead.js
/**
 * tdesign v1.17.1
 * (c) 2026 tdesign
 * @license MIT
 */

'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var _typeof = require('../_chunks/dep-c290284d.js');
var defineProperty = require('../_chunks/dep-c446aa11.js');
var React = require('react');
var classNames = require('classnames');
var table_Ellipsis = require('./Ellipsis.js');
var table_hooks_useClassName = require('./hooks/useClassName.js');
var table_hooks_useFixed = require('./hooks/useFixed.js');
var table_hooks_useTableHeader = require('./hooks/useTableHeader.js');
var table_utils = require('./utils.js');
var isFunction = require('../_chunks/dep-11347ea1.js');
require('../_chunks/dep-c60a3e4c.js');
require('../_chunks/dep-98e6819c.js');
require('../_chunks/dep-7f3fa935.js');
require('../_util/scroll.js');
require('raf');
require('../_chunks/dep-ee71ddca.js');
require('../_chunks/dep-427f8bc0.js');
require('../_chunks/dep-3d05de97.js');
require('../_chunks/dep-a54f202c.js');
require('../_chunks/dep-4f1729f0.js');
require('../_util/easing.js');
require('../hooks/useDebounce.js');
require('../hooks/usePersistFn.js');
require('../_chunks/dep-f2809376.js');
require('../_chunks/dep-a830ce1e.js');
require('../_chunks/dep-3be56e1e.js');
require('../_chunks/dep-36579bf6.js');
require('../tooltip/index.js');
require('../tooltip/Tooltip.js');
require('../_chunks/dep-e41137f7.js');
require('../hooks/useConfig.js');
require('../config-provider/ConfigContext.js');
require('../_chunks/dep-2ee09518.js');
require('../_chunks/dep-a6fa90dc.js');
require('dayjs');
require('../_chunks/dep-012d9a7c.js');
require('../_chunks/dep-150a112c.js');
require('../_chunks/dep-e360e17f.js');
require('../_chunks/dep-51284537.js');
require('../_chunks/dep-8a493834.js');
require('../_chunks/dep-9f5b2891.js');
require('../_chunks/dep-338f013a.js');
require('../_chunks/dep-c37c5e17.js');
require('../_chunks/dep-eee1aab9.js');
require('../_chunks/dep-770bc0ae.js');
require('../_chunks/dep-aaf27e39.js');
require('../_chunks/dep-b47d855f.js');
require('../_chunks/dep-81bce4e6.js');
require('../_chunks/dep-863f8f4d.js');
require('../_chunks/dep-84e4d0da.js');
require('../_chunks/dep-f78183d7.js');
require('../_chunks/dep-2f45f6c4.js');
require('../hooks/useDefaultProps.js');
require('../popup/index.js');
require('../popup/Popup.js');
require('react-transition-group');
require('../common/Portal.js');
require('react-dom');
require('../hooks/useLayoutEffect.js');
require('../hooks/useAnimation.js');
require('../hooks/useAttach.js');
require('../hooks/useControlled.js');
require('../_util/noop.js');
require('../_chunks/dep-45188d54.js');
require('../_chunks/dep-eccb6a94.js');
require('../_chunks/dep-f5e1d11c.js');
require('../_chunks/dep-b199860d.js');
require('../hooks/useMutationObserver.js');
require('../hooks/useLatest.js');
require('../_chunks/dep-aa9a2c54.js');
require('../_chunks/dep-77ef3011.js');
require('../_chunks/dep-b36779a4.js');
require('../_chunks/dep-551d5940.js');
require('../_chunks/dep-8005b808.js');
require('../_chunks/dep-2add8824.js');
require('../hooks/usePopper.js');
require('../_chunks/dep-05bc49e6.js');
require('../_chunks/dep-8c880a65.js');
require('react-fast-compare');
require('@popperjs/core');
require('../hooks/useWindowSize.js');
require('../popup/defaultProps.js');
require('../popup/hooks/useTrigger.js');
require('../_util/listener.js');
require('../_util/ref.js');
require('react-is');
require('../_util/isFragment.js');
require('../popup/utils/transition.js');
require('../popup/PopupPlugin.js');
require('../_chunks/dep-7bf14076.js');
require('../_util/react-render.js');
require('../common/PluginContainer.js');
require('../config-provider/ConfigProvider.js');
require('../_chunks/dep-ba90997f.js');
require('../_chunks/dep-22026fea.js');
require('../tooltip/defaultProps.js');
require('../tooltip/TooltipLite.js');
require('../hooks/useSwitch.js');
require('../_chunks/dep-d07eb641.js');
require('../_chunks/dep-3c628d18.js');
require('../_chunks/dep-ea91dbfd.js');
require('../_chunks/dep-6e0da644.js');
require('../_chunks/dep-b8d0b96c.js');
require('../hooks/useDeepEffect.js');
require('../hooks/usePrevious.js');
require('../_chunks/dep-677950af.js');
require('../_chunks/dep-5fd08885.js');
require('../_chunks/dep-f98232bb.js');
require('../_chunks/dep-7a4e0117.js');
require('../_chunks/dep-5ee57a9e.js');
require('../_chunks/dep-cdef7593.js');
require('../_chunks/dep-3784bbea.js');
require('../_chunks/dep-d4a6a9c1.js');
require('../_chunks/dep-bc160f5b.js');
require('../_chunks/dep-363864ae.js');
require('../_chunks/dep-204188de.js');
require('../_chunks/dep-f8c20fd8.js');
require('./hooks/useMultiHeader.js');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

var React__default = /*#__PURE__*/_interopDefaultLegacy(React);
var classNames__default = /*#__PURE__*/_interopDefaultLegacy(classNames);

function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { defineProperty._defineProperty(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
function THead(props) {
  var columnResizeParams = props.columnResizeParams,
    classPrefix = props.classPrefix,
    showColumnShadow = props.showColumnShadow;
  var theadRef = React.useRef(null);
  var classnames = table_hooks_useClassName["default"]();
  var tableHeaderClasses = classnames.tableHeaderClasses,
    tableBaseClass = classnames.tableBaseClass;
  var theadClasses = [tableHeaderClasses.header, defineProperty._defineProperty(defineProperty._defineProperty(defineProperty._defineProperty({}, tableHeaderClasses.fixed, Boolean(props.maxHeight || props.height)), tableBaseClass.bordered, props.bordered && props.isMultipleHeader), tableHeaderClasses.multipleHeader, props.isMultipleHeader)];
  var colspanSkipMap = React.useMemo(function () {
    var map = {};
    var list = props.thList[0];
    for (var i = 0, len = list.length; i < len; i++) {
      var item = list[i];
      if (item.colspan > 1) {
        for (var j = i + 1; j < i + item.colspan; j++) {
          if (list[j]) {
            map[list[j].colKey] = true;
          }
        }
      }
    }
    return map;
  }, [props.thList]);
  var getTableNode = function getTableNode(thead) {
    var parent = thead;
    while (parent) {
      var _parent;
      parent = parent.parentNode;
      if ((_parent = parent) !== null && _parent !== void 0 && (_parent = _parent.classList) !== null && _parent !== void 0 && _parent.contains("".concat(props.classPrefix, "-table"))) {
        break;
      }
    }
    return parent;
  };
  var renderThNodeList = function renderThNodeList(rowAndColFixedPosition, thWidthList) {
    var thBorderMap = /* @__PURE__ */new Map();
    var thRowspanAndColspan = props.spansAndLeafNodes.rowspanAndColspanMap;
    return props.thList.map(function (row, rowIndex) {
      var thRow = row.map(function (col, index) {
        var _col$children;
        if (colspanSkipMap[col.colKey]) return null;
        var rowspanAndColspan = thRowspanAndColspan.get(col);
        if (index === 0 && rowspanAndColspan.rowspan > 1) {
          for (var j = rowIndex + 1; j < rowIndex + rowspanAndColspan.rowspan; j++) {
            thBorderMap.set(props.thList[j][0], true);
          }
        }
        var thStyles = table_hooks_useFixed.getColumnFixedStyles(col, index, rowAndColFixedPosition, classnames.tableColFixedClasses);
        var colParams = {
          col: col,
          colIndex: index,
          row: {},
          rowIndex: -1
        };
        var isLeftFixedActive = showColumnShadow.left && col.fixed === "left";
        var isRightFixedActive = showColumnShadow.right && col.fixed === "right";
        var canDragSort = props.thDraggable && !(isLeftFixedActive || isRightFixedActive);
        var customClasses = table_utils.formatClassNames(col.className, _objectSpread(_objectSpread({}, colParams), {}, {
          type: "th"
        }));
        var thCustomClasses = table_utils.formatClassNames(col.thClassName, colParams);
        var thClasses = [thStyles.classes, customClasses, thCustomClasses, defineProperty._defineProperty(defineProperty._defineProperty(defineProperty._defineProperty(defineProperty._defineProperty({}, tableHeaderClasses.thBordered, thBorderMap.get(col)), "".concat(classPrefix, "-table__th-").concat(col.colKey), col.colKey), classnames.tdAlignClasses[col.align], col.align && col.align !== "left"), classnames.tableDraggableClasses.dragSortTh, canDragSort)];
        var withoutChildren = !((_col$children = col.children) !== null && _col$children !== void 0 && _col$children.length);
        var width = withoutChildren && thWidthList !== null && thWidthList !== void 0 && thWidthList[col.colKey] ? "".concat(thWidthList === null || thWidthList === void 0 ? void 0 : thWidthList[col.colKey], "px") : void 0;
        var styles = _objectSpread(_objectSpread({}, thStyles.style || {}), {}, {
          width: width
        });
        var innerTh = table_hooks_useTableHeader.renderTitle(col, index);
        if (!col.colKey) return null;
        var resizeColumnListener = props.resizable || !canDragSort ? {
          onMouseDown: function onMouseDown(e) {
            if (props.resizable) {
              var _columnResizeParams$o;
              columnResizeParams === null || columnResizeParams === void 0 || (_columnResizeParams$o = columnResizeParams.onColumnMousedown) === null || _columnResizeParams$o === void 0 || _columnResizeParams$o.call(columnResizeParams, e, col, index);
            }
            if (!canDragSort) {
              var timer = setTimeout(function () {
                var _thList$index;
                var thList = theadRef.current.querySelectorAll("th");
                (_thList$index = thList[index]) === null || _thList$index === void 0 || _thList$index.removeAttribute("draggable");
                clearTimeout(timer);
              }, 10);
            }
          },
          onMouseMove: function onMouseMove(e) {
            var _columnResizeParams$o2;
            props.resizable && (columnResizeParams === null || columnResizeParams === void 0 || (_columnResizeParams$o2 = columnResizeParams.onColumnMouseover) === null || _columnResizeParams$o2 === void 0 ? void 0 : _columnResizeParams$o2.call(columnResizeParams, e, col));
          }
        } : {};
        var content = isFunction.isFunction(col.ellipsisTitle) ? col.ellipsisTitle({
          col: col,
          colIndex: index
        }) : void 0;
        var isEllipsis = col.ellipsisTitle !== void 0 ? Boolean(col.ellipsisTitle) : Boolean(col.ellipsis);
        var attrs = (isFunction.isFunction(col.attrs) ? col.attrs(_objectSpread(_objectSpread({}, colParams), {}, {
          type: "th"
        })) : col.attrs) || {};
        if (col.colspan > 1) {
          attrs.colSpan = col.colspan;
        }
        return /* @__PURE__ */React__default["default"].createElement("th", _objectSpread(_objectSpread(_objectSpread({
          key: col.colKey,
          "data-colkey": col.colKey,
          className: classNames__default["default"](thClasses),
          style: styles
        }, {
          rowSpan: rowspanAndColspan.rowspan,
          colSpan: rowspanAndColspan.colspan
        }), attrs), resizeColumnListener), /* @__PURE__ */React__default["default"].createElement("div", {
          className: tableBaseClass.thCellInner
        }, isEllipsis ? /* @__PURE__ */React__default["default"].createElement(table_Ellipsis["default"], {
          placement: "bottom",
          attach: theadRef.current ? function () {
            return getTableNode(theadRef.current);
          } : void 0,
          popupContent: content,
          tooltipProps: _typeof._typeof(col.ellipsisTitle) === "object" ? col.ellipsisTitle : void 0,
          overlayClassName: props.ellipsisOverlayClassName,
          classPrefix: props.classPrefix
        }, innerTh) : innerTh));
      });
      return /* @__PURE__ */React__default["default"].createElement("tr", {
        key: rowIndex
      }, thRow);
    });
  };
  return /* @__PURE__ */React__default["default"].createElement("thead", {
    ref: theadRef,
    className: classNames__default["default"](theadClasses)
  }, renderThNodeList(props.rowAndColFixedPosition, props.thWidthList));
}

exports["default"] = THead;
