// tdesign-react/cjs/table/TR.js
/**
 * tdesign v1.17.1
 * (c) 2026 tdesign
 * @license MIT
 */

'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var defineProperty = require('../_chunks/dep-c446aa11.js');
var React = require('react');
var classNames = require('classnames');
var table_Cell = require('./Cell.js');
var table_hooks_useClassName = require('./hooks/useClassName.js');
var table_hooks_useFixed = require('./hooks/useFixed.js');
var table_hooks_useLazyLoad = require('./hooks/useLazyLoad.js');
var table_hooks_useRowspanAndColspan = require('./hooks/useRowspanAndColspan.js');
var table_utils = require('./utils.js');
var get = require('../_chunks/dep-f8c20fd8.js');
require('../_chunks/dep-c290284d.js');
require('./Ellipsis.js');
require('../_chunks/dep-c60a3e4c.js');
require('../_chunks/dep-98e6819c.js');
require('../_chunks/dep-7f3fa935.js');
require('../_util/scroll.js');
require('raf');
require('../_chunks/dep-ee71ddca.js');
require('../_chunks/dep-427f8bc0.js');
require('../_chunks/dep-3d05de97.js');
require('../_chunks/dep-a54f202c.js');
require('../_chunks/dep-4f1729f0.js');
require('../_util/easing.js');
require('../hooks/useDebounce.js');
require('../hooks/usePersistFn.js');
require('../_chunks/dep-f2809376.js');
require('../_chunks/dep-a830ce1e.js');
require('../_chunks/dep-3be56e1e.js');
require('../_chunks/dep-36579bf6.js');
require('../tooltip/index.js');
require('../tooltip/Tooltip.js');
require('../_chunks/dep-e41137f7.js');
require('../hooks/useConfig.js');
require('../config-provider/ConfigContext.js');
require('../_chunks/dep-2ee09518.js');
require('../_chunks/dep-a6fa90dc.js');
require('dayjs');
require('../_chunks/dep-012d9a7c.js');
require('../_chunks/dep-150a112c.js');
require('../_chunks/dep-e360e17f.js');
require('../_chunks/dep-51284537.js');
require('../_chunks/dep-8a493834.js');
require('../_chunks/dep-9f5b2891.js');
require('../_chunks/dep-338f013a.js');
require('../_chunks/dep-c37c5e17.js');
require('../_chunks/dep-eee1aab9.js');
require('../_chunks/dep-11347ea1.js');
require('../_chunks/dep-770bc0ae.js');
require('../_chunks/dep-aaf27e39.js');
require('../_chunks/dep-b47d855f.js');
require('../_chunks/dep-81bce4e6.js');
require('../_chunks/dep-863f8f4d.js');
require('../_chunks/dep-84e4d0da.js');
require('../_chunks/dep-f78183d7.js');
require('../_chunks/dep-2f45f6c4.js');
require('../hooks/useDefaultProps.js');
require('../popup/index.js');
require('../popup/Popup.js');
require('react-transition-group');
require('../common/Portal.js');
require('react-dom');
require('../hooks/useLayoutEffect.js');
require('../hooks/useAnimation.js');
require('../hooks/useAttach.js');
require('../hooks/useControlled.js');
require('../_util/noop.js');
require('../_chunks/dep-45188d54.js');
require('../_chunks/dep-eccb6a94.js');
require('../_chunks/dep-f5e1d11c.js');
require('../_chunks/dep-b199860d.js');
require('../hooks/useMutationObserver.js');
require('../hooks/useLatest.js');
require('../_chunks/dep-aa9a2c54.js');
require('../_chunks/dep-77ef3011.js');
require('../_chunks/dep-b36779a4.js');
require('../_chunks/dep-551d5940.js');
require('../_chunks/dep-8005b808.js');
require('../_chunks/dep-2add8824.js');
require('../hooks/usePopper.js');
require('../_chunks/dep-05bc49e6.js');
require('../_chunks/dep-8c880a65.js');
require('react-fast-compare');
require('@popperjs/core');
require('../hooks/useWindowSize.js');
require('../popup/defaultProps.js');
require('../popup/hooks/useTrigger.js');
require('../_util/listener.js');
require('../_util/ref.js');
require('react-is');
require('../_util/isFragment.js');
require('../popup/utils/transition.js');
require('../popup/PopupPlugin.js');
require('../_chunks/dep-7bf14076.js');
require('../_util/react-render.js');
require('../common/PluginContainer.js');
require('../config-provider/ConfigProvider.js');
require('../_chunks/dep-ba90997f.js');
require('../_chunks/dep-22026fea.js');
require('../tooltip/defaultProps.js');
require('../tooltip/TooltipLite.js');
require('../hooks/useSwitch.js');
require('../_chunks/dep-d07eb641.js');
require('../_chunks/dep-3c628d18.js');
require('../_chunks/dep-ea91dbfd.js');
require('../_chunks/dep-6e0da644.js');
require('../_chunks/dep-b8d0b96c.js');
require('../hooks/useDeepEffect.js');
require('../hooks/usePrevious.js');
require('../_chunks/dep-677950af.js');
require('../_chunks/dep-5fd08885.js');
require('../_chunks/dep-f98232bb.js');
require('../_chunks/dep-7a4e0117.js');
require('../_chunks/dep-5ee57a9e.js');
require('../_chunks/dep-cdef7593.js');
require('../_chunks/dep-3784bbea.js');
require('../_chunks/dep-d4a6a9c1.js');
require('../_chunks/dep-bc160f5b.js');
require('../_chunks/dep-363864ae.js');
require('../_chunks/dep-204188de.js');
require('../_chunks/dep-2bc69864.js');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

var React__default = /*#__PURE__*/_interopDefaultLegacy(React);
var classNames__default = /*#__PURE__*/_interopDefaultLegacy(classNames);

function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { defineProperty._defineProperty(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
var TABLE_PROPS = ["rowKey", "rowClassName", "columns", "fixedRows", "footData", "rowAttributes", "rowspanAndColspan", "scroll", "cellEmptyContent", "onCellClick", "onRowClick", "onRowDblclick", "onRowMouseover", "onRowMousedown", "onRowMouseenter", "onRowMouseleave", "onRowMouseup"];
var ROW_LISTENERS = ["click", "dblclick", "mouseover", "mousedown", "mouseenter", "mouseleave", "mouseup"];
function TR(props) {
  var _props$columns;
  var row = props.row,
    rowKey = props.rowKey,
    rowIndex = props.rowIndex,
    rowClassName = props.rowClassName,
    rowAttributes = props.rowAttributes,
    dataLength = props.dataLength,
    fixedRows = props.fixedRows,
    scroll = props.scroll,
    tableContentRef = props.tableContentRef,
    rowAndColFixedPosition = props.rowAndColFixedPosition,
    virtualConfig = props.virtualConfig,
    onRowMounted = props.onRowMounted;
  var trRef = React.useRef(null);
  var classNames = table_hooks_useClassName["default"]();
  var trStyles = table_hooks_useFixed.getRowFixedStyles(get.get(row, rowKey || "id"), rowIndex, dataLength, fixedRows, rowAndColFixedPosition, classNames.tableRowFixedClasses, virtualConfig.isVirtualScroll ? virtualConfig.translateY : 0);
  var trAttributes = React.useMemo(function () {
    return table_utils.formatRowAttributes(rowAttributes, {
      row: row,
      rowIndex: rowIndex,
      type: "body"
    }) || {};
  }, [row, rowAttributes, rowIndex]);
  var classes = React.useMemo(function () {
    var customClasses = table_utils.formatRowClassNames(rowClassName, {
      row: row,
      rowIndex: rowIndex,
      rowKey: rowKey,
      type: "body"
    }, rowKey || "id");
    return [trStyles === null || trStyles === void 0 ? void 0 : trStyles.classes, customClasses];
  }, [row, rowClassName, rowIndex, rowKey, trStyles === null || trStyles === void 0 ? void 0 : trStyles.classes]);
  var useLazyLoadParams = React.useMemo(function () {
    return _objectSpread(_objectSpread({}, scroll), {}, {
      rowIndex: rowIndex
    });
  }, [scroll, rowIndex]);
  var _useLazyLoad = table_hooks_useLazyLoad["default"](tableContentRef.current, trRef, useLazyLoadParams),
    hasLazyLoadHolder = _useLazyLoad.hasLazyLoadHolder,
    tRowHeight = _useLazyLoad.tRowHeight;
  React.useEffect(function () {
    if (virtualConfig.isVirtualScroll && trRef.current) {
      onRowMounted === null || onRowMounted === void 0 || onRowMounted({
        ref: trRef.current,
        data: row
      });
    }
  }, [virtualConfig.isVirtualScroll, trRef, row]);
  var columnVNodeList = (_props$columns = props.columns) === null || _props$columns === void 0 ? void 0 : _props$columns.map(function (col, colIndex) {
    var _props$skipSpansMap;
    var cellSpans = {};
    var params = {
      row: row,
      col: col,
      rowIndex: rowIndex,
      colIndex: colIndex
    };
    var spanState = null;
    if ((_props$skipSpansMap = props.skipSpansMap) !== null && _props$skipSpansMap !== void 0 && _props$skipSpansMap.size) {
      var _spanState, _spanState2;
      var cellKey = table_hooks_useRowspanAndColspan.getCellKey(row, props.rowKey, col.colKey, colIndex);
      spanState = props.skipSpansMap.get(cellKey) || {};
      ((_spanState = spanState) === null || _spanState === void 0 ? void 0 : _spanState.rowspan) > 1 && (cellSpans.rowspan = spanState.rowspan);
      ((_spanState2 = spanState) === null || _spanState2 === void 0 ? void 0 : _spanState2.colspan) > 1 && (cellSpans.colspan = spanState.colspan);
      if (spanState.skipped) return null;
    }
    var onClick = function onClick(e) {
      var _props$onCellClick;
      var p = _objectSpread(_objectSpread({}, params), {}, {
        e: e
      });
      if (col.stopPropagation) {
        e.stopPropagation();
      }
      (_props$onCellClick = props.onCellClick) === null || _props$onCellClick === void 0 || _props$onCellClick.call(props, p);
    };
    return /* @__PURE__ */React__default["default"].createElement(table_Cell["default"], {
      key: params.col.colKey,
      cellParams: params,
      dataLength: dataLength,
      rowAndColFixedPosition: rowAndColFixedPosition,
      columnLength: props.columns.length,
      cellSpans: cellSpans,
      cellEmptyContent: props.cellEmptyContent,
      tableClassNames: classNames,
      rowspanAndColspan: props.rowspanAndColspan,
      onClick: onClick,
      tableRef: props.tableRef,
      classPrefix: props.classPrefix,
      overlayClassName: props.ellipsisOverlayClassName,
      pagination: props.pagination
    });
  });
  var rowParams = {
    row: row,
    index: rowIndex
  };
  return /* @__PURE__ */React__default["default"].createElement("tr", _objectSpread(_objectSpread({
    ref: trRef
  }, trAttributes), {}, {
    style: trStyles === null || trStyles === void 0 ? void 0 : trStyles.style,
    className: classNames__default["default"](classes) || void 0,
    onClick: function onClick(e) {
      var _props$onRowClick;
      return (_props$onRowClick = props.onRowClick) === null || _props$onRowClick === void 0 ? void 0 : _props$onRowClick.call(props, _objectSpread(_objectSpread({}, rowParams), {}, {
        e: e
      }));
    },
    onDoubleClick: function onDoubleClick(e) {
      var _props$onRowDblclick;
      return (_props$onRowDblclick = props.onRowDblclick) === null || _props$onRowDblclick === void 0 ? void 0 : _props$onRowDblclick.call(props, _objectSpread(_objectSpread({}, rowParams), {}, {
        e: e
      }));
    },
    onMouseDown: function onMouseDown(e) {
      var _props$onRowMousedown;
      return (_props$onRowMousedown = props.onRowMousedown) === null || _props$onRowMousedown === void 0 ? void 0 : _props$onRowMousedown.call(props, _objectSpread(_objectSpread({}, rowParams), {}, {
        e: e
      }));
    },
    onMouseEnter: function onMouseEnter(e) {
      var _props$onRowMouseente;
      return (_props$onRowMouseente = props.onRowMouseenter) === null || _props$onRowMouseente === void 0 ? void 0 : _props$onRowMouseente.call(props, _objectSpread(_objectSpread({}, rowParams), {}, {
        e: e
      }));
    },
    onMouseLeave: function onMouseLeave(e) {
      var _props$onRowMouseleav;
      return (_props$onRowMouseleav = props.onRowMouseleave) === null || _props$onRowMouseleav === void 0 ? void 0 : _props$onRowMouseleav.call(props, _objectSpread(_objectSpread({}, rowParams), {}, {
        e: e
      }));
    },
    onMouseUp: function onMouseUp(e) {
      var _props$onRowMouseup;
      return (_props$onRowMouseup = props.onRowMouseup) === null || _props$onRowMouseup === void 0 ? void 0 : _props$onRowMouseup.call(props, _objectSpread(_objectSpread({}, rowParams), {}, {
        e: e
      }));
    },
    onMouseOver: function onMouseOver(e) {
      var _props$onRowMouseover;
      return (_props$onRowMouseover = props.onRowMouseover) === null || _props$onRowMouseover === void 0 ? void 0 : _props$onRowMouseover.call(props, _objectSpread(_objectSpread({}, rowParams), {}, {
        e: e
      }));
    }
  }), hasLazyLoadHolder ? [/* @__PURE__ */React__default["default"].createElement("td", {
    key: rowIndex,
    style: {
      height: "".concat(tRowHeight, "px"),
      border: "none"
    }
  })] : columnVNodeList);
}
TR.displayName = "TR";

exports.ROW_LISTENERS = ROW_LISTENERS;
exports.TABLE_PROPS = TABLE_PROPS;
exports["default"] = TR;
