// tdesign-react/cjs/transfer/TransferList.js
/**
 * tdesign v1.17.1
 * (c) 2026 tdesign
 * @license MIT
 */

'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var defineProperty = require('../_chunks/dep-c446aa11.js');
var slicedToArray = require('../_chunks/dep-c60a3e4c.js');
var React = require('react');
var classNames = require('classnames');
var tdesignIconsReact = require('tdesign-icons-react');
var _util_parseTNode = require('../_util/parseTNode.js');
var checkbox_index = require('../checkbox/index.js');
var hooks_useConfig = require('../hooks/useConfig.js');
var hooks_useGlobalIcon = require('../hooks/useGlobalIcon.js');
var input_index = require('../input/index.js');
var locale_LocalReceiver = require('../locale/LocalReceiver.js');
var pagination_index = require('../pagination/index.js');
var transfer_utils = require('./utils.js');
var isEmpty = require('../_chunks/dep-2be6b7d2.js');
var isFunction = require('../_chunks/dep-11347ea1.js');
var isString = require('../_chunks/dep-427f8bc0.js');
require('../_chunks/dep-c290284d.js');
require('../_chunks/dep-98e6819c.js');
require('../_chunks/dep-7f3fa935.js');
require('../_chunks/dep-3c628d18.js');
require('../_chunks/dep-3d05de97.js');
require('../_chunks/dep-a830ce1e.js');
require('../_chunks/dep-ab5baf4d.js');
require('../_util/forwardRefWithStatics.js');
require('hoist-non-react-statics');
require('../common/Check.js');
require('../_chunks/dep-e41137f7.js');
require('../_util/helper.js');
require('../_chunks/dep-b422817e.js');
require('../_chunks/dep-b199860d.js');
require('../_chunks/dep-a54f202c.js');
require('../_chunks/dep-36579bf6.js');
require('../_chunks/dep-4f1729f0.js');
require('../_chunks/dep-45188d54.js');
require('../_chunks/dep-eccb6a94.js');
require('../_chunks/dep-f5e1d11c.js');
require('../hooks/useControlled.js');
require('../_util/noop.js');
require('../_chunks/dep-fb263e64.js');
require('../config-provider/ConfigContext.js');
require('../_chunks/dep-2ee09518.js');
require('../_chunks/dep-a6fa90dc.js');
require('dayjs');
require('../_chunks/dep-012d9a7c.js');
require('../_chunks/dep-150a112c.js');
require('../_chunks/dep-e360e17f.js');
require('../_chunks/dep-51284537.js');
require('../_chunks/dep-8a493834.js');
require('../_chunks/dep-9f5b2891.js');
require('../_chunks/dep-338f013a.js');
require('../_chunks/dep-c37c5e17.js');
require('../_chunks/dep-eee1aab9.js');
require('../_chunks/dep-770bc0ae.js');
require('../_chunks/dep-aaf27e39.js');
require('../_chunks/dep-b47d855f.js');
require('../_chunks/dep-81bce4e6.js');
require('../_chunks/dep-863f8f4d.js');
require('../_chunks/dep-84e4d0da.js');
require('../_chunks/dep-f78183d7.js');
require('../_chunks/dep-2f45f6c4.js');
require('../hooks/useDefaultProps.js');
require('../_chunks/dep-05bc49e6.js');
require('../_chunks/dep-8c880a65.js');
require('../checkbox/defaultProps.js');
require('../_chunks/dep-b8d0b96c.js');
require('../input/Input.js');
require('../hooks/useLayoutEffect.js');
require('../_chunks/dep-ee71ddca.js');
require('../input/defaultProps.js');
require('../input/InputGroup.js');
require('../input/useLengthLimit.js');
require('../_chunks/dep-6e0da644.js');
require('../config-provider/ConfigProvider.js');
require('../_chunks/dep-ba90997f.js');
require('../_chunks/dep-22026fea.js');
require('../_chunks/dep-551d5940.js');
require('../_chunks/dep-8005b808.js');
require('../_chunks/dep-2add8824.js');
require('../pagination/Pagination.js');
require('../input-adornment/InputAdornment.js');
require('../input-number/index.js');
require('../input-number/InputNumber.js');
require('../button/index.js');
require('../button/Button.js');
require('../hooks/useDomRefCallback.js');
require('../hooks/useRipple.js');
require('../_chunks/dep-062bb1b1.js');
require('../hooks/useAnimation.js');
require('../loading/index.js');
require('../loading/Loading.js');
require('../common/Portal.js');
require('react-dom');
require('../loading/defaultProps.js');
require('../loading/gradient.js');
require('../loading/plugin.js');
require('../_util/react-render.js');
require('../_chunks/dep-7bf14076.js');
require('../common/PluginContainer.js');
require('../button/defaultProps.js');
require('../input-number/defaultProps.js');
require('../input-number/useInputNumber.js');
require('../_chunks/dep-f7c108a2.js');
require('../_chunks/dep-bfa39549.js');
require('../hooks/useCommonClassName.js');
require('../select/index.js');
require('../select/base/Select.js');
require('../_util/ref.js');
require('react-is');
require('../_util/isFragment.js');
require('../common/FakeArrow.js');
require('../select-input/index.js');
require('../select-input/SelectInput.js');
require('../popup/index.js');
require('../popup/Popup.js');
require('react-transition-group');
require('../hooks/useAttach.js');
require('../hooks/useMutationObserver.js');
require('../hooks/useLatest.js');
require('../_chunks/dep-aa9a2c54.js');
require('../_chunks/dep-77ef3011.js');
require('../_chunks/dep-b36779a4.js');
require('../_chunks/dep-f2809376.js');
require('../_chunks/dep-3be56e1e.js');
require('../hooks/usePopper.js');
require('react-fast-compare');
require('@popperjs/core');
require('../hooks/useWindowSize.js');
require('../popup/defaultProps.js');
require('../popup/hooks/useTrigger.js');
require('../_util/listener.js');
require('../popup/utils/transition.js');
require('../popup/PopupPlugin.js');
require('../select-input/defaultProps.js');
require('../select-input/useMultiple.js');
require('../tag-input/index.js');
require('../tag-input/TagInput.js');
require('../hooks/useDragSorter.js');
require('../hooks/useEventCallback.js');
require('../tag-input/defaultProps.js');
require('../tag-input/useHover.js');
require('../tag-input/useTagList.js');
require('../tag/index.js');
require('../tag/CheckTag.js');
require('../_chunks/dep-29222115.js');
require('../tag/defaultProps.js');
require('../tag/Tag.js');
require('../_chunks/dep-183643e1.js');
require('../tag/CheckTagGroup.js');
require('../tag-input/useTagScroll.js');
require('../select-input/useOverlayInnerStyle.js');
require('../hooks/useInnerPopupVisible.js');
require('../select-input/useSingle.js');
require('../_chunks/dep-3784bbea.js');
require('../_chunks/dep-d4a6a9c1.js');
require('../_chunks/dep-bc160f5b.js');
require('../_chunks/dep-363864ae.js');
require('../_chunks/dep-204188de.js');
require('../_chunks/dep-7a4e0117.js');
require('../select/defaultProps.js');
require('../select/hooks/useKeyboardControl.js');
require('../_chunks/dep-33de9795.js');
require('../select/base/OptionGroup.js');
require('../_chunks/dep-f8c20fd8.js');
require('../select/base/PopupContent.js');
require('../select/hooks/usePanelVirtualScroll.js');
require('../hooks/useVirtualScroll.js');
require('../pagination/defaultProps.js');
require('../pagination/hooks/useBoundaryJumper.js');
require('../pagination/hooks/usePageNumber.js');
require('../pagination/hooks/usePrevNextJumper.js');
require('../pagination/hooks/useTotal.js');
require('../pagination/validators.js');
require('../_chunks/dep-8de6ca9d.js');
require('../_chunks/dep-6e5e0470.js');
require('../_chunks/dep-cdef7593.js');
require('../pagination/PaginationMini.js');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

var React__default = /*#__PURE__*/_interopDefaultLegacy(React);
var classNames__default = /*#__PURE__*/_interopDefaultLegacy(classNames);

var TransferList = function TransferList(props) {
  var className = props.className,
    style = props.style,
    data = props.data,
    _props$search = props.search,
    search = _props$search === void 0 ? false : _props$search,
    _props$checked = props.checked,
    checked = _props$checked === void 0 ? [] : _props$checked,
    empty = props.empty,
    title = props.title,
    footer = props.footer,
    content = props.content,
    onCheckbox = props.onCheckbox,
    onSearch = props.onSearch,
    _props$disabled = props.disabled,
    disabled = _props$disabled === void 0 ? false : _props$disabled,
    pagination = props.pagination,
    transferItem = props.transferItem,
    treeNode = props.tree,
    showCheckAll = props.showCheckAll,
    listType = props.listType;
  var notDisabledData = !treeNode ? data.filter(function (item) {
    return !item.disabled;
  }) : transfer_utils.getLeafNodes(data).filter(function (item) {
    return !item.disabled;
  });
  var _useState = React.useState(1),
    _useState2 = slicedToArray._slicedToArray(_useState, 2),
    currentPage = _useState2[0],
    setCurrentPage = _useState2[1];
  var _useMemo = React.useMemo(function () {
      var activeLen = checked.length;
      var notLen = notDisabledData.length;
      return [activeLen && activeLen < notLen, activeLen && activeLen === notLen];
    }, [checked, notDisabledData]),
    _useMemo2 = slicedToArray._slicedToArray(_useMemo, 2),
    indeterminate = _useMemo2[0],
    allChecked = _useMemo2[1];
  var _useConfig = hooks_useConfig["default"](),
    classPrefix = _useConfig.classPrefix;
  var _useGlobalIcon = hooks_useGlobalIcon["default"]({
      SearchIcon: tdesignIconsReact.SearchIcon
    }),
    SearchIcon = _useGlobalIcon.SearchIcon;
  var CLASSPREFIX = "".concat(classPrefix, "-transfer__list");
  var _useLocaleReceiver = locale_LocalReceiver.useLocaleReceiver("transfer"),
    _useLocaleReceiver2 = slicedToArray._slicedToArray(_useLocaleReceiver, 2),
    local = _useLocaleReceiver2[0],
    t = _useLocaleReceiver2[1];
  var handleCheckbox = function handleCheckbox(vals) {
    if (isFunction.isFunction(onCheckbox)) onCheckbox(vals);
  };
  var handleAllCheckbox = function handleAllCheckbox(checked2) {
    if (isFunction.isFunction(onCheckbox)) onCheckbox(checked2 ? notDisabledData.map(function (item) {
      return item.value;
    }) : []);
  };
  var HeaderCmp = function HeaderCmp() {
    var total = treeNode ? transfer_utils.getLeafNodes(data).length : data.length;
    return /* @__PURE__ */React__default["default"].createElement("div", {
      className: "".concat(CLASSPREFIX, "-header")
    }, /* @__PURE__ */React__default["default"].createElement("div", null, showCheckAll ? /* @__PURE__ */React__default["default"].createElement(checkbox_index.Checkbox, {
      indeterminate: indeterminate,
      checked: allChecked,
      disabled: disabled,
      onChange: handleAllCheckbox
    }) : null, /* @__PURE__ */React__default["default"].createElement("span", null, t(local.title, {
      checked: checked.length,
      total: total
    }))), /* @__PURE__ */React__default["default"].createElement("span", null, title));
  };
  var SearchCmp = function SearchCmp() {
    return search ? /* @__PURE__ */React__default["default"].createElement("div", {
      className: "".concat(classPrefix, "-transfer__search-wrapper")
    }, /* @__PURE__ */React__default["default"].createElement(input_index.Input, {
      placeholder: local.placeholder,
      suffixIcon: /* @__PURE__ */React__default["default"].createElement(SearchIcon, null),
      onChange: onSearch
    })) : null;
  };
  var viewData = React.useMemo(function () {
    if (!isEmpty.isEmpty(pagination)) {
      var pageSize = pagination.pageSize || 10;
      var start = (currentPage - 1) * pageSize;
      return data.slice(start, start + pageSize);
    }
    return data;
  }, [currentPage, data, pagination]);
  var EmptyCmp = function EmptyCmp() {
    return isString.isString(empty) ? /* @__PURE__ */React__default["default"].createElement("div", {
      className: "".concat(classPrefix, "-transfer__empty")
    }, /* @__PURE__ */React__default["default"].createElement("span", null, empty || "\u6682\u65E0\u6570\u636E")) : empty;
  };
  var contentCmp = function contentCmp() {
    if (typeof treeNode === "function") {
      return treeNode({
        data: viewData,
        value: checked,
        onChange: handleCheckbox
      });
    }
    if (typeof content === "function") {
      return content({
        data: viewData
      });
    }
    return /* @__PURE__ */React__default["default"].createElement(checkbox_index.Checkbox.Group, {
      value: checked,
      onChange: handleCheckbox,
      disabled: disabled
    }, viewData.map(function (item, index) {
      return /* @__PURE__ */React__default["default"].createElement(checkbox_index.Checkbox, {
        key: item.value,
        value: item.value,
        disabled: item.disabled,
        className: "".concat(CLASSPREFIX, "-item")
      }, /* @__PURE__ */React__default["default"].createElement("span", null, transferItem ? _util_parseTNode.parseContentTNode(transferItem, {
        data: item,
        index: index,
        type: listType
      }) : item.label));
    }));
  };
  var BodyCmp = function BodyCmp() {
    return /* @__PURE__ */React__default["default"].createElement("div", {
      className: classNames__default["default"]("".concat(CLASSPREFIX, "-body"), defineProperty._defineProperty({}, "".concat(CLASSPREFIX, "--with-search"), search))
    }, SearchCmp(), /* @__PURE__ */React__default["default"].createElement("div", {
      className: "".concat(CLASSPREFIX, "-content narrow-scrollbar")
    }, viewData.length ? contentCmp() : EmptyCmp()));
  };
  var PaginationCmp = function PaginationCmp() {
    var handleCurrentPagination = function handleCurrentPagination(current) {
      setCurrentPage(current);
      if (isFunction.isFunction(pagination.onPageChange)) {
        pagination.onPageChange(current);
      }
    };
    return pagination ? /* @__PURE__ */React__default["default"].createElement("div", {
      className: "".concat(CLASSPREFIX, "-pagination")
    }, /* @__PURE__ */React__default["default"].createElement(pagination_index.Pagination, {
      size: "small",
      theme: "simple",
      pageSizeOptions: [],
      total: data.length,
      totalContent: false,
      onCurrentChange: handleCurrentPagination,
      pageSize: pagination.pageSize
    })) : null;
  };
  var footerCmp = function footerCmp() {
    return !isEmpty.isEmpty(footer) ? /* @__PURE__ */React__default["default"].createElement("div", {
      className: "".concat(classPrefix, "-transfer__footer")
    }, footer) : null;
  };
  return /* @__PURE__ */React__default["default"].createElement("div", {
    style: style,
    className: classNames__default["default"](CLASSPREFIX, className)
  }, HeaderCmp(), BodyCmp(), PaginationCmp(), footerCmp());
};

exports["default"] = TransferList;
