// tdesign-react/cjs/transfer/utils.js
/**
 * tdesign v1.17.1
 * (c) 2026 tdesign
 * @license MIT
 */

'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var defineProperty = require('../_chunks/dep-c446aa11.js');
var React = require('react');
var isEmpty = require('../_chunks/dep-2be6b7d2.js');
var isString = require('../_chunks/dep-427f8bc0.js');
var isFunction = require('../_chunks/dep-11347ea1.js');
require('../_chunks/dep-c290284d.js');
require('../_chunks/dep-2add8824.js');
require('../_chunks/dep-51284537.js');
require('../_chunks/dep-3d05de97.js');
require('../_chunks/dep-8a493834.js');
require('../_chunks/dep-4f1729f0.js');
require('../_chunks/dep-9f5b2891.js');
require('../_chunks/dep-eee1aab9.js');
require('../_chunks/dep-a830ce1e.js');
require('../_chunks/dep-e360e17f.js');
require('../_chunks/dep-a54f202c.js');
require('../_chunks/dep-863f8f4d.js');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

var React__default = /*#__PURE__*/_interopDefaultLegacy(React);

function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { defineProperty._defineProperty(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
var getDefaultValue = function getDefaultValue(value, defaultValue) {
  if (isEmpty.isEmpty(value)) {
    return Array.isArray(defaultValue) ? defaultValue : [defaultValue, defaultValue];
  }
  if (Array.isArray(value)) return value;
  return [defaultValue, defaultValue];
};
var getJSX = function getJSX(value) {
  if (isString.isString(value)) return value;
  if (isFunction.isFunction(value)) return value();
  if (/*#__PURE__*/React__default["default"].isValidElement(value)) return value;
  return value;
};
var getLeafNodes = function getLeafNodes(nodes) {
  var resData = [];
  var _map = function map(nodes2) {
    nodes2.forEach(function (child) {
      if (child.children && child.children.length > 0) {
        return _map(child.children);
      }
      resData.push(child);
    });
  };
  _map(nodes);
  return resData;
};
var _filterCheckedTreeNodes = function filterCheckedTreeNodes(nodes, checkeds) {
  return nodes.filter(function (item) {
    return !checkeds.includes(item.value);
  }).map(function (item) {
    var newItem = _objectSpread({}, item);
    if (item.children && item.children.length > 0) {
      newItem.children = _filterCheckedTreeNodes(item.children, checkeds);
    }
    return newItem;
  }).filter(function (item) {
    return item.children && item.children.length !== 0 || !item.children;
  });
};
var getTargetNodes = function getTargetNodes(sourceNodes, data) {
  var source = getLeafNodes(sourceNodes).map(function (item) {
    return item.value;
  });
  return _filterCheckedTreeNodes(data, source);
};

exports.filterCheckedTreeNodes = _filterCheckedTreeNodes;
exports.getDefaultValue = getDefaultValue;
exports.getJSX = getJSX;
exports.getLeafNodes = getLeafNodes;
exports.getTargetNodes = getTargetNodes;
