// tdesign-react/es/select/util/helper.js
/**
 * tdesign v1.17.1
 * (c) 2026 tdesign
 * @license MIT
 */

import '../../_chunks/dep-154fcb93.js';
import '../../_chunks/dep-55b77e26.js';
import 'lodash-es';
export { g as getKeyMapping, d as getLabel, e as getMultipleTags, b as getSelectValueArr, a as getSelectedOptions, c as getValueToOption, s as setValueToOptionFormOptionDom } from '../../_chunks/dep-c2de18fc.js';
import '../base/OptionGroup.js';
import '../../_chunks/dep-570a5adb.js';
import '../../_chunks/dep-e9c0bb63.js';
import '../../_chunks/dep-067cb22b.js';
import '../../_chunks/dep-d1eeb95c.js';
import '../../_chunks/dep-673e8854.js';
import 'react';
import 'classnames';
import '../../hooks/useConfig.js';
import '../../config-provider/ConfigContext.js';
import '../../locale/zh_CN.js';
import '../../_chunks/dep-e56ffacc.js';
import 'dayjs';
import '../../_chunks/dep-84036ea0.js';
import '../../hooks/useDomRefCallback.js';
import '../../hooks/useRipple.js';
import '../../_chunks/dep-e17ec8df.js';
import '../../_chunks/dep-8c226881.js';
import '../../hooks/useAnimation.js';
import '../../hooks/useDefaultProps.js';
import '../defaultProps.js';
