// tdesign-react/lib/common/FakeArrow.js
/**
 * tdesign v1.17.1
 * (c) 2026 tdesign
 * @license MIT
 */

import { _ as _defineProperty } from '../_chunks/dep-55b77e26.js';
import React from 'react';
import classNames from 'classnames';
import { i as isSafari } from '../_chunks/dep-b0cbea79.js';
import useConfig from '../hooks/useConfig.js';
import '../_chunks/dep-067cb22b.js';
import '../_chunks/dep-154fcb93.js';
import '../_chunks/dep-570a5adb.js';
import '../_chunks/dep-e9c0bb63.js';
import '../_chunks/dep-84926167.js';
import 'lodash-es';
import '../config-provider/ConfigContext.js';
import '../_chunks/dep-c3e5e988.js';
import '../_chunks/dep-65b05485.js';
import 'dayjs';

function FakeArrow(props) {
  var _useConfig = useConfig(),
    classPrefix = _useConfig.classPrefix;
  return /* @__PURE__ */React.createElement("svg", {
    style: props.style,
    className: classNames("".concat(classPrefix, "-fake-arrow"), _defineProperty(_defineProperty({}, "".concat(classPrefix, "-fake-arrow--transform"), isSafari()), "".concat(classPrefix, "-fake-arrow--active"), (props === null || props === void 0 ? void 0 : props.isActive) && !(props !== null && props !== void 0 && props.disabled)), props === null || props === void 0 ? void 0 : props.className),
    width: "16",
    height: "16",
    viewBox: "0 0 16 16",
    fill: "none",
    xmlns: "http://www.w3.org/2000/svg"
  }, /* @__PURE__ */React.createElement("path", {
    d: "M3.75 5.7998L7.99274 10.0425L12.2361 5.79921",
    stroke: "black",
    strokeOpacity: "0.9",
    strokeWidth: "1.3"
  }));
}

export { FakeArrow as default };
