// tdesign-react/es/tree/hooks/useControllable.js
/**
 * tdesign v1.17.1
 * (c) 2026 tdesign
 * @license MIT
 */

import { _ as _slicedToArray } from '../../_chunks/dep-d1eeb95c.js';
import { useState } from 'react';
import useControlled from '../../hooks/useControlled.js';
import '../../_chunks/dep-673e8854.js';
import '../../_chunks/dep-570a5adb.js';
import 'lodash-es';
import '../../_util/noop.js';

function useControllable(props) {
  var _useControlled = useControlled(props, "value", props.onChange),
    _useControlled2 = _slicedToArray(_useControlled, 2),
    value = _useControlled2[0],
    onChange = _useControlled2[1];
  var _useControlled3 = useControlled(props, "expanded", props.onExpand),
    _useControlled4 = _slicedToArray(_useControlled3, 2),
    expanded = _useControlled4[0],
    onExpand = _useControlled4[1];
  var _useControlled5 = useControlled(props, "actived", props.onActive),
    _useControlled6 = _slicedToArray(_useControlled5, 2),
    actived = _useControlled6[0],
    onActive = _useControlled6[1];
  var _useState = useState([]),
    _useState2 = _slicedToArray(_useState, 2),
    indeterminate = _useState2[0],
    setTreeIndeterminate = _useState2[1];
  return {
    value: value,
    onChange: onChange,
    expanded: expanded,
    onExpand: onExpand,
    actived: actived,
    onActive: onActive,
    setTreeIndeterminate: setTreeIndeterminate,
    indeterminate: indeterminate
  };
}

export { useControllable as default };
