// tdesign-react/cjs/tag/CheckTag.js
/**
 * tdesign v1.17.1
 * (c) 2026 tdesign
 * @license MIT
 */

'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var defineProperty = require('../_chunks/dep-c446aa11.js');
var slicedToArray = require('../_chunks/dep-c60a3e4c.js');
var objectWithoutProperties = require('../_chunks/dep-e41137f7.js');
var React = require('react');
var classNames = require('classnames');
var common = require('../_chunks/dep-29222115.js');
var _util_noop = require('../_util/noop.js');
var hooks_useConfig = require('../hooks/useConfig.js');
var hooks_useControlled = require('../hooks/useControlled.js');
var hooks_useDefaultProps = require('../hooks/useDefaultProps.js');
var tag_defaultProps = require('./defaultProps.js');
var tag_Tag = require('./Tag.js');
require('../_chunks/dep-c290284d.js');
require('../_chunks/dep-98e6819c.js');
require('../_chunks/dep-7f3fa935.js');
require('../config-provider/ConfigContext.js');
require('../_chunks/dep-2ee09518.js');
require('../_chunks/dep-a6fa90dc.js');
require('dayjs');
require('../_chunks/dep-012d9a7c.js');
require('../_chunks/dep-150a112c.js');
require('../_chunks/dep-e360e17f.js');
require('../_chunks/dep-3d05de97.js');
require('../_chunks/dep-4f1729f0.js');
require('../_chunks/dep-a54f202c.js');
require('../_chunks/dep-51284537.js');
require('../_chunks/dep-8a493834.js');
require('../_chunks/dep-9f5b2891.js');
require('../_chunks/dep-338f013a.js');
require('../_chunks/dep-c37c5e17.js');
require('../_chunks/dep-eee1aab9.js');
require('../_chunks/dep-11347ea1.js');
require('../_chunks/dep-a830ce1e.js');
require('../_chunks/dep-770bc0ae.js');
require('../_chunks/dep-aaf27e39.js');
require('../_chunks/dep-b47d855f.js');
require('../_chunks/dep-81bce4e6.js');
require('../_chunks/dep-863f8f4d.js');
require('../_chunks/dep-84e4d0da.js');
require('../_chunks/dep-f78183d7.js');
require('../_chunks/dep-2f45f6c4.js');
require('../_chunks/dep-45188d54.js');
require('../_chunks/dep-eccb6a94.js');
require('../_chunks/dep-f5e1d11c.js');
require('../_chunks/dep-b199860d.js');
require('../_chunks/dep-36579bf6.js');
require('tdesign-icons-react');
require('../_chunks/dep-183643e1.js');
require('../hooks/useGlobalIcon.js');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

var React__default = /*#__PURE__*/_interopDefaultLegacy(React);
var classNames__default = /*#__PURE__*/_interopDefaultLegacy(classNames);

var _excluded = ["value", "content", "onClick", "disabled", "children", "size", "checkedProps", "uncheckedProps", "onChange", "className"];
function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { defineProperty._defineProperty(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
var CheckTag = /*#__PURE__*/React.forwardRef(function (originalProps, ref) {
  var props = hooks_useDefaultProps["default"](originalProps, tag_defaultProps.checkTagDefaultProps);
  var value = props.value,
    content = props.content,
    _props$onClick = props.onClick,
    onClick = _props$onClick === void 0 ? _util_noop["default"] : _props$onClick,
    disabled = props.disabled,
    children = props.children,
    size = props.size,
    checkedProps = props.checkedProps,
    uncheckedProps = props.uncheckedProps,
    onChange = props.onChange,
    className = props.className,
    tagOtherProps = objectWithoutProperties._objectWithoutProperties(props, _excluded);
  var _useControlled = hooks_useControlled["default"](props, "checked", onChange),
    _useControlled2 = slicedToArray._slicedToArray(_useControlled, 2),
    innerChecked = _useControlled2[0],
    setInnerChecked = _useControlled2[1];
  var _useConfig = hooks_useConfig["default"](),
    classPrefix = _useConfig.classPrefix;
  var tagClassPrefix = "".concat(classPrefix, "-tag");
  var tagClass = React.useMemo(function () {
    return ["".concat(tagClassPrefix), "".concat(tagClassPrefix, "--check"), defineProperty._defineProperty(defineProperty._defineProperty(defineProperty._defineProperty(defineProperty._defineProperty({}, "".concat(tagClassPrefix, "--checked"), innerChecked), "".concat(tagClassPrefix, "--disabled"), disabled), "".concat(classPrefix, "-size-s"), size === "small"), "".concat(classPrefix, "-size-l"), size === "large"), className];
  }, [innerChecked, disabled, classPrefix, tagClassPrefix, size, className]);
  var checkTagProps = React.useMemo(function () {
    var tmpCheckedProps = _objectSpread({
      theme: "primary"
    }, checkedProps);
    var tmpUncheckedProps = _objectSpread({}, uncheckedProps);
    return innerChecked ? tmpCheckedProps : tmpUncheckedProps;
  }, [innerChecked, checkedProps, uncheckedProps]);
  var handleClick = function handleClick(_ref2) {
    var e = _ref2.e;
    if (!disabled) {
      onClick === null || onClick === void 0 || onClick({
        e: e
      });
      setInnerChecked(!innerChecked, {
        e: e,
        value: value
      });
    }
  };
  var keyboardEventListener = function keyboardEventListener(e) {
    var _e$key;
    var code = e.code || ((_e$key = e.key) === null || _e$key === void 0 ? void 0 : _e$key.trim());
    var isCheckedCode = common.SPACE_REG.test(code) || common.ENTER_REG.test(code);
    if (isCheckedCode) {
      e.preventDefault();
      setInnerChecked(!innerChecked, {
        e: e,
        value: value
      });
    }
  };
  var onCheckboxFocus = function onCheckboxFocus(e) {
    e.currentTarget.addEventListener("keydown", keyboardEventListener);
  };
  var onCheckboxBlur = function onCheckboxBlur(e) {
    e.currentTarget.removeEventListener("keydown", keyboardEventListener);
  };
  return /* @__PURE__ */React__default["default"].createElement(tag_Tag.Tag, _objectSpread(_objectSpread({
    ref: ref,
    className: classNames__default["default"](tagClass),
    disabled: props.disabled,
    tabIndex: props.disabled ? void 0 : 0,
    onFocus: onCheckboxFocus,
    onBlur: onCheckboxBlur
  }, checkTagProps), {}, {
    onClick: handleClick
  }, tagOtherProps), content || children);
});
CheckTag.displayName = "CheckTag";

exports["default"] = CheckTag;
