// tdesign-react/cjs/_util/helper.js
/**
 * tdesign v1.17.1
 * (c) 2026 tdesign
 * @license MIT
 */

'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var slicedToArray = require('../_chunks/dep-c60a3e4c.js');
var defineProperty = require('../_chunks/dep-c446aa11.js');
var camelCase = require('../_chunks/dep-b422817e.js');
require('../_chunks/dep-98e6819c.js');
require('../_chunks/dep-7f3fa935.js');
require('../_chunks/dep-c290284d.js');
require('../_chunks/dep-b199860d.js');
require('../_chunks/dep-3d05de97.js');
require('../_chunks/dep-a54f202c.js');
require('../_chunks/dep-36579bf6.js');
require('../_chunks/dep-4f1729f0.js');
require('../_chunks/dep-45188d54.js');
require('../_chunks/dep-eccb6a94.js');
require('../_chunks/dep-f5e1d11c.js');

function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { defineProperty._defineProperty(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
function omit(obj, fields) {
  var shallowCopy = _objectSpread({}, obj);
  for (var i = 0; i < fields.length; i++) {
    var key = fields[i];
    delete shallowCopy[key];
  }
  return shallowCopy;
}
function removeEmptyAttrs(obj) {
  var newObj = {};
  Object.keys(obj).forEach(function (key) {
    if (typeof obj[key] !== "undefined" || obj[key] === null) {
      newObj[key] = obj[key];
    }
  });
  return newObj;
}
function getTabElementByValue() {
  var tabs = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : [];
  var value = arguments.length > 1 ? arguments[1] : undefined;
  var _tabs$filter = tabs.filter(function (item) {
      var id = item.id;
      return id === value;
    }),
    _tabs$filter2 = slicedToArray._slicedToArray(_tabs$filter, 1),
    result = _tabs$filter2[0];
  return result || null;
}
function firstUpperCase(str) {
  return str.toLowerCase().replace(/( |^)[a-z]/g, function (_char) {
    return _char.toUpperCase();
  });
}
function getPropsApiByEvent(eventName) {
  return camelCase.camelCase("on-".concat(eventName));
}
function getOffsetTopToContainer(element, container) {
  var offsetTop = element.offsetTop;
  var current = element.offsetParent;
  while (current && current !== container) {
    offsetTop += current.offsetTop;
    current = current.offsetParent;
  }
  return offsetTop;
}

exports.firstUpperCase = firstUpperCase;
exports.getOffsetTopToContainer = getOffsetTopToContainer;
exports.getPropsApiByEvent = getPropsApiByEvent;
exports.getTabElementByValue = getTabElementByValue;
exports.omit = omit;
exports.removeEmptyAttrs = removeEmptyAttrs;
