// tdesign-react/cjs/hooks/useMutationObserver.js
/**
 * tdesign v1.17.1
 * (c) 2026 tdesign
 * @license MIT
 */

'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var defineProperty = require('../_chunks/dep-c446aa11.js');
var React = require('react');
var hooks_useLatest = require('./useLatest.js');
var isEqual = require('../_chunks/dep-aa9a2c54.js');
var debounce = require('../_chunks/dep-f2809376.js');
require('../_chunks/dep-c290284d.js');
require('../_chunks/dep-77ef3011.js');
require('../_chunks/dep-150a112c.js');
require('../_chunks/dep-e360e17f.js');
require('../_chunks/dep-3d05de97.js');
require('../_chunks/dep-4f1729f0.js');
require('../_chunks/dep-a54f202c.js');
require('../_chunks/dep-51284537.js');
require('../_chunks/dep-8a493834.js');
require('../_chunks/dep-9f5b2891.js');
require('../_chunks/dep-338f013a.js');
require('../_chunks/dep-c37c5e17.js');
require('../_chunks/dep-eee1aab9.js');
require('../_chunks/dep-11347ea1.js');
require('../_chunks/dep-a830ce1e.js');
require('../_chunks/dep-770bc0ae.js');
require('../_chunks/dep-b36779a4.js');
require('../_chunks/dep-551d5940.js');
require('../_chunks/dep-8005b808.js');
require('../_chunks/dep-2add8824.js');
require('../_chunks/dep-863f8f4d.js');
require('../_chunks/dep-3be56e1e.js');
require('../_chunks/dep-36579bf6.js');

function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { defineProperty._defineProperty(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
var DEFAULT_OPTIONS = {
  debounceTime: 0,
  config: {
    attributes: true,
    childList: true,
    characterData: true,
    subtree: true
  }
};
function useMutationObservable(targetEl, cb, options) {
  var _options$debounceTime;
  var optionsRef = React.useRef(null);
  var signalRef = React.useRef(0);
  var callbackRef = hooks_useLatest["default"](cb);
  var mergedOptions = {
    debounceTime: (_options$debounceTime = options === null || options === void 0 ? void 0 : options.debounceTime) !== null && _options$debounceTime !== void 0 ? _options$debounceTime : DEFAULT_OPTIONS.debounceTime,
    config: _objectSpread(_objectSpread({}, DEFAULT_OPTIONS.config), options === null || options === void 0 ? void 0 : options.config)
  };
  if (!isEqual.isEqual(mergedOptions, optionsRef.current)) {
    signalRef.current += 1;
  }
  optionsRef.current = mergedOptions;
  React.useEffect(function () {
    if (!targetEl || !(targetEl !== null && targetEl !== void 0 && targetEl.nodeType)) return;
    var observer = null;
    try {
      var _optionsRef$current = optionsRef.current,
        debounceTime = _optionsRef$current.debounceTime,
        config = _optionsRef$current.config;
      var mutationCallback = function mutationCallback() {
        callbackRef.current.apply(callbackRef, arguments);
      };
      observer = new MutationObserver(debounceTime > 0 ? debounce.debounce(mutationCallback, debounceTime) : mutationCallback);
      observer.observe(targetEl, config);
    } catch (e) {
      console.error(e);
    }
    return function () {
      if (observer) {
        observer.disconnect();
        observer = null;
      }
    };
  }, [targetEl, signalRef.current]);
}

exports["default"] = useMutationObservable;
