// tdesign-react/cjs/hooks/useVariables.js
/**
 * tdesign v1.17.1
 * (c) 2026 tdesign
 * @license MIT
 */

'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var slicedToArray = require('../_chunks/dep-c60a3e4c.js');
var React = require('react');
var common = require('../_chunks/dep-29222115.js');
var _util_style = require('../_chunks/dep-ee71ddca.js');
var hooks_useMutationObserver = require('./useMutationObserver.js');
require('../_chunks/dep-98e6819c.js');
require('../_chunks/dep-7f3fa935.js');
require('../_chunks/dep-427f8bc0.js');
require('../_chunks/dep-3d05de97.js');
require('../_chunks/dep-c290284d.js');
require('../_chunks/dep-a54f202c.js');
require('../_chunks/dep-4f1729f0.js');
require('../_chunks/dep-c446aa11.js');
require('./useLatest.js');
require('../_chunks/dep-aa9a2c54.js');
require('../_chunks/dep-77ef3011.js');
require('../_chunks/dep-150a112c.js');
require('../_chunks/dep-e360e17f.js');
require('../_chunks/dep-51284537.js');
require('../_chunks/dep-8a493834.js');
require('../_chunks/dep-9f5b2891.js');
require('../_chunks/dep-338f013a.js');
require('../_chunks/dep-c37c5e17.js');
require('../_chunks/dep-eee1aab9.js');
require('../_chunks/dep-11347ea1.js');
require('../_chunks/dep-a830ce1e.js');
require('../_chunks/dep-770bc0ae.js');
require('../_chunks/dep-b36779a4.js');
require('../_chunks/dep-551d5940.js');
require('../_chunks/dep-8005b808.js');
require('../_chunks/dep-2add8824.js');
require('../_chunks/dep-863f8f4d.js');
require('../_chunks/dep-f2809376.js');
require('../_chunks/dep-3be56e1e.js');
require('../_chunks/dep-36579bf6.js');

var getColorTokenColor = function getColorTokenColor(token) {
  var _document, _styles$getPropertyVa;
  if (typeof window === "undefined") return "";
  var targetElement = (_document = document) === null || _document === void 0 ? void 0 : _document.documentElement;
  var styles = getComputedStyle(targetElement);
  return (_styles$getPropertyVa = styles.getPropertyValue(token).trim()) !== null && _styles$getPropertyVa !== void 0 ? _styles$getPropertyVa : "";
};

function _createForOfIteratorHelper(r, e) { var t = "undefined" != typeof Symbol && r[Symbol.iterator] || r["@@iterator"]; if (!t) { if (Array.isArray(r) || (t = _unsupportedIterableToArray(r)) || e && r && "number" == typeof r.length) { t && (r = t); var _n = 0, F = function F() {}; return { s: F, n: function n() { return _n >= r.length ? { done: !0 } : { done: !1, value: r[_n++] }; }, e: function e(r) { throw r; }, f: F }; } throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); } var o, a = !0, u = !1; return { s: function s() { t = t.call(r); }, n: function n() { var r = t.next(); return a = r.done, r; }, e: function e(r) { u = !0, o = r; }, f: function f() { try { a || null == t["return"] || t["return"](); } finally { if (u) throw o; } } }; }
function _unsupportedIterableToArray(r, a) { if (r) { if ("string" == typeof r) return _arrayLikeToArray(r, a); var t = {}.toString.call(r).slice(8, -1); return "Object" === t && r.constructor && (t = r.constructor.name), "Map" === t || "Set" === t ? Array.from(r) : "Arguments" === t || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(t) ? _arrayLikeToArray(r, a) : void 0; } }
function _arrayLikeToArray(r, a) { (null == a || a > r.length) && (a = r.length); for (var e = 0, n = Array(a); e < a; e++) n[e] = r[e]; return n; }
var DEFAULT_OPTIONS = {
  debounceTime: 250,
  config: {
    attributes: true
  }
};
function useVariables(variables, targetElement) {
  var _useState = React.useState({}),
    _useState2 = slicedToArray._slicedToArray(_useState, 2),
    forceUpdate = _useState2[1];
  if (_util_style.canUseDocument && !targetElement) {
    var _document;
    targetElement = (_document = document) === null || _document === void 0 ? void 0 : _document.documentElement;
  }
  if (!variables || Object.keys(variables).length === 0) {
    throw new Error("useVariables: variables parameter cannot be empty");
  }
  var refs = React.useMemo(function () {
    var values = {};
    Object.entries(variables).forEach(function (_ref) {
      var _ref2 = slicedToArray._slicedToArray(_ref, 2),
        key = _ref2[0],
        varName = _ref2[1];
      try {
        var initialValue = getColorTokenColor(varName);
        values[key] = initialValue;
      } catch (error) {
        console.warn("Failed to get initial value for CSS variable ".concat(varName, ":"), error);
        values[key] = "";
      }
    });
    return values;
  }, [variables]);
  var updateVariables = function updateVariables() {
    try {
      Object.entries(variables).forEach(function (_ref3) {
        var _ref4 = slicedToArray._slicedToArray(_ref3, 2),
          key = _ref4[0],
          varName = _ref4[1];
        var newValue = getColorTokenColor(varName);
        if (refs[key] && refs[key] !== newValue) {
          refs[key] = newValue;
        }
      });
      forceUpdate({});
    } catch (error) {
      console.warn("Failed to update CSS variables:", error);
    }
  };
  hooks_useMutationObserver["default"](targetElement, function (mutationsList) {
    var _iterator = _createForOfIteratorHelper(mutationsList),
      _step;
    try {
      for (_iterator.s(); !(_step = _iterator.n()).done;) {
        var mutation = _step.value;
        if (mutation.type === "attributes" && mutation.attributeName === common.THEME_MODE) {
          updateVariables();
          return;
        }
      }
    } catch (err) {
      _iterator.e(err);
    } finally {
      _iterator.f();
    }
  }, DEFAULT_OPTIONS);
  if (!_util_style.canUseDocument) return {};
  return refs;
}

exports["default"] = useVariables;
