// tdesign-react/cjs/tree/hooks/useControllable.js
/**
 * tdesign v1.17.1
 * (c) 2026 tdesign
 * @license MIT
 */

'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var slicedToArray = require('../../_chunks/dep-c60a3e4c.js');
var React = require('react');
var hooks_useControlled = require('../../hooks/useControlled.js');
require('../../_chunks/dep-98e6819c.js');
require('../../_chunks/dep-7f3fa935.js');
require('../../_util/noop.js');
require('../../_chunks/dep-45188d54.js');
require('../../_chunks/dep-eccb6a94.js');
require('../../_chunks/dep-f5e1d11c.js');
require('../../_chunks/dep-b199860d.js');
require('../../_chunks/dep-3d05de97.js');
require('../../_chunks/dep-c290284d.js');
require('../../_chunks/dep-a54f202c.js');
require('../../_chunks/dep-36579bf6.js');
require('../../_chunks/dep-4f1729f0.js');

function useControllable(props) {
  var _useControlled = hooks_useControlled["default"](props, "value", props.onChange),
    _useControlled2 = slicedToArray._slicedToArray(_useControlled, 2),
    value = _useControlled2[0],
    onChange = _useControlled2[1];
  var _useControlled3 = hooks_useControlled["default"](props, "expanded", props.onExpand),
    _useControlled4 = slicedToArray._slicedToArray(_useControlled3, 2),
    expanded = _useControlled4[0],
    onExpand = _useControlled4[1];
  var _useControlled5 = hooks_useControlled["default"](props, "actived", props.onActive),
    _useControlled6 = slicedToArray._slicedToArray(_useControlled5, 2),
    actived = _useControlled6[0],
    onActive = _useControlled6[1];
  var _useState = React.useState([]),
    _useState2 = slicedToArray._slicedToArray(_useState, 2),
    indeterminate = _useState2[0],
    setTreeIndeterminate = _useState2[1];
  return {
    value: value,
    onChange: onChange,
    expanded: expanded,
    onExpand: onExpand,
    actived: actived,
    onActive: onActive,
    setTreeIndeterminate: setTreeIndeterminate,
    indeterminate: indeterminate
  };
}

exports["default"] = useControllable;
