// tdesign-react/cjs/tree/hooks/useStore.js
/**
 * tdesign v1.17.1
 * (c) 2026 tdesign
 * @license MIT
 */

'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var slicedToArray = require('../../_chunks/dep-c60a3e4c.js');
var React = require('react');
var treeStore = require('../../_chunks/dep-e4edf583.js');
var hooks_usePersistFn = require('../../hooks/usePersistFn.js');
var hooks_usePrevious = require('../../hooks/usePrevious.js');
var hooks_useUpdateLayoutEffect = require('../../hooks/useUpdateLayoutEffect.js');
var cloneDeep = require('../../_chunks/dep-ba90997f.js');
require('../../_chunks/dep-98e6819c.js');
require('../../_chunks/dep-7f3fa935.js');
require('../../_chunks/dep-c446aa11.js');
require('../../_chunks/dep-c290284d.js');
require('../../_chunks/dep-fb453fc0.js');
require('mitt');
require('../../_chunks/dep-ac66dc4e.js');
require('../../_chunks/dep-7bf14076.js');
require('../../_chunks/dep-05bc49e6.js');
require('../../_chunks/dep-8c880a65.js');
require('../../_chunks/dep-3c628d18.js');
require('../../_chunks/dep-77f065c8.js');
require('../../_chunks/dep-3784bbea.js');
require('../../_chunks/dep-d4a6a9c1.js');
require('../../_chunks/dep-a54f202c.js');
require('../../_chunks/dep-36579bf6.js');
require('../../_chunks/dep-3d05de97.js');
require('../../_chunks/dep-4f1729f0.js');
require('../../_chunks/dep-c37c5e17.js');
require('../../_chunks/dep-eee1aab9.js');
require('../../_chunks/dep-11347ea1.js');
require('../../_chunks/dep-a830ce1e.js');
require('../../_chunks/dep-770bc0ae.js');
require('../../_chunks/dep-b199860d.js');
require('../../_chunks/dep-bc160f5b.js');
require('../../_chunks/dep-aaf27e39.js');
require('../../_chunks/dep-b47d855f.js');
require('../../_chunks/dep-338f013a.js');
require('../../_chunks/dep-363864ae.js');
require('../../_chunks/dep-e360e17f.js');
require('../../_chunks/dep-8a493834.js');
require('../../_chunks/dep-204188de.js');
require('../../_chunks/dep-7a4e0117.js');
require('../../_chunks/dep-8005b808.js');
require('../../_chunks/dep-f78183d7.js');
require('../../_chunks/dep-fb263e64.js');
require('../../_chunks/dep-8de6ca9d.js');
require('../../_chunks/dep-22026fea.js');
require('../../_chunks/dep-150a112c.js');
require('../../_chunks/dep-51284537.js');
require('../../_chunks/dep-9f5b2891.js');
require('../../_chunks/dep-81bce4e6.js');
require('../../_chunks/dep-863f8f4d.js');
require('../../_chunks/dep-551d5940.js');
require('../../_chunks/dep-2add8824.js');
require('../../_chunks/dep-6e5e0470.js');
require('../../_chunks/dep-cdef7593.js');
require('../../_chunks/dep-f5e1d11c.js');
require('../../_chunks/dep-bfa39549.js');
require('../../_chunks/dep-b8d0b96c.js');
require('../../_chunks/dep-f8c20fd8.js');
require('../../_chunks/dep-84536fdc.js');
require('../../_chunks/dep-f4f0baa3.js');
require('../../_chunks/dep-427f8bc0.js');
require('../../_chunks/dep-d3176201.js');
require('../../_chunks/dep-5fd08885.js');
require('../../_chunks/dep-b36779a4.js');
require('../../_chunks/dep-f98232bb.js');
require('../../_chunks/dep-84e4d0da.js');
require('../../_chunks/dep-b422817e.js');
require('../../_chunks/dep-45188d54.js');
require('../../_chunks/dep-eccb6a94.js');
require('../../hooks/useIsFirstRender.js');
require('../../hooks/useLayoutEffect.js');
require('../../_chunks/dep-ee71ddca.js');

function useStore(props, refresh) {
  var storeRef = React.useRef(null);
  var _useState = React.useState(false),
    _useState2 = slicedToArray._slicedToArray(_useState, 2),
    filterChanged = _useState2[0],
    toggleFilterChanged = _useState2[1];
  var _useState3 = React.useState(null),
    _useState4 = slicedToArray._slicedToArray(_useState3, 2),
    prevExpanded = _useState4[0],
    changePrevExpanded = _useState4[1];
  var data = props.data,
    keys = props.keys,
    expandAll = props.expandAll,
    expandParent = props.expandParent,
    expanded = props.expanded,
    expandLevel = props.expandLevel,
    expandMutex = props.expandMutex,
    activable = props.activable,
    activeMultiple = props.activeMultiple,
    actived = props.actived,
    disabled = props.disabled,
    disableCheck = props.disableCheck,
    draggable = props.draggable,
    checkable = props.checkable,
    value = props.value,
    checkStrictly = props.checkStrictly,
    load = props.load,
    lazy = props.lazy,
    valueMode = props.valueMode,
    filter = props.filter,
    _onLoad = props.onLoad,
    indeterminate = props.indeterminate,
    setTreeIndeterminate = props.setTreeIndeterminate,
    _props$allowFoldNodeO = props.allowFoldNodeOnFilter,
    allowFoldNodeOnFilter = _props$allowFoldNodeO === void 0 ? false : _props$allowFoldNodeO;
  var preFilter = hooks_usePrevious["default"](filter);
  React.useEffect(function () {
    if (!allowFoldNodeOnFilter) return;
    toggleFilterChanged(JSON.stringify(preFilter) !== JSON.stringify(filter));
  }, [filter, allowFoldNodeOnFilter]);
  var expandFilterPath = function expandFilterPath() {
    if (!allowFoldNodeOnFilter || !filterChanged) return;
    toggleFilterChanged(false);
    var store2 = storeRef.current;
    if (props.filter) {
      if (!prevExpanded) changePrevExpanded(store2.getExpanded());
      var pathValues = [];
      var allNodes = store2.getNodes();
      allNodes.forEach(function (node) {
        if (node.vmIsLocked) {
          pathValues.push(node.value);
        }
      });
      store2.setExpanded(pathValues);
    } else if (prevExpanded) {
      store2.replaceExpanded(prevExpanded);
      changePrevExpanded(null);
    }
  };
  var handleUpdate = hooks_usePersistFn.usePersistFn(function () {
    expandFilterPath();
    refresh();
  });
  var getExpandedArr = function getExpandedArr(arr, store2) {
    var expandedMap = /* @__PURE__ */new Map();
    arr.forEach(function (val) {
      expandedMap.set(val, true);
      if (expandParent) {
        var node = store2.getNode(val);
        node === null || node === void 0 || node.getParents().forEach(function (tn) {
          expandedMap.set(tn.value, true);
        });
      }
    });
    return Array.from(expandedMap.keys());
  };
  var createStore = function createStore() {
    var store2 = new treeStore.TreeStore({
      keys: keys,
      activable: activable,
      activeMultiple: activeMultiple,
      checkable: checkable,
      checkStrictly: checkStrictly,
      expandAll: expandAll,
      expandLevel: expandLevel,
      expandMutex: expandMutex,
      expandParent: expandParent,
      disabled: disabled,
      disableCheck: disableCheck,
      draggable: draggable,
      load: load,
      lazy: lazy,
      valueMode: valueMode,
      filter: filter,
      onLoad: function onLoad(info) {
        var node = info.node;
        _onLoad === null || _onLoad === void 0 || _onLoad({
          node: node.getModel()
        });
      },
      onUpdate: handleUpdate,
      allowFoldNodeOnFilter: allowFoldNodeOnFilter
    });
    var list = cloneDeep.cloneDeep(data);
    if (!Array.isArray(list)) {
      list = [];
    }
    store2.append(list);
    store2.refreshNodes();
    if (Array.isArray(value)) {
      store2.setChecked(value);
    }
    if (Array.isArray(expanded)) {
      var expandedArr = getExpandedArr(expanded, store2);
      store2.setExpanded(expandedArr);
    }
    if (Array.isArray(actived)) {
      store2.setActived(actived);
    }
    store2.refreshNodes();
    return store2;
  };
  if (!storeRef.current) {
    storeRef.current = createStore();
  }
  var store = storeRef.current;
  hooks_useUpdateLayoutEffect["default"](function () {
    if (data && Array.isArray(data)) {
      var expanded2 = store.getExpanded();
      var checked = store.getChecked();
      var actived2 = store.getActived();
      store.removeAll();
      store.append(data);
      store.setChecked(checked);
      store.setActived(actived2);
      store.setExpanded(expanded2);
    }
  }, [data, store]);
  hooks_useUpdateLayoutEffect["default"](function () {
    store.setConfig({
      keys: keys,
      expandAll: expandAll,
      expandLevel: expandLevel,
      expandMutex: expandMutex,
      expandParent: expandParent,
      activable: activable,
      activeMultiple: activeMultiple,
      disabled: disabled,
      disableCheck: disableCheck,
      checkable: checkable,
      draggable: draggable,
      checkStrictly: checkStrictly,
      load: load,
      lazy: lazy,
      valueMode: valueMode
    });
    store.refreshState();
  }, [activable, activeMultiple, checkStrictly, draggable, checkable, disabled, disableCheck, expandAll, expandLevel, expandMutex, expandParent, keys, lazy, load, store, valueMode]);
  hooks_useUpdateLayoutEffect["default"](function () {
    if (expandAll) {
      var valueList = store.getNodes().filter(function (node) {
        return Array.isArray(node.children) && node.children.length;
      }).map(function (node) {
        return node.value;
      });
      store.setExpanded(valueList);
    } else {
      store.replaceExpanded(prevExpanded);
      changePrevExpanded(null);
    }
  }, [store, expandAll]);
  hooks_useUpdateLayoutEffect["default"](function () {
    if (Array.isArray(value)) {
      store.replaceChecked(value);
      var checkedValue = store.getCheckedNodes().map(function (v) {
        return v.data[(keys === null || keys === void 0 ? void 0 : keys.value) || "value"];
      });
      var indeterminateConflict = checkedValue.filter(function (v) {
        return indeterminate.includes(v);
      });
      if (indeterminateConflict.length) {
        setTreeIndeterminate(indeterminate.filter(function (v) {
          return !indeterminateConflict.includes(v);
        }));
      }
    }
  }, [store, value, data]);
  hooks_useUpdateLayoutEffect["default"](function () {
    if (Array.isArray(expanded)) {
      var expandedArr = getExpandedArr(expanded, store);
      store.replaceExpanded(expandedArr);
    }
  }, [expanded, store]);
  hooks_useUpdateLayoutEffect["default"](function () {
    if (Array.isArray(actived)) {
      store.replaceActived(actived);
    }
  }, [actived, store]);
  hooks_useUpdateLayoutEffect["default"](function () {
    if (Array.isArray(indeterminate)) {
      store.replaceIndeterminate(indeterminate);
    }
  }, [indeterminate, store, data]);
  hooks_useUpdateLayoutEffect["default"](function () {
    store.setConfig({
      filter: filter
    });
    store.updateAll();
  }, [filter, store]);
  return storeRef.current;
}

exports.useStore = useStore;
