// tdesign-react/cjs/descriptions/Row.js
/**
 * tdesign v1.17.1
 * (c) 2026 tdesign
 * @license MIT
 */

'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var slicedToArray = require('../_chunks/dep-c60a3e4c.js');
var React = require('react');
var hooks_useConfig = require('../hooks/useConfig.js');
var locale_LocalReceiver = require('../locale/LocalReceiver.js');
var descriptions_DescriptionsContext = require('./DescriptionsContext.js');
require('../_chunks/dep-98e6819c.js');
require('../_chunks/dep-7f3fa935.js');
require('../config-provider/ConfigContext.js');
require('../_chunks/dep-c446aa11.js');
require('../_chunks/dep-c290284d.js');
require('../_chunks/dep-2ee09518.js');
require('../_chunks/dep-a6fa90dc.js');
require('dayjs');
require('../_chunks/dep-012d9a7c.js');
require('../_chunks/dep-150a112c.js');
require('../_chunks/dep-e360e17f.js');
require('../_chunks/dep-3d05de97.js');
require('../_chunks/dep-4f1729f0.js');
require('../_chunks/dep-a54f202c.js');
require('../_chunks/dep-51284537.js');
require('../_chunks/dep-8a493834.js');
require('../_chunks/dep-9f5b2891.js');
require('../_chunks/dep-338f013a.js');
require('../_chunks/dep-c37c5e17.js');
require('../_chunks/dep-eee1aab9.js');
require('../_chunks/dep-11347ea1.js');
require('../_chunks/dep-a830ce1e.js');
require('../_chunks/dep-770bc0ae.js');
require('../_chunks/dep-aaf27e39.js');
require('../_chunks/dep-b47d855f.js');
require('../_chunks/dep-81bce4e6.js');
require('../_chunks/dep-863f8f4d.js');
require('../_chunks/dep-84e4d0da.js');
require('../_chunks/dep-f78183d7.js');
require('../_chunks/dep-2f45f6c4.js');
require('../_chunks/dep-427f8bc0.js');
require('../config-provider/ConfigProvider.js');
require('../_chunks/dep-ba90997f.js');
require('../_chunks/dep-22026fea.js');
require('../_chunks/dep-551d5940.js');
require('../_chunks/dep-8005b808.js');
require('../_chunks/dep-2add8824.js');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

var React__default = /*#__PURE__*/_interopDefaultLegacy(React);

var Row = function Row(props) {
  var row = props.row;
  var _useConfig = hooks_useConfig["default"](),
    classPrefix = _useConfig.classPrefix;
  var _useLocaleReceiver = locale_LocalReceiver.useLocaleReceiver("descriptions"),
    _useLocaleReceiver2 = slicedToArray._slicedToArray(_useLocaleReceiver, 2),
    locale = _useLocaleReceiver2[0],
    t = _useLocaleReceiver2[1];
  var descriptionsContext = React.useContext(descriptions_DescriptionsContext.DescriptionsContext);
  var COMPONENT_NAME = "".concat(classPrefix, "-descriptions");
  var label = function label(node) {
    var layout = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : "horizontal";
    var rowKey = arguments.length > 2 ? arguments[2] : undefined;
    var span = node.span;
    var labelSpan = layout === "horizontal" ? 1 : span;
    return /* @__PURE__ */React__default["default"].createElement("td", {
      key: rowKey,
      colSpan: labelSpan,
      className: "".concat(COMPONENT_NAME, "__label"),
      style: descriptionsContext.labelStyle
    }, node.label, descriptionsContext.colon && t(locale.colonText));
  };
  var content = function content(node) {
    var layout = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : "horizontal";
    var rowKey = arguments.length > 2 ? arguments[2] : undefined;
    var span = node.span;
    var contentSpan = span > 1 && layout === "horizontal" ? span * 2 - 1 : span;
    return /* @__PURE__ */React__default["default"].createElement("td", {
      key: rowKey,
      colSpan: contentSpan,
      className: "".concat(COMPONENT_NAME, "__content"),
      style: descriptionsContext.contentStyle
    }, node.content);
  };
  var hh = function hh() {
    return /* @__PURE__ */React__default["default"].createElement("tr", null, row.map(function (node, i) {
      return /* @__PURE__ */React__default["default"].createElement(React__default["default"].Fragment, {
        key: i
      }, label(node), content(node));
    }));
  };
  var hv = function hv() {
    return /* @__PURE__ */React__default["default"].createElement(React__default["default"].Fragment, null, /* @__PURE__ */React__default["default"].createElement("tr", null, row.map(function (node, i) {
      return label(node, "vertical", "top_".concat(i));
    })), /* @__PURE__ */React__default["default"].createElement("tr", null, row.map(function (node, i) {
      return content(node, "vertical", "bottom_".concat(i));
    })));
  };
  var vh = function vh() {
    return /* @__PURE__ */React__default["default"].createElement(React__default["default"].Fragment, null, row.map(function (node, i) {
      return /* @__PURE__ */React__default["default"].createElement("tr", {
        key: i
      }, label(node), content(node));
    }));
  };
  var vv = function vv() {
    return /* @__PURE__ */React__default["default"].createElement(React__default["default"].Fragment, null, row.map(function (node, i) {
      return /* @__PURE__ */React__default["default"].createElement(React__default["default"].Fragment, {
        key: i
      }, /* @__PURE__ */React__default["default"].createElement("tr", null, label(node)), /* @__PURE__ */React__default["default"].createElement("tr", null, content(node)));
    }));
  };
  if (descriptionsContext.layout === "horizontal") {
    if (descriptionsContext.itemLayout === "horizontal") {
      return hh();
    }
    return hv();
  }
  if (descriptionsContext.itemLayout === "horizontal") {
    return vh();
  }
  return vv();
};
Row.displayName = "DescriptionsRow";

exports["default"] = Row;
