// tdesign-react/lib/date-picker/base/Footer.js
/**
 * tdesign v1.17.1
 * (c) 2026 tdesign
 * @license MIT
 */

import { _ as _defineProperty } from '../../_chunks/dep-55b77e26.js';
import { _ as _slicedToArray } from '../../_chunks/dep-d1eeb95c.js';
import React from 'react';
import classNames from 'classnames';
import { isFunction } from 'lodash-es';
import { Button } from '../../button/index.js';
import useConfig from '../../hooks/useConfig.js';
import { useLocaleReceiver } from '../../locale/LocalReceiver.js';
import '../../_chunks/dep-067cb22b.js';
import '../../_chunks/dep-673e8854.js';
import '../../_chunks/dep-570a5adb.js';
import '../../button/Button.js';
import '../../_chunks/dep-84926167.js';
import '../../_util/parseTNode.js';
import '../../_chunks/dep-65b369a5.js';
import '../../hooks/useDefaultProps.js';
import '../../hooks/useDomRefCallback.js';
import '../../hooks/useRipple.js';
import '../../_chunks/dep-e17ec8df.js';
import '../../_chunks/dep-68899e89.js';
import '../../hooks/useAnimation.js';
import '../../config-provider/ConfigContext.js';
import '../../_chunks/dep-c3e5e988.js';
import '../../_chunks/dep-65b05485.js';
import 'dayjs';
import '../../loading/index.js';
import '../../loading/Loading.js';
import '../../common/Portal.js';
import 'react-dom';
import '../../hooks/useLayoutEffect.js';
import '../../loading/defaultProps.js';
import '../../loading/gradient.js';
import '../../_chunks/dep-b0cbea79.js';
import '../../_chunks/dep-154fcb93.js';
import '../../_chunks/dep-e9c0bb63.js';
import '../../loading/plugin.js';
import '../../_util/react-render.js';
import '../../_chunks/dep-e29eee31.js';
import '../../common/PluginContainer.js';
import '../../config-provider/ConfigProvider.js';
import '../../button/defaultProps.js';

var DatePickerFooter = function DatePickerFooter(props) {
  var _useLocaleReceiver = useLocaleReceiver("datePicker"),
    _useLocaleReceiver2 = _slicedToArray(_useLocaleReceiver, 2),
    local = _useLocaleReceiver2[0],
    t = _useLocaleReceiver2[1];
  var confirmText = t(local.confirm);
  var _useConfig = useConfig(),
    classPrefix = _useConfig.classPrefix;
  var enableTimePicker = props.enableTimePicker,
    onConfirmClick = props.onConfirmClick,
    _props$presetsPlaceme = props.presetsPlacement,
    presetsPlacement = _props$presetsPlaceme === void 0 ? "bottom" : _props$presetsPlaceme,
    presets = props.presets,
    onPresetClick = props.onPresetClick,
    selectedValue = props.selectedValue,
    needConfirm = props.needConfirm,
    onTimePanelChange = props.onTimePanelChange,
    isSwitchTimeMode = props.isSwitchTimeMode;
  var footerClass = classNames("".concat(classPrefix, "-date-picker__footer"), "".concat(classPrefix, "-date-picker__footer--").concat(presetsPlacement));
  var renderPresets = function renderPresets() {
    if (presets) {
      if (/*#__PURE__*/React.isValidElement(presets)) return presets;
      if (isFunction(presets)) return presets();
      return Object.keys(presets).map(function (key) {
        return /* @__PURE__ */React.createElement(Button, {
          key: key,
          size: "small",
          variant: "text",
          onClick: function onClick(e) {
            return onPresetClick(presets[key], {
              e: e,
              preset: _defineProperty({}, key, presets[key])
            });
          }
        }, key);
      });
    }
    return null;
  };
  return /* @__PURE__ */React.createElement("div", {
    className: footerClass
  }, /* @__PURE__ */React.createElement("div", {
    className: "".concat(classPrefix, "-date-picker__presets")
  }, renderPresets()), /* @__PURE__ */React.createElement("div", null, isSwitchTimeMode && /* @__PURE__ */React.createElement(Button, {
    style: {
      marginRight: 16
    },
    size: "small",
    theme: "primary",
    variant: "text",
    onClick: onTimePanelChange
  }, props.isDateRangeContent ? t(local.selectTime) : t(local.selectDate)), enableTimePicker && needConfirm && /* @__PURE__ */React.createElement(Button, {
    disabled: !selectedValue,
    size: "small",
    theme: "primary",
    onClick: function onClick(e) {
      return onConfirmClick({
        e: e
      });
    }
  }, confirmText)));
};
DatePickerFooter.displayName = "DatePickerFooter";

export { DatePickerFooter as default };
