// tdesign-react/cjs/color-picker/components/panel/hue.js
/**
 * tdesign v1.17.1
 * (c) 2026 tdesign
 * @license MIT
 */

'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var React = require('react');
var colorPicker_components_panel_slider = require('./slider.js');
require('classnames');
require('../../../_chunks/dep-d925eb42.js');
require('../../../hooks/useMouseEvent.js');
require('../../../hooks/useLatest.js');
require('../../hooks/useStyles.js');
require('../../../_chunks/dep-c60a3e4c.js');
require('../../../_chunks/dep-98e6819c.js');
require('../../../_chunks/dep-7f3fa935.js');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

var React__default = /*#__PURE__*/_interopDefaultLegacy(React);

var HueSlider = function HueSlider(props) {
  var color = props.color,
    baseClassName = props.baseClassName,
    disabled = props.disabled,
    onChange = props.onChange;
  return /* @__PURE__ */React__default["default"].createElement(colorPicker_components_panel_slider["default"], {
    disabled: disabled,
    baseClassName: baseClassName,
    className: "".concat(baseClassName, "__hue"),
    color: color,
    value: color.hue,
    type: "hue",
    onChange: onChange
  });
};
var HueSlider$1 = /*#__PURE__*/React__default["default"].memo(HueSlider);

exports["default"] = HueSlider$1;
