// tdesign-react/cjs/color-picker/components/panel/swatches.js
/**
 * tdesign v1.17.1
 * (c) 2026 tdesign
 * @license MIT
 */

'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var React = require('react');
var classNames = require('classnames');
var tdesignIconsReact = require('tdesign-icons-react');
var color = require('../../../_chunks/dep-9bbe9605.js');
var hooks_useCommonClassName = require('../../../hooks/useCommonClassName.js');
var hooks_useGlobalIcon = require('../../../hooks/useGlobalIcon.js');
require('../../../_chunks/dep-c60a3e4c.js');
require('../../../_chunks/dep-98e6819c.js');
require('../../../_chunks/dep-7f3fa935.js');
require('../../../_chunks/dep-fb453fc0.js');
require('../../../_chunks/dep-c446aa11.js');
require('../../../_chunks/dep-c290284d.js');
require('../../../_chunks/dep-183643e1.js');
require('../../../_chunks/dep-d925eb42.js');
require('../../../_chunks/dep-427f8bc0.js');
require('../../../_chunks/dep-3d05de97.js');
require('../../../_chunks/dep-a54f202c.js');
require('../../../_chunks/dep-4f1729f0.js');
require('../../../_chunks/dep-84536fdc.js');
require('../../../hooks/useConfig.js');
require('../../../config-provider/ConfigContext.js');
require('../../../_chunks/dep-2ee09518.js');
require('../../../_chunks/dep-a6fa90dc.js');
require('dayjs');
require('../../../_chunks/dep-012d9a7c.js');
require('../../../_chunks/dep-150a112c.js');
require('../../../_chunks/dep-e360e17f.js');
require('../../../_chunks/dep-51284537.js');
require('../../../_chunks/dep-8a493834.js');
require('../../../_chunks/dep-9f5b2891.js');
require('../../../_chunks/dep-338f013a.js');
require('../../../_chunks/dep-c37c5e17.js');
require('../../../_chunks/dep-eee1aab9.js');
require('../../../_chunks/dep-11347ea1.js');
require('../../../_chunks/dep-a830ce1e.js');
require('../../../_chunks/dep-770bc0ae.js');
require('../../../_chunks/dep-aaf27e39.js');
require('../../../_chunks/dep-b47d855f.js');
require('../../../_chunks/dep-81bce4e6.js');
require('../../../_chunks/dep-863f8f4d.js');
require('../../../_chunks/dep-84e4d0da.js');
require('../../../_chunks/dep-f78183d7.js');
require('../../../_chunks/dep-2f45f6c4.js');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

var React__default = /*#__PURE__*/_interopDefaultLegacy(React);
var classNames__default = /*#__PURE__*/_interopDefaultLegacy(classNames);

var Swatches = function Swatches(props) {
  var baseClassName = props.baseClassName,
    _props$colors = props.colors,
    colors = _props$colors === void 0 ? [] : _props$colors,
    _props$editable = props.editable,
    editable = _props$editable === void 0 ? false : _props$editable,
    title = props.title,
    onChange = props.onChange,
    disabled = props.disabled,
    onSetColor = props.onSetColor,
    handleAddColor = props.handleAddColor;
  var _useGlobalIcon = hooks_useGlobalIcon["default"]({
      DeleteIcon: tdesignIconsReact.DeleteIcon,
      AddIcon: tdesignIconsReact.AddIcon
    }),
    DeleteIcon = _useGlobalIcon.DeleteIcon,
    AddIcon = _useGlobalIcon.AddIcon;
  var swatchesClass = "".concat(baseClassName, "__swatches");
  var _useCommonClassName = hooks_useCommonClassName["default"](),
    statusClassNames = _useCommonClassName.STATUS;
  var isEqualCurrentColor = function isEqualCurrentColor(color$1) {
    return color.Color.compare(color$1, props.color.css);
  };
  var selectedColorIndex = function selectedColorIndex() {
    return colors.findIndex(function (color) {
      return isEqualCurrentColor(color);
    });
  };
  var handleRemoveColor = function handleRemoveColor() {
    var selectedIndex = selectedColorIndex();
    if (selectedIndex > -1) {
      var newColors = colors.filter(function (item, index) {
        return index !== selectedIndex;
      });
      onChange(newColors);
    }
  };
  var handleClick = function handleClick(color) {
    return onSetColor(color);
  };
  return /* @__PURE__ */React__default["default"].createElement("div", {
    className: swatchesClass
  }, title ? /* @__PURE__ */React__default["default"].createElement("h3", {
    className: "".concat(swatchesClass, "--title")
  }, /* @__PURE__ */React__default["default"].createElement("span", null, title), editable && /* @__PURE__ */React__default["default"].createElement("div", {
    className: "".concat(swatchesClass, "--actions")
  }, /* @__PURE__ */React__default["default"].createElement("span", {
    role: "button",
    className: "".concat(baseClassName, "__icon"),
    onClick: function onClick() {
      return handleAddColor();
    }
  }, /* @__PURE__ */React__default["default"].createElement(AddIcon, null)), colors.length > 0 ? /* @__PURE__ */React__default["default"].createElement("span", {
    role: "button",
    className: "".concat(baseClassName, "__icon"),
    onClick: function onClick() {
      return handleRemoveColor();
    }
  }, /* @__PURE__ */React__default["default"].createElement(DeleteIcon, null)) : null)) : /* @__PURE__ */React__default["default"].createElement(React__default["default"].Fragment, null), /* @__PURE__ */React__default["default"].createElement("ul", {
    className: classNames__default["default"]("".concat(swatchesClass, "--items"), "narrow-scrollbar")
  }, colors.map(function (color) {
    return /* @__PURE__ */React__default["default"].createElement("li", {
      className: classNames__default["default"]("".concat(swatchesClass, "--item"), isEqualCurrentColor(color) && editable ? statusClassNames.active : ""),
      key: color,
      onClick: function onClick() {
        if (disabled) {
          return;
        }
        handleClick(color);
      }
    }, /* @__PURE__ */React__default["default"].createElement("div", {
      className: classNames__default["default"]("".concat(swatchesClass, "--item__color"), "".concat(baseClassName, "--bg-alpha"))
    }, /* @__PURE__ */React__default["default"].createElement("span", {
      className: "".concat(swatchesClass, "--item__inner"),
      style: {
        background: color
      }
    })));
  })));
};
var SwatchesPanel = /*#__PURE__*/React__default["default"].memo(Swatches);

exports["default"] = SwatchesPanel;
