// tdesign-react/es/time-picker/hooks/useTimePickerTextConfig.js
/**
 * tdesign v1.17.1
 * (c) 2026 tdesign
 * @license MIT
 */

import { _ as _slicedToArray } from '../../_chunks/dep-d1eeb95c.js';
import { useLocaleReceiver } from '../../locale/LocalReceiver.js';
import '../../_chunks/dep-673e8854.js';
import '../../_chunks/dep-570a5adb.js';
import '../../_chunks/dep-55b77e26.js';
import '../../_chunks/dep-067cb22b.js';
import 'react';
import 'lodash-es';
import '../../config-provider/index.js';
import '../../config-provider/ConfigContext.js';
import '../../locale/zh_CN.js';
import '../../_chunks/dep-e56ffacc.js';
import 'dayjs';
import '../../_chunks/dep-84036ea0.js';
import '../../config-provider/ConfigProvider.js';
import '../../config-provider/type.js';

var useTimePickerTextConfig = function useTimePickerTextConfig() {
  var _useLocaleReceiver = useLocaleReceiver("timePicker"),
    _useLocaleReceiver2 = _slicedToArray(_useLocaleReceiver, 2),
    local = _useLocaleReceiver2[0],
    t = _useLocaleReceiver2[1];
  return {
    nowTime: t(local.now),
    confirm: t(local.confirm),
    am: t(local.anteMeridiem),
    pm: t(local.postMeridiem),
    placeholder: t(local.placeholder),
    rangePlaceholder: [t(local.placeholder), t(local.placeholder)]
  };
};

export { useTimePickerTextConfig };
