// tdesign-react/cjs/date-picker/base/Cell.js
/**
 * tdesign v1.17.1
 * (c) 2026 tdesign
 * @license MIT
 */

'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var defineProperty = require('../../_chunks/dep-c446aa11.js');
var React = require('react');
var classNames = require('classnames');
var utils = require('../../_chunks/dep-39cd4265.js');
var _util_parseTNode = require('../../_util/parseTNode.js');
var hooks_useConfig = require('../../hooks/useConfig.js');
require('../../_chunks/dep-c290284d.js');
require('../../_chunks/dep-c60a3e4c.js');
require('../../_chunks/dep-98e6819c.js');
require('../../_chunks/dep-7f3fa935.js');
require('dayjs');
require('../../_chunks/dep-a6fa90dc.js');
require('../../_chunks/dep-c1575286.js');
require('../../_chunks/dep-3be56e1e.js');
require('../../_chunks/dep-a830ce1e.js');
require('../../_chunks/dep-36579bf6.js');
require('../../_chunks/dep-3d05de97.js');
require('../../_chunks/dep-4f1729f0.js');
require('../../_chunks/dep-34b7c4f7.js');
require('../../_chunks/dep-b21e3f45.js');
require('../../_chunks/dep-3c628d18.js');
require('../../_chunks/dep-427f8bc0.js');
require('../../_chunks/dep-a54f202c.js');
require('../../_chunks/dep-11347ea1.js');
require('../../_chunks/dep-f5e1d11c.js');
require('../../_chunks/dep-2f45f6c4.js');
require('../../_chunks/dep-770bc0ae.js');
require('../../_chunks/dep-863f8f4d.js');
require('../../_chunks/dep-8a493834.js');
require('../../_chunks/dep-338f013a.js');
require('../../config-provider/ConfigContext.js');
require('../../_chunks/dep-2ee09518.js');
require('../../_chunks/dep-012d9a7c.js');
require('../../_chunks/dep-150a112c.js');
require('../../_chunks/dep-e360e17f.js');
require('../../_chunks/dep-51284537.js');
require('../../_chunks/dep-9f5b2891.js');
require('../../_chunks/dep-c37c5e17.js');
require('../../_chunks/dep-eee1aab9.js');
require('../../_chunks/dep-aaf27e39.js');
require('../../_chunks/dep-b47d855f.js');
require('../../_chunks/dep-81bce4e6.js');
require('../../_chunks/dep-84e4d0da.js');
require('../../_chunks/dep-f78183d7.js');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

var React__default = /*#__PURE__*/_interopDefaultLegacy(React);
var classNames__default = /*#__PURE__*/_interopDefaultLegacy(classNames);

var DatePickerCell = function DatePickerCell(props) {
  var _classNames, _parseTNode;
  var _useConfig = hooks_useConfig["default"](),
    classPrefix = _useConfig.classPrefix;
  var cellClass = classNames__default["default"]("".concat(classPrefix, "-date-picker__cell"), (_classNames = {}, defineProperty._defineProperty(defineProperty._defineProperty(defineProperty._defineProperty(defineProperty._defineProperty(defineProperty._defineProperty(defineProperty._defineProperty(defineProperty._defineProperty(defineProperty._defineProperty(defineProperty._defineProperty(defineProperty._defineProperty(_classNames, "".concat(classPrefix, "-date-picker__cell--now"), props.now), "".concat(classPrefix, "-date-picker__cell--active"), props.active), "".concat(classPrefix, "-date-picker__cell--disabled"), props.disabled), "".concat(classPrefix, "-date-picker__cell--highlight"), props.highlight), "".concat(classPrefix, "-date-picker__cell--hover-highlight"), props.hoverHighlight), "".concat(classPrefix, "-date-picker__cell--active-start"), props.startOfRange), "".concat(classPrefix, "-date-picker__cell--active-end"), props.endOfRange), "".concat(classPrefix, "-date-picker__cell--hover-start"), props.hoverStartOfRange), "".concat(classPrefix, "-date-picker__cell--hover-end"), props.hoverEndOfRange), "".concat(classPrefix, "-date-picker__cell--additional"), props.additional), defineProperty._defineProperty(defineProperty._defineProperty(defineProperty._defineProperty(_classNames, "".concat(classPrefix, "-date-picker__cell--first-day-of-month"), props.firstDayOfMonth), "".concat(classPrefix, "-date-picker__cell--last-day-of-month"), props.lastDayOfMonth), "".concat(classPrefix, "-date-picker__cell--week-of-year"), props.weekOfYear)));
  function handleClick(e) {
    var _props$onClick;
    if (props.disabled) return;
    if (props.time) {
      var _extractTimeObj = utils.extractTimeObj(props.time),
        hours = _extractTimeObj.hours,
        minutes = _extractTimeObj.minutes,
        seconds = _extractTimeObj.seconds,
        milliseconds = _extractTimeObj.milliseconds,
        meridiem = _extractTimeObj.meridiem;
      var nextHours = hours;
      if (/am/i.test(meridiem) && nextHours === 12) nextHours -= 12;
      if (/pm/i.test(meridiem) && nextHours < 12) nextHours += 12;
      props.value.setHours(nextHours);
      props.value.setMinutes(minutes);
      props.value.setSeconds(seconds);
      props.value.setMilliseconds(milliseconds);
    }
    (_props$onClick = props.onClick) === null || _props$onClick === void 0 || _props$onClick.call(props, props.value, {
      e: e
    });
  }
  function handleMouseEnter() {
    var _props$onMouseEnter;
    if (props.disabled) return;
    if (props.time) {
      var _extractTimeObj2 = utils.extractTimeObj(props.time),
        hours = _extractTimeObj2.hours,
        minutes = _extractTimeObj2.minutes,
        seconds = _extractTimeObj2.seconds,
        milliseconds = _extractTimeObj2.milliseconds,
        meridiem = _extractTimeObj2.meridiem;
      var nextHours = hours;
      if (/am/i.test(meridiem) && nextHours === 12) nextHours -= 12;
      if (/pm/i.test(meridiem) && nextHours < 12) nextHours += 12;
      props.value.setHours(nextHours);
      props.value.setMinutes(minutes);
      props.value.setSeconds(seconds);
      props.value.setMilliseconds(milliseconds);
    }
    (_props$onMouseEnter = props.onMouseEnter) === null || _props$onMouseEnter === void 0 || _props$onMouseEnter.call(props, props.value);
  }
  return /* @__PURE__ */React__default["default"].createElement("td", {
    className: cellClass,
    onClick: handleClick,
    onMouseEnter: handleMouseEnter
  }, /* @__PURE__ */React__default["default"].createElement("div", {
    className: "".concat(classPrefix, "-date-picker__cell-inner")
  }, (_parseTNode = _util_parseTNode["default"](props.cell, {
    value: props.value
  })) !== null && _parseTNode !== void 0 ? _parseTNode : props.text));
};
DatePickerCell.displayName = "DatePickerCell";

exports["default"] = DatePickerCell;
