// tdesign-react/cjs/date-picker/base/Footer.js
/**
 * tdesign v1.17.1
 * (c) 2026 tdesign
 * @license MIT
 */

'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var defineProperty = require('../../_chunks/dep-c446aa11.js');
var slicedToArray = require('../../_chunks/dep-c60a3e4c.js');
var React = require('react');
var classNames = require('classnames');
var button_index = require('../../button/index.js');
var hooks_useConfig = require('../../hooks/useConfig.js');
var locale_LocalReceiver = require('../../locale/LocalReceiver.js');
var isFunction = require('../../_chunks/dep-11347ea1.js');
require('../../_chunks/dep-c290284d.js');
require('../../_chunks/dep-98e6819c.js');
require('../../_chunks/dep-7f3fa935.js');
require('../../button/Button.js');
require('../../_chunks/dep-e41137f7.js');
require('../../_util/parseTNode.js');
require('../../_chunks/dep-3c628d18.js');
require('../../_chunks/dep-3d05de97.js');
require('../../_chunks/dep-a830ce1e.js');
require('../../hooks/useDefaultProps.js');
require('../../hooks/useDomRefCallback.js');
require('../../hooks/useRipple.js');
require('../../_chunks/dep-062bb1b1.js');
require('../../_chunks/dep-ee71ddca.js');
require('../../_chunks/dep-427f8bc0.js');
require('../../_chunks/dep-a54f202c.js');
require('../../_chunks/dep-4f1729f0.js');
require('../../hooks/useAnimation.js');
require('../../config-provider/ConfigContext.js');
require('../../_chunks/dep-2ee09518.js');
require('../../_chunks/dep-a6fa90dc.js');
require('dayjs');
require('../../_chunks/dep-012d9a7c.js');
require('../../_chunks/dep-150a112c.js');
require('../../_chunks/dep-e360e17f.js');
require('../../_chunks/dep-51284537.js');
require('../../_chunks/dep-8a493834.js');
require('../../_chunks/dep-9f5b2891.js');
require('../../_chunks/dep-338f013a.js');
require('../../_chunks/dep-c37c5e17.js');
require('../../_chunks/dep-eee1aab9.js');
require('../../_chunks/dep-770bc0ae.js');
require('../../_chunks/dep-aaf27e39.js');
require('../../_chunks/dep-b47d855f.js');
require('../../_chunks/dep-81bce4e6.js');
require('../../_chunks/dep-863f8f4d.js');
require('../../_chunks/dep-84e4d0da.js');
require('../../_chunks/dep-f78183d7.js');
require('../../_chunks/dep-2f45f6c4.js');
require('../../loading/index.js');
require('../../loading/Loading.js');
require('../../common/Portal.js');
require('react-dom');
require('../../hooks/useLayoutEffect.js');
require('../../loading/defaultProps.js');
require('../../loading/gradient.js');
require('../../_chunks/dep-6e0da644.js');
require('../../_chunks/dep-05bc49e6.js');
require('../../_chunks/dep-8c880a65.js');
require('../../_chunks/dep-b8d0b96c.js');
require('../../loading/plugin.js');
require('../../_util/react-render.js');
require('../../_chunks/dep-7bf14076.js');
require('../../common/PluginContainer.js');
require('../../config-provider/ConfigProvider.js');
require('../../_chunks/dep-ba90997f.js');
require('../../_chunks/dep-22026fea.js');
require('../../_chunks/dep-551d5940.js');
require('../../_chunks/dep-8005b808.js');
require('../../_chunks/dep-2add8824.js');
require('../../button/defaultProps.js');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

var React__default = /*#__PURE__*/_interopDefaultLegacy(React);
var classNames__default = /*#__PURE__*/_interopDefaultLegacy(classNames);

var DatePickerFooter = function DatePickerFooter(props) {
  var _useLocaleReceiver = locale_LocalReceiver.useLocaleReceiver("datePicker"),
    _useLocaleReceiver2 = slicedToArray._slicedToArray(_useLocaleReceiver, 2),
    local = _useLocaleReceiver2[0],
    t = _useLocaleReceiver2[1];
  var confirmText = t(local.confirm);
  var _useConfig = hooks_useConfig["default"](),
    classPrefix = _useConfig.classPrefix;
  var enableTimePicker = props.enableTimePicker,
    onConfirmClick = props.onConfirmClick,
    _props$presetsPlaceme = props.presetsPlacement,
    presetsPlacement = _props$presetsPlaceme === void 0 ? "bottom" : _props$presetsPlaceme,
    presets = props.presets,
    onPresetClick = props.onPresetClick,
    selectedValue = props.selectedValue,
    needConfirm = props.needConfirm,
    onTimePanelChange = props.onTimePanelChange,
    isSwitchTimeMode = props.isSwitchTimeMode;
  var footerClass = classNames__default["default"]("".concat(classPrefix, "-date-picker__footer"), "".concat(classPrefix, "-date-picker__footer--").concat(presetsPlacement));
  var renderPresets = function renderPresets() {
    if (presets) {
      if (/*#__PURE__*/React__default["default"].isValidElement(presets)) return presets;
      if (isFunction.isFunction(presets)) return presets();
      return Object.keys(presets).map(function (key) {
        return /* @__PURE__ */React__default["default"].createElement(button_index.Button, {
          key: key,
          size: "small",
          variant: "text",
          onClick: function onClick(e) {
            return onPresetClick(presets[key], {
              e: e,
              preset: defineProperty._defineProperty({}, key, presets[key])
            });
          }
        }, key);
      });
    }
    return null;
  };
  return /* @__PURE__ */React__default["default"].createElement("div", {
    className: footerClass
  }, /* @__PURE__ */React__default["default"].createElement("div", {
    className: "".concat(classPrefix, "-date-picker__presets")
  }, renderPresets()), /* @__PURE__ */React__default["default"].createElement("div", null, isSwitchTimeMode && /* @__PURE__ */React__default["default"].createElement(button_index.Button, {
    style: {
      marginRight: 16
    },
    size: "small",
    theme: "primary",
    variant: "text",
    onClick: onTimePanelChange
  }, props.isDateRangeContent ? t(local.selectTime) : t(local.selectDate)), enableTimePicker && needConfirm && /* @__PURE__ */React__default["default"].createElement(button_index.Button, {
    disabled: !selectedValue,
    size: "small",
    theme: "primary",
    onClick: function onClick(e) {
      return onConfirmClick({
        e: e
      });
    }
  }, confirmText)));
};
DatePickerFooter.displayName = "DatePickerFooter";

exports["default"] = DatePickerFooter;
