// tdesign-react/lib/select/util/helper.js
/**
 * tdesign v1.17.1
 * (c) 2026 tdesign
 * @license MIT
 */

import '../../_chunks/dep-154fcb93.js';
import '../../_chunks/dep-55b77e26.js';
import 'lodash-es';
export { g as getKeyMapping, d as getLabel, e as getMultipleTags, b as getSelectValueArr, a as getSelectedOptions, c as getValueToOption, s as setValueToOptionFormOptionDom } from '../../_chunks/dep-0210405c.js';
import '../base/OptionGroup.js';
import '../../_chunks/dep-570a5adb.js';
import '../../_chunks/dep-e9c0bb63.js';
import '../../_chunks/dep-067cb22b.js';
import '../../_chunks/dep-d1eeb95c.js';
import '../../_chunks/dep-673e8854.js';
import 'react';
import 'classnames';
import '../../hooks/useConfig.js';
import '../../config-provider/ConfigContext.js';
import '../../_chunks/dep-c3e5e988.js';
import '../../_chunks/dep-65b05485.js';
import 'dayjs';
import '../../hooks/useDomRefCallback.js';
import '../../hooks/useRipple.js';
import '../../_chunks/dep-e17ec8df.js';
import '../../_chunks/dep-68899e89.js';
import '../../hooks/useAnimation.js';
import '../../hooks/useDefaultProps.js';
import '../defaultProps.js';
