// tdesign-react/cjs/steps/StepItem.js
/**
 * tdesign v1.17.1
 * (c) 2026 tdesign
 * @license MIT
 */

'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var defineProperty = require('../_chunks/dep-c446aa11.js');
var React = require('react');
var classNames = require('classnames');
var tdesignIconsReact = require('tdesign-icons-react');
var hooks_useConfig = require('../hooks/useConfig.js');
var hooks_useDefaultProps = require('../hooks/useDefaultProps.js');
var hooks_useGlobalIcon = require('../hooks/useGlobalIcon.js');
var steps_defaultProps = require('./defaultProps.js');
var steps_StepsContext = require('./StepsContext.js');
require('../_chunks/dep-c290284d.js');
require('../config-provider/ConfigContext.js');
require('../_chunks/dep-2ee09518.js');
require('../_chunks/dep-a6fa90dc.js');
require('dayjs');
require('../_chunks/dep-012d9a7c.js');
require('../_chunks/dep-150a112c.js');
require('../_chunks/dep-e360e17f.js');
require('../_chunks/dep-3d05de97.js');
require('../_chunks/dep-4f1729f0.js');
require('../_chunks/dep-a54f202c.js');
require('../_chunks/dep-51284537.js');
require('../_chunks/dep-8a493834.js');
require('../_chunks/dep-9f5b2891.js');
require('../_chunks/dep-338f013a.js');
require('../_chunks/dep-c37c5e17.js');
require('../_chunks/dep-eee1aab9.js');
require('../_chunks/dep-11347ea1.js');
require('../_chunks/dep-a830ce1e.js');
require('../_chunks/dep-770bc0ae.js');
require('../_chunks/dep-aaf27e39.js');
require('../_chunks/dep-b47d855f.js');
require('../_chunks/dep-81bce4e6.js');
require('../_chunks/dep-863f8f4d.js');
require('../_chunks/dep-84e4d0da.js');
require('../_chunks/dep-f78183d7.js');
require('../_chunks/dep-2f45f6c4.js');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

var React__default = /*#__PURE__*/_interopDefaultLegacy(React);
var classNames__default = /*#__PURE__*/_interopDefaultLegacy(classNames);

var StepItem = function StepItem(originalProps) {
  var props = hooks_useDefaultProps["default"](originalProps, steps_defaultProps.stepItemDefaultProps);
  var index = props.index,
    icon = props.icon,
    title = props.title,
    content = props.content,
    value = props.value,
    children = props.children,
    style = props.style,
    status = props.status;
  var _useContext = React.useContext(steps_StepsContext["default"]),
    current = _useContext.current,
    theme = _useContext.theme,
    onChange = _useContext.onChange,
    readOnly = _useContext.readOnly;
  var _useConfig = hooks_useConfig["default"](),
    classPrefix = _useConfig.classPrefix,
    globalStepsConfig = _useConfig.steps;
  var _useGlobalIcon = hooks_useGlobalIcon["default"]({
      CloseIcon: tdesignIconsReact.CloseIcon,
      CheckIcon: tdesignIconsReact.CheckIcon
    }),
    CloseIcon = _useGlobalIcon.CloseIcon,
    CheckIcon = _useGlobalIcon.CheckIcon;
  var canClick = status !== "process" && !readOnly;
  var iconNode = React__default["default"].useMemo(function () {
    if (!icon || theme !== "default") {
      return null;
    }
    if (icon !== true) {
      return icon;
    }
    var iconCls = "".concat(classPrefix, "-steps-item__icon--number");
    var getDefaultIconByStatus = function getDefaultIconByStatus() {
      switch (status) {
        case "error":
          return globalStepsConfig.errorIcon || /* @__PURE__ */React__default["default"].createElement(CloseIcon, null);
        case "finish":
          return globalStepsConfig.checkIcon || /* @__PURE__ */React__default["default"].createElement(CheckIcon, null);
        default:
          return Number(index) + 1;
      }
    };
    return /* @__PURE__ */React__default["default"].createElement("span", {
      className: iconCls
    }, getDefaultIconByStatus());
  }, [icon, classPrefix, theme, status, globalStepsConfig, index, CloseIcon, CheckIcon]);
  function onStepClick(e) {
    if (!canClick) {
      return;
    }
    var currentValue = value !== null && value !== void 0 ? value : index;
    onChange(currentValue, current, {
      e: e
    });
  }
  return /* @__PURE__ */React__default["default"].createElement("div", {
    style: style,
    className: classNames__default["default"](defineProperty._defineProperty(defineProperty._defineProperty(defineProperty._defineProperty(defineProperty._defineProperty(defineProperty._defineProperty(defineProperty._defineProperty({}, "".concat(classPrefix, "-steps-item"), true), "".concat(classPrefix, "-steps-item--wait"), status === "default"), "".concat(classPrefix, "-steps-item--error"), status === "error"), "".concat(classPrefix, "-steps-item--finish"), status === "finish"), "".concat(classPrefix, "-steps-item--process"), status === "process"), props.className, !!props.className))
  }, /* @__PURE__ */React__default["default"].createElement("div", {
    className: classNames__default["default"]("".concat(classPrefix, "-steps-item__inner"), defineProperty._defineProperty({}, "".concat(classPrefix, "-steps-item--clickable"), canClick)),
    onClick: onStepClick
  }, /* @__PURE__ */React__default["default"].createElement("div", {
    className: classNames__default["default"]("".concat(classPrefix, "-steps-item__icon"), defineProperty._defineProperty({}, "".concat(classPrefix, "-steps-item-").concat(status), status))
  }, iconNode), /* @__PURE__ */React__default["default"].createElement("div", {
    className: "".concat(classPrefix, "-steps-item__content")
  }, /* @__PURE__ */React__default["default"].createElement("div", {
    className: "".concat(classPrefix, "-steps-item__title")
  }, title), /* @__PURE__ */React__default["default"].createElement("div", {
    className: "".concat(classPrefix, "-steps-item__description")
  }, content), children ? /* @__PURE__ */React__default["default"].createElement("div", {
    className: "".concat(classPrefix, "-steps-item__extra")
  }, children) : null)));
};
StepItem.displayName = "StepItem";

exports["default"] = StepItem;
