// tdesign-react/cjs/steps/Steps.js
/**
 * tdesign v1.17.1
 * (c) 2026 tdesign
 * @license MIT
 */

'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var defineProperty = require('../_chunks/dep-c446aa11.js');
var slicedToArray = require('../_chunks/dep-c60a3e4c.js');
var React = require('react');
var classNames = require('classnames');
var _util_forwardRefWithStatics = require('../_util/forwardRefWithStatics.js');
var hooks_useConfig = require('../hooks/useConfig.js');
var hooks_useControlled = require('../hooks/useControlled.js');
var hooks_useDefaultProps = require('../hooks/useDefaultProps.js');
var steps_defaultProps = require('./defaultProps.js');
var steps_StepItem = require('./StepItem.js');
var steps_StepsContext = require('./StepsContext.js');
require('../_chunks/dep-c290284d.js');
require('../_chunks/dep-98e6819c.js');
require('../_chunks/dep-7f3fa935.js');
require('hoist-non-react-statics');
require('../config-provider/ConfigContext.js');
require('../_chunks/dep-2ee09518.js');
require('../_chunks/dep-a6fa90dc.js');
require('dayjs');
require('../_chunks/dep-012d9a7c.js');
require('../_chunks/dep-150a112c.js');
require('../_chunks/dep-e360e17f.js');
require('../_chunks/dep-3d05de97.js');
require('../_chunks/dep-4f1729f0.js');
require('../_chunks/dep-a54f202c.js');
require('../_chunks/dep-51284537.js');
require('../_chunks/dep-8a493834.js');
require('../_chunks/dep-9f5b2891.js');
require('../_chunks/dep-338f013a.js');
require('../_chunks/dep-c37c5e17.js');
require('../_chunks/dep-eee1aab9.js');
require('../_chunks/dep-11347ea1.js');
require('../_chunks/dep-a830ce1e.js');
require('../_chunks/dep-770bc0ae.js');
require('../_chunks/dep-aaf27e39.js');
require('../_chunks/dep-b47d855f.js');
require('../_chunks/dep-81bce4e6.js');
require('../_chunks/dep-863f8f4d.js');
require('../_chunks/dep-84e4d0da.js');
require('../_chunks/dep-f78183d7.js');
require('../_chunks/dep-2f45f6c4.js');
require('../_util/noop.js');
require('../_chunks/dep-45188d54.js');
require('../_chunks/dep-eccb6a94.js');
require('../_chunks/dep-f5e1d11c.js');
require('../_chunks/dep-b199860d.js');
require('../_chunks/dep-36579bf6.js');
require('tdesign-icons-react');
require('../hooks/useGlobalIcon.js');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

var React__default = /*#__PURE__*/_interopDefaultLegacy(React);
var classNames__default = /*#__PURE__*/_interopDefaultLegacy(classNames);

function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { defineProperty._defineProperty(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
var Steps = _util_forwardRefWithStatics["default"](function (originalProps, ref) {
  var props = hooks_useDefaultProps["default"](originalProps, steps_defaultProps.stepsDefaultProps);
  var style = props.style,
    layout = props.layout,
    theme = props.theme,
    sequence = props.sequence,
    separator = props.separator,
    children = props.children,
    options = props.options;
  var _useConfig = hooks_useConfig["default"](),
    classPrefix = _useConfig.classPrefix;
  var _useControlled = hooks_useControlled["default"](props, "current", props.onChange),
    _useControlled2 = slicedToArray._slicedToArray(_useControlled, 2),
    current = _useControlled2[0],
    onChange = _useControlled2[1];
  var indexMap = React.useMemo(function () {
    var map = {};
    if (options) {
      options.forEach(function (item, index) {
        if (item.value !== void 0) {
          map[item.value] = index;
        }
      });
    } else {
      React__default["default"].Children.forEach(children, function (child, index) {
        if (! /*#__PURE__*/React__default["default"].isValidElement(child)) {
          return;
        }
        if (child.props.value !== void 0) {
          map[child.props.value] = index;
        }
      });
    }
    return map;
  }, [options, children]);
  var handleStatus = React.useCallback(function (item, index) {
    var _item$value;
    if (current === "FINISH") {
      return "finish";
    }
    if (item.status && item.status !== "default") {
      return item.status;
    }
    if (item.value === void 0) {
      if (sequence === "positive" && typeof current === "number" && index < current) {
        return "finish";
      }
      if (sequence === "reverse" && typeof current === "number" && index > current) {
        return "finish";
      }
    }
    if (item.value !== void 0) {
      var matchIndex = indexMap[current];
      if (matchIndex === void 0) {
        console.warn("TDesign Steps Warn: The current `value` is not exist.");
        return "default";
      }
      if (sequence === "positive" && index < matchIndex) {
        return "finish";
      }
      if (sequence === "reverse" && index > matchIndex) {
        return "finish";
      }
    }
    var key = (_item$value = item.value) !== null && _item$value !== void 0 ? _item$value : index;
    if (key === current) {
      return "process";
    }
    return "default";
  }, [current, sequence, indexMap]);
  var stepItemList = React.useMemo(function () {
    if (options) {
      var optionsDisplayList = sequence === "reverse" ? options.reverse() : options;
      return options.map(function (item, index) {
        var stepIndex = sequence === "reverse" ? optionsDisplayList.length - index - 1 : index;
        return /* @__PURE__ */React__default["default"].createElement(steps_StepItem["default"], _objectSpread(_objectSpread({
          key: index
        }, item), {}, {
          index: stepIndex,
          status: handleStatus(item, index)
        }));
      });
    }
    var childrenList = React__default["default"].Children.toArray(children);
    var childrenDisplayList = sequence === "reverse" ? childrenList.reverse() : childrenList;
    return childrenList.map(function (child, index) {
      var stepIndex = sequence === "reverse" ? childrenDisplayList.length - index - 1 : index;
      return /*#__PURE__*/React__default["default"].cloneElement(child, _objectSpread(_objectSpread({}, child.props), {}, {
        index: stepIndex,
        status: handleStatus(child.props, index)
      }));
    });
  }, [options, children, sequence, handleStatus]);
  return /* @__PURE__ */React__default["default"].createElement(steps_StepsContext["default"].Provider, {
    value: {
      current: current,
      theme: theme,
      readOnly: props.readOnly || props.readonly,
      onChange: onChange
    }
  }, /* @__PURE__ */React__default["default"].createElement("div", {
    ref: ref,
    style: style,
    className: classNames__default["default"](defineProperty._defineProperty(defineProperty._defineProperty(defineProperty._defineProperty(defineProperty._defineProperty(defineProperty._defineProperty(defineProperty._defineProperty({}, "".concat(classPrefix, "-steps"), true), "".concat(classPrefix, "-steps--").concat(theme, "-anchor"), theme), "".concat(classPrefix, "-steps--").concat(layout), layout), "".concat(classPrefix, "-steps--").concat(sequence), sequence), "".concat(classPrefix, "-steps--").concat(separator, "-separator"), separator), props.className, !!props.className))
  }, stepItemList));
}, {
  StepItem: steps_StepItem["default"]
});
Steps.displayName = "Steps";

exports["default"] = Steps;
