// tdesign-react/lib/table/hooks/useAsyncLoading.js
/**
 * tdesign v1.17.1
 * (c) 2026 tdesign
 * @license MIT
 */

import { _ as _defineProperty } from '../../_chunks/dep-55b77e26.js';
import { _ as _slicedToArray } from '../../_chunks/dep-d1eeb95c.js';
import React from 'react';
import classNames from 'classnames';
import { isString } from 'lodash-es';
import { Loading } from '../../loading/index.js';
import { useLocaleReceiver } from '../../locale/LocalReceiver.js';
import useClassName from './useClassName.js';
import '../../_chunks/dep-067cb22b.js';
import '../../_chunks/dep-673e8854.js';
import '../../_chunks/dep-570a5adb.js';
import '../../loading/Loading.js';
import '../../_chunks/dep-68899e89.js';
import '../../common/Portal.js';
import 'react-dom';
import '../../hooks/useConfig.js';
import '../../config-provider/ConfigContext.js';
import '../../_chunks/dep-c3e5e988.js';
import '../../_chunks/dep-65b05485.js';
import 'dayjs';
import '../../hooks/useLayoutEffect.js';
import '../../hooks/useDefaultProps.js';
import '../../loading/defaultProps.js';
import '../../loading/gradient.js';
import '../../_chunks/dep-e17ec8df.js';
import '../../_chunks/dep-b0cbea79.js';
import '../../_chunks/dep-154fcb93.js';
import '../../_chunks/dep-e9c0bb63.js';
import '../../_chunks/dep-84926167.js';
import '../../hooks/useDomRefCallback.js';
import '../../loading/plugin.js';
import '../../_util/react-render.js';
import '../../_chunks/dep-e29eee31.js';
import '../../common/PluginContainer.js';
import '../../config-provider/ConfigProvider.js';

function useAsyncLoading(props) {
  var _useLocaleReceiver = useLocaleReceiver("table"),
    _useLocaleReceiver2 = _slicedToArray(_useLocaleReceiver, 1),
    local = _useLocaleReceiver2[0];
  var _useClassName = useClassName(),
    isLoadingClass = _useClassName.isLoadingClass,
    isLoadMoreClass = _useClassName.isLoadMoreClass,
    asyncLoadingClass = _useClassName.asyncLoadingClass;
  var classes = [asyncLoadingClass, _defineProperty(_defineProperty({}, isLoadingClass, props.asyncLoading === "loading"), isLoadMoreClass, props.asyncLoading === "load-more")];
  function onLoadClick() {
    var _props$onAsyncLoading;
    if (typeof props.asyncLoading !== "string") return;
    var status = props.asyncLoading === "loading" ? "loading" : "load-more";
    (_props$onAsyncLoading = props.onAsyncLoadingClick) === null || _props$onAsyncLoading === void 0 || _props$onAsyncLoading.call(props, {
      status: status
    });
  }
  function renderAsyncLoading() {
    var asyncLoadingNode = props.asyncLoading;
    if (isString(asyncLoadingNode)) {
      var asyncLoading = props.asyncLoading;
      var loadingText = {
        "load-more": local.loadingMoreText,
        loading: local.loadingText
      }[String(asyncLoading)];
      return /* @__PURE__ */React.createElement("div", {
        className: classNames(classes),
        onClick: onLoadClick
      }, /* @__PURE__ */React.createElement(Loading, {
        loading: true,
        size: "small",
        text: loadingText,
        indicator: asyncLoading === "loading"
      }));
    }
    if (asyncLoadingNode) {
      return /* @__PURE__ */React.createElement("div", {
        className: classNames(classes),
        onClick: onLoadClick
      }, asyncLoadingNode);
    }
    return null;
  }
  return {
    renderAsyncLoading: renderAsyncLoading
  };
}

export { useAsyncLoading as default };
