// tdesign-react/esm/date-picker/base/Cell.js
/**
 * tdesign v1.17.1
 * (c) 2026 tdesign
 * @license MIT
 */

import { _ as _defineProperty } from '../../_chunks/dep-55b77e26.js';
import React from 'react';
import classNames from 'classnames';
import { e as extractTimeObj } from '../../_chunks/dep-10236bdf.js';
import parseTNode from '../../_util/parseTNode.js';
import useConfig from '../../hooks/useConfig.js';
import '../../_chunks/dep-067cb22b.js';
import '../../_chunks/dep-d1eeb95c.js';
import '../../_chunks/dep-673e8854.js';
import '../../_chunks/dep-570a5adb.js';
import 'lodash-es';
import 'dayjs';
import '../../_chunks/dep-e56ffacc.js';
import '../../_chunks/dep-ec64076a.js';
import '../../_chunks/dep-ca17be23.js';
import '../../_chunks/dep-dc41ae40.js';
import '../../_chunks/dep-0f787ecd.js';
import '../../_chunks/dep-65b369a5.js';
import '../../config-provider/ConfigContext.js';
import '../../locale/zh_CN.js';
import '../../_chunks/dep-84036ea0.js';

var DatePickerCell = function DatePickerCell(props) {
  var _classNames, _parseTNode;
  var _useConfig = useConfig(),
    classPrefix = _useConfig.classPrefix;
  var cellClass = classNames("".concat(classPrefix, "-date-picker__cell"), (_classNames = {}, _defineProperty(_defineProperty(_defineProperty(_defineProperty(_defineProperty(_defineProperty(_defineProperty(_defineProperty(_defineProperty(_defineProperty(_classNames, "".concat(classPrefix, "-date-picker__cell--now"), props.now), "".concat(classPrefix, "-date-picker__cell--active"), props.active), "".concat(classPrefix, "-date-picker__cell--disabled"), props.disabled), "".concat(classPrefix, "-date-picker__cell--highlight"), props.highlight), "".concat(classPrefix, "-date-picker__cell--hover-highlight"), props.hoverHighlight), "".concat(classPrefix, "-date-picker__cell--active-start"), props.startOfRange), "".concat(classPrefix, "-date-picker__cell--active-end"), props.endOfRange), "".concat(classPrefix, "-date-picker__cell--hover-start"), props.hoverStartOfRange), "".concat(classPrefix, "-date-picker__cell--hover-end"), props.hoverEndOfRange), "".concat(classPrefix, "-date-picker__cell--additional"), props.additional), _defineProperty(_defineProperty(_defineProperty(_classNames, "".concat(classPrefix, "-date-picker__cell--first-day-of-month"), props.firstDayOfMonth), "".concat(classPrefix, "-date-picker__cell--last-day-of-month"), props.lastDayOfMonth), "".concat(classPrefix, "-date-picker__cell--week-of-year"), props.weekOfYear)));
  function handleClick(e) {
    var _props$onClick;
    if (props.disabled) return;
    if (props.time) {
      var _extractTimeObj = extractTimeObj(props.time),
        hours = _extractTimeObj.hours,
        minutes = _extractTimeObj.minutes,
        seconds = _extractTimeObj.seconds,
        milliseconds = _extractTimeObj.milliseconds,
        meridiem = _extractTimeObj.meridiem;
      var nextHours = hours;
      if (/am/i.test(meridiem) && nextHours === 12) nextHours -= 12;
      if (/pm/i.test(meridiem) && nextHours < 12) nextHours += 12;
      props.value.setHours(nextHours);
      props.value.setMinutes(minutes);
      props.value.setSeconds(seconds);
      props.value.setMilliseconds(milliseconds);
    }
    (_props$onClick = props.onClick) === null || _props$onClick === void 0 || _props$onClick.call(props, props.value, {
      e: e
    });
  }
  function handleMouseEnter() {
    var _props$onMouseEnter;
    if (props.disabled) return;
    if (props.time) {
      var _extractTimeObj2 = extractTimeObj(props.time),
        hours = _extractTimeObj2.hours,
        minutes = _extractTimeObj2.minutes,
        seconds = _extractTimeObj2.seconds,
        milliseconds = _extractTimeObj2.milliseconds,
        meridiem = _extractTimeObj2.meridiem;
      var nextHours = hours;
      if (/am/i.test(meridiem) && nextHours === 12) nextHours -= 12;
      if (/pm/i.test(meridiem) && nextHours < 12) nextHours += 12;
      props.value.setHours(nextHours);
      props.value.setMinutes(minutes);
      props.value.setSeconds(seconds);
      props.value.setMilliseconds(milliseconds);
    }
    (_props$onMouseEnter = props.onMouseEnter) === null || _props$onMouseEnter === void 0 || _props$onMouseEnter.call(props, props.value);
  }
  return /* @__PURE__ */React.createElement("td", {
    className: cellClass,
    onClick: handleClick,
    onMouseEnter: handleMouseEnter
  }, /* @__PURE__ */React.createElement("div", {
    className: "".concat(classPrefix, "-date-picker__cell-inner")
  }, (_parseTNode = parseTNode(props.cell, {
    value: props.value
  })) !== null && _parseTNode !== void 0 ? _parseTNode : props.text));
};
DatePickerCell.displayName = "DatePickerCell";

export { DatePickerCell as default };
