// tdesign-react/cjs/form/Form.js
/**
 * tdesign v1.17.1
 * (c) 2026 tdesign
 * @license MIT
 */

'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var toConsumableArray = require('../_chunks/dep-05bc49e6.js');
var slicedToArray = require('../_chunks/dep-c60a3e4c.js');
var defineProperty = require('../_chunks/dep-c446aa11.js');
var React = require('react');
var classNames = require('classnames');
var _util_forwardRefWithStatics = require('../_util/forwardRefWithStatics.js');
var _util_noop = require('../_util/noop.js');
var hooks_useConfig = require('../hooks/useConfig.js');
var hooks_useDefaultProps = require('../hooks/useDefaultProps.js');
var form_defaultProps = require('./defaultProps.js');
var form_FormContext = require('./FormContext.js');
var form_FormItem = require('./FormItem.js');
var form_FormList = require('./FormList.js');
var form_hooks_useForm = require('./hooks/useForm.js');
var form_hooks_useInstance = require('./hooks/useInstance.js');
var form_hooks_useWatch = require('./hooks/useWatch.js');
require('../_chunks/dep-7f3fa935.js');
require('../_chunks/dep-8c880a65.js');
require('../_chunks/dep-98e6819c.js');
require('../_chunks/dep-c290284d.js');
require('hoist-non-react-statics');
require('../config-provider/ConfigContext.js');
require('../_chunks/dep-2ee09518.js');
require('../_chunks/dep-a6fa90dc.js');
require('dayjs');
require('../_chunks/dep-012d9a7c.js');
require('../_chunks/dep-150a112c.js');
require('../_chunks/dep-e360e17f.js');
require('../_chunks/dep-3d05de97.js');
require('../_chunks/dep-4f1729f0.js');
require('../_chunks/dep-a54f202c.js');
require('../_chunks/dep-51284537.js');
require('../_chunks/dep-8a493834.js');
require('../_chunks/dep-9f5b2891.js');
require('../_chunks/dep-338f013a.js');
require('../_chunks/dep-c37c5e17.js');
require('../_chunks/dep-eee1aab9.js');
require('../_chunks/dep-11347ea1.js');
require('../_chunks/dep-a830ce1e.js');
require('../_chunks/dep-770bc0ae.js');
require('../_chunks/dep-aaf27e39.js');
require('../_chunks/dep-b47d855f.js');
require('../_chunks/dep-81bce4e6.js');
require('../_chunks/dep-863f8f4d.js');
require('../_chunks/dep-84e4d0da.js');
require('../_chunks/dep-f78183d7.js');
require('../_chunks/dep-2f45f6c4.js');
require('../_chunks/dep-7bf14076.js');
require('tdesign-icons-react');
require('../hooks/useGlobalIcon.js');
require('../locale/LocalReceiver.js');
require('../_chunks/dep-427f8bc0.js');
require('../config-provider/ConfigProvider.js');
require('../_chunks/dep-ba90997f.js');
require('../_chunks/dep-22026fea.js');
require('../_chunks/dep-551d5940.js');
require('../_chunks/dep-8005b808.js');
require('../_chunks/dep-2add8824.js');
require('./const.js');
require('./formModel.js');
require('../_chunks/dep-6e0da644.js');
require('../_chunks/dep-e41137f7.js');
require('../_chunks/dep-b8d0b96c.js');
require('../_chunks/dep-2be6b7d2.js');
require('./hooks/useFormItemInitialData.js');
require('../_chunks/dep-ce83473b.js');
require('../_chunks/dep-f8c20fd8.js');
require('../_chunks/dep-d4a6a9c1.js');
require('../_chunks/dep-36579bf6.js');
require('../_chunks/dep-b199860d.js');
require('../_chunks/dep-cabd787d.js');
require('../_chunks/dep-363864ae.js');
require('../_chunks/dep-980a01d9.js');
require('../_chunks/dep-6e5e0470.js');
require('../_chunks/dep-cdef7593.js');
require('../_chunks/dep-f5e1d11c.js');
require('./hooks/useFormItemStyle.js');
require('../_util/parseTNode.js');
require('../_chunks/dep-3c628d18.js');
require('./utils/index.js');
require('../_chunks/dep-aa9a2c54.js');
require('../_chunks/dep-77ef3011.js');
require('../_chunks/dep-b36779a4.js');
require('../_chunks/dep-643d7472.js');
require('../_chunks/dep-bc160f5b.js');
require('../_chunks/dep-fb453fc0.js');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

var React__default = /*#__PURE__*/_interopDefaultLegacy(React);
var classNames__default = /*#__PURE__*/_interopDefaultLegacy(classNames);

function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { defineProperty._defineProperty(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
var Form = _util_forwardRefWithStatics["default"](function (originalProps, ref) {
  var _form$getInternalHook, _form$getInternalHook2;
  var _useConfig = hooks_useConfig["default"](),
    classPrefix = _useConfig.classPrefix,
    globalFormConfig = _useConfig.form;
  var props = hooks_useDefaultProps["default"](originalProps, form_defaultProps.formDefaultProps);
  var style = props.style,
    className = props.className,
    labelWidth = props.labelWidth,
    statusIcon = props.statusIcon,
    labelAlign = props.labelAlign,
    layout = props.layout,
    colon = props.colon,
    initialData = props.initialData,
    _props$requiredMark = props.requiredMark,
    requiredMark = _props$requiredMark === void 0 ? globalFormConfig.requiredMark : _props$requiredMark,
    _props$requiredMarkPo = props.requiredMarkPosition,
    requiredMarkPosition = _props$requiredMarkPo === void 0 ? globalFormConfig.requiredMarkPosition : _props$requiredMarkPo,
    scrollToFirstError = props.scrollToFirstError,
    showErrorMessage = props.showErrorMessage,
    resetType = props.resetType,
    rules = props.rules,
    _props$errorMessage = props.errorMessage,
    errorMessage = _props$errorMessage === void 0 ? globalFormConfig.errorMessage : _props$errorMessage,
    disabled = props.disabled,
    children = props.children,
    id = props.id,
    onReset = props.onReset,
    _props$onValuesChange = props.onValuesChange,
    onValuesChange = _props$onValuesChange === void 0 ? _util_noop["default"] : _props$onValuesChange;
  var formClass = classNames__default["default"]("".concat(classPrefix, "-form"), className, defineProperty._defineProperty({}, "".concat(classPrefix, "-form-inline"), layout === "inline"));
  var _useForm = form_hooks_useForm["default"](props.form),
    _useForm2 = slicedToArray._slicedToArray(_useForm, 1),
    form = _useForm2[0];
  var formRef = React.useRef(null);
  var formMapRef = React.useRef(/* @__PURE__ */new Map());
  var floatingFormDataRef = React.useRef({});
  var formInstance = form_hooks_useInstance["default"](props, formRef, formMapRef, floatingFormDataRef, form);
  React.useImperativeHandle(ref, function () {
    return formInstance;
  });
  Object.assign(form, _objectSpread({}, formInstance));
  form === null || form === void 0 || (_form$getInternalHook = form.getInternalHooks) === null || _form$getInternalHook === void 0 || (_form$getInternalHook = _form$getInternalHook.call(form, form_hooks_useForm.HOOK_MARK)) === null || _form$getInternalHook === void 0 || (_form$getInternalHook2 = _form$getInternalHook.setForm) === null || _form$getInternalHook2 === void 0 || _form$getInternalHook2.call(_form$getInternalHook, formInstance);
  React.useEffect(function () {
    var _form$getInternalHook3, _form$getInternalHook4;
    form === null || form === void 0 || (_form$getInternalHook3 = form.getInternalHooks) === null || _form$getInternalHook3 === void 0 || (_form$getInternalHook3 = _form$getInternalHook3.call(form, form_hooks_useForm.HOOK_MARK)) === null || _form$getInternalHook3 === void 0 || (_form$getInternalHook4 = _form$getInternalHook3.flashQueue) === null || _form$getInternalHook4 === void 0 || _form$getInternalHook4.call(_form$getInternalHook3);
  }, [form]);
  function isEventFromSelf(e) {
    return (e === null || e === void 0 ? void 0 : e.target) === formRef.current;
  }
  function onResetHandler(e) {
    var _form$getInternalHook5, _form$getInternalHook6;
    if (!isEventFromSelf(e)) return;
    toConsumableArray._toConsumableArray(formMapRef.current.values()).forEach(function (formItemRef) {
      formItemRef === null || formItemRef === void 0 || formItemRef.current.resetField();
    });
    form === null || form === void 0 || (_form$getInternalHook5 = form.getInternalHooks) === null || _form$getInternalHook5 === void 0 || (_form$getInternalHook5 = _form$getInternalHook5.call(form, form_hooks_useForm.HOOK_MARK)) === null || _form$getInternalHook5 === void 0 || (_form$getInternalHook6 = _form$getInternalHook5.notifyWatch) === null || _form$getInternalHook6 === void 0 || _form$getInternalHook6.call(_form$getInternalHook5, []);
    form.store = {};
    floatingFormDataRef.current = {};
    onReset === null || onReset === void 0 || onReset({
      e: e
    });
  }
  function onSubmitHandler(e) {
    if (!isEventFromSelf(e)) return;
    formInstance.submit(e);
  }
  function onFormItemValueChange(changedValue) {
    var allFields = formInstance.getFieldsValue(true);
    onValuesChange(changedValue, allFields);
  }
  return /* @__PURE__ */React__default["default"].createElement(form_FormContext["default"].Provider, {
    value: {
      form: form,
      labelWidth: labelWidth,
      statusIcon: statusIcon,
      labelAlign: labelAlign,
      layout: layout,
      colon: colon,
      initialData: initialData,
      requiredMark: requiredMark,
      requiredMarkPosition: requiredMarkPosition,
      errorMessage: errorMessage,
      showErrorMessage: showErrorMessage,
      scrollToFirstError: scrollToFirstError,
      resetType: resetType,
      rules: rules,
      disabled: disabled,
      readOnly: props.readOnly || props.readonly,
      formMapRef: formMapRef,
      floatingFormDataRef: floatingFormDataRef,
      onFormItemValueChange: onFormItemValueChange
    }
  }, /* @__PURE__ */React__default["default"].createElement("form", {
    ref: formRef,
    id: id,
    style: style,
    className: formClass,
    onSubmit: onSubmitHandler,
    onReset: onResetHandler
  }, children));
}, {
  useForm: form_hooks_useForm["default"],
  useWatch: form_hooks_useWatch["default"],
  FormItem: form_FormItem["default"],
  FormList: form_FormList["default"]
});
Form.displayName = "Form";

exports["default"] = Form;
