// tdesign-react/cjs/dialog/DialogCard.js
/**
 * tdesign v1.17.1
 * (c) 2026 tdesign
 * @license MIT
 */

'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var objectWithoutProperties = require('../_chunks/dep-e41137f7.js');
var slicedToArray = require('../_chunks/dep-c60a3e4c.js');
var defineProperty = require('../_chunks/dep-c446aa11.js');
var React = require('react');
var classNames = require('classnames');
var tdesignIconsReact = require('tdesign-icons-react');
var _util_parseTNode = require('../_util/parseTNode.js');
var button_index = require('../button/index.js');
var hooks_useConfig = require('../hooks/useConfig.js');
var hooks_useDefaultProps = require('../hooks/useDefaultProps.js');
var hooks_useGlobalIcon = require('../hooks/useGlobalIcon.js');
var locale_LocalReceiver = require('../locale/LocalReceiver.js');
var dialog_defaultProps = require('./defaultProps.js');
var isString = require('../_chunks/dep-427f8bc0.js');
var isObject = require('../_chunks/dep-a830ce1e.js');
var isFunction = require('../_chunks/dep-11347ea1.js');
require('../_chunks/dep-98e6819c.js');
require('../_chunks/dep-7f3fa935.js');
require('../_chunks/dep-c290284d.js');
require('../_chunks/dep-3c628d18.js');
require('../button/Button.js');
require('../hooks/useDomRefCallback.js');
require('../hooks/useRipple.js');
require('../_chunks/dep-062bb1b1.js');
require('../_chunks/dep-ee71ddca.js');
require('../hooks/useAnimation.js');
require('../config-provider/ConfigContext.js');
require('../_chunks/dep-2ee09518.js');
require('../_chunks/dep-a6fa90dc.js');
require('dayjs');
require('../_chunks/dep-012d9a7c.js');
require('../_chunks/dep-150a112c.js');
require('../_chunks/dep-e360e17f.js');
require('../_chunks/dep-3d05de97.js');
require('../_chunks/dep-4f1729f0.js');
require('../_chunks/dep-a54f202c.js');
require('../_chunks/dep-51284537.js');
require('../_chunks/dep-8a493834.js');
require('../_chunks/dep-9f5b2891.js');
require('../_chunks/dep-338f013a.js');
require('../_chunks/dep-c37c5e17.js');
require('../_chunks/dep-eee1aab9.js');
require('../_chunks/dep-770bc0ae.js');
require('../_chunks/dep-aaf27e39.js');
require('../_chunks/dep-b47d855f.js');
require('../_chunks/dep-81bce4e6.js');
require('../_chunks/dep-863f8f4d.js');
require('../_chunks/dep-84e4d0da.js');
require('../_chunks/dep-f78183d7.js');
require('../_chunks/dep-2f45f6c4.js');
require('../loading/index.js');
require('../loading/Loading.js');
require('../common/Portal.js');
require('react-dom');
require('../hooks/useLayoutEffect.js');
require('../loading/defaultProps.js');
require('../loading/gradient.js');
require('../_chunks/dep-6e0da644.js');
require('../_chunks/dep-05bc49e6.js');
require('../_chunks/dep-8c880a65.js');
require('../_chunks/dep-b8d0b96c.js');
require('../loading/plugin.js');
require('../_util/react-render.js');
require('../_chunks/dep-7bf14076.js');
require('../common/PluginContainer.js');
require('../config-provider/ConfigProvider.js');
require('../_chunks/dep-ba90997f.js');
require('../_chunks/dep-22026fea.js');
require('../_chunks/dep-551d5940.js');
require('../_chunks/dep-8005b808.js');
require('../_chunks/dep-2add8824.js');
require('../button/defaultProps.js');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

var React__default = /*#__PURE__*/_interopDefaultLegacy(React);
var classNames__default = /*#__PURE__*/_interopDefaultLegacy(classNames);

var _excluded = ["theme", "header", "closeBtn", "footer", "body", "children", "className", "onCancel", "onConfirm", "onCloseBtnClick", "cancelBtn", "confirmBtn", "confirmLoading", "mode"];
function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { defineProperty._defineProperty(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
var renderDialogButton = function renderDialogButton(btn, defaultProps) {
  var result = null;
  if (isString.isString(btn)) {
    result = /* @__PURE__ */React__default["default"].createElement(button_index.Button, _objectSpread({}, defaultProps), btn);
  } else if (/*#__PURE__*/React.isValidElement(btn)) {
    result = btn;
  } else if (isObject.isObject(btn)) {
    result = /* @__PURE__ */React__default["default"].createElement(button_index.Button, _objectSpread(_objectSpread({}, defaultProps), btn));
  } else if (isFunction.isFunction(btn)) {
    result = btn();
  }
  return result;
};
var DialogCard = /*#__PURE__*/React.forwardRef(function (props, ref) {
  var _useConfig = hooks_useConfig["default"](),
    classPrefix = _useConfig.classPrefix;
  var componentCls = "".concat(classPrefix, "-dialog");
  var _useGlobalIcon = hooks_useGlobalIcon["default"]({
      CloseIcon: tdesignIconsReact.CloseIcon,
      InfoCircleFilledIcon: tdesignIconsReact.InfoCircleFilledIcon,
      CheckCircleFilledIcon: tdesignIconsReact.CheckCircleFilledIcon
    }),
    CloseIcon = _useGlobalIcon.CloseIcon,
    InfoCircleFilledIcon = _useGlobalIcon.InfoCircleFilledIcon,
    CheckCircleFilledIcon = _useGlobalIcon.CheckCircleFilledIcon;
  var _useLocaleReceiver = locale_LocalReceiver.useLocaleReceiver("dialog"),
    _useLocaleReceiver2 = slicedToArray._slicedToArray(_useLocaleReceiver, 2),
    local = _useLocaleReceiver2[0],
    t = _useLocaleReceiver2[1];
  var confirmText = t(local.confirm);
  var cancelText = t(local.cancel);
  var _useDefaultProps = hooks_useDefaultProps["default"](props, dialog_defaultProps.dialogCardDefaultProps),
    theme = _useDefaultProps.theme,
    header = _useDefaultProps.header,
    closeBtn = _useDefaultProps.closeBtn,
    footer = _useDefaultProps.footer,
    body = _useDefaultProps.body,
    children = _useDefaultProps.children,
    className = _useDefaultProps.className,
    onCancel = _useDefaultProps.onCancel,
    onConfirm = _useDefaultProps.onConfirm,
    onCloseBtnClick = _useDefaultProps.onCloseBtnClick,
    _useDefaultProps$canc = _useDefaultProps.cancelBtn,
    cancelBtn = _useDefaultProps$canc === void 0 ? cancelText : _useDefaultProps$canc,
    _useDefaultProps$conf = _useDefaultProps.confirmBtn,
    confirmBtn = _useDefaultProps$conf === void 0 ? confirmText : _useDefaultProps$conf,
    confirmLoading = _useDefaultProps.confirmLoading,
    mode = _useDefaultProps.mode,
    otherProps = objectWithoutProperties._objectWithoutProperties(_useDefaultProps, _excluded);
  var isFullScreen = mode === "full-screen";
  var renderHeaderContent = function renderHeaderContent() {
    var iconMap = {
      info: /* @__PURE__ */React__default["default"].createElement(InfoCircleFilledIcon, {
        className: "".concat(classPrefix, "-is-info")
      }),
      warning: /* @__PURE__ */React__default["default"].createElement(InfoCircleFilledIcon, {
        className: "".concat(classPrefix, "-is-warning")
      }),
      error: /* @__PURE__ */React__default["default"].createElement(InfoCircleFilledIcon, {
        className: "".concat(classPrefix, "-is-error")
      }),
      danger: /* @__PURE__ */React__default["default"].createElement(InfoCircleFilledIcon, {
        className: "".concat(classPrefix, "-is-error")
      }),
      success: /* @__PURE__ */React__default["default"].createElement(CheckCircleFilledIcon, {
        className: "".concat(classPrefix, "-is-success")
      })
    };
    return /* @__PURE__ */React__default["default"].createElement("div", {
      className: "".concat(componentCls, "__header-content")
    }, iconMap[theme], header);
  };
  var renderCloseBtn = function renderCloseBtn() {
    if (!closeBtn) {
      return null;
    }
    var closeIcon = function closeIcon() {
      return closeBtn === true ? /* @__PURE__ */React__default["default"].createElement(CloseIcon, null) : closeBtn;
    };
    return /* @__PURE__ */React__default["default"].createElement("span", {
      className: classNames__default["default"]("".concat(componentCls, "__close"), defineProperty._defineProperty({}, "".concat(componentCls, "__close--fullscreen"), isFullScreen)),
      style: {
        marginLeft: "auto"
      },
      onClick: function onClick(e) {
        return onCloseBtnClick === null || onCloseBtnClick === void 0 ? void 0 : onCloseBtnClick({
          e: e
        });
      }
    }, closeIcon());
  };
  var renderHeader = function renderHeader() {
    return /* @__PURE__ */React__default["default"].createElement("div", {
      className: classNames__default["default"]("".concat(componentCls, "__header"), defineProperty._defineProperty({}, "".concat(componentCls, "__header--fullscreen"), isFullScreen))
    }, renderHeaderContent(), renderCloseBtn());
  };
  var renderFooter = function renderFooter() {
    var defaultFooter = function defaultFooter() {
      var _cancelBtn$props;
      var renderCancelBtn = renderDialogButton(cancelBtn, {
        variant: "outline",
        onClick: function onClick(e) {
          return onCancel === null || onCancel === void 0 ? void 0 : onCancel({
            e: e
          });
        },
        className: classNames__default["default"]("".concat(componentCls, "__cancel"), cancelBtn === null || cancelBtn === void 0 || (_cancelBtn$props = cancelBtn.props) === null || _cancelBtn$props === void 0 ? void 0 : _cancelBtn$props.className)
      });
      var renderConfirmBtn = renderDialogButton(confirmBtn, {
        theme: "primary",
        loading: confirmLoading,
        onClick: function onClick(e) {
          return onConfirm === null || onConfirm === void 0 ? void 0 : onConfirm({
            e: e
          });
        },
        className: classNames__default["default"]("".concat(componentCls, "__confirm"), confirmBtn === null || confirmBtn === void 0 ? void 0 : confirmBtn.className)
      });
      return /* @__PURE__ */React__default["default"].createElement(React__default["default"].Fragment, null, renderCancelBtn, renderConfirmBtn);
    };
    return /* @__PURE__ */React__default["default"].createElement("div", {
      className: classNames__default["default"]("".concat(componentCls, "__footer"), defineProperty._defineProperty({}, "".concat(componentCls, "__footer--fullscreen"), isFullScreen))
    }, _util_parseTNode["default"](footer, null, defaultFooter()));
  };
  return /* @__PURE__ */React__default["default"].createElement("div", _objectSpread(_objectSpread({
    ref: ref
  }, otherProps), {}, {
    className: classNames__default["default"](componentCls, "".concat(componentCls, "--default"), defineProperty._defineProperty({}, "".concat(componentCls, "__fullscreen"), isFullScreen), className)
  }), !!header && renderHeader(), /* @__PURE__ */React__default["default"].createElement("div", {
    className: classNames__default["default"]("".concat(componentCls, "__body"), defineProperty._defineProperty(defineProperty._defineProperty({}, "".concat(componentCls, "__body--fullscreen"), isFullScreen && !!footer), "".concat(componentCls, "__body--fullscreen--without-footer"), isFullScreen && !footer))
  }, body || children), !!footer && renderFooter());
});
DialogCard.displayName = "DialogCard";

exports["default"] = DialogCard;
