// tdesign-react/cjs/dialog/Dialog.js
/**
 * tdesign v1.17.1
 * (c) 2026 tdesign
 * @license MIT
 */

'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var defineProperty = require('../_chunks/dep-c446aa11.js');
var slicedToArray = require('../_chunks/dep-c60a3e4c.js');
var objectWithoutProperties = require('../_chunks/dep-e41137f7.js');
var React = require('react');
var reactTransitionGroup = require('react-transition-group');
var classNames = require('classnames');
var log = require('../_chunks/dep-3c628d18.js');
var helper = require('../_chunks/dep-6e0da644.js');
var common_Portal = require('../common/Portal.js');
var hooks_useAttach = require('../hooks/useAttach.js');
var hooks_useConfig = require('../hooks/useConfig.js');
var hooks_useDeepEffect = require('../hooks/useDeepEffect.js');
var hooks_useDefaultProps = require('../hooks/useDefaultProps.js');
var hooks_useSetState = require('../hooks/useSetState.js');
var locale_LocalReceiver = require('../locale/LocalReceiver.js');
var dialog_defaultProps = require('./defaultProps.js');
var dialog_DialogCard = require('./DialogCard.js');
var dialog_hooks_useDialogDrag = require('./hooks/useDialogDrag.js');
var dialog_hooks_useDialogEsc = require('./hooks/useDialogEsc.js');
var dialog_hooks_useDialogPosition = require('./hooks/useDialogPosition.js');
var dialog_hooks_useLockStyle = require('./hooks/useLockStyle.js');
var isUndefined = require('../_chunks/dep-bfa39549.js');
require('../_chunks/dep-c290284d.js');
require('../_chunks/dep-98e6819c.js');
require('../_chunks/dep-7f3fa935.js');
require('../_chunks/dep-05bc49e6.js');
require('../_chunks/dep-8c880a65.js');
require('../_chunks/dep-b8d0b96c.js');
require('../_chunks/dep-3d05de97.js');
require('../_chunks/dep-4f1729f0.js');
require('../_chunks/dep-427f8bc0.js');
require('../_chunks/dep-a54f202c.js');
require('react-dom');
require('../_chunks/dep-ee71ddca.js');
require('../hooks/useLayoutEffect.js');
require('../config-provider/ConfigContext.js');
require('../_chunks/dep-2ee09518.js');
require('../_chunks/dep-a6fa90dc.js');
require('dayjs');
require('../_chunks/dep-012d9a7c.js');
require('../_chunks/dep-150a112c.js');
require('../_chunks/dep-e360e17f.js');
require('../_chunks/dep-51284537.js');
require('../_chunks/dep-8a493834.js');
require('../_chunks/dep-9f5b2891.js');
require('../_chunks/dep-338f013a.js');
require('../_chunks/dep-c37c5e17.js');
require('../_chunks/dep-eee1aab9.js');
require('../_chunks/dep-11347ea1.js');
require('../_chunks/dep-a830ce1e.js');
require('../_chunks/dep-770bc0ae.js');
require('../_chunks/dep-aaf27e39.js');
require('../_chunks/dep-b47d855f.js');
require('../_chunks/dep-81bce4e6.js');
require('../_chunks/dep-863f8f4d.js');
require('../_chunks/dep-84e4d0da.js');
require('../_chunks/dep-f78183d7.js');
require('../_chunks/dep-2f45f6c4.js');
require('../_chunks/dep-77ef3011.js');
require('../_chunks/dep-b36779a4.js');
require('../_chunks/dep-551d5940.js');
require('../_chunks/dep-8005b808.js');
require('../_chunks/dep-2add8824.js');
require('../config-provider/ConfigProvider.js');
require('../_chunks/dep-ba90997f.js');
require('../_chunks/dep-22026fea.js');
require('tdesign-icons-react');
require('../_util/parseTNode.js');
require('../button/index.js');
require('../button/Button.js');
require('../hooks/useDomRefCallback.js');
require('../hooks/useRipple.js');
require('../_chunks/dep-062bb1b1.js');
require('../hooks/useAnimation.js');
require('../loading/index.js');
require('../loading/Loading.js');
require('../loading/defaultProps.js');
require('../loading/gradient.js');
require('../loading/plugin.js');
require('../_util/react-render.js');
require('../_chunks/dep-7bf14076.js');
require('../common/PluginContainer.js');
require('../button/defaultProps.js');
require('../hooks/useGlobalIcon.js');
require('../hooks/useMouseEvent.js');
require('../hooks/useLatest.js');
require('../_chunks/dep-ea91dbfd.js');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

var React__default = /*#__PURE__*/_interopDefaultLegacy(React);
var classNames__default = /*#__PURE__*/_interopDefaultLegacy(classNames);

var _excluded = ["children"],
  _excluded2 = ["className", "dialogClassName", "style", "width", "mode", "zIndex", "visible", "attach", "onBeforeOpen", "onBeforeClose", "onOpened", "onCancel", "onConfirm", "onClose", "onClosed", "isPlugin", "draggable", "onOverlayClick", "onEscKeydown", "closeOnEscKeydown", "confirmOnEnter", "showOverlay", "showInAttachedElement", "closeOnOverlayClick", "destroyOnClose", "preventScrollThrough", "onCloseBtnClick", "forceRender", "lazy"];
function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { defineProperty._defineProperty(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
var Dialog = /*#__PURE__*/React.forwardRef(function (originalProps, ref) {
  var props = hooks_useDefaultProps["default"](originalProps, dialog_defaultProps.dialogDefaultProps);
  var children = props.children,
    restProps = objectWithoutProperties._objectWithoutProperties(props, _excluded);
  var _useConfig = hooks_useConfig["default"](),
    classPrefix = _useConfig.classPrefix;
  var componentCls = "".concat(classPrefix, "-dialog");
  var wrapRef = React.useRef(null);
  var maskRef = React.useRef(null);
  var contentClickRef = React.useRef(false);
  var dialogCardRef = React.useRef(null);
  var dialogPosition = React.useRef(null);
  var portalRef = React.useRef(null);
  var _useSetState = hooks_useSetState["default"](_objectSpread({
      isPlugin: false
    }, restProps)),
    _useSetState2 = slicedToArray._slicedToArray(_useSetState, 2),
    state = _useSetState2[0],
    setState = _useSetState2[1];
  var _useLocaleReceiver = locale_LocalReceiver.useLocaleReceiver("dialog"),
    _useLocaleReceiver2 = slicedToArray._slicedToArray(_useLocaleReceiver, 1),
    local = _useLocaleReceiver2[0];
  var className = state.className,
    dialogClassName = state.dialogClassName,
    style = state.style,
    width = state.width,
    mode = state.mode,
    zIndex = state.zIndex,
    visible = state.visible,
    attach = state.attach,
    onBeforeOpen = state.onBeforeOpen,
    onBeforeClose = state.onBeforeClose,
    onOpened = state.onOpened,
    onCancel = state.onCancel,
    onConfirm = state.onConfirm,
    onClose = state.onClose,
    onClosed = state.onClosed,
    isPlugin = state.isPlugin,
    draggable = state.draggable,
    onOverlayClick = state.onOverlayClick,
    onEscKeydown = state.onEscKeydown,
    closeOnEscKeydown = state.closeOnEscKeydown,
    confirmOnEnter = state.confirmOnEnter,
    showOverlay = state.showOverlay,
    showInAttachedElement = state.showInAttachedElement,
    closeOnOverlayClick = state.closeOnOverlayClick,
    destroyOnClose = state.destroyOnClose,
    preventScrollThrough = state.preventScrollThrough,
    onCloseBtnClick = state.onCloseBtnClick,
    forceRender = state.forceRender,
    lazy = state.lazy,
    restState = objectWithoutProperties._objectWithoutProperties(state, _excluded2);
  var isModeless = mode === "modeless";
  var isFullScreen = mode === "full-screen";
  var dialogAttach = hooks_useAttach["default"]("dialog", attach);
  var _useState = React.useState(false),
    _useState2 = slicedToArray._slicedToArray(_useState, 2),
    dialogAnimationVisible = _useState2[0],
    setDialogAnimationVisible = _useState2[1];
  dialog_hooks_useLockStyle["default"]({
    preventScrollThrough: preventScrollThrough,
    visible: visible,
    mode: mode,
    showInAttachedElement: showInAttachedElement
  });
  var _useDialogEsc = dialog_hooks_useDialogEsc["default"](visible, wrapRef),
    activateDialog = _useDialogEsc.activateDialog;
  var _useDialogPosition = dialog_hooks_useDialogPosition["default"](dialogCardRef),
    applyTransform = _useDialogPosition.applyTransform;
  var _useDialogDrag = dialog_hooks_useDialogDrag["default"]({
      dialogCardRef: dialogCardRef,
      canDraggable: !isFullScreen && draggable
    }),
    isInputInteracting = _useDialogDrag.isInputInteracting;
  hooks_useDeepEffect["default"](function () {
    if (isPlugin) return;
    setState(function (prevState) {
      return _objectSpread(_objectSpread({}, prevState), props);
    });
  }, [props, setState]);
  React.useEffect(function () {
    if (!dialogCardRef.current) return;
    dialogCardRef.current.style.display = dialogAnimationVisible ? "block" : "none";
    if (dialogAnimationVisible) {
      activateDialog();
      applyTransform();
    }
  }, [dialogAnimationVisible]);
  React.useImperativeHandle(ref, function () {
    return {
      show: function show() {
        setState({
          visible: true
        });
      },
      hide: function hide() {
        setState({
          visible: false
        });
      },
      setConfirmLoading: function setConfirmLoading(loading) {
        setState({
          confirmLoading: loading
        });
      },
      destroy: function destroy() {
        setState({
          visible: false,
          destroyOnClose: true
        });
      },
      update: function update(newOptions) {
        setState(function (prevState) {
          return _objectSpread(_objectSpread({}, prevState), newOptions);
        });
      }
    };
  });
  if (props.mode === "normal") {
    log.log.error("Dialog", 'mode="normal" is not supported, please use DialogCard.');
    return /* @__PURE__ */React__default["default"].createElement(dialog_DialogCard["default"], _objectSpread({}, props));
  }
  var onMaskClick = function onMaskClick(e) {
    if (!showOverlay || isModeless || isInputInteracting.current) return;
    if (contentClickRef.current) {
      contentClickRef.current = false;
      return;
    }
    if (e.target !== dialogPosition.current) return;
    onOverlayClick === null || onOverlayClick === void 0 || onOverlayClick({
      e: e
    });
    if (closeOnOverlayClick !== null && closeOnOverlayClick !== void 0 ? closeOnOverlayClick : local.closeOnOverlayClick) {
      onClose === null || onClose === void 0 || onClose({
        e: e,
        trigger: "overlay"
      });
    }
  };
  var handleCancel = function handleCancel(_ref) {
    var e = _ref.e;
    onCancel === null || onCancel === void 0 || onCancel({
      e: e
    });
    onClose === null || onClose === void 0 || onClose({
      e: e,
      trigger: "cancel"
    });
  };
  var handleClose = function handleClose(_ref2) {
    var e = _ref2.e;
    onCloseBtnClick === null || onCloseBtnClick === void 0 || onCloseBtnClick({
      e: e
    });
    onClose === null || onClose === void 0 || onClose({
      e: e,
      trigger: "close-btn"
    });
  };
  var handleKeyDown = function handleKeyDown(e) {
    if (e.key === "Escape") {
      e.stopPropagation();
      onEscKeydown === null || onEscKeydown === void 0 || onEscKeydown({
        e: e
      });
      if (closeOnEscKeydown !== null && closeOnEscKeydown !== void 0 ? closeOnEscKeydown : local.closeOnEscKeydown) {
        onClose === null || onClose === void 0 || onClose({
          e: e,
          trigger: "esc"
        });
      }
    } else if (e.key === "Enter" || e.key === "NumpadEnter") {
      e.stopPropagation();
      confirmOnEnter && (onConfirm === null || onConfirm === void 0 ? void 0 : onConfirm({
        e: e
      }));
    }
  };
  var onAnimateStart = function onAnimateStart() {
    onBeforeOpen === null || onBeforeOpen === void 0 || onBeforeOpen();
    if (!wrapRef.current) return;
    wrapRef.current.style.display = "block";
  };
  var onAnimateLeave = function onAnimateLeave() {
    onClosed === null || onClosed === void 0 || onClosed();
    if (!wrapRef.current) return;
    wrapRef.current.style.display = "none";
  };
  var onInnerAnimateStart = function onInnerAnimateStart() {
    setDialogAnimationVisible(true);
  };
  var onInnerAnimateLeave = function onInnerAnimateLeave() {
    setDialogAnimationVisible(false);
  };
  var renderMask = function renderMask() {
    if (mode !== "modal") return null;
    return showOverlay ? /* @__PURE__ */React__default["default"].createElement(reactTransitionGroup.CSSTransition, {
      "in": visible,
      appear: true,
      timeout: 300,
      classNames: "".concat(componentCls, "-fade"),
      mountOnEnter: true,
      unmountOnExit: true,
      nodeRef: maskRef
    }, /* @__PURE__ */React__default["default"].createElement("div", {
      ref: maskRef,
      className: "".concat(componentCls, "__mask")
    })) : null;
  };
  return /* @__PURE__ */React__default["default"].createElement(reactTransitionGroup.CSSTransition, {
    "in": visible,
    appear: true,
    timeout: 300,
    mountOnEnter: isUndefined.isUndefined(forceRender) ? lazy : !forceRender,
    unmountOnExit: destroyOnClose,
    nodeRef: portalRef,
    onEnter: onAnimateStart,
    onEntered: onOpened,
    onExit: onBeforeClose,
    onExited: onAnimateLeave
  }, /* @__PURE__ */React__default["default"].createElement(common_Portal["default"], {
    attach: dialogAttach,
    ref: portalRef
  }, /* @__PURE__ */React__default["default"].createElement("div", {
    ref: wrapRef,
    className: classNames__default["default"](className, "".concat(componentCls, "__ctx"), "".concat(componentCls, "__").concat(mode), defineProperty._defineProperty(defineProperty._defineProperty(defineProperty._defineProperty({}, "".concat(componentCls, "__ctx--fixed"), !showInAttachedElement), "".concat(componentCls, "__ctx--absolute"), showInAttachedElement), "".concat(componentCls, "__ctx--modeless"), isModeless)),
    style: {
      zIndex: zIndex
    },
    onKeyDown: handleKeyDown,
    tabIndex: 0
  }, renderMask(), /* @__PURE__ */React__default["default"].createElement("div", {
    className: "".concat(componentCls, "__wrap")
  }, /* @__PURE__ */React__default["default"].createElement("div", {
    ref: dialogPosition,
    className: classNames__default["default"](isFullScreen ? "".concat(componentCls, "__position_fullscreen") : "".concat(componentCls, "__position"), defineProperty._defineProperty(defineProperty._defineProperty({}, "".concat(componentCls, "--top"), !isFullScreen && (!!props.top || props.placement === "top")), "".concat(componentCls, "--center"), !isFullScreen && props.placement === "center" && !props.top)),
    style: {
      paddingTop: isFullScreen ? void 0 : helper.pxCompat(props.top)
    },
    onClick: onMaskClick
  }, /* @__PURE__ */React__default["default"].createElement(reactTransitionGroup.CSSTransition, {
    "in": visible,
    appear: true,
    timeout: 300,
    classNames: "".concat(componentCls, "-zoom"),
    nodeRef: dialogCardRef,
    onEnter: onInnerAnimateStart,
    onExited: onInnerAnimateLeave
  }, /* @__PURE__ */React__default["default"].createElement(dialog_DialogCard["default"], _objectSpread(_objectSpread({
    ref: dialogCardRef
  }, restState), {}, {
    mode: mode,
    className: classNames__default["default"](dialogClassName),
    style: _objectSpread(_objectSpread({}, style), {}, {
      width: helper.pxCompat(width || (style === null || style === void 0 ? void 0 : style.width))
    }),
    onConfirm: onConfirm,
    onCancel: handleCancel,
    onCloseBtnClick: handleClose
  }), children)))))));
});
Dialog.displayName = "Dialog";

exports["default"] = Dialog;
