// tdesign-react/cjs/button/Button.js
/**
 * tdesign v1.17.1
 * (c) 2026 tdesign
 * @license MIT
 */

'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var defineProperty = require('../_chunks/dep-c446aa11.js');
var slicedToArray = require('../_chunks/dep-c60a3e4c.js');
var objectWithoutProperties = require('../_chunks/dep-e41137f7.js');
var React = require('react');
var classNames = require('classnames');
var _util_parseTNode = require('../_util/parseTNode.js');
var hooks_useConfig = require('../hooks/useConfig.js');
var hooks_useDefaultProps = require('../hooks/useDefaultProps.js');
var hooks_useDomRefCallback = require('../hooks/useDomRefCallback.js');
var hooks_useRipple = require('../hooks/useRipple.js');
var loading_index = require('../loading/index.js');
var button_defaultProps = require('./defaultProps.js');
require('../_chunks/dep-c290284d.js');
require('../_chunks/dep-98e6819c.js');
require('../_chunks/dep-7f3fa935.js');
require('../_chunks/dep-3c628d18.js');
require('../_chunks/dep-11347ea1.js');
require('../_chunks/dep-3d05de97.js');
require('../_chunks/dep-a830ce1e.js');
require('../config-provider/ConfigContext.js');
require('../_chunks/dep-2ee09518.js');
require('../_chunks/dep-a6fa90dc.js');
require('dayjs');
require('../_chunks/dep-012d9a7c.js');
require('../_chunks/dep-150a112c.js');
require('../_chunks/dep-e360e17f.js');
require('../_chunks/dep-4f1729f0.js');
require('../_chunks/dep-a54f202c.js');
require('../_chunks/dep-51284537.js');
require('../_chunks/dep-8a493834.js');
require('../_chunks/dep-9f5b2891.js');
require('../_chunks/dep-338f013a.js');
require('../_chunks/dep-c37c5e17.js');
require('../_chunks/dep-eee1aab9.js');
require('../_chunks/dep-770bc0ae.js');
require('../_chunks/dep-aaf27e39.js');
require('../_chunks/dep-b47d855f.js');
require('../_chunks/dep-81bce4e6.js');
require('../_chunks/dep-863f8f4d.js');
require('../_chunks/dep-84e4d0da.js');
require('../_chunks/dep-f78183d7.js');
require('../_chunks/dep-2f45f6c4.js');
require('../_chunks/dep-062bb1b1.js');
require('../_chunks/dep-ee71ddca.js');
require('../_chunks/dep-427f8bc0.js');
require('../hooks/useAnimation.js');
require('../loading/Loading.js');
require('../common/Portal.js');
require('react-dom');
require('../hooks/useLayoutEffect.js');
require('../loading/defaultProps.js');
require('../loading/gradient.js');
require('../_chunks/dep-6e0da644.js');
require('../_chunks/dep-05bc49e6.js');
require('../_chunks/dep-8c880a65.js');
require('../_chunks/dep-b8d0b96c.js');
require('../loading/plugin.js');
require('../_util/react-render.js');
require('../_chunks/dep-7bf14076.js');
require('../common/PluginContainer.js');
require('../config-provider/ConfigProvider.js');
require('../_chunks/dep-ba90997f.js');
require('../_chunks/dep-22026fea.js');
require('../_chunks/dep-551d5940.js');
require('../_chunks/dep-8005b808.js');
require('../_chunks/dep-2add8824.js');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

var React__default = /*#__PURE__*/_interopDefaultLegacy(React);
var classNames__default = /*#__PURE__*/_interopDefaultLegacy(classNames);

var _excluded = ["type", "theme", "variant", "icon", "disabled", "loading", "size", "block", "ghost", "shape", "children", "content", "className", "suffix", "href", "tag", "onClick"];
function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { defineProperty._defineProperty(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
var Button = /*#__PURE__*/React.forwardRef(function (originProps, ref) {
  var props = hooks_useDefaultProps["default"](originProps, button_defaultProps.buttonDefaultProps);
  var type = props.type,
    theme = props.theme,
    variant = props.variant,
    icon = props.icon,
    disabled = props.disabled,
    loading = props.loading,
    size = props.size,
    block = props.block,
    ghost = props.ghost,
    shape = props.shape,
    children = props.children,
    content = props.content,
    className = props.className,
    suffix = props.suffix,
    href = props.href,
    tag = props.tag,
    onClick = props.onClick,
    buttonProps = objectWithoutProperties._objectWithoutProperties(props, _excluded);
  var _useConfig = hooks_useConfig["default"](),
    classPrefix = _useConfig.classPrefix;
  var _useDomRefCallback = hooks_useDomRefCallback["default"](),
    _useDomRefCallback2 = slicedToArray._slicedToArray(_useDomRefCallback, 2),
    btnDom = _useDomRefCallback2[0],
    setRefCurrent = _useDomRefCallback2[1];
  hooks_useRipple["default"]((ref === null || ref === void 0 ? void 0 : ref.current) || btnDom);
  var renderChildren = content !== null && content !== void 0 ? content : children;
  var iconNode = icon;
  if (loading) iconNode = /* @__PURE__ */React__default["default"].createElement(loading_index.Loading, {
    loading: loading,
    inheritColor: true
  });
  var renderTheme = React.useMemo(function () {
    if (!theme) {
      if (variant === "base") return "primary";
      return "default";
    }
    return theme;
  }, [theme, variant]);
  var renderTag = React.useMemo(function () {
    if (!tag && href && !disabled) return "a";
    if (!tag && disabled) return "div";
    return tag || "button";
  }, [tag, href, disabled]);
  return /*#__PURE__*/React__default["default"].createElement(renderTag, _objectSpread(_objectSpread({}, buttonProps), {}, {
    href: href,
    type: type,
    ref: ref || setRefCurrent,
    disabled: disabled || loading,
    className: classNames__default["default"](className, ["".concat(classPrefix, "-button"), "".concat(classPrefix, "-button--theme-").concat(renderTheme), "".concat(classPrefix, "-button--variant-").concat(variant)], defineProperty._defineProperty(defineProperty._defineProperty(defineProperty._defineProperty(defineProperty._defineProperty(defineProperty._defineProperty(defineProperty._defineProperty(defineProperty._defineProperty({}, "".concat(classPrefix, "-button--shape-").concat(shape), shape !== "rectangle"), "".concat(classPrefix, "-button--ghost"), ghost), "".concat(classPrefix, "-is-loading"), loading), "".concat(classPrefix, "-is-disabled"), disabled), "".concat(classPrefix, "-size-s"), size === "small"), "".concat(classPrefix, "-size-l"), size === "large"), "".concat(classPrefix, "-size-full-width"), block)),
    onClick: !disabled && !loading ? onClick : void 0
  }), /* @__PURE__ */React__default["default"].createElement(React__default["default"].Fragment, null, iconNode, renderChildren && /* @__PURE__ */React__default["default"].createElement("span", {
    className: "".concat(classPrefix, "-button__text")
  }, renderChildren), suffix && /* @__PURE__ */React__default["default"].createElement("span", {
    className: "".concat(classPrefix, "-button__suffix")
  }, _util_parseTNode["default"](suffix))));
});
Button.displayName = "Button";

exports["default"] = Button;
