// tdesign-react/cjs/select/hooks/useKeyboardControl.js
/**
 * tdesign v1.17.1
 * (c) 2026 tdesign
 * @license MIT
 */

'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var slicedToArray = require('../../_chunks/dep-c60a3e4c.js');
var React = require('react');
var hooks_useConfig = require('../../hooks/useConfig.js');
var select_base_Option = require('../../_chunks/dep-33de9795.js');
require('../../_chunks/dep-98e6819c.js');
require('../../_chunks/dep-7f3fa935.js');
require('../../config-provider/ConfigContext.js');
require('../../_chunks/dep-c446aa11.js');
require('../../_chunks/dep-c290284d.js');
require('../../_chunks/dep-2ee09518.js');
require('../../_chunks/dep-a6fa90dc.js');
require('dayjs');
require('../../_chunks/dep-012d9a7c.js');
require('../../_chunks/dep-150a112c.js');
require('../../_chunks/dep-e360e17f.js');
require('../../_chunks/dep-3d05de97.js');
require('../../_chunks/dep-4f1729f0.js');
require('../../_chunks/dep-a54f202c.js');
require('../../_chunks/dep-51284537.js');
require('../../_chunks/dep-8a493834.js');
require('../../_chunks/dep-9f5b2891.js');
require('../../_chunks/dep-338f013a.js');
require('../../_chunks/dep-c37c5e17.js');
require('../../_chunks/dep-eee1aab9.js');
require('../../_chunks/dep-11347ea1.js');
require('../../_chunks/dep-a830ce1e.js');
require('../../_chunks/dep-770bc0ae.js');
require('../../_chunks/dep-aaf27e39.js');
require('../../_chunks/dep-b47d855f.js');
require('../../_chunks/dep-81bce4e6.js');
require('../../_chunks/dep-863f8f4d.js');
require('../../_chunks/dep-84e4d0da.js');
require('../../_chunks/dep-f78183d7.js');
require('../../_chunks/dep-2f45f6c4.js');
require('../../_chunks/dep-05bc49e6.js');
require('../../_chunks/dep-8c880a65.js');
require('classnames');
require('../../hooks/useDomRefCallback.js');
require('../../hooks/useRipple.js');
require('../../_chunks/dep-062bb1b1.js');
require('../../_chunks/dep-ee71ddca.js');
require('../../_chunks/dep-427f8bc0.js');
require('../../hooks/useAnimation.js');
require('../base/OptionGroup.js');
require('../../hooks/useDefaultProps.js');
require('../defaultProps.js');
require('../../_chunks/dep-f8c20fd8.js');
require('../../_chunks/dep-d4a6a9c1.js');
require('../../_chunks/dep-36579bf6.js');
require('../../_chunks/dep-b199860d.js');
require('../../_chunks/dep-b8d0b96c.js');

function useKeyboardControl(_ref) {
  var max = _ref.max,
    multiple = _ref.multiple,
    keys = _ref.keys,
    value = _ref.value,
    valueType = _ref.valueType,
    handleChange = _ref.handleChange,
    innerPopupVisible = _ref.innerPopupVisible,
    handlePopupVisibleChange = _ref.handlePopupVisibleChange,
    displayOptions = _ref.displayOptions,
    onCheckAllChange = _ref.onCheckAllChange,
    selectInputRef = _ref.selectInputRef,
    toggleIsScrolling = _ref.toggleIsScrolling;
  var _useConfig = hooks_useConfig["default"](),
    classPrefix = _useConfig.classPrefix;
  var isCheckAll = React.useRef(false);
  var _useState = React.useState(-1),
    _useState2 = slicedToArray._slicedToArray(_useState, 2),
    hoverIndex = _useState2[0],
    changeHoverIndex = _useState2[1];
  var isObjectType = React.useMemo(function () {
    return valueType === "object";
  }, [valueType]);
  var _useMemo = React.useMemo(function () {
      return select_base_Option.getKeyMapping(keys);
    }, [keys]),
    valueKey = _useMemo.valueKey,
    disabledKey = _useMemo.disabledKey;
  React.useEffect(function () {
    if (!innerPopupVisible) {
      changeHoverIndex(-1);
    } else if (!multiple) {
      var index = displayOptions.findIndex(function (option) {
        return option.value === value;
      });
      changeHoverIndex(index >= 0 ? index : -1);
    } else {
      changeHoverIndex(-1);
    }
  }, [innerPopupVisible, displayOptions]);
  React.useEffect(function () {
    if (!Array.isArray(value)) return;
    isCheckAll.current = value.length === displayOptions.filter(function (v) {
      return !((v.disabled || v.checkAll) && !value.includes(v.value));
    }).length;
  }, [value, displayOptions]);
  var handleKeyboardScroll = function handleKeyboardScroll(hoverIndex2) {
    var popupContent = selectInputRef.current.getPopupContentElement();
    var optionSelector = ".".concat(classPrefix, "-select-option");
    var selector = ".".concat(classPrefix, "-select-option__hover");
    var firstSelectedNode = popupContent.querySelector(selector);
    if (firstSelectedNode) {
      toggleIsScrolling(true);
      var scrollHeight = popupContent.querySelector(optionSelector).clientHeight * hoverIndex2;
      popupContent.scrollTo({
        top: scrollHeight,
        behavior: "smooth"
      });
    }
  };
  var handleKeyDown = function handleKeyDown(_value, _ref2) {
    var _displayOptions$newIn, _displayOptions$newIn2;
    var e = _ref2.e;
    var optionsListLength = displayOptions.length;
    var newIndex = hoverIndex;
    switch (e.code) {
      case "ArrowUp":
        e.preventDefault();
        if (hoverIndex === -1) newIndex = 0;else if (hoverIndex === 0 || hoverIndex > optionsListLength - 1) newIndex = optionsListLength - 1;else newIndex -= 1;
        if ((_displayOptions$newIn = displayOptions[newIndex]) !== null && _displayOptions$newIn !== void 0 && _displayOptions$newIn[disabledKey]) newIndex -= 1;
        changeHoverIndex(newIndex);
        handleKeyboardScroll(newIndex);
        break;
      case "ArrowDown":
        e.preventDefault();
        if (hoverIndex === -1 || hoverIndex >= optionsListLength - 1) newIndex = 0;else newIndex += 1;
        if ((_displayOptions$newIn2 = displayOptions[newIndex]) !== null && _displayOptions$newIn2 !== void 0 && _displayOptions$newIn2.disabled) newIndex += 1;
        changeHoverIndex(newIndex);
        handleKeyboardScroll(newIndex);
        break;
      case "Enter":
        {
          if (!innerPopupVisible) {
            handlePopupVisibleChange(true, {
              e: e
            });
            break;
          }
          if (hoverIndex === -1) return;
          if (displayOptions[hoverIndex].checkAll) {
            onCheckAllChange(!isCheckAll.current, e);
            return;
          }
          var selectedOptions = displayOptions[hoverIndex];
          var optionValue = isObjectType ? selectedOptions : selectedOptions[valueKey];
          if (!multiple) {
            handleChange(optionValue, {
              trigger: "check",
              e: e
            });
            handlePopupVisibleChange(false, {
              e: e
            });
            changeHoverIndex(-1);
            handleKeyboardScroll(0);
          } else {
            var valueIndex = value.indexOf(optionValue);
            var isSelected = valueIndex > -1;
            var values = select_base_Option.getSelectValueArr(value, optionValue, isSelected, valueType, keys, selectedOptions);
            if (max > 0 && values.length > max) return;
            handleChange(values, {
              trigger: !isSelected ? "check" : "uncheck",
              e: e
            });
          }
          break;
        }
      case "Escape":
        handlePopupVisibleChange(false, {
          e: e
        });
        changeHoverIndex(-1);
        handleKeyboardScroll(0);
    }
  };
  return {
    hoverIndex: hoverIndex,
    handleKeyDown: handleKeyDown
  };
}

exports["default"] = useKeyboardControl;
