// tdesign-react/cjs/select/hooks/usePanelVirtualScroll.js
/**
 * tdesign v1.17.1
 * (c) 2026 tdesign
 * @license MIT
 */

'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var React = require('react');
var hooks_useVirtualScroll = require('../../hooks/useVirtualScroll.js');
require('../../_chunks/dep-c60a3e4c.js');
require('../../_chunks/dep-98e6819c.js');
require('../../_chunks/dep-7f3fa935.js');
require('../../_chunks/dep-aa9a2c54.js');
require('../../_chunks/dep-77ef3011.js');
require('../../_chunks/dep-150a112c.js');
require('../../_chunks/dep-e360e17f.js');
require('../../_chunks/dep-3d05de97.js');
require('../../_chunks/dep-c290284d.js');
require('../../_chunks/dep-4f1729f0.js');
require('../../_chunks/dep-a54f202c.js');
require('../../_chunks/dep-51284537.js');
require('../../_chunks/dep-8a493834.js');
require('../../_chunks/dep-9f5b2891.js');
require('../../_chunks/dep-338f013a.js');
require('../../_chunks/dep-c37c5e17.js');
require('../../_chunks/dep-eee1aab9.js');
require('../../_chunks/dep-11347ea1.js');
require('../../_chunks/dep-a830ce1e.js');
require('../../_chunks/dep-770bc0ae.js');
require('../../_chunks/dep-b36779a4.js');
require('../../_chunks/dep-551d5940.js');
require('../../_chunks/dep-8005b808.js');
require('../../_chunks/dep-2add8824.js');
require('../../_chunks/dep-863f8f4d.js');

var usePanelVirtualScroll = function usePanelVirtualScroll(_ref) {
  var popupContentRef = _ref.popupContentRef,
    scroll = _ref.scroll,
    options = _ref.options,
    size = _ref.size;
  var scrollThreshold = (scroll === null || scroll === void 0 ? void 0 : scroll.threshold) || 100;
  var scrollType = scroll === null || scroll === void 0 ? void 0 : scroll.type;
  var enableVirtual = React.useMemo(function () {
    return scrollType === "virtual";
  }, [scrollType]);
  var isVirtual = React.useMemo(function () {
    return enableVirtual && (options === null || options === void 0 ? void 0 : options.length) > scrollThreshold;
  }, [enableVirtual, options === null || options === void 0 ? void 0 : options.length, scrollThreshold]);
  var scrollParams = React.useMemo(function () {
    var heightMap = {
      small: 20,
      medium: 28,
      large: 36
    };
    var rowHeight = heightMap[size] || 28;
    return {
      type: "virtual",
      isFixedRowHeight: (scroll === null || scroll === void 0 ? void 0 : scroll.isFixedRowHeight) || false,
      rowHeight: (scroll === null || scroll === void 0 ? void 0 : scroll.rowHeight) || rowHeight,
      bufferSize: (scroll === null || scroll === void 0 ? void 0 : scroll.bufferSize) || 20,
      threshold: scrollThreshold
    };
  }, [scroll, scrollThreshold, size]);
  var _useVirtualScroll = hooks_useVirtualScroll["default"](popupContentRef, {
      enable: enableVirtual,
      data: options || [],
      scroll: scrollParams
    }),
    _useVirtualScroll$vis = _useVirtualScroll.visibleData,
    visibleData = _useVirtualScroll$vis === void 0 ? null : _useVirtualScroll$vis,
    _useVirtualScroll$han = _useVirtualScroll.handleScroll,
    handleVirtualScroll = _useVirtualScroll$han === void 0 ? null : _useVirtualScroll$han,
    _useVirtualScroll$scr = _useVirtualScroll.scrollHeight,
    scrollHeight = _useVirtualScroll$scr === void 0 ? null : _useVirtualScroll$scr,
    _useVirtualScroll$tra = _useVirtualScroll.translateY,
    translateY = _useVirtualScroll$tra === void 0 ? null : _useVirtualScroll$tra,
    _useVirtualScroll$han2 = _useVirtualScroll.handleRowMounted,
    handleRowMounted = _useVirtualScroll$han2 === void 0 ? null : _useVirtualScroll$han2;
  var lastScrollY = -1;
  var onInnerVirtualScroll = React.useCallback(function (e) {
    if (!isVirtual) {
      return;
    }
    var target = e.target;
    var top = target.scrollTop;
    if (Math.abs(lastScrollY - top) > 5) {
      handleVirtualScroll();
      lastScrollY = top;
    } else {
      lastScrollY = -1;
    }
  }, [isVirtual, handleVirtualScroll]);
  React.useEffect(function () {
    var popupContentDom = popupContentRef === null || popupContentRef === void 0 ? void 0 : popupContentRef.current;
    if (isVirtual) {
      var _popupContentDom$addE;
      popupContentDom === null || popupContentDom === void 0 || (_popupContentDom$addE = popupContentDom.addEventListener) === null || _popupContentDom$addE === void 0 || _popupContentDom$addE.call(popupContentDom, "scroll", onInnerVirtualScroll);
    }
    return function () {
      if (isVirtual) {
        var _popupContentDom$remo;
        popupContentDom === null || popupContentDom === void 0 || (_popupContentDom$remo = popupContentDom.removeEventListener) === null || _popupContentDom$remo === void 0 || _popupContentDom$remo.call(popupContentDom, "scroll", onInnerVirtualScroll);
      }
    };
  }, [isVirtual, onInnerVirtualScroll, popupContentRef.current]);
  var cursorStyle = {
    position: "absolute",
    width: "1px",
    height: "1px",
    transition: "transform 0.2s",
    transform: "translate(0, ".concat(scrollHeight, "px)"),
    MsTransform: "translate(0, ".concat(scrollHeight, "px)"),
    MozTransform: "translate(0, ".concat(scrollHeight, "px)"),
    WebkitTransform: "translate(0, ".concat(scrollHeight, "px)")
  };
  var panelStyle = {
    transform: "translate(0, ".concat(translateY, "px)"),
    MsTransform: "translate(0, ".concat(translateY, "px)"),
    MozTransform: "translate(0, ".concat(translateY, "px)"),
    WebkitTransform: "translate(0, ".concat(translateY, "px)")
  };
  return {
    scrollHeight: scrollHeight,
    translateY: translateY,
    visibleData: visibleData,
    handleRowMounted: handleRowMounted,
    isVirtual: isVirtual,
    cursorStyle: cursorStyle,
    panelStyle: panelStyle
  };
};

exports["default"] = usePanelVirtualScroll;
