// tdesign-react/cjs/cascader/core/effect.js
/**
 * tdesign v1.17.1
 * (c) 2026 tdesign
 * @license MIT
 */

'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var slicedToArray = require('../../_chunks/dep-c60a3e4c.js');
var treeNodeModel = require('../../_chunks/dep-77f065c8.js');
var cascader_core_helper = require('./helper.js');
var isNumber = require('../../_chunks/dep-b8d0b96c.js');
var cloneDeep = require('../../_chunks/dep-ba90997f.js');
var isFunction = require('../../_chunks/dep-11347ea1.js');
require('../../_chunks/dep-98e6819c.js');
require('../../_chunks/dep-7f3fa935.js');
require('../../_chunks/dep-c446aa11.js');
require('../../_chunks/dep-c290284d.js');
require('../../_chunks/dep-3c628d18.js');
require('../../_chunks/dep-3784bbea.js');
require('../../_chunks/dep-d4a6a9c1.js');
require('../../_chunks/dep-a54f202c.js');
require('../../_chunks/dep-36579bf6.js');
require('../../_chunks/dep-3d05de97.js');
require('../../_chunks/dep-4f1729f0.js');
require('../../_chunks/dep-c37c5e17.js');
require('../../_chunks/dep-eee1aab9.js');
require('../../_chunks/dep-a830ce1e.js');
require('../../_chunks/dep-770bc0ae.js');
require('../../_chunks/dep-b199860d.js');
require('../../_chunks/dep-bc160f5b.js');
require('../../_chunks/dep-aaf27e39.js');
require('../../_chunks/dep-b47d855f.js');
require('../../_chunks/dep-338f013a.js');
require('../../_chunks/dep-363864ae.js');
require('../../_chunks/dep-e360e17f.js');
require('../../_chunks/dep-8a493834.js');
require('../../_chunks/dep-204188de.js');
require('../../_chunks/dep-7a4e0117.js');
require('../../_chunks/dep-8005b808.js');
require('../../_chunks/dep-f78183d7.js');
require('../../_chunks/dep-fb263e64.js');
require('../../_chunks/dep-8de6ca9d.js');
require('../../_chunks/dep-22026fea.js');
require('../../_chunks/dep-150a112c.js');
require('../../_chunks/dep-51284537.js');
require('../../_chunks/dep-9f5b2891.js');
require('../../_chunks/dep-81bce4e6.js');
require('../../_chunks/dep-863f8f4d.js');
require('../../_chunks/dep-551d5940.js');
require('../../_chunks/dep-2add8824.js');
require('../../_chunks/dep-6e5e0470.js');
require('../../_chunks/dep-cdef7593.js');
require('../../_chunks/dep-f5e1d11c.js');
require('../../_chunks/dep-bfa39549.js');
require('../../_chunks/dep-2be6b7d2.js');

function expendClickEffect(propsTrigger, trigger, node, cascaderContext) {
  var checkStrictly = cascaderContext.checkStrictly,
    multiple = cascaderContext.multiple,
    treeStore = cascaderContext.treeStore,
    setVisible = cascaderContext.setVisible,
    setValue = cascaderContext.setValue,
    setTreeNodes = cascaderContext.setTreeNodes,
    setExpend = cascaderContext.setExpend,
    value = cascaderContext.value,
    max = cascaderContext.max,
    valueType = cascaderContext.valueType,
    isParentFilterable = cascaderContext.isParentFilterable;
  var isDisabled = node.disabled || multiple && value.length >= max && max !== 0;
  if (isDisabled) return;
  if (propsTrigger === trigger && node.children !== null) {
    var expanded = node.setExpanded(true);
    treeStore.refreshNodes();
    treeStore.replaceExpanded(expanded);
    var nodes = treeStore.getNodes().filter(function (node2) {
      return node2.visible;
    });
    setTreeNodes(nodes);
    if (multiple && !isParentFilterable) {
      setExpend(expanded);
    }
  }
  if (!multiple && (node.isLeaf() || checkStrictly) && trigger === "click") {
    var shouldClosePanel = !checkStrictly || propsTrigger === "hover";
    if (node.checked) {
      if (shouldClosePanel) {
        setVisible(false, {});
      }
      return;
    }
    treeStore.resetChecked();
    var checked = node.setChecked(true);
    var _checked = slicedToArray._slicedToArray(checked, 1),
      value2 = _checked[0];
    setValue(valueType === "single" ? value2 : node.getPath().map(function (item) {
      return item.value;
    }), "check", node.getModel());
    if (shouldClosePanel) {
      setVisible(false, {});
    }
  }
}
function valueChangeEffect(node, cascaderContext) {
  var disabled = cascaderContext.disabled,
    max = cascaderContext.max,
    inputVal = cascaderContext.inputVal,
    multiple = cascaderContext.multiple,
    setVisible = cascaderContext.setVisible,
    setValue = cascaderContext.setValue,
    treeNodes = cascaderContext.treeNodes,
    treeStore = cascaderContext.treeStore,
    valueType = cascaderContext.valueType;
  if (!node || disabled || node.disabled) {
    return;
  }
  var checked = node.setChecked(!node.isChecked());
  if (isNumber.isNumber(max) && max < 0) {
    console.warn("TDesign Warn:", "max should > 0");
  }
  if (checked.length > max && isNumber.isNumber(max) && max > 0) {
    return;
  }
  if (checked.length === 0) {
    var expanded = treeStore.getExpanded();
    setTimeout(function () {
      treeStore.replaceExpanded(expanded);
      treeStore.refreshNodes();
    }, 0);
  }
  if (!multiple) {
    setVisible(false, {});
  }
  var isSelectAll = treeNodes.every(function (item) {
    return checked.indexOf(item.value) > -1;
  });
  if (inputVal && isSelectAll) {
    setVisible(false, {});
  }
  var resValue = valueType === "single" ? checked : checked.map(function (val) {
    return treeStore.getNode(val).getPath().map(function (item) {
      return item.value;
    });
  });
  setValue(resValue, node.checked ? "uncheck" : "check", node.getModel());
}
function closeIconClickEffect(cascaderContext) {
  var setVisible = cascaderContext.setVisible,
    multiple = cascaderContext.multiple,
    setExpend = cascaderContext.setExpend,
    setValue = cascaderContext.setValue,
    treeStore = cascaderContext.treeStore;
  var expanded = treeStore.getExpanded();
  setTimeout(function () {
    treeStore.replaceExpanded(expanded);
    treeStore.refreshNodes();
  }, 0);
  setVisible(false, {});
  if (multiple) {
    setExpend([]);
  }
  setValue(multiple ? [] : "", "clear");
}
function handleRemoveTagEffect(cascaderContext, index, onRemove) {
  var disabled = cascaderContext.disabled,
    setValue = cascaderContext.setValue,
    value = cascaderContext.value,
    valueType = cascaderContext.valueType,
    treeStore = cascaderContext.treeStore;
  if (disabled) return;
  var newValue = cloneDeep.cloneDeep(value);
  var res = newValue.splice(index, 1);
  var node = treeStore.getNodes(res[0])[0];
  var checked = node.setChecked(!node.isChecked());
  if (valueType === "single") {
    setValue(newValue, "uncheck", node.getModel());
  } else {
    var resValue = checked.map(function (val) {
      return treeStore.getNode(val).getPath().map(function (item) {
        return item.value;
      });
    });
    setValue(resValue, "uncheck", node.getModel());
  }
  if (isFunction.isFunction(onRemove)) {
    onRemove({
      value: checked,
      node: node
    });
  }
}
var treeNodesEffect = function treeNodesEffect(inputVal, treeStore, setTreeNodes, filter, isParentFilterable) {
  if (!treeStore) return;
  var nodes = [];
  if (inputVal) {
    var filterMethods = function filterMethods(node) {
      if (!node.isLeaf() && !isParentFilterable) return;
      if (isFunction.isFunction(filter)) {
        return filter("".concat(inputVal), node);
      }
      var fullPathLabel = cascader_core_helper.getFullPathLabel(node, "");
      return fullPathLabel.indexOf("".concat(inputVal)) > -1;
    };
    nodes = treeStore.nodes.filter(filterMethods);
  } else {
    nodes = treeStore.getNodes().filter(function (node) {
      return node.visible;
    });
  }
  setTreeNodes(nodes);
};
var treeStoreExpendEffect = function treeStoreExpendEffect(treeStore, value, expend, valueType) {
  var _treeStore$config;
  var treeValue = cascader_core_helper.getTreeValue(value);
  if (!treeStore) return;
  var allowDuplicateValue = (_treeStore$config = treeStore.config) === null || _treeStore$config === void 0 ? void 0 : _treeStore$config.allowDuplicateValue;
  if (Array.isArray(treeValue) && expend.length === 0) {
    var expandedMap = /* @__PURE__ */new Map();
    var node = null;
    if (valueType === "full" && Array.isArray(value) && value.length > 0) {
      node = treeStore.getNode(value);
    } else {
      var _treeValue = slicedToArray._slicedToArray(treeValue, 1),
        val = _treeValue[0];
      if (val) {
        expandedMap.set(val, true);
        node = treeStore.getNode(val);
      }
    }
    if (node) {
      node.getParents().forEach(function (tn) {
        var expandedKey = allowDuplicateValue ? treeNodeModel.pathToKey(tn.getPath().map(function (n) {
          return n.value;
        })) : tn.value;
        expandedMap.set(expandedKey, true);
      });
      var expandedArr = Array.from(expandedMap.keys());
      treeStore.replaceExpanded(expandedArr);
    } else if (treeValue.length > 0) {
      treeStore.refreshNodes();
      return;
    } else {
      treeStore.resetExpanded();
    }
  }
  if (treeStore.getExpanded() && expend.length) {
    treeStore.replaceExpanded(expend);
  }
  treeStore.refreshNodes();
};

exports.closeIconClickEffect = closeIconClickEffect;
exports.expendClickEffect = expendClickEffect;
exports.handleRemoveTagEffect = handleRemoveTagEffect;
exports.treeNodesEffect = treeNodesEffect;
exports.treeStoreExpendEffect = treeStoreExpendEffect;
exports.valueChangeEffect = valueChangeEffect;
