// tdesign-react/cjs/cascader/core/helper.js
/**
 * tdesign v1.17.1
 * (c) 2026 tdesign
 * @license MIT
 */

'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var _typeof = require('../../_chunks/dep-c290284d.js');
var treeNodeModel = require('../../_chunks/dep-77f065c8.js');
var isEmpty = require('../../_chunks/dep-2be6b7d2.js');
require('../../_chunks/dep-c446aa11.js');
require('../../_chunks/dep-3c628d18.js');
require('../../_chunks/dep-3784bbea.js');
require('../../_chunks/dep-d4a6a9c1.js');
require('../../_chunks/dep-a54f202c.js');
require('../../_chunks/dep-36579bf6.js');
require('../../_chunks/dep-3d05de97.js');
require('../../_chunks/dep-4f1729f0.js');
require('../../_chunks/dep-c37c5e17.js');
require('../../_chunks/dep-eee1aab9.js');
require('../../_chunks/dep-11347ea1.js');
require('../../_chunks/dep-a830ce1e.js');
require('../../_chunks/dep-770bc0ae.js');
require('../../_chunks/dep-b199860d.js');
require('../../_chunks/dep-bc160f5b.js');
require('../../_chunks/dep-aaf27e39.js');
require('../../_chunks/dep-b47d855f.js');
require('../../_chunks/dep-338f013a.js');
require('../../_chunks/dep-363864ae.js');
require('../../_chunks/dep-e360e17f.js');
require('../../_chunks/dep-8a493834.js');
require('../../_chunks/dep-204188de.js');
require('../../_chunks/dep-7a4e0117.js');
require('../../_chunks/dep-8005b808.js');
require('../../_chunks/dep-f78183d7.js');
require('../../_chunks/dep-fb263e64.js');
require('../../_chunks/dep-8de6ca9d.js');
require('../../_chunks/dep-22026fea.js');
require('../../_chunks/dep-150a112c.js');
require('../../_chunks/dep-51284537.js');
require('../../_chunks/dep-9f5b2891.js');
require('../../_chunks/dep-81bce4e6.js');
require('../../_chunks/dep-863f8f4d.js');
require('../../_chunks/dep-551d5940.js');
require('../../_chunks/dep-2add8824.js');
require('../../_chunks/dep-6e5e0470.js');
require('../../_chunks/dep-cdef7593.js');
require('../../_chunks/dep-f5e1d11c.js');
require('../../_chunks/dep-bfa39549.js');

function getSingleContent(cascaderContext) {
  var value = cascaderContext.value,
    multiple = cascaderContext.multiple,
    treeStore = cascaderContext.treeStore,
    showAllLevels = cascaderContext.showAllLevels,
    valueType = cascaderContext.valueType;
  var isEmpty2 = !value && value !== 0 || Array.isArray(value) && value.length === 0;
  var isInvalidFullPathValue = valueType === "full" && !Array.isArray(value);
  if (multiple || isEmpty2 || isInvalidFullPathValue) return "";
  var formatContent = function formatContent(path2) {
    if (!(path2 !== null && path2 !== void 0 && path2.length)) return "";
    return showAllLevels ? path2.map(function (node) {
      return node.label;
    }).join(" ".concat(treeNodeModel.PATH_SEPARATOR, " ")) : path2[path2.length - 1].label;
  };
  var getDefaultDisplay = function getDefaultDisplay(val) {
    return Array.isArray(val) ? val.join(" ".concat(treeNodeModel.PATH_SEPARATOR, " ")) : String(val);
  };
  if (valueType === "full" && Array.isArray(value) && value.length > 0) {
    var node = treeStore === null || treeStore === void 0 ? void 0 : treeStore.getNode(value);
    var path2 = node === null || node === void 0 ? void 0 : node.getPath();
    return path2 !== null && path2 !== void 0 && path2.length ? formatContent(path2) : getDefaultDisplay(value);
  }
  var nodes = treeStore === null || treeStore === void 0 ? void 0 : treeStore.getNodes(value);
  if (!(nodes !== null && nodes !== void 0 && nodes.length)) {
    return getDefaultDisplay(value);
  }
  var path = nodes[0].getPath();
  return path !== null && path !== void 0 && path.length ? formatContent(path) : getDefaultDisplay(value);
}
function getMultipleContent(cascaderContext) {
  var value = cascaderContext.value,
    multiple = cascaderContext.multiple,
    treeStore = cascaderContext.treeStore,
    showAllLevels = cascaderContext.showAllLevels,
    valueType = cascaderContext.valueType;
  if (!multiple) return [];
  if (multiple && !Array.isArray(value)) return [];
  return value.map(function (item) {
    var _node;
    var node;
    if (valueType === "full" && Array.isArray(item)) {
      node = treeStore === null || treeStore === void 0 ? void 0 : treeStore.getNode(item);
    } else {
      var nodes = treeStore === null || treeStore === void 0 ? void 0 : treeStore.getNodes(item);
      node = nodes === null || nodes === void 0 ? void 0 : nodes[0];
    }
    if (!node) return void 0;
    return showAllLevels ? getFullPathLabel(node) : (_node = node) === null || _node === void 0 ? void 0 : _node.label;
  }).filter(function (item) {
    return !!item;
  });
}
function getPanels(treeNodes) {
  var panels = [];
  treeNodes.forEach(function (node) {
    if (panels[node.level]) {
      panels[node.level].push(node);
    } else {
      panels[node.level] = [node];
    }
  });
  return panels;
}
function getFullPathLabel(node) {
  var separator = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : "/";
  return node === null || node === void 0 ? void 0 : node.getPath().map(function (node2) {
    return node2.label;
  }).join(separator);
}
var getTreeValue = function getTreeValue(value) {
  var treeValue = [];
  if (Array.isArray(value)) {
    if (value.length > 0 && _typeof._typeof(value[0]) === "object") {
      treeValue = value.map(function (val) {
        return val.value;
      });
    } else if (value.length) {
      treeValue = value;
    }
  } else if (value) {
    if (_typeof._typeof(value) === "object") {
      treeValue = [value.value];
    } else {
      treeValue = [value];
    }
  }
  return treeValue;
};
var getExpandValue = function getExpandValue(value, valueType) {
  if (valueType === "full" && Array.isArray(value) && value.length > 0) {
    return value[value.length - 1];
  }
  var treeValue = getTreeValue(value);
  return treeValue[0];
};
function isEmptyValues(value) {
  if (typeof value === "number" && !isNaN(value)) return false;
  return isEmpty.isEmpty(value);
}
function isValueInvalid(value, cascaderContext) {
  var multiple = cascaderContext.multiple,
    showAllLevels = cascaderContext.showAllLevels,
    valueType = cascaderContext.valueType;
  if (multiple) {
    return !Array.isArray(value);
  }
  if (valueType === "full") {
    return !Array.isArray(value);
  }
  if (Array.isArray(value)) {
    return !showAllLevels;
  }
  return false;
}

exports.getExpandValue = getExpandValue;
exports.getFullPathLabel = getFullPathLabel;
exports.getMultipleContent = getMultipleContent;
exports.getPanels = getPanels;
exports.getSingleContent = getSingleContent;
exports.getTreeValue = getTreeValue;
exports.isEmptyValues = isEmptyValues;
exports.isValueInvalid = isValueInvalid;
