// tdesign-react/esm/image-viewer/hooks/usePosition.js
/**
 * tdesign v1.17.1
 * (c) 2026 tdesign
 * @license MIT
 */

import { _ as _slicedToArray } from '../../_chunks/dep-d1eeb95c.js';
import { useState, useRef, useCallback } from 'react';
import useMouseEvent from '../../hooks/useMouseEvent.js';
import '../../_chunks/dep-673e8854.js';
import '../../_chunks/dep-570a5adb.js';
import '../../hooks/useLatest.js';

var usePosition = function usePosition(imgRef, options) {
  var _ref = options || {},
    _ref$initPosition = _ref.initPosition,
    initPosition = _ref$initPosition === void 0 ? [0, 0] : _ref$initPosition;
  var _useState = useState(initPosition),
    _useState2 = _slicedToArray(_useState, 2),
    position = _useState2[0],
    setPosition = _useState2[1];
  var _useState3 = useState(false),
    _useState4 = _slicedToArray(_useState3, 2),
    isDragging = _useState4[0],
    setIsDragging = _useState4[1];
  var lastScreenPositionRef = useRef(null);
  var positionRef = useRef(position);
  positionRef.current = position;
  var isDraggingRef = useRef(isDragging);
  isDraggingRef.current = isDragging;
  useMouseEvent(imgRef, {
    onDown: function onDown(e) {
      var screenX = e.screenX,
        screenY = e.screenY;
      lastScreenPositionRef.current = {
        x: screenX,
        y: screenY
      };
      setIsDragging(true);
    },
    onMove: function onMove(e) {
      if (!lastScreenPositionRef.current) return;
      var screenX = e.screenX,
        screenY = e.screenY;
      var _lastScreenPositionRe = lastScreenPositionRef.current,
        lastX = _lastScreenPositionRe.x,
        lastY = _lastScreenPositionRe.y;
      setPosition(function (_ref2) {
        var _ref3 = _slicedToArray(_ref2, 2),
          x = _ref3[0],
          y = _ref3[1];
        return [x + screenX - lastX, y + screenY - lastY];
      });
      lastScreenPositionRef.current = {
        x: screenX,
        y: screenY
      };
    },
    onUp: function onUp() {
      lastScreenPositionRef.current = null;
      setIsDragging(false);
    }
  });
  var resetPosition = useCallback(function () {
    setPosition(initPosition);
  }, [initPosition]);
  return {
    position: position,
    positionRef: positionRef,
    setPosition: setPosition,
    resetPosition: resetPosition,
    isDragging: isDragging,
    isDraggingRef: isDraggingRef
  };
};

export { usePosition as default };
