// tdesign-react/cjs/loading/Loading.js
/**
 * tdesign v1.17.1
 * (c) 2026 tdesign
 * @license MIT
 */

'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var defineProperty = require('../_chunks/dep-c446aa11.js');
var slicedToArray = require('../_chunks/dep-c60a3e4c.js');
var React = require('react');
var classNames = require('classnames');
var _util_style = require('../_chunks/dep-ee71ddca.js');
var common_Portal = require('../common/Portal.js');
var hooks_useConfig = require('../hooks/useConfig.js');
var hooks_useDefaultProps = require('../hooks/useDefaultProps.js');
var loading_defaultProps = require('./defaultProps.js');
var loading_gradient = require('./gradient.js');
require('../_chunks/dep-c290284d.js');
require('../_chunks/dep-98e6819c.js');
require('../_chunks/dep-7f3fa935.js');
require('../_chunks/dep-427f8bc0.js');
require('../_chunks/dep-3d05de97.js');
require('../_chunks/dep-a54f202c.js');
require('../_chunks/dep-4f1729f0.js');
require('react-dom');
require('../hooks/useLayoutEffect.js');
require('../config-provider/ConfigContext.js');
require('../_chunks/dep-2ee09518.js');
require('../_chunks/dep-a6fa90dc.js');
require('dayjs');
require('../_chunks/dep-012d9a7c.js');
require('../_chunks/dep-150a112c.js');
require('../_chunks/dep-e360e17f.js');
require('../_chunks/dep-51284537.js');
require('../_chunks/dep-8a493834.js');
require('../_chunks/dep-9f5b2891.js');
require('../_chunks/dep-338f013a.js');
require('../_chunks/dep-c37c5e17.js');
require('../_chunks/dep-eee1aab9.js');
require('../_chunks/dep-11347ea1.js');
require('../_chunks/dep-a830ce1e.js');
require('../_chunks/dep-770bc0ae.js');
require('../_chunks/dep-aaf27e39.js');
require('../_chunks/dep-b47d855f.js');
require('../_chunks/dep-81bce4e6.js');
require('../_chunks/dep-863f8f4d.js');
require('../_chunks/dep-84e4d0da.js');
require('../_chunks/dep-f78183d7.js');
require('../_chunks/dep-2f45f6c4.js');
require('../_chunks/dep-062bb1b1.js');
require('../_chunks/dep-6e0da644.js');
require('../_chunks/dep-05bc49e6.js');
require('../_chunks/dep-8c880a65.js');
require('../_chunks/dep-e41137f7.js');
require('../_chunks/dep-b8d0b96c.js');
require('../hooks/useDomRefCallback.js');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

var React__default = /*#__PURE__*/_interopDefaultLegacy(React);
var classNames__default = /*#__PURE__*/_interopDefaultLegacy(classNames);

function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { defineProperty._defineProperty(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
var Loading = function Loading(props) {
  var _useDefaultProps = hooks_useDefaultProps["default"](props, loading_defaultProps.loadingDefaultProps),
    attach = _useDefaultProps.attach,
    indicator = _useDefaultProps.indicator,
    text = _useDefaultProps.text,
    loading = _useDefaultProps.loading,
    size = _useDefaultProps.size,
    delay = _useDefaultProps.delay,
    fullscreen = _useDefaultProps.fullscreen,
    preventScrollThrough = _useDefaultProps.preventScrollThrough,
    showOverlay = _useDefaultProps.showOverlay,
    content = _useDefaultProps.content,
    children = _useDefaultProps.children,
    inheritColor = _useDefaultProps.inheritColor,
    zIndex = _useDefaultProps.zIndex,
    className = _useDefaultProps.className,
    style = _useDefaultProps.style;
  var _useState = React.useState(function () {
      return delay ? false : loading;
    }),
    _useState2 = slicedToArray._slicedToArray(_useState, 2),
    showLoading = _useState2[0],
    setShowLoading = _useState2[1];
  var _useConfig = hooks_useConfig["default"](),
    classPrefix = _useConfig.classPrefix;
  var name = "".concat(classPrefix, "-loading");
  var centerClass = "".concat(classPrefix, "-loading--center");
  var inheritColorClass = "".concat(classPrefix, "-loading--inherit-color");
  var fullClass = "".concat(classPrefix, "-loading--full");
  var fullscreenClass = "".concat(classPrefix, "-loading__fullscreen");
  var lockClass = "".concat(classPrefix, "-loading--lock");
  var overlayClass = "".concat(classPrefix, "-loading__overlay");
  var relativeClass = "".concat(classPrefix, "-loading__parent");
  var textClass = "".concat(classPrefix, "-loading__text");
  React.useEffect(function () {
    var timer;
    if (delay && loading) {
      timer = setTimeout(function () {
        setShowLoading(loading);
      }, delay);
    } else {
      setShowLoading(loading);
    }
    return function () {
      clearTimeout(timer);
    };
  }, [delay, loading]);
  var calcStyles = React.useMemo(function () {
    var styles = {};
    if (zIndex !== void 0) {
      styles.zIndex = zIndex;
    }
    if (!["small", "medium", "large"].includes(size)) {
      styles.fontSize = size;
    }
    return styles;
  }, [size, zIndex]);
  var sizeMap = {
    large: "".concat(classPrefix, "-size-l"),
    small: "".concat(classPrefix, "-size-s"),
    medium: "".concat(classPrefix, "-size-m")
  };
  var baseClasses = classNames__default["default"](centerClass, sizeMap[size], defineProperty._defineProperty({}, inheritColorClass, inheritColor), className);
  React.useEffect(function () {
    if (preventScrollThrough && fullscreen && _util_style.canUseDocument && loading) {
      _util_style.addClass(document.body, lockClass);
    }
    return function () {
      _util_style.removeClass(document.body, lockClass);
    };
  }, [loading, preventScrollThrough, fullscreen, lockClass]);
  var commonContent = function commonContent() {
    var renderIndicator = /* @__PURE__ */React__default["default"].createElement(loading_gradient["default"], null);
    if (indicator && typeof indicator !== "boolean") {
      renderIndicator = indicator;
    }
    return /* @__PURE__ */React__default["default"].createElement(React__default["default"].Fragment, null, indicator ? renderIndicator : null, text ? /* @__PURE__ */React__default["default"].createElement("div", {
      className: textClass
    }, text) : null);
  };
  if (fullscreen) {
    return showLoading ? /* @__PURE__ */React__default["default"].createElement("div", {
      className: classNames__default["default"](name, fullscreenClass, centerClass, overlayClass),
      style: _objectSpread(_objectSpread({}, calcStyles), style)
    }, /* @__PURE__ */React__default["default"].createElement("div", {
      className: baseClasses
    }, commonContent())) : null;
  }
  if (content || children) {
    return /* @__PURE__ */React__default["default"].createElement("div", {
      className: relativeClass,
      style: style
    }, content || children, showLoading ? /* @__PURE__ */React__default["default"].createElement("div", {
      className: classNames__default["default"](name, baseClasses, fullClass, defineProperty._defineProperty({}, overlayClass, showOverlay)),
      style: calcStyles
    }, commonContent()) : null);
  }
  if (attach) {
    return /* @__PURE__ */React__default["default"].createElement(common_Portal["default"], {
      attach: attach
    }, showLoading ? /* @__PURE__ */React__default["default"].createElement("div", {
      className: classNames__default["default"](name, baseClasses, fullClass, defineProperty._defineProperty({}, overlayClass, showOverlay)),
      style: _objectSpread(_objectSpread({}, calcStyles), style)
    }, commonContent()) : null);
  }
  return showLoading ? /* @__PURE__ */React__default["default"].createElement("div", {
    className: classNames__default["default"](name, baseClasses),
    style: _objectSpread(_objectSpread({}, calcStyles), style)
  }, commonContent()) : null;
};
Loading.displayName = "Loading";

exports["default"] = Loading;
