// tdesign-react/cjs/message/Message.js
/**
 * tdesign v1.17.1
 * (c) 2026 tdesign
 * @license MIT
 */

'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var slicedToArray = require('../_chunks/dep-c60a3e4c.js');
var defineProperty = require('../_chunks/dep-c446aa11.js');
var React = require('react');
var classNames = require('classnames');
var message_MessageComponent = require('../_chunks/dep-5094d657.js');
var constants = require('../_chunks/dep-147ad148.js');
var _util_style = require('../_chunks/dep-ee71ddca.js');
var _util_reactRender = require('../_util/react-render.js');
var common_PluginContainer = require('../common/PluginContainer.js');
require('../config-provider/ConfigContext.js');
var configProvider_ConfigProvider = require('../config-provider/ConfigProvider.js');
var message_config = require('./config.js');
var message_useMessageClass = require('./useMessageClass.js');
var camelCase = require('../_chunks/dep-b422817e.js');
require('../_chunks/dep-98e6819c.js');
require('../_chunks/dep-7f3fa935.js');
require('../_chunks/dep-c290284d.js');
require('../_chunks/dep-e41137f7.js');
require('../_util/noop.js');
require('../hooks/usePersistFn.js');
require('./MessageClose.js');
require('tdesign-icons-react');
require('../hooks/useGlobalIcon.js');
require('../hooks/useConfig.js');
require('./MessageIcon.js');
require('../loading/index.js');
require('../loading/Loading.js');
require('../common/Portal.js');
require('react-dom');
require('../hooks/useLayoutEffect.js');
require('../hooks/useDefaultProps.js');
require('../loading/defaultProps.js');
require('../loading/gradient.js');
require('../_chunks/dep-062bb1b1.js');
require('../_chunks/dep-6e0da644.js');
require('../_chunks/dep-05bc49e6.js');
require('../_chunks/dep-8c880a65.js');
require('../_chunks/dep-b8d0b96c.js');
require('../_chunks/dep-3d05de97.js');
require('../_chunks/dep-4f1729f0.js');
require('../_chunks/dep-427f8bc0.js');
require('../_chunks/dep-a54f202c.js');
require('../hooks/useDomRefCallback.js');
require('../_chunks/dep-2ee09518.js');
require('../_chunks/dep-a6fa90dc.js');
require('dayjs');
require('../_chunks/dep-012d9a7c.js');
require('../_chunks/dep-150a112c.js');
require('../_chunks/dep-e360e17f.js');
require('../_chunks/dep-51284537.js');
require('../_chunks/dep-8a493834.js');
require('../_chunks/dep-9f5b2891.js');
require('../_chunks/dep-338f013a.js');
require('../_chunks/dep-c37c5e17.js');
require('../_chunks/dep-eee1aab9.js');
require('../_chunks/dep-11347ea1.js');
require('../_chunks/dep-a830ce1e.js');
require('../_chunks/dep-770bc0ae.js');
require('../_chunks/dep-aaf27e39.js');
require('../_chunks/dep-b47d855f.js');
require('../_chunks/dep-81bce4e6.js');
require('../_chunks/dep-863f8f4d.js');
require('../_chunks/dep-84e4d0da.js');
require('../_chunks/dep-f78183d7.js');
require('../_chunks/dep-2f45f6c4.js');
require('../loading/plugin.js');
require('../_chunks/dep-7bf14076.js');
require('../_chunks/dep-ba90997f.js');
require('../_chunks/dep-22026fea.js');
require('../_chunks/dep-551d5940.js');
require('../_chunks/dep-8005b808.js');
require('../_chunks/dep-2add8824.js');
require('./defaultProps.js');
require('../_chunks/dep-b199860d.js');
require('../_chunks/dep-36579bf6.js');
require('../_chunks/dep-45188d54.js');
require('../_chunks/dep-eccb6a94.js');
require('../_chunks/dep-f5e1d11c.js');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

var React__default = /*#__PURE__*/_interopDefaultLegacy(React);
var classNames__default = /*#__PURE__*/_interopDefaultLegacy(classNames);

function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { defineProperty._defineProperty(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
var messageKey = 1;
var MessageContainerMaps = /* @__PURE__ */new Map();
var MessageContainer = function MessageContainer(props) {
  var placement = props.placement,
    children = props.children,
    zIndex = props.zIndex;
  var ref = React.useRef(null);
  var style = {};
  Object.keys(constants.PLACEMENT_OFFSET[placement]).forEach(function (key) {
    style[camelCase.camelCase(key)] = constants.PLACEMENT_OFFSET[placement][key];
  });
  if (placement.includes("top")) {
    style.top = "".concat(message_config.globalConfig.top, "px");
  }
  if (typeof zIndex !== "undefined") {
    style.zIndex = zIndex;
  }
  var _useMessageClass = message_useMessageClass.useMessageClass(),
    tdMessagePlacementClassGenerator = _useMessageClass.tdMessagePlacementClassGenerator,
    tdMessageListClass = _useMessageClass.tdMessageListClass;
  return /* @__PURE__ */React__default["default"].createElement("div", {
    ref: ref,
    className: classNames__default["default"](tdMessageListClass, tdMessagePlacementClassGenerator(placement)),
    style: style
  }, children);
};
function getAttachNodeMap(attachNode) {
  if (!MessageContainerMaps.has(attachNode)) {
    MessageContainerMaps.set(attachNode, /* @__PURE__ */new Map());
  }
  return MessageContainerMaps.get(attachNode);
}
function createContainer(placement, attachNode, zIndex) {
  var mGlobalConfig = configProvider_ConfigProvider["default"].getGlobalConfig();
  var containerWrapper = document.createElement("div");
  attachNode.appendChild(containerWrapper);
  _util_reactRender.render(/* @__PURE__ */React__default["default"].createElement(common_PluginContainer["default"], {
    globalConfig: mGlobalConfig
  }, /* @__PURE__ */React__default["default"].createElement(MessageContainer, {
    placement: placement,
    zIndex: zIndex
  })), containerWrapper);
  return containerWrapper;
}
function renderContainer(containerInstance) {
  var mGlobalConfig = configProvider_ConfigProvider["default"].getGlobalConfig();
  _util_reactRender.render(/* @__PURE__ */React__default["default"].createElement(common_PluginContainer["default"], {
    globalConfig: mGlobalConfig
  }, /* @__PURE__ */React__default["default"].createElement(MessageContainer, {
    placement: containerInstance.placement,
    zIndex: containerInstance.zIndex
  }, containerInstance.messages.map(function (item) {
    return /* @__PURE__ */React__default["default"].createElement(message_MessageComponent.MessageComponent, _objectSpread(_objectSpread({
      key: item.key,
      ref: item.ref
    }, item.config), {}, {
      theme: item.theme
    }));
  }))), containerInstance.container);
}
function destroyContainer(containerInstance, attachNode) {
  if (containerInstance.messages.length > 0) return;
  var container = containerInstance.container,
    placement = containerInstance.placement;
  _util_reactRender.render(null, container);
  if (container.parentNode) {
    container.parentNode.removeChild(container);
  }
  var attachNodeMap = MessageContainerMaps.get(attachNode);
  if (attachNodeMap) {
    attachNodeMap["delete"](placement);
    if (attachNodeMap.size === 0) {
      MessageContainerMaps["delete"](attachNode);
    }
  }
}
function renderElement(theme, config) {
  var attach = config.attach,
    _config$placement = config.placement,
    placement = _config$placement === void 0 ? "top" : _config$placement,
    zIndex = config.zIndex;
  var attachNode = _util_style.getAttach(attach);
  var attachNodeMap = getAttachNodeMap(attachNode);
  var containerInstance = attachNodeMap.get(placement);
  if (!containerInstance) {
    var container = createContainer(placement, attachNode, zIndex);
    containerInstance = {
      container: container,
      messages: [],
      placement: placement,
      zIndex: zIndex
    };
    attachNodeMap.set(placement, containerInstance);
  } else if (typeof zIndex !== "undefined") {
    containerInstance.zIndex = zIndex;
  }
  var style = _objectSpread({}, config.style);
  if (Array.isArray(config.offset) && config.offset.length === 2) {
    var _config$offset = slicedToArray._slicedToArray(config.offset, 2),
      left = _config$offset[0],
      top = _config$offset[1];
    style = _objectSpread(_objectSpread({
      left: left,
      top: top
    }, style), {}, {
      position: "relative"
    });
  }
  messageKey += 1;
  var currentKey = messageKey;
  var messageRef = /*#__PURE__*/React.createRef();
  var message = {
    key: currentKey,
    ref: messageRef,
    theme: theme,
    content: config.content,
    config: _objectSpread(_objectSpread({}, config), {}, {
      style: style,
      placement: placement,
      onClose: function onClose(context) {
        closeMessage(context);
      }
    }),
    close: function close() {
      var index = containerInstance.messages.indexOf(message);
      if (index === -1) return;
      message_MessageComponent.fadeOut(messageRef.current, placement, function () {
        closeMessage();
      });
    }
  };
  function closeMessage(context) {
    var index = containerInstance.messages.indexOf(message);
    if (index === -1) return;
    message_MessageComponent.fadeOut(messageRef.current, placement, function () {
      containerInstance.messages.splice(index, 1);
      renderContainer(containerInstance);
      destroyContainer(containerInstance, attachNode);
      if (context) {
        var _config$onClose;
        (_config$onClose = config.onClose) === null || _config$onClose === void 0 || _config$onClose.call(config, context);
      }
    });
  }
  containerInstance.messages.push(message);
  renderContainer(containerInstance);
  return message;
}
function isConfig(content) {
  return Object.prototype.toString.call(content) === "[object Object]" && "content" in content;
}
var messageMethod = function messageMethod(theme, content, duration) {
  var config = {};
  if (isConfig(content)) {
    config = _objectSpread({
      duration: duration
    }, content);
  } else {
    config = {
      content: content,
      duration: duration
    };
  }
  return Promise.resolve(renderElement(theme, message_config.getMessageConfig(config)));
};
var MessagePlugin = function MessagePlugin(theme, message, duration) {
  return messageMethod(theme, message, duration);
};
MessagePlugin.info = function (content, duration) {
  return messageMethod("info", content, duration);
};
MessagePlugin.error = function (content, duration) {
  return messageMethod("error", content, duration);
};
MessagePlugin.warning = function (content, duration) {
  return messageMethod("warning", content, duration);
};
MessagePlugin.success = function (content, duration) {
  return messageMethod("success", content, duration);
};
MessagePlugin.question = function (content, duration) {
  return messageMethod("question", content, duration);
};
MessagePlugin.loading = function (content, duration) {
  return messageMethod("loading", content, duration);
};
MessagePlugin.config = function (options) {
  return message_config.setGlobalConfig(options);
};
MessagePlugin.close = function (messageInstance) {
  messageInstance.then(function (instance) {
    return instance.close();
  });
};
MessagePlugin.closeAll = function () {
  MessageContainerMaps.forEach(function (placementMap, attachNode) {
    placementMap.forEach(function (container) {
      if (container.messages.length) {
        container.messages.splice(0, container.messages.length);
        destroyContainer(container, attachNode);
      }
    });
  });
};

exports["default"] = message_MessageComponent.MessageComponent;
exports.MessagePlugin = MessagePlugin;
