// tdesign-react/cjs/form/hooks/useFormItemStyle.js
/**
 * tdesign v1.17.1
 * (c) 2026 tdesign
 * @license MIT
 */

'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var defineProperty = require('../../_chunks/dep-c446aa11.js');
var React = require('react');
var classNames = require('classnames');
var _util_parseTNode = require('../../_util/parseTNode.js');
var hooks_useConfig = require('../../hooks/useConfig.js');
var form_const = require('../const.js');
require('../../_chunks/dep-c290284d.js');
require('../../_chunks/dep-3c628d18.js');
require('../../_chunks/dep-11347ea1.js');
require('../../_chunks/dep-3d05de97.js');
require('../../_chunks/dep-a830ce1e.js');
require('../../config-provider/ConfigContext.js');
require('../../_chunks/dep-2ee09518.js');
require('../../_chunks/dep-a6fa90dc.js');
require('dayjs');
require('../../_chunks/dep-012d9a7c.js');
require('../../_chunks/dep-150a112c.js');
require('../../_chunks/dep-e360e17f.js');
require('../../_chunks/dep-4f1729f0.js');
require('../../_chunks/dep-a54f202c.js');
require('../../_chunks/dep-51284537.js');
require('../../_chunks/dep-8a493834.js');
require('../../_chunks/dep-9f5b2891.js');
require('../../_chunks/dep-338f013a.js');
require('../../_chunks/dep-c37c5e17.js');
require('../../_chunks/dep-eee1aab9.js');
require('../../_chunks/dep-770bc0ae.js');
require('../../_chunks/dep-aaf27e39.js');
require('../../_chunks/dep-b47d855f.js');
require('../../_chunks/dep-81bce4e6.js');
require('../../_chunks/dep-863f8f4d.js');
require('../../_chunks/dep-84e4d0da.js');
require('../../_chunks/dep-f78183d7.js');
require('../../_chunks/dep-2f45f6c4.js');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

var React__default = /*#__PURE__*/_interopDefaultLegacy(React);
var classNames__default = /*#__PURE__*/_interopDefaultLegacy(classNames);

function useFormItemStyle(props) {
  var _useConfig = hooks_useConfig["default"](),
    classPrefix = _useConfig.classPrefix;
  var className = props.className,
    help = props.help,
    tips = props.tips,
    snakeName = props.snakeName,
    status = props.status,
    successBorder = props.successBorder,
    errorList = props.errorList,
    successList = props.successList,
    layout = props.layout,
    verifyStatus = props.verifyStatus,
    label = props.label,
    labelWidth = props.labelWidth,
    labelAlign = props.labelAlign,
    requiredMark = props.requiredMark,
    requiredMarkPosition = props.requiredMarkPosition,
    showErrorMessage = props.showErrorMessage,
    innerRules = props.innerRules;
  var renderStatus = status || verifyStatus;
  var helpNode = help && /* @__PURE__ */React__default["default"].createElement("div", {
    className: "".concat(classPrefix, "-input__help")
  }, _util_parseTNode["default"](help));
  var needRequiredMark = requiredMark || (requiredMark !== null && requiredMark !== void 0 ? requiredMark : innerRules.filter(function (rule) {
    return rule.required;
  }).length > 0);
  var extraNode = React.useMemo(function () {
    var _errorList$;
    var extra = tips ? /* @__PURE__ */React__default["default"].createElement("div", {
      className: "".concat(classPrefix, "-input__extra"),
      title: tips
    }, tips) : null;
    if (showErrorMessage && errorList !== null && errorList !== void 0 && (_errorList$ = errorList[0]) !== null && _errorList$ !== void 0 && _errorList$.message) {
      extra = /* @__PURE__ */React__default["default"].createElement("div", {
        className: "".concat(classPrefix, "-input__extra"),
        title: errorList[0].message
      }, errorList[0].message);
    } else if (successList.length) {
      extra = /* @__PURE__ */React__default["default"].createElement("div", {
        className: "".concat(classPrefix, "-input__extra"),
        title: successList[0].message
      }, successList[0].message);
    }
    return extra;
  }, [showErrorMessage, errorList, successList, tips, classPrefix]);
  var formSnakeName = snakeName.split(",").join("_");
  var formItemClass = classNames__default["default"]("".concat(classPrefix, "-form__item"), className, defineProperty._defineProperty(defineProperty._defineProperty(defineProperty._defineProperty({}, "".concat(classPrefix, "-form-item__").concat(formSnakeName), formSnakeName), "".concat(classPrefix, "-form__item-with-help"), helpNode), "".concat(classPrefix, "-form__item-with-extra"), extraNode));
  var formItemLabelClass = classNames__default["default"]("".concat(classPrefix, "-form__label"), defineProperty._defineProperty(defineProperty._defineProperty(defineProperty._defineProperty(defineProperty._defineProperty(defineProperty._defineProperty({}, "".concat(classPrefix, "-form__label--required"), needRequiredMark), "".concat(classPrefix, "-form__label--required-right"), needRequiredMark && requiredMarkPosition === "right"), "".concat(classPrefix, "-form__label--top"), labelAlign === "top" || !labelWidth), "".concat(classPrefix, "-form__label--left"), labelAlign === "left" && labelWidth), "".concat(classPrefix, "-form__label--right"), labelAlign === "right" && labelWidth));
  var contentClass = function contentClass() {
    var controlCls = "".concat(classPrefix, "-form__controls");
    if (!showErrorMessage) return controlCls;
    var isSuccess = renderStatus === form_const.ValidateStatus.SUCCESS;
    if (isSuccess) {
      return classNames__default["default"](controlCls, "".concat(classPrefix, "-is-success"), defineProperty._defineProperty({}, "".concat(classPrefix, "-form--success-border"), successBorder));
    }
    return classNames__default["default"](controlCls, defineProperty._defineProperty(defineProperty._defineProperty(defineProperty._defineProperty({}, "".concat(classPrefix, "-is-warning"), renderStatus === "warning"), "".concat(classPrefix, "-is-error"), ["fail", "error"].includes(renderStatus)), "".concat(classPrefix, "-form--has-error"), renderStatus === form_const.ValidateStatus.ERROR || renderStatus === form_const.ValidateStatus.WARNING));
  };
  var labelStyle = {};
  var contentStyle = {};
  if (label && labelWidth && labelAlign !== "top") {
    if (typeof labelWidth === "number") {
      labelStyle = {
        width: "".concat(labelWidth, "px")
      };
      contentStyle = {
        marginLeft: layout !== "inline" ? "".concat(labelWidth, "px") : ""
      };
    } else {
      labelStyle = {
        width: labelWidth
      };
      contentStyle = {
        marginLeft: layout !== "inline" ? labelWidth : ""
      };
    }
  }
  return {
    formItemClass: formItemClass,
    formItemLabelClass: formItemLabelClass,
    contentClass: contentClass,
    labelStyle: labelStyle,
    contentStyle: contentStyle,
    helpNode: helpNode,
    extraNode: extraNode
  };
}

exports["default"] = useFormItemStyle;
