// tdesign-react/lib/color-picker/components/trigger.js
/**
 * tdesign v1.17.1
 * (c) 2026 tdesign
 * @license MIT
 */

import { _ as _defineProperty } from '../../_chunks/dep-55b77e26.js';
import React from 'react';
import classNames from 'classnames';
import { g as getColorObject, C as Color } from '../../_chunks/dep-1fdc790e.js';
import noop from '../../_util/noop.js';
import { Input } from '../../input/index.js';
import useClassNames from '../hooks/useClassNames.js';
import '../../_chunks/dep-067cb22b.js';
import '../../_chunks/dep-d1eeb95c.js';
import '../../_chunks/dep-673e8854.js';
import '../../_chunks/dep-570a5adb.js';
import '../../_chunks/dep-c4dcddfa.js';
import '../../_chunks/dep-0c178c7f.js';
import '../../_chunks/dep-5997fa49.js';
import 'lodash-es';
import '../../input/Input.js';
import '../../_chunks/dep-84926167.js';
import 'tdesign-icons-react';
import '../../_util/forwardRefWithStatics.js';
import 'hoist-non-react-statics';
import '../../_util/parseTNode.js';
import '../../_chunks/dep-65b369a5.js';
import '../../hooks/useConfig.js';
import '../../config-provider/ConfigContext.js';
import '../../_chunks/dep-c3e5e988.js';
import '../../_chunks/dep-65b05485.js';
import 'dayjs';
import '../../hooks/useControlled.js';
import '../../hooks/useDefaultProps.js';
import '../../hooks/useGlobalIcon.js';
import '../../hooks/useLayoutEffect.js';
import '../../_chunks/dep-68899e89.js';
import '../../locale/LocalReceiver.js';
import '../../config-provider/ConfigProvider.js';
import '../../input/defaultProps.js';
import '../../input/InputGroup.js';
import '../../input/useLengthLimit.js';
import '../../_chunks/dep-b0cbea79.js';
import '../../_chunks/dep-154fcb93.js';
import '../../_chunks/dep-e9c0bb63.js';

function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { _defineProperty(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
var ColorPickerTrigger = function ColorPickerTrigger(props) {
  var baseClassName = useClassNames();
  var _props$disabled = props.disabled,
    disabled = _props$disabled === void 0 ? false : _props$disabled,
    _props$borderless = props.borderless,
    borderless = _props$borderless === void 0 ? false : _props$borderless,
    _props$inputProps = props.inputProps,
    inputProps = _props$inputProps === void 0 ? {
      autoWidth: true
    } : _props$inputProps,
    clearable = props.clearable,
    onClear = props.onClear;
  var handleChange = function handleChange(input) {
    if (input !== props.value) {
      var _props$onChange;
      (_props$onChange = props.onChange) === null || _props$onChange === void 0 || _props$onChange.call(props, input, {
        color: getColorObject(new Color(input)),
        trigger: "input"
      });
    }
  };
  return /* @__PURE__ */React.createElement("div", {
    className: "".concat(baseClassName, "__trigger--default")
  }, /* @__PURE__ */React.createElement(Input, _objectSpread(_objectSpread({
    borderless: borderless,
    clearable: clearable
  }, inputProps), {}, {
    value: props.value,
    disabled: disabled,
    label: /* @__PURE__ */React.createElement("div", {
      className: classNames("".concat(baseClassName, "__trigger--default__color"), "".concat(baseClassName, "--bg-alpha"))
    }, /* @__PURE__ */React.createElement("span", {
      className: "color-inner",
      style: {
        background: props.value
      }
    })),
    onChange: handleChange,
    onClear: onClear || noop
  })));
};
var ColorTrigger = /*#__PURE__*/React.memo(ColorPickerTrigger);

export { ColorTrigger as default };
