// tdesign-react/es/date-picker/hooks/useSelectRange.js
/**
 * tdesign v1.17.1
 * (c) 2026 tdesign
 * @license MIT
 */

import { useMemo } from 'react';
import { isArray, isFunction } from 'lodash-es';
import { g as getRangeBounds, m as monthHasAnyAllowed, y as yearHasAnyAllowed, l as decadeHasAnyAllowed, n as computePaginationDisabled } from '../../_chunks/dep-10236bdf.js';
import '../../_chunks/dep-d1eeb95c.js';
import '../../_chunks/dep-673e8854.js';
import '../../_chunks/dep-570a5adb.js';
import '../../_chunks/dep-55b77e26.js';
import '../../_chunks/dep-067cb22b.js';
import 'dayjs';
import '../../_chunks/dep-e56ffacc.js';
import '../../_chunks/dep-ec64076a.js';
import '../../_chunks/dep-ca17be23.js';
import '../../_chunks/dep-dc41ae40.js';
import '../../_chunks/dep-0f787ecd.js';
import '../../_chunks/dep-65b369a5.js';

function useSelectRange(props) {
  var rangeBounds = useMemo(function () {
    return getRangeBounds(props.range);
  }, [props.range]);
  var monthAllowed = function monthAllowed(year, month) {
    return monthHasAnyAllowed(props.range, year, month);
  };
  var yearAllowed = function yearAllowed(year) {
    return yearHasAnyAllowed(props.range, year);
  };
  var decadeAllowed = function decadeAllowed(decadeEndYear) {
    return decadeHasAnyAllowed(props.range, decadeEndYear);
  };
  var paginationDisabled = useMemo(function () {
    return computePaginationDisabled(props.range, props.mode, props.year, props.month);
  }, [props]);
  var canLoadMoreTop = function canLoadMoreTop(firstValue) {
    var min = rangeBounds.min;
    if (isArray(props.range)) {
      if (!min) return true;
      var minYear = min.getFullYear();
      if (props.mode === "year") {
        var minDecadeEnd = Math.floor(minYear / 10) * 10 + 9;
        return firstValue > minDecadeEnd;
      }
      return firstValue > minYear;
    }
    if (isFunction(props.range)) {
      if (props.mode === "year") {
        for (var i = firstValue - 10; i >= firstValue - 50; i -= 10) {
          if (decadeAllowed(i)) return true;
        }
        return false;
      }
      for (var _i = firstValue - 1; _i > firstValue - 10; _i--) {
        if (yearAllowed(_i)) return true;
      }
      return false;
    }
    return true;
  };
  var canLoadMoreBottom = function canLoadMoreBottom(lastValue) {
    var max = rangeBounds.max;
    if (isArray(props.range)) {
      if (!max) return true;
      var maxYear = max.getFullYear();
      if (props.mode === "year") {
        var maxDecadeEnd = Math.floor(maxYear / 10) * 10 + 9;
        return lastValue < maxDecadeEnd;
      }
      return lastValue < maxYear;
    }
    if (isFunction(props.range)) {
      if (props.mode === "year") {
        for (var i = lastValue + 10; i <= lastValue + 50; i += 10) {
          if (decadeAllowed(i)) return true;
        }
        return false;
      }
      for (var _i2 = lastValue + 1; _i2 <= lastValue + 10; _i2++) {
        if (yearAllowed(_i2)) return true;
      }
      return false;
    }
    return true;
  };
  return {
    rangeBounds: rangeBounds,
    monthHasAnyAllowed: monthAllowed,
    yearHasAnyAllowed: yearAllowed,
    decadeHasAnyAllowed: decadeAllowed,
    paginationDisabled: paginationDisabled,
    canLoadMoreTop: canLoadMoreTop,
    canLoadMoreBottom: canLoadMoreBottom
  };
}

export { useSelectRange };
