// tdesign-react/cjs/breadcrumb/Breadcrumb.js
/**
 * tdesign v1.17.1
 * (c) 2026 tdesign
 * @license MIT
 */

'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var toConsumableArray = require('../_chunks/dep-05bc49e6.js');
var defineProperty = require('../_chunks/dep-c446aa11.js');
var objectWithoutProperties = require('../_chunks/dep-e41137f7.js');
var React = require('react');
var classNames = require('classnames');
var tdesignIconsReact = require('tdesign-icons-react');
var log = require('../_chunks/dep-3c628d18.js');
var _util_forwardRefWithStatics = require('../_util/forwardRefWithStatics.js');
var hooks_useConfig = require('../hooks/useConfig.js');
var hooks_useDefaultProps = require('../hooks/useDefaultProps.js');
var breadcrumb_BreadcrumbContext = require('./BreadcrumbContext.js');
var breadcrumb_BreadcrumbItem = require('./BreadcrumbItem.js');
var breadcrumb_defaultProps = require('./defaultProps.js');
var isFunction = require('../_chunks/dep-11347ea1.js');
require('../_chunks/dep-7f3fa935.js');
require('../_chunks/dep-8c880a65.js');
require('../_chunks/dep-c290284d.js');
require('hoist-non-react-statics');
require('../config-provider/ConfigContext.js');
require('../_chunks/dep-2ee09518.js');
require('../_chunks/dep-a6fa90dc.js');
require('dayjs');
require('../_chunks/dep-012d9a7c.js');
require('../_chunks/dep-150a112c.js');
require('../_chunks/dep-e360e17f.js');
require('../_chunks/dep-3d05de97.js');
require('../_chunks/dep-4f1729f0.js');
require('../_chunks/dep-a54f202c.js');
require('../_chunks/dep-51284537.js');
require('../_chunks/dep-8a493834.js');
require('../_chunks/dep-9f5b2891.js');
require('../_chunks/dep-338f013a.js');
require('../_chunks/dep-c37c5e17.js');
require('../_chunks/dep-eee1aab9.js');
require('../_chunks/dep-a830ce1e.js');
require('../_chunks/dep-770bc0ae.js');
require('../_chunks/dep-aaf27e39.js');
require('../_chunks/dep-b47d855f.js');
require('../_chunks/dep-81bce4e6.js');
require('../_chunks/dep-863f8f4d.js');
require('../_chunks/dep-84e4d0da.js');
require('../_chunks/dep-f78183d7.js');
require('../_chunks/dep-2f45f6c4.js');
require('../_chunks/dep-c60a3e4c.js');
require('../_chunks/dep-98e6819c.js');
require('../_util/parseTNode.js');
require('../_util/scroll.js');
require('raf');
require('../_chunks/dep-ee71ddca.js');
require('../_chunks/dep-427f8bc0.js');
require('../_util/easing.js');
require('../hooks/useCommonClassName.js');
require('../hooks/useGlobalIcon.js');
require('../tooltip/index.js');
require('../tooltip/Tooltip.js');
require('../popup/index.js');
require('../popup/Popup.js');
require('react-transition-group');
require('../common/Portal.js');
require('react-dom');
require('../hooks/useLayoutEffect.js');
require('../hooks/useAnimation.js');
require('../hooks/useAttach.js');
require('../hooks/useControlled.js');
require('../_util/noop.js');
require('../_chunks/dep-45188d54.js');
require('../_chunks/dep-eccb6a94.js');
require('../_chunks/dep-f5e1d11c.js');
require('../_chunks/dep-b199860d.js');
require('../_chunks/dep-36579bf6.js');
require('../hooks/useMutationObserver.js');
require('../hooks/useLatest.js');
require('../_chunks/dep-aa9a2c54.js');
require('../_chunks/dep-77ef3011.js');
require('../_chunks/dep-b36779a4.js');
require('../_chunks/dep-551d5940.js');
require('../_chunks/dep-8005b808.js');
require('../_chunks/dep-2add8824.js');
require('../_chunks/dep-f2809376.js');
require('../_chunks/dep-3be56e1e.js');
require('../hooks/usePopper.js');
require('react-fast-compare');
require('@popperjs/core');
require('../hooks/useWindowSize.js');
require('../popup/defaultProps.js');
require('../popup/hooks/useTrigger.js');
require('../_util/listener.js');
require('../_util/ref.js');
require('react-is');
require('../_util/isFragment.js');
require('../popup/utils/transition.js');
require('../popup/PopupPlugin.js');
require('../_chunks/dep-7bf14076.js');
require('../_util/react-render.js');
require('../common/PluginContainer.js');
require('../config-provider/ConfigProvider.js');
require('../_chunks/dep-ba90997f.js');
require('../_chunks/dep-22026fea.js');
require('../tooltip/defaultProps.js');
require('../tooltip/TooltipLite.js');
require('../hooks/useSwitch.js');
require('../hooks/usePersistFn.js');
require('../_chunks/dep-d07eb641.js');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

var React__default = /*#__PURE__*/_interopDefaultLegacy(React);
var classNames__default = /*#__PURE__*/_interopDefaultLegacy(classNames);

var _excluded = ["children", "options", "separator", "maxItemWidth", "className", "maxItems", "itemsAfterCollapse", "itemsBeforeCollapse", "ellipsis"];
function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { defineProperty._defineProperty(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
var Breadcrumb = _util_forwardRefWithStatics["default"](function (props, ref) {
  var _useDefaultProps = hooks_useDefaultProps["default"](props, breadcrumb_defaultProps.breadcrumbDefaultProps),
    children = _useDefaultProps.children,
    options = _useDefaultProps.options,
    separator = _useDefaultProps.separator,
    maxItemWidth = _useDefaultProps.maxItemWidth,
    className = _useDefaultProps.className,
    maxItems = _useDefaultProps.maxItems,
    itemsAfterCollapse = _useDefaultProps.itemsAfterCollapse,
    itemsBeforeCollapse = _useDefaultProps.itemsBeforeCollapse,
    ellipsis = _useDefaultProps.ellipsis,
    restProps = objectWithoutProperties._objectWithoutProperties(_useDefaultProps, _excluded);
  var _useConfig = hooks_useConfig["default"](),
    classPrefix = _useConfig.classPrefix;
  var renderOptionsToItems = React.useCallback(function (options2) {
    if (!options2 || !options2.length) {
      return void 0;
    }
    return options2.map(function (option, index) {
      var targetProps = {};
      if (option.target) {
        targetProps.target = option.target;
      }
      return /* @__PURE__ */React__default["default"].createElement(breadcrumb_BreadcrumbItem["default"], _objectSpread(_objectSpread({}, targetProps), {}, {
        key: index,
        maxWidth: option.maxWidth,
        disabled: option.disabled,
        href: option.href,
        to: option.to,
        router: option.router,
        replace: option.replace,
        separator: separator,
        maxItemWidth: maxItemWidth,
        icon: option.icon,
        content: option.content
      }), option.content || option.children);
    });
  }, [maxItemWidth, separator]);
  var renderEllipsis = React.useCallback(function (items) {
    if (!ellipsis) {
      return /* @__PURE__ */React__default["default"].createElement(tdesignIconsReact.EllipsisIcon, null);
    }
    if (isFunction.isFunction(ellipsis)) {
      return ellipsis({
        items: items,
        separator: separator
      });
    }
    return ellipsis;
  }, [ellipsis, separator]);
  var content = React.useMemo(function () {
    var _items;
    var items = renderOptionsToItems(options) || React.Children.toArray(children);
    var itemsCount = ((_items = items) === null || _items === void 0 ? void 0 : _items.length) || 0;
    if (!items || !maxItems) {
      return items;
    }
    if (maxItems >= itemsCount) {
      return items;
    }
    if (maxItems && (!itemsBeforeCollapse || !itemsAfterCollapse)) {
      log.log.error("Breadcrumb", "\u9700\u8981\u8BBE\u7F6E itemsBeforeCollapse \u548C itemsAfterCollapse \u5C5E\u6027\u6765\u63A7\u5236\u7701\u7565\u53F7\u524D\u540E\u7684\u663E\u793A\u9879\u6570\u3002");
      return items;
    }
    if (maxItems && maxItems < (itemsAfterCollapse || 0) + (itemsBeforeCollapse || 0)) {
      log.log.error("Breadcrumb", "maxItems={".concat(maxItems, "} \u5FC5\u987B\u5927\u4E8E\u6216\u7B49\u4E8E itemsBeforeCollapse={").concat(itemsBeforeCollapse, "} + itemsAfterCollapse={").concat(itemsAfterCollapse, "}"));
      return items;
    }
    var startCount = itemsBeforeCollapse;
    var endCount = itemsAfterCollapse;
    var collapsedItems = items.slice(startCount, itemsCount - endCount);
    var beforeItems = items.slice(0, startCount);
    var afterItems = items.slice(itemsCount - endCount);
    var ellipsis2 = renderEllipsis(collapsedItems.map(function (item) {
      return item.props;
    }));
    items = [].concat(toConsumableArray._toConsumableArray(beforeItems), [/* @__PURE__ */React__default["default"].createElement(breadcrumb_BreadcrumbItem["default"], {
      key: "ellipsis",
      readOnly: true
    }, ellipsis2)], toConsumableArray._toConsumableArray(afterItems));
    return items;
  }, [renderOptionsToItems, options, children, maxItems, itemsBeforeCollapse, itemsAfterCollapse, renderEllipsis]);
  return /* @__PURE__ */React__default["default"].createElement(breadcrumb_BreadcrumbContext.BreadcrumbContext.Provider, {
    value: {
      maxItemWidthInContext: maxItemWidth,
      separator: separator
    }
  }, /* @__PURE__ */React__default["default"].createElement("div", _objectSpread({
    ref: ref,
    className: classNames__default["default"]("".concat(classPrefix, "-breadcrumb"), className)
  }, restProps), content));
}, {
  BreadcrumbItem: breadcrumb_BreadcrumbItem["default"]
});
Breadcrumb.displayName = "Breadcrumb";

exports["default"] = Breadcrumb;
