// tdesign-react/cjs/breadcrumb/BreadcrumbItem.js
/**
 * tdesign v1.17.1
 * (c) 2026 tdesign
 * @license MIT
 */

'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var defineProperty = require('../_chunks/dep-c446aa11.js');
var slicedToArray = require('../_chunks/dep-c60a3e4c.js');
var objectWithoutProperties = require('../_chunks/dep-e41137f7.js');
var React = require('react');
var classNames = require('classnames');
var tdesignIconsReact = require('tdesign-icons-react');
var _util_parseTNode = require('../_util/parseTNode.js');
var _util_scroll = require('../_util/scroll.js');
var hooks_useCommonClassName = require('../hooks/useCommonClassName.js');
var hooks_useConfig = require('../hooks/useConfig.js');
var hooks_useDefaultProps = require('../hooks/useDefaultProps.js');
var hooks_useGlobalIcon = require('../hooks/useGlobalIcon.js');
var tooltip_index = require('../tooltip/index.js');
var breadcrumb_BreadcrumbContext = require('./BreadcrumbContext.js');
var breadcrumb_defaultProps = require('./defaultProps.js');
var isFunction = require('../_chunks/dep-11347ea1.js');
require('../_chunks/dep-c290284d.js');
require('../_chunks/dep-98e6819c.js');
require('../_chunks/dep-7f3fa935.js');
require('../_chunks/dep-3c628d18.js');
require('../_chunks/dep-3d05de97.js');
require('../_chunks/dep-a830ce1e.js');
require('raf');
require('../_chunks/dep-ee71ddca.js');
require('../_chunks/dep-427f8bc0.js');
require('../_chunks/dep-a54f202c.js');
require('../_chunks/dep-4f1729f0.js');
require('../_util/easing.js');
require('../config-provider/ConfigContext.js');
require('../_chunks/dep-2ee09518.js');
require('../_chunks/dep-a6fa90dc.js');
require('dayjs');
require('../_chunks/dep-012d9a7c.js');
require('../_chunks/dep-150a112c.js');
require('../_chunks/dep-e360e17f.js');
require('../_chunks/dep-51284537.js');
require('../_chunks/dep-8a493834.js');
require('../_chunks/dep-9f5b2891.js');
require('../_chunks/dep-338f013a.js');
require('../_chunks/dep-c37c5e17.js');
require('../_chunks/dep-eee1aab9.js');
require('../_chunks/dep-770bc0ae.js');
require('../_chunks/dep-aaf27e39.js');
require('../_chunks/dep-b47d855f.js');
require('../_chunks/dep-81bce4e6.js');
require('../_chunks/dep-863f8f4d.js');
require('../_chunks/dep-84e4d0da.js');
require('../_chunks/dep-f78183d7.js');
require('../_chunks/dep-2f45f6c4.js');
require('../tooltip/Tooltip.js');
require('../popup/index.js');
require('../popup/Popup.js');
require('react-transition-group');
require('../common/Portal.js');
require('react-dom');
require('../hooks/useLayoutEffect.js');
require('../hooks/useAnimation.js');
require('../hooks/useAttach.js');
require('../hooks/useControlled.js');
require('../_util/noop.js');
require('../_chunks/dep-45188d54.js');
require('../_chunks/dep-eccb6a94.js');
require('../_chunks/dep-f5e1d11c.js');
require('../_chunks/dep-b199860d.js');
require('../_chunks/dep-36579bf6.js');
require('../hooks/useMutationObserver.js');
require('../hooks/useLatest.js');
require('../_chunks/dep-aa9a2c54.js');
require('../_chunks/dep-77ef3011.js');
require('../_chunks/dep-b36779a4.js');
require('../_chunks/dep-551d5940.js');
require('../_chunks/dep-8005b808.js');
require('../_chunks/dep-2add8824.js');
require('../_chunks/dep-f2809376.js');
require('../_chunks/dep-3be56e1e.js');
require('../hooks/usePopper.js');
require('../_chunks/dep-05bc49e6.js');
require('../_chunks/dep-8c880a65.js');
require('react-fast-compare');
require('@popperjs/core');
require('../hooks/useWindowSize.js');
require('../popup/defaultProps.js');
require('../popup/hooks/useTrigger.js');
require('../_util/listener.js');
require('../_util/ref.js');
require('react-is');
require('../_util/isFragment.js');
require('../popup/utils/transition.js');
require('../popup/PopupPlugin.js');
require('../_chunks/dep-7bf14076.js');
require('../_util/react-render.js');
require('../common/PluginContainer.js');
require('../config-provider/ConfigProvider.js');
require('../_chunks/dep-ba90997f.js');
require('../_chunks/dep-22026fea.js');
require('../tooltip/defaultProps.js');
require('../tooltip/TooltipLite.js');
require('../hooks/useSwitch.js');
require('../hooks/usePersistFn.js');
require('../_chunks/dep-d07eb641.js');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

var React__default = /*#__PURE__*/_interopDefaultLegacy(React);
var classNames__default = /*#__PURE__*/_interopDefaultLegacy(classNames);

var _excluded = ["children", "separator", "disabled", "maxItemWidth", "maxWidth", "icon", "href", "to", "target", "router", "replace", "className", "content", "onClick", "tooltipProps", "readOnly"];
function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { defineProperty._defineProperty(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
var BreadcrumbItem = /*#__PURE__*/React.forwardRef(function (props, ref) {
  var _useConfig = hooks_useConfig["default"](),
    classPrefix = _useConfig.classPrefix;
  var commonClassNames = hooks_useCommonClassName["default"]();
  var _useGlobalIcon = hooks_useGlobalIcon["default"]({
      ChevronRightIcon: tdesignIconsReact.ChevronRightIcon
    }),
    ChevronRightIcon = _useGlobalIcon.ChevronRightIcon;
  var _useDefaultProps = hooks_useDefaultProps["default"](props, breadcrumb_defaultProps.breadcrumbItemDefaultProps),
    children = _useDefaultProps.children,
    separator = _useDefaultProps.separator,
    disabled = _useDefaultProps.disabled,
    maxItemWidth = _useDefaultProps.maxItemWidth,
    maxWidth = _useDefaultProps.maxWidth,
    icon = _useDefaultProps.icon,
    href = _useDefaultProps.href,
    to = _useDefaultProps.to,
    target = _useDefaultProps.target,
    router = _useDefaultProps.router,
    replace = _useDefaultProps.replace,
    className = _useDefaultProps.className,
    content = _useDefaultProps.content,
    onClick = _useDefaultProps.onClick,
    tooltipProps = _useDefaultProps.tooltipProps,
    readonly = _useDefaultProps.readOnly,
    restProps = objectWithoutProperties._objectWithoutProperties(_useDefaultProps, _excluded);
  var _useContext = React.useContext(breadcrumb_BreadcrumbContext.BreadcrumbContext),
    maxItemWidthInContext = _useContext.maxItemWidthInContext,
    separatorInContext = _useContext.separator;
  var breadcrumbText = React.useRef(null);
  var _useState = React.useState(false),
    _useState2 = slicedToArray._slicedToArray(_useState, 2),
    isCutOff = _useState2[0],
    setIsCutOff = _useState2[1];
  var breadcrumbItemClassNames = classNames__default["default"]("".concat(classPrefix, "-breadcrumb__item"));
  var textWrapperClassName = "".concat(classPrefix, "-breadcrumb__inner");
  var textClassName = "".concat(classPrefix, "-breadcrumb__inner-text");
  var textClassNames = classNames__default["default"](defineProperty._defineProperty(defineProperty._defineProperty({}, "".concat(classPrefix, "-breadcrumb--text-overflow"), !readonly), commonClassNames.STATUS.disabled, disabled));
  var separatorClassName = "".concat(classPrefix, "-breadcrumb__separator");
  var linkClassName = "".concat(classPrefix, "-link");
  var maxWidthForItem = React.useMemo(function () {
    return {
      maxWidth: maxWidth || maxItemWidth || maxItemWidthInContext || "120px"
    };
  }, [maxItemWidth, maxWidth, maxItemWidthInContext]);
  React.useEffect(function () {
    if (!breadcrumbText.current) return;
    return setIsCutOff(_util_scroll.isNodeOverflow(breadcrumbText.current));
  }, [breadcrumbText]);
  var textContent = /* @__PURE__ */React__default["default"].createElement("span", {
    className: textWrapperClassName,
    style: maxWidthForItem
  }, isFunction.isFunction(icon) ? icon() : icon, /* @__PURE__ */React__default["default"].createElement("span", {
    ref: breadcrumbText,
    className: textClassName
  }, children || content));
  var itemContent = /* @__PURE__ */React__default["default"].createElement("span", {
    className: textClassNames
  }, textContent);
  if ((href || to) && !disabled) {
    var handleRouting = function handleRouting() {
      if (href || !router) return;
      replace ? router.replace(to) : router.push(to);
    };
    itemContent = /* @__PURE__ */React__default["default"].createElement("a", {
      className: classNames__default["default"](textClassNames, linkClassName),
      href: href,
      target: target,
      onClick: handleRouting
    }, textContent);
  }
  var separatorInProps = _util_parseTNode["default"](separator);
  var separatorContent = separatorInProps || separatorInContext || /* @__PURE__ */React__default["default"].createElement(ChevronRightIcon, {
    style: {
      opacity: ".5"
    }
  });
  var handleClick = function handleClick(e) {
    if (disabled) return;
    onClick === null || onClick === void 0 || onClick(e);
  };
  return /* @__PURE__ */React__default["default"].createElement("div", _objectSpread({
    className: classNames__default["default"](breadcrumbItemClassNames, className),
    ref: ref,
    onClick: handleClick
  }, restProps), isCutOff ? /* @__PURE__ */React__default["default"].createElement(tooltip_index.TooltipLite, _objectSpread({
    content: children || content
  }, tooltipProps), itemContent) : itemContent, /* @__PURE__ */React__default["default"].createElement("span", {
    className: separatorClassName
  }, separatorContent));
});
BreadcrumbItem.displayName = "BreadcrumbItem";

exports["default"] = BreadcrumbItem;
