// tdesign-react/cjs/anchor/Anchor.js
/**
 * tdesign v1.17.1
 * (c) 2026 tdesign
 * @license MIT
 */

'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var defineProperty = require('../_chunks/dep-c446aa11.js');
var slicedToArray = require('../_chunks/dep-c60a3e4c.js');
var objectWithoutProperties = require('../_chunks/dep-e41137f7.js');
var React = require('react');
var classNames = require('classnames');
var _util_style = require('../_chunks/dep-ee71ddca.js');
var _util_forwardRefWithStatics = require('../_util/forwardRefWithStatics.js');
var _util_scroll = require('../_util/scroll.js');
var affix_index = require('../affix/index.js');
var hooks_useConfig = require('../hooks/useConfig.js');
var hooks_useDefaultProps = require('../hooks/useDefaultProps.js');
var anchor_AnchorContext = require('./AnchorContext.js');
var anchor_AnchorItem = require('./AnchorItem.js');
var anchor_AnchorTarget = require('./AnchorTarget.js');
var anchor_defaultProps = require('./defaultProps.js');
var isEmpty = require('../_chunks/dep-2be6b7d2.js');
var isFunction = require('../_chunks/dep-11347ea1.js');
require('../_chunks/dep-c290284d.js');
require('../_chunks/dep-98e6819c.js');
require('../_chunks/dep-7f3fa935.js');
require('../_chunks/dep-427f8bc0.js');
require('../_chunks/dep-3d05de97.js');
require('../_chunks/dep-a54f202c.js');
require('../_chunks/dep-4f1729f0.js');
require('hoist-non-react-statics');
require('raf');
require('../_util/easing.js');
require('../affix/Affix.js');
require('../affix/defaultProps.js');
require('../config-provider/ConfigContext.js');
require('../_chunks/dep-2ee09518.js');
require('../_chunks/dep-a6fa90dc.js');
require('dayjs');
require('../_chunks/dep-012d9a7c.js');
require('../_chunks/dep-150a112c.js');
require('../_chunks/dep-e360e17f.js');
require('../_chunks/dep-51284537.js');
require('../_chunks/dep-8a493834.js');
require('../_chunks/dep-9f5b2891.js');
require('../_chunks/dep-338f013a.js');
require('../_chunks/dep-c37c5e17.js');
require('../_chunks/dep-eee1aab9.js');
require('../_chunks/dep-a830ce1e.js');
require('../_chunks/dep-770bc0ae.js');
require('../_chunks/dep-aaf27e39.js');
require('../_chunks/dep-b47d855f.js');
require('../_chunks/dep-81bce4e6.js');
require('../_chunks/dep-863f8f4d.js');
require('../_chunks/dep-84e4d0da.js');
require('../_chunks/dep-f78183d7.js');
require('../_chunks/dep-2f45f6c4.js');
require('../_util/noop.js');
require('tdesign-icons-react');
require('../_util/copyText.js');
require('../hooks/useGlobalIcon.js');
require('../message/index.js');
require('../message/Message.js');
require('../_chunks/dep-5094d657.js');
require('../_chunks/dep-147ad148.js');
require('../hooks/usePersistFn.js');
require('../message/MessageClose.js');
require('../message/useMessageClass.js');
require('../message/MessageIcon.js');
require('../loading/index.js');
require('../loading/Loading.js');
require('../common/Portal.js');
require('react-dom');
require('../hooks/useLayoutEffect.js');
require('../loading/defaultProps.js');
require('../loading/gradient.js');
require('../_chunks/dep-062bb1b1.js');
require('../_chunks/dep-6e0da644.js');
require('../_chunks/dep-05bc49e6.js');
require('../_chunks/dep-8c880a65.js');
require('../_chunks/dep-b8d0b96c.js');
require('../hooks/useDomRefCallback.js');
require('../loading/plugin.js');
require('../_util/react-render.js');
require('../_chunks/dep-7bf14076.js');
require('../common/PluginContainer.js');
require('../config-provider/ConfigProvider.js');
require('../_chunks/dep-ba90997f.js');
require('../_chunks/dep-22026fea.js');
require('../_chunks/dep-551d5940.js');
require('../_chunks/dep-8005b808.js');
require('../_chunks/dep-2add8824.js');
require('../message/config.js');
require('../message/defaultProps.js');
require('../_chunks/dep-b422817e.js');
require('../_chunks/dep-b199860d.js');
require('../_chunks/dep-36579bf6.js');
require('../_chunks/dep-45188d54.js');
require('../_chunks/dep-eccb6a94.js');
require('../_chunks/dep-f5e1d11c.js');
require('../popup/index.js');
require('../popup/Popup.js');
require('react-transition-group');
require('../hooks/useAnimation.js');
require('../hooks/useAttach.js');
require('../hooks/useControlled.js');
require('../hooks/useMutationObserver.js');
require('../hooks/useLatest.js');
require('../_chunks/dep-aa9a2c54.js');
require('../_chunks/dep-77ef3011.js');
require('../_chunks/dep-b36779a4.js');
require('../_chunks/dep-f2809376.js');
require('../_chunks/dep-3be56e1e.js');
require('../hooks/usePopper.js');
require('react-fast-compare');
require('@popperjs/core');
require('../hooks/useWindowSize.js');
require('../popup/defaultProps.js');
require('../popup/hooks/useTrigger.js');
require('../_util/listener.js');
require('../_util/ref.js');
require('react-is');
require('../_util/isFragment.js');
require('../popup/utils/transition.js');
require('../popup/PopupPlugin.js');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

var React__default = /*#__PURE__*/_interopDefaultLegacy(React);
var classNames__default = /*#__PURE__*/_interopDefaultLegacy(classNames);

var _excluded = ["affixProps", "bounds", "targetOffset", "container", "size", "children", "cursor", "onClick", "onChange", "className", "getCurrentAnchor"];
function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { defineProperty._defineProperty(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
var ANCHOR_SHARP_REGEXP = /#(\S+)$/;
var Anchor = _util_forwardRefWithStatics["default"](function (props, ref) {
  var _useDefaultProps = hooks_useDefaultProps["default"](props, anchor_defaultProps.anchorDefaultProps),
    affixProps = _useDefaultProps.affixProps,
    bounds = _useDefaultProps.bounds,
    targetOffset = _useDefaultProps.targetOffset,
    container = _useDefaultProps.container,
    size = _useDefaultProps.size,
    children = _useDefaultProps.children,
    cursor = _useDefaultProps.cursor,
    onClick = _useDefaultProps.onClick,
    onChange = _useDefaultProps.onChange,
    className = _useDefaultProps.className,
    getCurrentAnchor = _useDefaultProps.getCurrentAnchor,
    rest = objectWithoutProperties._objectWithoutProperties(_useDefaultProps, _excluded);
  var _useConfig = hooks_useConfig["default"](),
    classPrefix = _useConfig.classPrefix;
  var _useState = React.useState(""),
    _useState2 = slicedToArray._slicedToArray(_useState, 2),
    activeItem = _useState2[0],
    setActiveItem = _useState2[1];
  var _useState3 = React.useState({
      top: "0px",
      height: "0px",
      opacity: 0
    }),
    _useState4 = slicedToArray._slicedToArray(_useState3, 2),
    cursorStyle = _useState4[0],
    setCursorStyle = _useState4[1];
  var anchorEl = React.useRef(null);
  var intervalRef = React.useRef({
    items: [],
    scrollContainer: _util_style.canUseDocument ? window : null,
    handleScrollLock: false
  });
  React.useImperativeHandle(ref, function () {
    return anchorEl.current;
  });
  var registerItem = function registerItem(href) {
    var items = intervalRef.current.items;
    if (ANCHOR_SHARP_REGEXP.test(href) && items.indexOf(href) < 0) items.push(href);
  };
  var unregisterItem = function unregisterItem(href) {
    var items = intervalRef.current.items;
    intervalRef.current.items = items.filter(function (item) {
      return href !== item;
    });
  };
  var getAnchorTarget = function getAnchorTarget(href) {
    return document.querySelector(href);
  };
  var handleScrollTo = function handleScrollTo(link) {
    var anchor = getAnchorTarget(link);
    if (!anchor) return;
    onChange === null || onChange === void 0 || onChange(link, activeItem);
    setActiveItem(link);
    intervalRef.current.handleScrollLock = true;
    var scrollContainer = intervalRef.current.scrollContainer;
    var scrollTop = _util_scroll.getScroll(scrollContainer);
    var offsetTop = _util_scroll.getOffsetTop(anchor, scrollContainer);
    var top = scrollTop + offsetTop - targetOffset;
    _util_scroll.scrollTo(top, {
      container: scrollContainer
    }).then(function () {
      intervalRef.current.handleScrollLock = false;
    });
  };
  var handleClick = function handleClick(item, e) {
    onClick === null || onClick === void 0 || onClick(_objectSpread({
      e: e
    }, item));
    handleScrollTo(item.href);
  };
  React.useEffect(function () {
    var _anchorEl$current$que, _anchorEl$current;
    var pointEl = (_anchorEl$current$que = (_anchorEl$current = anchorEl.current).querySelector) === null || _anchorEl$current$que === void 0 ? void 0 : _anchorEl$current$que.call(_anchorEl$current, ".".concat(classPrefix, "-is-active>a"));
    if (!pointEl) {
      setCursorStyle(null);
    } else {
      var top = pointEl.offsetTop,
        height = pointEl.offsetHeight;
      setCursorStyle({
        top: "".concat(top, "px"),
        height: "".concat(height, "px"),
        opacity: 1
      });
    }
  }, [activeItem, classPrefix]);
  var handleScroll = React.useCallback(function () {
    var _intervalRef$current = intervalRef.current,
      scrollContainer = _intervalRef$current.scrollContainer,
      handleScrollLock = _intervalRef$current.handleScrollLock;
    if (handleScrollLock) return;
    var items = intervalRef.current.items;
    var filters = [];
    var active = "";
    items.forEach(function (href) {
      var anchor = getAnchorTarget(href);
      if (!anchor) return;
      var top = _util_scroll.getOffsetTop(anchor, scrollContainer);
      if (top <= bounds + targetOffset) {
        filters.push({
          href: href,
          top: top
        });
      }
    });
    if (filters.length) {
      var latest = filters.reduce(function (prev, cur) {
        return prev.top > cur.top ? prev : cur;
      });
      active = latest.href;
    }
    if (active !== activeItem) {
      onChange === null || onChange === void 0 || onChange(active, activeItem);
      setActiveItem(active);
    }
  }, [activeItem, bounds, onChange, targetOffset]);
  React.useEffect(function () {
    intervalRef.current.scrollContainer = _util_scroll.getScrollContainer(container);
    var scrollContainer = intervalRef.current.scrollContainer;
    handleScroll();
    scrollContainer === null || scrollContainer === void 0 || scrollContainer.addEventListener("scroll", handleScroll);
    return function () {
      scrollContainer === null || scrollContainer === void 0 || scrollContainer.removeEventListener("scroll", handleScroll);
    };
  }, [container, handleScroll]);
  var anchorClass = classNames__default["default"]("".concat(classPrefix, "-anchor"), defineProperty._defineProperty(defineProperty._defineProperty(defineProperty._defineProperty({}, "".concat(classPrefix, "-size-s"), size === "small"), "".concat(classPrefix, "-size-m"), size === "medium"), "".concat(classPrefix, "-size-l"), size === "large"), className);
  var CursorCmp = function CursorCmp() {
    if (isFunction.isFunction(cursor)) return cursor();
    if (isEmpty.isEmpty(cursor)) return /* @__PURE__ */React__default["default"].createElement("div", {
      className: "".concat(classPrefix, "-anchor__line-cursor")
    });
    return cursor;
  };
  var Cmp = /* @__PURE__ */React__default["default"].createElement(anchor_AnchorContext.AnchorContext.Provider, {
    value: {
      onClick: handleClick,
      activeItem: (getCurrentAnchor === null || getCurrentAnchor === void 0 ? void 0 : getCurrentAnchor(activeItem)) || activeItem,
      registerItem: registerItem,
      unregisterItem: unregisterItem
    }
  }, /* @__PURE__ */React__default["default"].createElement("div", _objectSpread(_objectSpread({}, rest), {}, {
    className: anchorClass,
    ref: anchorEl
  }), /* @__PURE__ */React__default["default"].createElement("div", {
    className: "".concat(classPrefix, "-anchor__line")
  }, /* @__PURE__ */React__default["default"].createElement("div", {
    className: "".concat(classPrefix, "-anchor__line-cursor-wrapper"),
    style: cursorStyle
  }, CursorCmp())), children));
  return isEmpty.isEmpty(affixProps) ? Cmp : /* @__PURE__ */React__default["default"].createElement(affix_index.Affix, _objectSpread({}, affixProps), Cmp);
}, {
  AnchorItem: anchor_AnchorItem["default"],
  AnchorTarget: anchor_AnchorTarget["default"]
});
Anchor.displayName = "Anchor";

exports["default"] = Anchor;
