// tdesign-react/cjs/anchor/AnchorItem.js
/**
 * tdesign v1.17.1
 * (c) 2026 tdesign
 * @license MIT
 */

'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var defineProperty = require('../_chunks/dep-c446aa11.js');
var objectWithoutProperties = require('../_chunks/dep-e41137f7.js');
var React = require('react');
var classNames = require('classnames');
var hooks_useConfig = require('../hooks/useConfig.js');
var hooks_useDefaultProps = require('../hooks/useDefaultProps.js');
var anchor_AnchorContext = require('./AnchorContext.js');
var anchor_defaultProps = require('./defaultProps.js');
require('../_chunks/dep-c290284d.js');
require('../config-provider/ConfigContext.js');
require('../_chunks/dep-2ee09518.js');
require('../_chunks/dep-a6fa90dc.js');
require('dayjs');
require('../_chunks/dep-012d9a7c.js');
require('../_chunks/dep-150a112c.js');
require('../_chunks/dep-e360e17f.js');
require('../_chunks/dep-3d05de97.js');
require('../_chunks/dep-4f1729f0.js');
require('../_chunks/dep-a54f202c.js');
require('../_chunks/dep-51284537.js');
require('../_chunks/dep-8a493834.js');
require('../_chunks/dep-9f5b2891.js');
require('../_chunks/dep-338f013a.js');
require('../_chunks/dep-c37c5e17.js');
require('../_chunks/dep-eee1aab9.js');
require('../_chunks/dep-11347ea1.js');
require('../_chunks/dep-a830ce1e.js');
require('../_chunks/dep-770bc0ae.js');
require('../_chunks/dep-aaf27e39.js');
require('../_chunks/dep-b47d855f.js');
require('../_chunks/dep-81bce4e6.js');
require('../_chunks/dep-863f8f4d.js');
require('../_chunks/dep-84e4d0da.js');
require('../_chunks/dep-f78183d7.js');
require('../_chunks/dep-2f45f6c4.js');
require('../_util/noop.js');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

var React__default = /*#__PURE__*/_interopDefaultLegacy(React);
var classNames__default = /*#__PURE__*/_interopDefaultLegacy(classNames);

var _excluded = ["href", "title", "target", "children", "className"];
function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { defineProperty._defineProperty(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
var AnchorItem = function AnchorItem(props) {
  var _useContext = React.useContext(anchor_AnchorContext.AnchorContext),
    onClick = _useContext.onClick,
    activeItem = _useContext.activeItem,
    registerItem = _useContext.registerItem,
    unregisterItem = _useContext.unregisterItem;
  var _useDefaultProps = hooks_useDefaultProps["default"](props, anchor_defaultProps.anchorItemDefaultProps),
    href = _useDefaultProps.href,
    title = _useDefaultProps.title,
    target = _useDefaultProps.target,
    children = _useDefaultProps.children,
    className = _useDefaultProps.className,
    rest = objectWithoutProperties._objectWithoutProperties(_useDefaultProps, _excluded);
  var _useConfig = hooks_useConfig["default"](),
    classPrefix = _useConfig.classPrefix;
  var titleAttr = typeof title === "string" ? title : null;
  var handleClick = function handleClick(e) {
    onClick({
      title: titleAttr,
      href: href
    }, e);
  };
  React.useEffect(function () {
    registerItem(href);
    return function () {
      return unregisterItem(href);
    };
  }, [href, registerItem, unregisterItem]);
  return /* @__PURE__ */React__default["default"].createElement("div", _objectSpread(_objectSpread({}, rest), {}, {
    className: classNames__default["default"]("".concat(classPrefix, "-anchor__item"), defineProperty._defineProperty({}, "".concat(classPrefix, "-is-active"), activeItem === href), className)
  }), /* @__PURE__ */React__default["default"].createElement("a", {
    href: href,
    className: classNames__default["default"]("".concat(classPrefix, "-anchor__item-link")),
    title: titleAttr,
    target: target,
    onClick: function onClick(e) {
      return handleClick(e);
    }
  }, title), children);
};
AnchorItem.displayName = "AnchorItem";

exports["default"] = AnchorItem;
