// tdesign-react/cjs/common/Check.js
/**
 * tdesign v1.17.1
 * (c) 2026 tdesign
 * @license MIT
 */

'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var defineProperty = require('../_chunks/dep-c446aa11.js');
var slicedToArray = require('../_chunks/dep-c60a3e4c.js');
var objectWithoutProperties = require('../_chunks/dep-e41137f7.js');
var React = require('react');
var classNames = require('classnames');
var _util_helper = require('../_util/helper.js');
var hooks_useConfig = require('../hooks/useConfig.js');
var hooks_useControlled = require('../hooks/useControlled.js');
var isBoolean = require('../_chunks/dep-fb263e64.js');
require('../_chunks/dep-c290284d.js');
require('../_chunks/dep-98e6819c.js');
require('../_chunks/dep-7f3fa935.js');
require('../_chunks/dep-b422817e.js');
require('../_chunks/dep-b199860d.js');
require('../_chunks/dep-3d05de97.js');
require('../_chunks/dep-a54f202c.js');
require('../_chunks/dep-36579bf6.js');
require('../_chunks/dep-4f1729f0.js');
require('../_chunks/dep-45188d54.js');
require('../_chunks/dep-eccb6a94.js');
require('../_chunks/dep-f5e1d11c.js');
require('../config-provider/ConfigContext.js');
require('../_chunks/dep-2ee09518.js');
require('../_chunks/dep-a6fa90dc.js');
require('dayjs');
require('../_chunks/dep-012d9a7c.js');
require('../_chunks/dep-150a112c.js');
require('../_chunks/dep-e360e17f.js');
require('../_chunks/dep-51284537.js');
require('../_chunks/dep-8a493834.js');
require('../_chunks/dep-9f5b2891.js');
require('../_chunks/dep-338f013a.js');
require('../_chunks/dep-c37c5e17.js');
require('../_chunks/dep-eee1aab9.js');
require('../_chunks/dep-11347ea1.js');
require('../_chunks/dep-a830ce1e.js');
require('../_chunks/dep-770bc0ae.js');
require('../_chunks/dep-aaf27e39.js');
require('../_chunks/dep-b47d855f.js');
require('../_chunks/dep-81bce4e6.js');
require('../_chunks/dep-863f8f4d.js');
require('../_chunks/dep-84e4d0da.js');
require('../_chunks/dep-f78183d7.js');
require('../_chunks/dep-2f45f6c4.js');
require('../_util/noop.js');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

var React__default = /*#__PURE__*/_interopDefaultLegacy(React);
var classNames__default = /*#__PURE__*/_interopDefaultLegacy(classNames);

var _excluded = ["allowUncheck", "type", "disabled", "name", "value", "onChange", "indeterminate", "children", "label", "className", "style", "onClick"];
function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { defineProperty._defineProperty(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
var CheckContext = /*#__PURE__*/React__default["default"].createContext(null);
var Check = /*#__PURE__*/React.forwardRef(function (_props, ref) {
  var context = React.useContext(CheckContext);
  var props = context ? context.inject(_props) : _props;
  var _props$allowUncheck = props.allowUncheck,
    allowUncheck = _props$allowUncheck === void 0 ? false : _props$allowUncheck,
    type = props.type,
    disabled = props.disabled,
    name = props.name,
    value = props.value,
    onChange = props.onChange,
    indeterminate = props.indeterminate,
    children = props.children,
    label = props.label,
    className = props.className,
    style = props.style,
    onClick = props.onClick,
    htmlProps = objectWithoutProperties._objectWithoutProperties(props, _excluded);
  var _useConfig = hooks_useConfig["default"](),
    classPrefix = _useConfig.classPrefix;
  var TOnChange = onChange;
  var _useControlled = hooks_useControlled["default"](props, "checked", TOnChange),
    _useControlled2 = slicedToArray._slicedToArray(_useControlled, 2),
    internalChecked = _useControlled2[0],
    setInternalChecked = _useControlled2[1];
  var labelClassName = classNames__default["default"]("".concat(classPrefix, "-").concat(type), className, defineProperty._defineProperty(defineProperty._defineProperty(defineProperty._defineProperty({}, "".concat(classPrefix, "-is-checked"), internalChecked), "".concat(classPrefix, "-is-disabled"), disabled), "".concat(classPrefix, "-is-indeterminate"), indeterminate));
  var readOnly = props.readOnly || props.readonly;
  var isDisabled = disabled || readOnly;
  var input = /* @__PURE__ */React__default["default"].createElement("input", {
    readOnly: readOnly,
    type: type === "radio-button" ? "radio" : type,
    className: "".concat(classPrefix, "-").concat(type, "__former"),
    checked: internalChecked,
    disabled: disabled,
    name: name,
    tabIndex: -1,
    value: isBoolean.isBoolean(value) ? Number(value) : value,
    "data-value": typeof value === "string" ? "'".concat(value, "'") : value,
    "data-allow-uncheck": allowUncheck || void 0,
    onClick: function onClick(e) {
      e.stopPropagation();
      if ((type === "radio-button" || type === "radio") && allowUncheck && internalChecked) {
        setInternalChecked(!e.currentTarget.checked, {
          e: e
        });
      }
    },
    onChange: function onChange(e) {
      if (isDisabled) return;
      setInternalChecked(e.currentTarget.checked, {
        e: e
      });
    }
  });
  var showLabel = !!(children || label);
  var handleLabelClick = function handleLabelClick(event) {
    if (props.stopLabelTrigger) {
      event.preventDefault();
    }
  };
  var onInnerClick = function onInnerClick(e) {
    if (isDisabled) return;
    onClick === null || onClick === void 0 || onClick({
      e: e
    });
  };
  return /* @__PURE__ */React__default["default"].createElement("label", _objectSpread(_objectSpread({
    ref: ref,
    tabIndex: disabled ? void 0 : 0,
    className: labelClassName,
    title: props.title,
    style: style
  }, _util_helper.omit(htmlProps, ["checkAll", "stopLabelTrigger", "readonly"])), {}, {
    onClick: onInnerClick
  }), input, /* @__PURE__ */React__default["default"].createElement("span", {
    className: "".concat(classPrefix, "-").concat(type, "__input")
  }), showLabel && /* @__PURE__ */React__default["default"].createElement("span", {
    key: "label",
    className: "".concat(classPrefix, "-").concat(type, "__label"),
    onClick: handleLabelClick
  }, children || label));
});
Check.displayName = "Check";

exports.CheckContext = CheckContext;
exports["default"] = Check;
