// tdesign-react/cjs/list/ListItemMeta.js
/**
 * tdesign v1.17.1
 * (c) 2026 tdesign
 * @license MIT
 */

'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var React = require('react');
var classNames = require('classnames');
var hooks_useConfig = require('../hooks/useConfig.js');
require('../config-provider/ConfigContext.js');
require('../_chunks/dep-c446aa11.js');
require('../_chunks/dep-c290284d.js');
require('../_chunks/dep-2ee09518.js');
require('../_chunks/dep-a6fa90dc.js');
require('dayjs');
require('../_chunks/dep-012d9a7c.js');
require('../_chunks/dep-150a112c.js');
require('../_chunks/dep-e360e17f.js');
require('../_chunks/dep-3d05de97.js');
require('../_chunks/dep-4f1729f0.js');
require('../_chunks/dep-a54f202c.js');
require('../_chunks/dep-51284537.js');
require('../_chunks/dep-8a493834.js');
require('../_chunks/dep-9f5b2891.js');
require('../_chunks/dep-338f013a.js');
require('../_chunks/dep-c37c5e17.js');
require('../_chunks/dep-eee1aab9.js');
require('../_chunks/dep-11347ea1.js');
require('../_chunks/dep-a830ce1e.js');
require('../_chunks/dep-770bc0ae.js');
require('../_chunks/dep-aaf27e39.js');
require('../_chunks/dep-b47d855f.js');
require('../_chunks/dep-81bce4e6.js');
require('../_chunks/dep-863f8f4d.js');
require('../_chunks/dep-84e4d0da.js');
require('../_chunks/dep-f78183d7.js');
require('../_chunks/dep-2f45f6c4.js');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

var React__default = /*#__PURE__*/_interopDefaultLegacy(React);
var classNames__default = /*#__PURE__*/_interopDefaultLegacy(classNames);

var ListItemMeta = /*#__PURE__*/React.forwardRef(function (props, ref) {
  var title = props.title,
    image = props.image,
    description = props.description,
    className = props.className,
    style = props.style;
  var _useConfig = hooks_useConfig["default"](),
    classPrefix = _useConfig.classPrefix;
  var renderAvatar = function renderAvatar() {
    if (image && typeof image === "string") {
      return /* @__PURE__ */React__default["default"].createElement("div", {
        className: "".concat(classPrefix, "-list-item__meta-avatar")
      }, /* @__PURE__ */React__default["default"].createElement("img", {
        src: image,
        alt: ""
      }));
    }
    return /* @__PURE__ */React__default["default"].createElement("div", {
      className: "".concat(classPrefix, "-list-item__meta-avatar")
    }, image);
  };
  return /* @__PURE__ */React__default["default"].createElement("div", {
    ref: ref,
    className: classNames__default["default"]("".concat(classPrefix, "-list-item__meta"), className),
    style: style
  }, image && renderAvatar(), /* @__PURE__ */React__default["default"].createElement("div", {
    className: "".concat(classPrefix, "-list-item__meta-content")
  }, /* @__PURE__ */React__default["default"].createElement("h3", {
    className: "".concat(classPrefix, "-list-item__meta-title")
  }, title), /* @__PURE__ */React__default["default"].createElement("div", {
    className: "".concat(classPrefix, "-list-item__meta-description")
  }, typeof description === "string" ? /* @__PURE__ */React__default["default"].createElement("p", null, description) : description)));
});
ListItemMeta.displayName = "ListItemMeta";

exports["default"] = ListItemMeta;
