// tdesign-react/es/menu/Menu.js
/**
 * tdesign v1.17.1
 * (c) 2026 tdesign
 * @license MIT
 */

import { _ as _defineProperty } from '../_chunks/dep-55b77e26.js';
import React from 'react';
import classNames from 'classnames';
import forwardRefWithStatics from '../_util/forwardRefWithStatics.js';
import useConfig from '../hooks/useConfig.js';
import useDefaultProps from '../hooks/useDefaultProps.js';
import { DEFAULT_MENU_WIDTH } from './_util/constant.js';
import { menuDefaultProps } from './defaultProps.js';
import HeadMenu from './HeadMenu.js';
import useMenuContext from './hooks/useMenuContext.js';
import { MenuContext } from './MenuContext.js';
import MenuGroup from './MenuGroup.js';
import MenuItem from './MenuItem.js';
import SubMenu from './SubMenu.js';
import '../_chunks/dep-067cb22b.js';
import 'hoist-non-react-statics';
import '../config-provider/ConfigContext.js';
import 'lodash-es';
import '../locale/zh_CN.js';
import '../_chunks/dep-e56ffacc.js';
import 'dayjs';
import '../_chunks/dep-84036ea0.js';
import '../_chunks/dep-84926167.js';
import '../tabs/index.js';
import '../tabs/style/css.js';
import '../tabs/Tabs.js';
import '../_chunks/dep-d1eeb95c.js';
import '../_chunks/dep-673e8854.js';
import '../_chunks/dep-570a5adb.js';
import '../hooks/useControlled.js';
import '../_util/noop.js';
import '../hooks/useDragSorter.js';
import '../hooks/useEventCallback.js';
import '../tabs/defaultProps.js';
import '../tabs/TabNav.js';
import 'tdesign-icons-react';
import '../_util/parseTNode.js';
import '../_chunks/dep-0f787ecd.js';
import '../_chunks/dep-65b369a5.js';
import '../hooks/useGlobalIcon.js';
import '../tabs/TabBar.js';
import '../hooks/useMutationObserver.js';
import '../hooks/useLatest.js';
import '../hooks/useResizeObserver.js';
import '../_chunks/dep-8c226881.js';
import '../hooks/useLayoutEffect.js';
import '../tabs/TabNavItem.js';
import '../hooks/useDomRefCallback.js';
import '../hooks/useRipple.js';
import '../_chunks/dep-e17ec8df.js';
import '../hooks/useAnimation.js';
import '../tabs/useTabClass.js';
import '../tabs/TabPanel.js';
import '../tabs/type.js';
import './_util/checkSubMenuActive.js';
import './_util/checkMenuType.js';
import './_util/type.js';
import './_util/checkSubMenuChildrenActive.js';
import './_util/getMenuTree.js';
import '../_chunks/dep-154fcb93.js';
import '../_chunks/dep-e9c0bb63.js';
import '../_chunks/dep-6ce6d9a9.js';
import './_util/calculatePaddingLeft.js';
import '../tooltip/index.js';
import '../tooltip/style/css.js';
import '../tooltip/Tooltip.js';
import '../popup/index.js';
import '../popup/style/css.js';
import '../popup/Popup.js';
import 'react-transition-group';
import '../common/Portal.js';
import 'react-dom';
import '../hooks/useAttach.js';
import '../hooks/usePopper.js';
import 'react-fast-compare';
import '@popperjs/core';
import '../hooks/useWindowSize.js';
import '../popup/defaultProps.js';
import '../popup/hooks/useTrigger.js';
import '../_util/listener.js';
import '../_util/ref.js';
import 'react-is';
import '../_util/isFragment.js';
import '../popup/utils/transition.js';
import '../popup/PopupPlugin.js';
import '../_chunks/dep-55f620a2.js';
import '../_util/react-render.js';
import '../common/PluginContainer.js';
import '../config-provider/index.js';
import '../config-provider/ConfigProvider.js';
import '../config-provider/type.js';
import '../popup/type.js';
import '../tooltip/defaultProps.js';
import '../tooltip/TooltipLite.js';
import '../hooks/useSwitch.js';
import '../hooks/usePersistFn.js';
import '../tooltip/type.js';
import '../common/FakeArrow.js';
import '../_chunks/dep-89bc2167.js';
import './_util/getSubMenuChildStyle.js';

function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { _defineProperty(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
var Menu = forwardRefWithStatics(function (originalProps, ref) {
  var props = useDefaultProps(originalProps, menuDefaultProps);
  var className = props.className,
    _props$width = props.width,
    width = _props$width === void 0 ? DEFAULT_MENU_WIDTH : _props$width,
    children = props.children,
    theme = props.theme,
    logo = props.logo,
    operations = props.operations,
    style = props.style;
  var _useConfig = useConfig(),
    classPrefix = _useConfig.classPrefix;
  var _useMenuContext = useMenuContext(_objectSpread(_objectSpread({}, props), {}, {
      children: children,
      mode: "accordion"
    })),
    value = _useMenuContext.value;
  var menuWidthArr = Array.isArray(width) ? width : [width, DEFAULT_MENU_WIDTH[1]];
  var collapsed = value.collapsed;
  return /* @__PURE__ */React.createElement(MenuContext.Provider, {
    value: value
  }, /* @__PURE__ */React.createElement("div", {
    ref: ref,
    className: classNames("".concat(classPrefix, "-default-menu"), className, _defineProperty(_defineProperty({}, "".concat(classPrefix, "-is-collapsed"), collapsed), "".concat(classPrefix, "-menu--dark"), theme === "dark")),
    style: _objectSpread({
      width: collapsed ? menuWidthArr[1] : menuWidthArr[0]
    }, style)
  }, /* @__PURE__ */React.createElement("div", {
    className: "".concat(classPrefix, "-default-menu__inner")
  }, logo && /* @__PURE__ */React.createElement("div", {
    className: "".concat(classPrefix, "-menu__logo")
  }, logo), /* @__PURE__ */React.createElement("ul", {
    className: classNames("".concat(classPrefix, "-menu"), "".concat(classPrefix, "-menu--scroll"))
  }, children), operations && /* @__PURE__ */React.createElement("div", {
    className: "".concat(classPrefix, "-menu__operations")
  }, operations))));
}, {
  HeadMenu: HeadMenu,
  SubMenu: SubMenu,
  MenuItem: MenuItem,
  MenuGroup: MenuGroup
});
Menu.displayName = "Menu";

export { Menu as default };
