// tdesign-react/cjs/grid/Row.js
/**
 * tdesign v1.17.1
 * (c) 2026 tdesign
 * @license MIT
 */

'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var defineProperty = require('../_chunks/dep-c446aa11.js');
var slicedToArray = require('../_chunks/dep-c60a3e4c.js');
var objectWithoutProperties = require('../_chunks/dep-e41137f7.js');
var React = require('react');
var classNames = require('classnames');
var _util_style = require('../_chunks/dep-ee71ddca.js');
var hooks_useConfig = require('../hooks/useConfig.js');
var hooks_useDefaultProps = require('../hooks/useDefaultProps.js');
var grid_defaultProps = require('./defaultProps.js');
var isObject = require('../_chunks/dep-a830ce1e.js');
require('../_chunks/dep-c290284d.js');
require('../_chunks/dep-98e6819c.js');
require('../_chunks/dep-7f3fa935.js');
require('../_chunks/dep-427f8bc0.js');
require('../_chunks/dep-3d05de97.js');
require('../_chunks/dep-a54f202c.js');
require('../_chunks/dep-4f1729f0.js');
require('../config-provider/ConfigContext.js');
require('../_chunks/dep-2ee09518.js');
require('../_chunks/dep-a6fa90dc.js');
require('dayjs');
require('../_chunks/dep-012d9a7c.js');
require('../_chunks/dep-150a112c.js');
require('../_chunks/dep-e360e17f.js');
require('../_chunks/dep-51284537.js');
require('../_chunks/dep-8a493834.js');
require('../_chunks/dep-9f5b2891.js');
require('../_chunks/dep-338f013a.js');
require('../_chunks/dep-c37c5e17.js');
require('../_chunks/dep-eee1aab9.js');
require('../_chunks/dep-11347ea1.js');
require('../_chunks/dep-770bc0ae.js');
require('../_chunks/dep-aaf27e39.js');
require('../_chunks/dep-b47d855f.js');
require('../_chunks/dep-81bce4e6.js');
require('../_chunks/dep-863f8f4d.js');
require('../_chunks/dep-84e4d0da.js');
require('../_chunks/dep-f78183d7.js');
require('../_chunks/dep-2f45f6c4.js');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

var React__default = /*#__PURE__*/_interopDefaultLegacy(React);
var classNames__default = /*#__PURE__*/_interopDefaultLegacy(classNames);

var _excluded = ["align", "gutter", "justify", "tag", "style", "className", "children"];
function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { defineProperty._defineProperty(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
var calcSize = function calcSize(width) {
  var smWidth = parseFloat(_util_style.getCssVarsValue("--td-screen-sm") || "768");
  var mdWidth = parseFloat(_util_style.getCssVarsValue("--td-screen-md") || "992");
  var lgWidth = parseFloat(_util_style.getCssVarsValue("--td-screen-lg") || "1200");
  var xlWidth = parseFloat(_util_style.getCssVarsValue("--td-screen-xl") || "1400");
  var xxlWidth = parseFloat(_util_style.getCssVarsValue("--td-screen-xxl") || "1880");
  var size = "xs";
  if (width >= xxlWidth) {
    size = "xxl";
  } else if (width >= xlWidth) {
    size = "xl";
  } else if (width >= lgWidth) {
    size = "lg";
  } else if (width >= mdWidth) {
    size = "md";
  } else if (width >= smWidth) {
    size = "sm";
  } else {
    size = "xs";
  }
  return size;
};
var calcRowStyle = function calcRowStyle(gutter, currentSize) {
  var rowStyle = {};
  if (typeof gutter === "number") {
    Object.assign(rowStyle, {
      marginLeft: "".concat(gutter / -2, "px"),
      marginRight: "".concat(gutter / -2, "px")
    });
  } else if (Array.isArray(gutter) && gutter.length) {
    if (typeof gutter[0] === "number") {
      Object.assign(rowStyle, {
        marginLeft: "".concat(gutter[0] / -2, "px"),
        marginRight: "".concat(gutter[0] / -2, "px")
      });
    }
    if (typeof gutter[1] === "number") {
      Object.assign(rowStyle, {
        rowGap: "".concat(gutter[1], "px")
      });
    }
    if (isObject.isObject(gutter[0]) && gutter[0][currentSize] !== void 0) {
      Object.assign(rowStyle, {
        marginLeft: "".concat(gutter[0][currentSize] / -2, "px"),
        marginRight: "".concat(gutter[0][currentSize] / -2, "px")
      });
    }
    if (isObject.isObject(gutter[1]) && gutter[1][currentSize] !== void 0) {
      Object.assign(rowStyle, {
        rowGap: "".concat(gutter[1][currentSize], "px")
      });
    }
  } else if (isObject.isObject(gutter) && gutter[currentSize]) {
    if (Array.isArray(gutter[currentSize]) && gutter[currentSize].length) {
      Object.assign(rowStyle, {
        marginLeft: "".concat(gutter[currentSize][0] / -2, "px"),
        marginRight: "".concat(gutter[currentSize][0] / -2, "px")
      });
      Object.assign(rowStyle, {
        rowGap: "".concat(gutter[currentSize][1], "px")
      });
    } else {
      Object.assign(rowStyle, {
        marginLeft: "".concat(gutter[currentSize] / -2, "px"),
        marginRight: "".concat(gutter[currentSize] / -2, "px")
      });
    }
  }
  return rowStyle;
};
var RowContext = /*#__PURE__*/React.createContext({
  gutter: void 0,
  size: void 0
});
var Row = /*#__PURE__*/React.forwardRef(function (props, ref) {
  var _useDefaultProps = hooks_useDefaultProps["default"](props, grid_defaultProps.rowDefaultProps),
    align = _useDefaultProps.align,
    gutter = _useDefaultProps.gutter,
    justify = _useDefaultProps.justify,
    tag = _useDefaultProps.tag,
    propStyle = _useDefaultProps.style,
    className = _useDefaultProps.className,
    children = _useDefaultProps.children,
    otherRowProps = objectWithoutProperties._objectWithoutProperties(_useDefaultProps, _excluded);
  var _useState = React.useState(function () {
      return _util_style.canUseDocument ? calcSize(window.innerWidth) : "md";
    }),
    _useState2 = slicedToArray._slicedToArray(_useState, 2),
    size = _useState2[0],
    setSize = _useState2[1];
  var updateSize = function updateSize() {
    var currentSize = calcSize(window.innerWidth);
    if (currentSize !== size) {
      setSize(size);
    }
  };
  var _useConfig = hooks_useConfig["default"](),
    classPrefix = _useConfig.classPrefix;
  var rowClassNames = classNames__default["default"]("".concat(classPrefix, "-row"), className, defineProperty._defineProperty(defineProperty._defineProperty({}, "".concat(classPrefix, "-row--").concat(justify), true), "".concat(classPrefix, "-row--").concat(align), true));
  var rowStyle = _objectSpread(_objectSpread({}, calcRowStyle(gutter, size)), propStyle);
  React.useEffect(function () {
    window.addEventListener("resize", updateSize);
    return function () {
      window.removeEventListener("resize", updateSize);
    };
  });
  return /*#__PURE__*/React__default["default"].createElement(tag, _objectSpread({
    ref: ref,
    className: rowClassNames,
    style: rowStyle
  }, otherRowProps), /* @__PURE__ */React__default["default"].createElement(RowContext.Provider, {
    value: {
      gutter: gutter,
      size: size
    }
  }, children));
});
Row.displayName = "Row";

exports.RowContext = RowContext;
exports["default"] = Row;
