// tdesign-react/cjs/upload/themes/DraggerFile.js
/**
 * tdesign v1.17.1
 * (c) 2026 tdesign
 * @license MIT
 */

'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var toConsumableArray = require('../../_chunks/dep-05bc49e6.js');
var defineProperty = require('../../_chunks/dep-c446aa11.js');
var React = require('react');
var classNames = require('classnames');
var tdesignIconsReact = require('tdesign-icons-react');
var utils = require('../../_chunks/dep-24740dfb.js');
var _util_parseTNode = require('../../_util/parseTNode.js');
var hooks_useCommonClassName = require('../../hooks/useCommonClassName.js');
var hooks_useGlobalIcon = require('../../hooks/useGlobalIcon.js');
var image_index = require('../../image/index.js');
var imageViewer_index = require('../../image-viewer/index.js');
var link_index = require('../../link/index.js');
var loading_index = require('../../loading/index.js');
var upload_hooks_useDrag = require('../hooks/useDrag.js');
require('../../_chunks/dep-7f3fa935.js');
require('../../_chunks/dep-8c880a65.js');
require('../../_chunks/dep-c290284d.js');
require('../../_chunks/dep-3c628d18.js');
require('../../_chunks/dep-11347ea1.js');
require('../../_chunks/dep-3d05de97.js');
require('../../_chunks/dep-a830ce1e.js');
require('../../hooks/useConfig.js');
require('../../config-provider/ConfigContext.js');
require('../../_chunks/dep-2ee09518.js');
require('../../_chunks/dep-a6fa90dc.js');
require('dayjs');
require('../../_chunks/dep-012d9a7c.js');
require('../../_chunks/dep-150a112c.js');
require('../../_chunks/dep-e360e17f.js');
require('../../_chunks/dep-4f1729f0.js');
require('../../_chunks/dep-a54f202c.js');
require('../../_chunks/dep-51284537.js');
require('../../_chunks/dep-8a493834.js');
require('../../_chunks/dep-9f5b2891.js');
require('../../_chunks/dep-338f013a.js');
require('../../_chunks/dep-c37c5e17.js');
require('../../_chunks/dep-eee1aab9.js');
require('../../_chunks/dep-770bc0ae.js');
require('../../_chunks/dep-aaf27e39.js');
require('../../_chunks/dep-b47d855f.js');
require('../../_chunks/dep-81bce4e6.js');
require('../../_chunks/dep-863f8f4d.js');
require('../../_chunks/dep-84e4d0da.js');
require('../../_chunks/dep-f78183d7.js');
require('../../_chunks/dep-2f45f6c4.js');
require('../../image/Image.js');
require('../../_chunks/dep-c60a3e4c.js');
require('../../_chunks/dep-98e6819c.js');
require('../../_chunks/dep-e41137f7.js');
require('../../_chunks/dep-2bc69864.js');
require('../../hooks/useDefaultProps.js');
require('../../hooks/useImagePreviewUrl.js');
require('../../locale/LocalReceiver.js');
require('../../_chunks/dep-427f8bc0.js');
require('../../config-provider/ConfigProvider.js');
require('../../_chunks/dep-ba90997f.js');
require('../../_chunks/dep-22026fea.js');
require('../../_chunks/dep-551d5940.js');
require('../../_chunks/dep-8005b808.js');
require('../../_chunks/dep-2add8824.js');
require('../../space/index.js');
require('../../space/Space.js');
require('react-is');
require('../../_chunks/dep-6e0da644.js');
require('../../_chunks/dep-b8d0b96c.js');
require('../../space/defaultProps.js');
require('../../image/defaultProps.js');
require('../../image-viewer/ImageViewer.js');
require('react-dom');
require('../../_chunks/dep-4f5ff67a.js');
require('../../_chunks/dep-7bf14076.js');
require('../../_chunks/dep-98b1e4fa.js');
require('../../_chunks/dep-f7c108a2.js');
require('../../_chunks/dep-bfa39549.js');
require('../../tooltip/index.js');
require('../../tooltip/Tooltip.js');
require('../../popup/index.js');
require('../../popup/Popup.js');
require('react-transition-group');
require('../../_chunks/dep-ee71ddca.js');
require('../../common/Portal.js');
require('../../hooks/useLayoutEffect.js');
require('../../hooks/useAnimation.js');
require('../../hooks/useAttach.js');
require('../../hooks/useControlled.js');
require('../../_util/noop.js');
require('../../_chunks/dep-45188d54.js');
require('../../_chunks/dep-eccb6a94.js');
require('../../_chunks/dep-f5e1d11c.js');
require('../../_chunks/dep-b199860d.js');
require('../../_chunks/dep-36579bf6.js');
require('../../hooks/useMutationObserver.js');
require('../../hooks/useLatest.js');
require('../../_chunks/dep-aa9a2c54.js');
require('../../_chunks/dep-77ef3011.js');
require('../../_chunks/dep-b36779a4.js');
require('../../_chunks/dep-f2809376.js');
require('../../_chunks/dep-3be56e1e.js');
require('../../hooks/usePopper.js');
require('react-fast-compare');
require('@popperjs/core');
require('../../hooks/useWindowSize.js');
require('../../popup/defaultProps.js');
require('../../popup/hooks/useTrigger.js');
require('../../_util/listener.js');
require('../../_util/ref.js');
require('../../_util/isFragment.js');
require('../../popup/utils/transition.js');
require('../../popup/PopupPlugin.js');
require('../../_util/react-render.js');
require('../../common/PluginContainer.js');
require('../../tooltip/defaultProps.js');
require('../../tooltip/TooltipLite.js');
require('../../hooks/useSwitch.js');
require('../../hooks/usePersistFn.js');
require('../../_chunks/dep-d07eb641.js');
require('../../image-viewer/hooks/useIconMap.js');
require('../../image-viewer/hooks/useIndex.js');
require('../../image-viewer/hooks/useMirror.js');
require('../../image-viewer/hooks/usePosition.js');
require('../../hooks/useMouseEvent.js');
require('../../image-viewer/hooks/useRotate.js');
require('../../image-viewer/hooks/useScale.js');
require('../../dialog/index.js');
require('../../dialog/Dialog.js');
require('../../hooks/useDeepEffect.js');
require('../../hooks/useSetState.js');
require('../../dialog/defaultProps.js');
require('../../dialog/DialogCard.js');
require('../../button/index.js');
require('../../button/Button.js');
require('../../hooks/useDomRefCallback.js');
require('../../hooks/useRipple.js');
require('../../_chunks/dep-062bb1b1.js');
require('../../button/defaultProps.js');
require('../../loading/Loading.js');
require('../../loading/defaultProps.js');
require('../../loading/gradient.js');
require('../../loading/plugin.js');
require('../../dialog/hooks/useDialogDrag.js');
require('../../dialog/hooks/useDialogEsc.js');
require('../../dialog/hooks/useDialogPosition.js');
require('../../dialog/hooks/useLockStyle.js');
require('../../_chunks/dep-ea91dbfd.js');
require('../../dialog/plugin.js');
require('../../image-viewer/defaultProps.js');
require('../../image-viewer/defaultTrigger.js');
require('../../image-viewer/hooks/useImageScale.js');
require('../../image-viewer/hooks/useViewerScale.js');
require('../../link/Link.js');
require('../../link/defaultProps.js');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

var React__default = /*#__PURE__*/_interopDefaultLegacy(React);
var classNames__default = /*#__PURE__*/_interopDefaultLegacy(classNames);

function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { defineProperty._defineProperty(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
var DraggerFile = function DraggerFile(props) {
  var displayFiles = props.displayFiles,
    locale = props.locale,
    disabled = props.disabled,
    trigger = props.trigger,
    accept = props.accept;
  var _useCommonClassName = hooks_useCommonClassName["default"](),
    SIZE = _useCommonClassName.SIZE;
  var uploadPrefix = "".concat(props.classPrefix, "-upload");
  var drag = upload_hooks_useDrag["default"](_objectSpread(_objectSpread({}, props.dragEvents), {}, {
    accept: accept
  }));
  var dragActive = drag.dragActive;
  var draggerFileRef = React.useRef(null);
  var _useGlobalIcon = hooks_useGlobalIcon["default"]({
      CheckCircleFilledIcon: tdesignIconsReact.CheckCircleFilledIcon,
      ErrorCircleFilledIcon: tdesignIconsReact.ErrorCircleFilledIcon
    }),
    CheckCircleFilledIcon = _useGlobalIcon.CheckCircleFilledIcon,
    ErrorCircleFilledIcon = _useGlobalIcon.ErrorCircleFilledIcon;
  var firstFile = displayFiles[0];
  var firstFileStatus = firstFile === null || firstFile === void 0 ? void 0 : firstFile.status;
  var classes = React.useMemo(function () {
    return ["".concat(uploadPrefix, "__dragger"), defineProperty._defineProperty({}, "".concat(uploadPrefix, "__dragger-center"), !firstFile), defineProperty._defineProperty({}, "".concat(uploadPrefix, "__dragger-error"), firstFileStatus === "fail")];
  }, [firstFile, firstFileStatus, uploadPrefix]);
  var renderImage = function renderImage() {
    var _file$response;
    var file = displayFiles[0];
    var url = (file === null || file === void 0 ? void 0 : file.url) || (file === null || file === void 0 || (_file$response = file.response) === null || _file$response === void 0 ? void 0 : _file$response.url);
    return /* @__PURE__ */React__default["default"].createElement("div", {
      className: "".concat(uploadPrefix, "__dragger-img-wrap")
    }, /* @__PURE__ */React__default["default"].createElement(imageViewer_index.ImageViewer, _objectSpread({
      images: [url],
      trigger: function trigger(_ref3) {
        var open = _ref3.open;
        return /* @__PURE__ */React__default["default"].createElement(image_index.Image, {
          src: url || file.raw,
          onClick: function onClick() {
            return open();
          }
        });
      }
    }, props.imageViewerProps)));
  };
  var renderUploading = function renderUploading() {
    var file = displayFiles[0];
    if ((file === null || file === void 0 ? void 0 : file.status) === "progress") {
      return /* @__PURE__ */React__default["default"].createElement("div", {
        className: "".concat(uploadPrefix, "__single-progress")
      }, /* @__PURE__ */React__default["default"].createElement(loading_index.Loading, null), /* @__PURE__ */React__default["default"].createElement("span", {
        className: "".concat(uploadPrefix, "__single-percent")
      }, file.percent, "%"));
    }
  };
  var renderMainPreview = function renderMainPreview() {
    var file = displayFiles[0];
    var fileName = props.abridgeName ? utils.abridgeName.apply(void 0, [file.name].concat(toConsumableArray._toConsumableArray(props.abridgeName))) : file.name;
    var fileInfo = /* @__PURE__ */React__default["default"].createElement(React__default["default"].Fragment, null, /* @__PURE__ */React__default["default"].createElement("div", {
      className: "".concat(uploadPrefix, "__dragger-text")
    }, /* @__PURE__ */React__default["default"].createElement("span", {
      className: "".concat(uploadPrefix, "__single-name")
    }, fileName), file.status === "progress" && renderUploading(), file.status === "success" && /* @__PURE__ */React__default["default"].createElement(CheckCircleFilledIcon, null), file.status === "fail" && /* @__PURE__ */React__default["default"].createElement(ErrorCircleFilledIcon, null)), /* @__PURE__ */React__default["default"].createElement("small", {
      className: "".concat(SIZE.small)
    }, locale.file.fileSizeText, "\uFF1A", utils.getFileSizeText(file.size)), /* @__PURE__ */React__default["default"].createElement("small", {
      className: "".concat(SIZE.small)
    }, locale.file.fileOperationDateText, "\uFF1A", file.uploadTime || "-"));
    return /* @__PURE__ */React__default["default"].createElement("div", {
      className: "".concat(uploadPrefix, "__dragger-progress")
    }, props.theme === "image" && renderImage(), /* @__PURE__ */React__default["default"].createElement("div", {
      className: "".concat(uploadPrefix, "__dragger-progress-info")
    }, props.fileListDisplay ? _util_parseTNode.parseContentTNode(props.fileListDisplay, {
      files: displayFiles
    }) : fileInfo, /* @__PURE__ */React__default["default"].createElement("div", {
      className: "".concat(uploadPrefix, "__dragger-btns")
    }, ["progress", "waiting"].includes(file.status) && !disabled && /* @__PURE__ */React__default["default"].createElement(link_index.Link, {
      theme: "primary",
      hover: "color",
      disabled: disabled,
      className: "".concat(uploadPrefix, "__dragger-progress-cancel"),
      onClick: function onClick(e) {
        var _props$cancelUpload;
        return (_props$cancelUpload = props.cancelUpload) === null || _props$cancelUpload === void 0 ? void 0 : _props$cancelUpload.call(props, {
          e: e,
          file: props.toUploadFiles[0] || props.files[0]
        });
      }
    }, locale === null || locale === void 0 ? void 0 : locale.cancelUploadText), !props.autoUpload && file.status === "waiting" && /* @__PURE__ */React__default["default"].createElement(link_index.Link, {
      theme: "primary",
      hover: "color",
      disabled: disabled,
      onClick: function onClick() {
        var _props$uploadFiles;
        return (_props$uploadFiles = props.uploadFiles) === null || _props$uploadFiles === void 0 ? void 0 : _props$uploadFiles.call(props);
      },
      className: "".concat(uploadPrefix, "__dragger-upload-btn")
    }, locale.triggerUploadText.normal)), ["fail", "success"].includes(file === null || file === void 0 ? void 0 : file.status) && !disabled && /* @__PURE__ */React__default["default"].createElement("div", {
      className: "".concat(uploadPrefix, "__dragger-btns")
    }, /* @__PURE__ */React__default["default"].createElement(link_index.Link, {
      theme: "primary",
      hover: "color",
      disabled: disabled,
      className: "".concat(uploadPrefix, "__dragger-progress-cancel"),
      onClick: props.triggerUpload
    }, locale.triggerUploadText.reupload), /* @__PURE__ */React__default["default"].createElement(link_index.Link, {
      theme: "danger",
      hover: "color",
      disabled: disabled,
      className: "".concat(uploadPrefix, "__dragger-delete-btn"),
      onClick: function onClick(e) {
        return props.onRemove({
          e: e,
          index: 0,
          file: file
        });
      }
    }, locale.triggerUploadText["delete"]))));
  };
  var renderDefaultDragElement = function renderDefaultDragElement() {
    var _locale$triggerUpload;
    var unActiveElement = /* @__PURE__ */React__default["default"].createElement("div", null, /* @__PURE__ */React__default["default"].createElement("span", {
      className: "".concat(uploadPrefix, "--highlight")
    }, (_locale$triggerUpload = locale.triggerUploadText) === null || _locale$triggerUpload === void 0 ? void 0 : _locale$triggerUpload.normal), /* @__PURE__ */React__default["default"].createElement("span", null, "\xA0\xA0/\xA0\xA0", locale.dragger.draggingText));
    var activeElement = /* @__PURE__ */React__default["default"].createElement("div", null, locale.dragger.dragDropText);
    return dragActive ? activeElement : unActiveElement;
  };
  var getContent = function getContent() {
    var file = displayFiles[0];
    if (file && (["progress", "success", "fail", "waiting"].includes(file.status) || !file.status)) {
      return renderMainPreview();
    }
    return /* @__PURE__ */React__default["default"].createElement("div", {
      className: "".concat(uploadPrefix, "__trigger"),
      onClick: props.triggerUpload
    }, props.children || renderDefaultDragElement());
  };
  return /* @__PURE__ */React__default["default"].createElement("div", {
    ref: draggerFileRef,
    className: classNames__default["default"](classes),
    onDrop: drag.handleDrop,
    onDragEnter: drag.handleDragenter,
    onDragOver: drag.handleDragover,
    onDragLeave: drag.handleDragleave
  }, (_util_parseTNode.parseContentTNode === null || _util_parseTNode.parseContentTNode === void 0 ? void 0 : _util_parseTNode.parseContentTNode(trigger, {
    files: displayFiles,
    dragActive: dragActive
  })) || getContent());
};
DraggerFile.displayName = "DraggerFile";

exports["default"] = DraggerFile;
