// tdesign-react/cjs/upload/themes/ImageCard.js
/**
 * tdesign v1.17.1
 * (c) 2026 tdesign
 * @license MIT
 */

'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var toConsumableArray = require('../../_chunks/dep-05bc49e6.js');
var defineProperty = require('../../_chunks/dep-c446aa11.js');
var objectWithoutProperties = require('../../_chunks/dep-e41137f7.js');
var React = require('react');
var classNames = require('classnames');
var tdesignIconsReact = require('tdesign-icons-react');
var utils = require('../../_chunks/dep-24740dfb.js');
var _util_parseTNode = require('../../_util/parseTNode.js');
var hooks_useGlobalIcon = require('../../hooks/useGlobalIcon.js');
var image_index = require('../../image/index.js');
var imageViewer_index = require('../../image-viewer/index.js');
var link_index = require('../../link/index.js');
var loading_index = require('../../loading/index.js');
require('../../_chunks/dep-7f3fa935.js');
require('../../_chunks/dep-8c880a65.js');
require('../../_chunks/dep-c290284d.js');
require('../../_chunks/dep-3c628d18.js');
require('../../_chunks/dep-11347ea1.js');
require('../../_chunks/dep-3d05de97.js');
require('../../_chunks/dep-a830ce1e.js');
require('../../hooks/useConfig.js');
require('../../config-provider/ConfigContext.js');
require('../../_chunks/dep-2ee09518.js');
require('../../_chunks/dep-a6fa90dc.js');
require('dayjs');
require('../../_chunks/dep-012d9a7c.js');
require('../../_chunks/dep-150a112c.js');
require('../../_chunks/dep-e360e17f.js');
require('../../_chunks/dep-4f1729f0.js');
require('../../_chunks/dep-a54f202c.js');
require('../../_chunks/dep-51284537.js');
require('../../_chunks/dep-8a493834.js');
require('../../_chunks/dep-9f5b2891.js');
require('../../_chunks/dep-338f013a.js');
require('../../_chunks/dep-c37c5e17.js');
require('../../_chunks/dep-eee1aab9.js');
require('../../_chunks/dep-770bc0ae.js');
require('../../_chunks/dep-aaf27e39.js');
require('../../_chunks/dep-b47d855f.js');
require('../../_chunks/dep-81bce4e6.js');
require('../../_chunks/dep-863f8f4d.js');
require('../../_chunks/dep-84e4d0da.js');
require('../../_chunks/dep-f78183d7.js');
require('../../_chunks/dep-2f45f6c4.js');
require('../../image/Image.js');
require('../../_chunks/dep-c60a3e4c.js');
require('../../_chunks/dep-98e6819c.js');
require('../../_chunks/dep-2bc69864.js');
require('../../hooks/useDefaultProps.js');
require('../../hooks/useImagePreviewUrl.js');
require('../../locale/LocalReceiver.js');
require('../../_chunks/dep-427f8bc0.js');
require('../../config-provider/ConfigProvider.js');
require('../../_chunks/dep-ba90997f.js');
require('../../_chunks/dep-22026fea.js');
require('../../_chunks/dep-551d5940.js');
require('../../_chunks/dep-8005b808.js');
require('../../_chunks/dep-2add8824.js');
require('../../space/index.js');
require('../../space/Space.js');
require('react-is');
require('../../_chunks/dep-6e0da644.js');
require('../../_chunks/dep-b8d0b96c.js');
require('../../space/defaultProps.js');
require('../../image/defaultProps.js');
require('../../image-viewer/ImageViewer.js');
require('react-dom');
require('../../_chunks/dep-4f5ff67a.js');
require('../../_chunks/dep-7bf14076.js');
require('../../_chunks/dep-98b1e4fa.js');
require('../../_chunks/dep-f7c108a2.js');
require('../../_chunks/dep-bfa39549.js');
require('../../tooltip/index.js');
require('../../tooltip/Tooltip.js');
require('../../popup/index.js');
require('../../popup/Popup.js');
require('react-transition-group');
require('../../_chunks/dep-ee71ddca.js');
require('../../common/Portal.js');
require('../../hooks/useLayoutEffect.js');
require('../../hooks/useAnimation.js');
require('../../hooks/useAttach.js');
require('../../hooks/useControlled.js');
require('../../_util/noop.js');
require('../../_chunks/dep-45188d54.js');
require('../../_chunks/dep-eccb6a94.js');
require('../../_chunks/dep-f5e1d11c.js');
require('../../_chunks/dep-b199860d.js');
require('../../_chunks/dep-36579bf6.js');
require('../../hooks/useMutationObserver.js');
require('../../hooks/useLatest.js');
require('../../_chunks/dep-aa9a2c54.js');
require('../../_chunks/dep-77ef3011.js');
require('../../_chunks/dep-b36779a4.js');
require('../../_chunks/dep-f2809376.js');
require('../../_chunks/dep-3be56e1e.js');
require('../../hooks/usePopper.js');
require('react-fast-compare');
require('@popperjs/core');
require('../../hooks/useWindowSize.js');
require('../../popup/defaultProps.js');
require('../../popup/hooks/useTrigger.js');
require('../../_util/listener.js');
require('../../_util/ref.js');
require('../../_util/isFragment.js');
require('../../popup/utils/transition.js');
require('../../popup/PopupPlugin.js');
require('../../_util/react-render.js');
require('../../common/PluginContainer.js');
require('../../tooltip/defaultProps.js');
require('../../tooltip/TooltipLite.js');
require('../../hooks/useSwitch.js');
require('../../hooks/usePersistFn.js');
require('../../_chunks/dep-d07eb641.js');
require('../../image-viewer/hooks/useIconMap.js');
require('../../image-viewer/hooks/useIndex.js');
require('../../image-viewer/hooks/useMirror.js');
require('../../image-viewer/hooks/usePosition.js');
require('../../hooks/useMouseEvent.js');
require('../../image-viewer/hooks/useRotate.js');
require('../../image-viewer/hooks/useScale.js');
require('../../dialog/index.js');
require('../../dialog/Dialog.js');
require('../../hooks/useDeepEffect.js');
require('../../hooks/useSetState.js');
require('../../dialog/defaultProps.js');
require('../../dialog/DialogCard.js');
require('../../button/index.js');
require('../../button/Button.js');
require('../../hooks/useDomRefCallback.js');
require('../../hooks/useRipple.js');
require('../../_chunks/dep-062bb1b1.js');
require('../../button/defaultProps.js');
require('../../loading/Loading.js');
require('../../loading/defaultProps.js');
require('../../loading/gradient.js');
require('../../loading/plugin.js');
require('../../dialog/hooks/useDialogDrag.js');
require('../../dialog/hooks/useDialogEsc.js');
require('../../dialog/hooks/useDialogPosition.js');
require('../../dialog/hooks/useLockStyle.js');
require('../../_chunks/dep-ea91dbfd.js');
require('../../dialog/plugin.js');
require('../../image-viewer/defaultProps.js');
require('../../image-viewer/defaultTrigger.js');
require('../../image-viewer/hooks/useImageScale.js');
require('../../image-viewer/hooks/useViewerScale.js');
require('../../link/Link.js');
require('../../link/defaultProps.js');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

var React__default = /*#__PURE__*/_interopDefaultLegacy(React);
var classNames__default = /*#__PURE__*/_interopDefaultLegacy(classNames);

var _excluded = ["className"];
function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { defineProperty._defineProperty(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
var ImageCard = function ImageCard(props) {
  var _locale$triggerUpload;
  var displayFiles = props.displayFiles,
    locale = props.locale,
    classPrefix = props.classPrefix,
    multiple = props.multiple,
    _props$max = props.max,
    max = _props$max === void 0 ? 0 : _props$max,
    onRemove = props.onRemove,
    disabled = props.disabled,
    fileListDisplay = props.fileListDisplay,
    _props$imageProps = props.imageProps,
    imageProps = _props$imageProps === void 0 ? {} : _props$imageProps;
  var _useGlobalIcon = hooks_useGlobalIcon["default"]({
      AddIcon: tdesignIconsReact.AddIcon,
      BrowseIcon: tdesignIconsReact.BrowseIcon,
      DeleteIcon: tdesignIconsReact.DeleteIcon,
      ErrorCircleFilledIcon: tdesignIconsReact.ErrorCircleFilledIcon
    }),
    BrowseIcon = _useGlobalIcon.BrowseIcon,
    DeleteIcon = _useGlobalIcon.DeleteIcon,
    AddIcon = _useGlobalIcon.AddIcon,
    ErrorCircleFilledIcon = _useGlobalIcon.ErrorCircleFilledIcon;
  var imageClassName = imageProps.className,
    restImageProps = objectWithoutProperties._objectWithoutProperties(imageProps, _excluded);
  var showTrigger = React__default["default"].useMemo(function () {
    if (multiple) {
      return !max || displayFiles.length < max;
    }
    return !(displayFiles !== null && displayFiles !== void 0 && displayFiles[0]);
  }, [displayFiles, max, multiple]);
  var renderMainContent = function renderMainContent(file, index) {
    return /* @__PURE__ */React__default["default"].createElement("div", {
      className: "".concat(classPrefix, "-upload__card-content ").concat(classPrefix, "-upload__card-box")
    }, /* @__PURE__ */React__default["default"].createElement(image_index.Image, _objectSpread(_objectSpread({
      fit: "contain",
      className: classNames__default["default"]("".concat(classPrefix, "-upload__card-image"), imageClassName)
    }, restImageProps), {}, {
      src: file.url || file.raw
    })), /* @__PURE__ */React__default["default"].createElement("div", {
      className: "".concat(classPrefix, "-upload__card-mask")
    }, /* @__PURE__ */React__default["default"].createElement("span", {
      className: "".concat(classPrefix, "-upload__card-mask-item"),
      onClick: function onClick(e) {
        return e.stopPropagation();
      }
    }, /* @__PURE__ */React__default["default"].createElement(imageViewer_index.ImageViewer, _objectSpread({
      trigger: function trigger(_ref) {
        var open = _ref.open;
        return /* @__PURE__ */React__default["default"].createElement(BrowseIcon, {
          onClick: function onClick(e) {
            var _props$onPreview;
            (_props$onPreview = props.onPreview) === null || _props$onPreview === void 0 || _props$onPreview.call(props, {
              file: file,
              index: index,
              e: e
            });
            open();
          }
        });
      },
      images: displayFiles.map(function (t) {
        return t.url || t.raw;
      }),
      defaultIndex: index
    }, props.imageViewerProps))), !disabled && /* @__PURE__ */React__default["default"].createElement(React__default["default"].Fragment, null, /* @__PURE__ */React__default["default"].createElement("span", {
      className: "".concat(classPrefix, "-upload__card-mask-item-divider")
    }), /* @__PURE__ */React__default["default"].createElement("span", {
      className: "".concat(classPrefix, "-upload__card-mask-item"),
      onClick: function onClick(e) {
        return e.stopPropagation();
      }
    }, /* @__PURE__ */React__default["default"].createElement(DeleteIcon, {
      onClick: function onClick(e) {
        return onRemove === null || onRemove === void 0 ? void 0 : onRemove({
          e: e,
          file: file,
          index: index
        });
      }
    })))));
  };
  var renderProgressFile = function renderProgressFile(file, loadCard) {
    var _locale$progress;
    return /* @__PURE__ */React__default["default"].createElement("div", {
      className: classNames__default["default"]([loadCard, "".concat(classPrefix, "-upload__").concat(props.theme, "-").concat(file.status)])
    }, /* @__PURE__ */React__default["default"].createElement(loading_index.Loading, {
      loading: true,
      size: "medium"
    }), /* @__PURE__ */React__default["default"].createElement("p", null, locale === null || locale === void 0 || (_locale$progress = locale.progress) === null || _locale$progress === void 0 ? void 0 : _locale$progress.uploadingText, props.showUploadProgress ? " ".concat(file.percent, "%") : ""));
  };
  var renderFailFile = function renderFailFile(file, index, loadCard) {
    var _file$response, _locale$progress2;
    return /* @__PURE__ */React__default["default"].createElement("div", {
      className: loadCard
    }, /* @__PURE__ */React__default["default"].createElement(ErrorCircleFilledIcon, null), /* @__PURE__ */React__default["default"].createElement("p", null, ((_file$response = file.response) === null || _file$response === void 0 ? void 0 : _file$response.error) || (locale === null || locale === void 0 || (_locale$progress2 = locale.progress) === null || _locale$progress2 === void 0 ? void 0 : _locale$progress2.failText)), /* @__PURE__ */React__default["default"].createElement("div", {
      className: "".concat(classPrefix, "-upload__card-mask")
    }, /* @__PURE__ */React__default["default"].createElement("span", {
      className: "".concat(classPrefix, "-upload__card-mask-item"),
      onClick: function onClick(e) {
        return e.stopPropagation();
      }
    }, /* @__PURE__ */React__default["default"].createElement(DeleteIcon, {
      onClick: function onClick(e) {
        var _props$onRemove;
        return (_props$onRemove = props.onRemove) === null || _props$onRemove === void 0 ? void 0 : _props$onRemove.call(props, {
          e: e,
          file: file,
          index: index
        });
      }
    }))));
  };
  var cardItemClasses = "".concat(classPrefix, "-upload__card-item ").concat(classPrefix, "-is-background");
  if (fileListDisplay) {
    return /* @__PURE__ */React__default["default"].createElement("div", null, _util_parseTNode["default"](fileListDisplay, {
      triggerUpload: props.triggerUpload,
      uploadFiles: props.uploadFiles,
      cancelUpload: props.cancelUpload,
      onPreview: props.onPreview,
      onRemove: props.onRemove,
      toUploadFiles: props.toUploadFiles,
      sizeOverLimitMessage: props.sizeOverLimitMessage,
      locale: props.locale,
      files: displayFiles
    }));
  }
  return /* @__PURE__ */React__default["default"].createElement("div", null, /* @__PURE__ */React__default["default"].createElement("ul", {
    className: "".concat(classPrefix, "-upload__card")
  }, displayFiles === null || displayFiles === void 0 ? void 0 : displayFiles.map(function (file, index) {
    var loadCard = "".concat(classPrefix, "-upload__card-container ").concat(classPrefix, "-upload__card-box");
    var fileName = props.abridgeName ? utils.abridgeName.apply(void 0, [file.name].concat(toConsumableArray._toConsumableArray(props.abridgeName))) : file.name;
    var fileNameClassName = "".concat(classPrefix, "-upload__card-name");
    return /* @__PURE__ */React__default["default"].createElement("li", {
      className: cardItemClasses,
      key: index
    }, file.status === "progress" && renderProgressFile(file, loadCard), file.status === "fail" && renderFailFile(file, index, loadCard), !["progress", "fail"].includes(file.status) && renderMainContent(file, index), fileName && props.showImageFileName && (file.url ? /* @__PURE__ */React__default["default"].createElement(link_index.Link, {
      href: file.url,
      className: fileNameClassName,
      target: "_blank",
      hover: "color",
      size: "small"
    }, fileName) : /* @__PURE__ */React__default["default"].createElement("span", {
      className: fileNameClassName
    }, fileName)));
  }), showTrigger && /* @__PURE__ */React__default["default"].createElement("li", {
    className: cardItemClasses,
    onClick: props.triggerUpload
  }, /* @__PURE__ */React__default["default"].createElement("div", {
    className: classNames__default["default"](["".concat(classPrefix, "-upload__image-add"), "".concat(classPrefix, "-upload__card-container"), "".concat(classPrefix, "-upload__card-box"), defineProperty._defineProperty({}, "".concat(classPrefix, "-is-disabled"), props.disabled)])
  }, /* @__PURE__ */React__default["default"].createElement(AddIcon, null), /* @__PURE__ */React__default["default"].createElement("p", {
    className: classNames__default["default"](["".concat(classPrefix, "-size-s"), "".concat(classPrefix, "-upload__add-text")])
  }, locale === null || locale === void 0 || (_locale$triggerUpload = locale.triggerUploadText) === null || _locale$triggerUpload === void 0 ? void 0 : _locale$triggerUpload.image)))));
};
ImageCard.displayName = "ImageCard";

exports["default"] = ImageCard;
