// tdesign-react/cjs/upload/themes/NormalFile.js
/**
 * tdesign v1.17.1
 * (c) 2026 tdesign
 * @license MIT
 */

'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var defineProperty = require('../../_chunks/dep-c446aa11.js');
var toConsumableArray = require('../../_chunks/dep-05bc49e6.js');
var React = require('react');
var classNames = require('classnames');
var tdesignIconsReact = require('tdesign-icons-react');
var utils = require('../../_chunks/dep-24740dfb.js');
var _util_parseTNode = require('../../_util/parseTNode.js');
var hooks_useGlobalIcon = require('../../hooks/useGlobalIcon.js');
var link_index = require('../../link/index.js');
var loading_index = require('../../loading/index.js');
require('../../_chunks/dep-c290284d.js');
require('../../_chunks/dep-7f3fa935.js');
require('../../_chunks/dep-8c880a65.js');
require('../../_chunks/dep-3c628d18.js');
require('../../_chunks/dep-11347ea1.js');
require('../../_chunks/dep-3d05de97.js');
require('../../_chunks/dep-a830ce1e.js');
require('../../hooks/useConfig.js');
require('../../config-provider/ConfigContext.js');
require('../../_chunks/dep-2ee09518.js');
require('../../_chunks/dep-a6fa90dc.js');
require('dayjs');
require('../../_chunks/dep-012d9a7c.js');
require('../../_chunks/dep-150a112c.js');
require('../../_chunks/dep-e360e17f.js');
require('../../_chunks/dep-4f1729f0.js');
require('../../_chunks/dep-a54f202c.js');
require('../../_chunks/dep-51284537.js');
require('../../_chunks/dep-8a493834.js');
require('../../_chunks/dep-9f5b2891.js');
require('../../_chunks/dep-338f013a.js');
require('../../_chunks/dep-c37c5e17.js');
require('../../_chunks/dep-eee1aab9.js');
require('../../_chunks/dep-770bc0ae.js');
require('../../_chunks/dep-aaf27e39.js');
require('../../_chunks/dep-b47d855f.js');
require('../../_chunks/dep-81bce4e6.js');
require('../../_chunks/dep-863f8f4d.js');
require('../../_chunks/dep-84e4d0da.js');
require('../../_chunks/dep-f78183d7.js');
require('../../_chunks/dep-2f45f6c4.js');
require('../../link/Link.js');
require('../../_chunks/dep-e41137f7.js');
require('../../hooks/useDefaultProps.js');
require('../../link/defaultProps.js');
require('../../loading/Loading.js');
require('../../_chunks/dep-c60a3e4c.js');
require('../../_chunks/dep-98e6819c.js');
require('../../_chunks/dep-ee71ddca.js');
require('../../_chunks/dep-427f8bc0.js');
require('../../common/Portal.js');
require('react-dom');
require('../../hooks/useLayoutEffect.js');
require('../../loading/defaultProps.js');
require('../../loading/gradient.js');
require('../../_chunks/dep-062bb1b1.js');
require('../../_chunks/dep-6e0da644.js');
require('../../_chunks/dep-b8d0b96c.js');
require('../../hooks/useDomRefCallback.js');
require('../../loading/plugin.js');
require('../../_util/react-render.js');
require('../../_chunks/dep-7bf14076.js');
require('../../common/PluginContainer.js');
require('../../config-provider/ConfigProvider.js');
require('../../_chunks/dep-ba90997f.js');
require('../../_chunks/dep-22026fea.js');
require('../../_chunks/dep-551d5940.js');
require('../../_chunks/dep-8005b808.js');
require('../../_chunks/dep-2add8824.js');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

var React__default = /*#__PURE__*/_interopDefaultLegacy(React);
var classNames__default = /*#__PURE__*/_interopDefaultLegacy(classNames);

function NormalFile(props) {
  var _displayFiles$, _displayFiles$0$respo;
  var theme = props.theme,
    disabled = props.disabled,
    classPrefix = props.classPrefix,
    locale = props.locale;
  var _useGlobalIcon = hooks_useGlobalIcon["default"]({
      CloseIcon: tdesignIconsReact.CloseIcon,
      TimeFilledIcon: tdesignIconsReact.TimeFilledIcon,
      CheckCircleFilledIcon: tdesignIconsReact.CheckCircleFilledIcon,
      ErrorCircleFilledIcon: tdesignIconsReact.ErrorCircleFilledIcon,
      CloseCircleFilledIcon: tdesignIconsReact.CloseCircleFilledIcon
    }),
    CloseIcon = _useGlobalIcon.CloseIcon,
    TimeFilledIcon = _useGlobalIcon.TimeFilledIcon,
    CheckCircleFilledIcon = _useGlobalIcon.CheckCircleFilledIcon,
    ErrorCircleFilledIcon = _useGlobalIcon.ErrorCircleFilledIcon,
    CloseCircleFilledIcon = _useGlobalIcon.CloseCircleFilledIcon;
  var uploadPrefix = "".concat(classPrefix, "-upload");
  var renderProgress = function renderProgress(percent) {
    return /* @__PURE__ */React__default["default"].createElement("div", {
      className: "".concat(uploadPrefix, "__single-progress")
    }, /* @__PURE__ */React__default["default"].createElement(loading_index.Loading, null), /* @__PURE__ */React__default["default"].createElement("span", {
      className: "".concat(uploadPrefix, "__single-percent")
    }, percent || 0, "%"));
  };
  var renderFilePreviewAsText = function renderFilePreviewAsText(files) {
    var _files$;
    if (theme !== "file") return null;
    if (!props.multiple && ((_files$ = files[0]) === null || _files$ === void 0 ? void 0 : _files$.status) === "fail" && props.autoUpload) {
      return null;
    }
    return files.map(function (file, index) {
      var fileName = props.abridgeName && file.name ? utils.abridgeName.apply(void 0, [file.name].concat(toConsumableArray._toConsumableArray(props.abridgeName))) : file.name;
      return /* @__PURE__ */React__default["default"].createElement("div", {
        className: "".concat(uploadPrefix, "__single-display-text ").concat(uploadPrefix, "__display-text--margin"),
        key: file.name + index + file.percent + file.status
      }, file.url ? /* @__PURE__ */React__default["default"].createElement(link_index.Link, {
        href: file.url,
        target: "_blank",
        hover: "color",
        size: "small",
        className: "".concat(uploadPrefix, "__single-name")
      }, fileName) : /* @__PURE__ */React__default["default"].createElement("span", {
        className: "".concat(uploadPrefix, "__single-name")
      }, fileName), file.status === "fail" && /* @__PURE__ */React__default["default"].createElement("div", {
        className: "".concat(uploadPrefix, "__flow-status ").concat(uploadPrefix, "__file-fail")
      }, /* @__PURE__ */React__default["default"].createElement(ErrorCircleFilledIcon, null)), file.status === "waiting" && /* @__PURE__ */React__default["default"].createElement("div", {
        className: "".concat(uploadPrefix, "__flow-status ").concat(uploadPrefix, "__file-waiting")
      }, /* @__PURE__ */React__default["default"].createElement(TimeFilledIcon, null)), file.status === "progress" && renderProgress(file.percent), !disabled && file.status !== "progress" && /* @__PURE__ */React__default["default"].createElement(CloseIcon, {
        className: "".concat(uploadPrefix, "__icon-delete"),
        onClick: function onClick(e) {
          return props.onRemove({
            e: e,
            file: file,
            index: index
          });
        }
      }));
    });
  };
  var renderFilePreviewAsInput = function renderFilePreviewAsInput() {
    var _props$abridgeName;
    var file = props.displayFiles[0];
    var inputTextClass = ["".concat(classPrefix, "-input__inner"), defineProperty._defineProperty({}, "".concat(uploadPrefix, "__placeholder"), !props.displayFiles[0])];
    var disabledClass = disabled ? "".concat(classPrefix, "-is-disabled") : "";
    var fileName = (_props$abridgeName = props.abridgeName) !== null && _props$abridgeName !== void 0 && _props$abridgeName.length && file !== null && file !== void 0 && file.name ? utils.abridgeName.apply(void 0, [file.name].concat(toConsumableArray._toConsumableArray(props.abridgeName))) : file === null || file === void 0 ? void 0 : file.name;
    return /* @__PURE__ */React__default["default"].createElement("div", {
      className: "".concat(uploadPrefix, "__single-input-preview ").concat(classPrefix, "-input ").concat(disabledClass)
    }, /* @__PURE__ */React__default["default"].createElement("div", {
      className: classNames__default["default"](inputTextClass)
    }, /* @__PURE__ */React__default["default"].createElement("span", {
      className: classNames__default["default"]("".concat(uploadPrefix, "__single-input-text"), defineProperty._defineProperty({}, props.placeholderClass, props.placeholder && !(file !== null && file !== void 0 && file.name)))
    }, file !== null && file !== void 0 && file.name ? fileName : props.placeholder), (file === null || file === void 0 ? void 0 : file.status) === "progress" && renderProgress(file.percent), (file === null || file === void 0 ? void 0 : file.status) === "waiting" && /* @__PURE__ */React__default["default"].createElement(TimeFilledIcon, {
      className: "".concat(uploadPrefix, "__status-icon ").concat(uploadPrefix, "__file-waiting")
    }), (file === null || file === void 0 ? void 0 : file.name) && file.status === "success" && /* @__PURE__ */React__default["default"].createElement(CheckCircleFilledIcon, {
      className: "".concat(uploadPrefix, "__status-icon")
    }), (file === null || file === void 0 ? void 0 : file.name) && file.status === "fail" && /* @__PURE__ */React__default["default"].createElement(ErrorCircleFilledIcon, {
      className: "".concat(uploadPrefix, "__status-icon ").concat(uploadPrefix, "__file-fail")
    }), Boolean(!disabled && (file === null || file === void 0 ? void 0 : file.name)) && /* @__PURE__ */React__default["default"].createElement(CloseCircleFilledIcon, {
      className: "".concat(uploadPrefix, "__single-input-clear"),
      onClick: function onClick(e) {
        return props.onRemove({
          e: e,
          file: file,
          index: 0
        });
      }
    })));
  };
  var displayFiles = props.displayFiles,
    fileListDisplay = props.fileListDisplay;
  var fileListDisplayNode = _util_parseTNode["default"](fileListDisplay, {
    onRemove: props.onRemove,
    toUploadFiles: props.toUploadFiles,
    sizeOverLimitMessage: props.sizeOverLimitMessage,
    locale: props.locale,
    files: displayFiles
  });
  var classes = ["".concat(uploadPrefix, "__single"), "".concat(uploadPrefix, "__single-").concat(theme)];
  return /* @__PURE__ */React__default["default"].createElement("div", {
    className: classNames__default["default"](classes)
  }, theme === "file-input" && renderFilePreviewAsInput(), props.children, theme === "file" && props.placeholder && !displayFiles[0] && /* @__PURE__ */React__default["default"].createElement("small", {
    className: classNames__default["default"]([props.tipsClasses, props.placeholderClass])
  }, props.placeholder), fileListDisplayNode === null ? null : fileListDisplayNode || renderFilePreviewAsText(displayFiles), !props.multiple && ((_displayFiles$ = displayFiles[0]) === null || _displayFiles$ === void 0 ? void 0 : _displayFiles$.status) === "fail" && theme === "file" ? /* @__PURE__ */React__default["default"].createElement("small", {
    className: classNames__default["default"](props.errorClasses)
  }, ((_displayFiles$0$respo = displayFiles[0].response) === null || _displayFiles$0$respo === void 0 ? void 0 : _displayFiles$0$respo.error) || locale.progress.failText) : null);
}
NormalFile.displayName = "NormalFile";

exports["default"] = NormalFile;
