// tdesign-react/cjs/cascader/components/Item.js
/**
 * tdesign v1.17.1
 * (c) 2026 tdesign
 * @license MIT
 */

'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var defineProperty = require('../../_chunks/dep-c446aa11.js');
var slicedToArray = require('../../_chunks/dep-c60a3e4c.js');
var React = require('react');
var classNames = require('classnames');
var tdesignIconsReact = require('tdesign-icons-react');
var checkbox_index = require('../../checkbox/index.js');
var hooks_useCommonClassName = require('../../hooks/useCommonClassName.js');
var hooks_useConfig = require('../../hooks/useConfig.js');
var hooks_useDomRefCallback = require('../../hooks/useDomRefCallback.js');
var hooks_useGlobalIcon = require('../../hooks/useGlobalIcon.js');
var loading_index = require('../../loading/index.js');
var cascader_core_className = require('../core/className.js');
var cascader_core_helper = require('../core/helper.js');
var isFunction = require('../../_chunks/dep-11347ea1.js');
require('../../_chunks/dep-c290284d.js');
require('../../_chunks/dep-98e6819c.js');
require('../../_chunks/dep-7f3fa935.js');
require('../../_chunks/dep-ab5baf4d.js');
require('../../_util/forwardRefWithStatics.js');
require('hoist-non-react-statics');
require('../../common/Check.js');
require('../../_chunks/dep-e41137f7.js');
require('../../_util/helper.js');
require('../../_chunks/dep-b422817e.js');
require('../../_chunks/dep-b199860d.js');
require('../../_chunks/dep-3d05de97.js');
require('../../_chunks/dep-a54f202c.js');
require('../../_chunks/dep-36579bf6.js');
require('../../_chunks/dep-4f1729f0.js');
require('../../_chunks/dep-45188d54.js');
require('../../_chunks/dep-eccb6a94.js');
require('../../_chunks/dep-f5e1d11c.js');
require('../../hooks/useControlled.js');
require('../../_util/noop.js');
require('../../_chunks/dep-fb263e64.js');
require('../../config-provider/ConfigContext.js');
require('../../_chunks/dep-2ee09518.js');
require('../../_chunks/dep-a6fa90dc.js');
require('dayjs');
require('../../_chunks/dep-012d9a7c.js');
require('../../_chunks/dep-150a112c.js');
require('../../_chunks/dep-e360e17f.js');
require('../../_chunks/dep-51284537.js');
require('../../_chunks/dep-8a493834.js');
require('../../_chunks/dep-9f5b2891.js');
require('../../_chunks/dep-338f013a.js');
require('../../_chunks/dep-c37c5e17.js');
require('../../_chunks/dep-eee1aab9.js');
require('../../_chunks/dep-a830ce1e.js');
require('../../_chunks/dep-770bc0ae.js');
require('../../_chunks/dep-aaf27e39.js');
require('../../_chunks/dep-b47d855f.js');
require('../../_chunks/dep-81bce4e6.js');
require('../../_chunks/dep-863f8f4d.js');
require('../../_chunks/dep-84e4d0da.js');
require('../../_chunks/dep-f78183d7.js');
require('../../_chunks/dep-2f45f6c4.js');
require('../../hooks/useDefaultProps.js');
require('../../_chunks/dep-05bc49e6.js');
require('../../_chunks/dep-8c880a65.js');
require('../../checkbox/defaultProps.js');
require('../../_chunks/dep-b8d0b96c.js');
require('../../loading/Loading.js');
require('../../_chunks/dep-ee71ddca.js');
require('../../_chunks/dep-427f8bc0.js');
require('../../common/Portal.js');
require('react-dom');
require('../../hooks/useLayoutEffect.js');
require('../../loading/defaultProps.js');
require('../../loading/gradient.js');
require('../../_chunks/dep-062bb1b1.js');
require('../../_chunks/dep-6e0da644.js');
require('../../loading/plugin.js');
require('../../_util/react-render.js');
require('../../_chunks/dep-7bf14076.js');
require('../../common/PluginContainer.js');
require('../../config-provider/ConfigProvider.js');
require('../../_chunks/dep-ba90997f.js');
require('../../_chunks/dep-22026fea.js');
require('../../_chunks/dep-551d5940.js');
require('../../_chunks/dep-8005b808.js');
require('../../_chunks/dep-2add8824.js');
require('../../_chunks/dep-77f065c8.js');
require('../../_chunks/dep-3c628d18.js');
require('../../_chunks/dep-3784bbea.js');
require('../../_chunks/dep-d4a6a9c1.js');
require('../../_chunks/dep-bc160f5b.js');
require('../../_chunks/dep-363864ae.js');
require('../../_chunks/dep-204188de.js');
require('../../_chunks/dep-7a4e0117.js');
require('../../_chunks/dep-8de6ca9d.js');
require('../../_chunks/dep-6e5e0470.js');
require('../../_chunks/dep-cdef7593.js');
require('../../_chunks/dep-bfa39549.js');
require('../../_chunks/dep-2be6b7d2.js');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

var React__default = /*#__PURE__*/_interopDefaultLegacy(React);
var classNames__default = /*#__PURE__*/_interopDefaultLegacy(classNames);

function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { defineProperty._defineProperty(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
var Item = /*#__PURE__*/React.forwardRef(function (props, ref) {
  var node = props.node,
    optionChild = props.optionChild,
    multiple = props.cascaderContext.multiple,
    _onClick = props.onClick,
    _onChange = props.onChange,
    _onMouseEnter = props.onMouseEnter,
    cascaderContext = props.cascaderContext;
  var _useConfig = hooks_useConfig["default"](),
    prefix = _useConfig.classPrefix;
  var _useGlobalIcon = hooks_useGlobalIcon["default"]({
      ChevronRightIcon: tdesignIconsReact.ChevronRightIcon
    }),
    ChevronRightIcon = _useGlobalIcon.ChevronRightIcon;
  var COMPONENT_NAME = "".concat(prefix, "-cascader__item");
  var _useDomRefCallback = hooks_useDomRefCallback["default"](),
    _useDomRefCallback2 = slicedToArray._slicedToArray(_useDomRefCallback, 2);
    _useDomRefCallback2[0];
    _useDomRefCallback2[1];
  var _useCommonClassName = hooks_useCommonClassName["default"](),
    STATUS = _useCommonClassName.STATUS,
    SIZE = _useCommonClassName.SIZE;
  var itemClass = React.useMemo(function () {
    return classNames__default["default"](cascader_core_className.getCascaderItemClass(prefix, node, SIZE, STATUS, cascaderContext));
  }, [prefix, node, SIZE, STATUS, cascaderContext]);
  var iconClass = React.useMemo(function () {
    return classNames__default["default"](cascader_core_className.getCascaderItemIconClass(prefix, node, STATUS, cascaderContext));
  }, [prefix, node, STATUS, cascaderContext]);
  var RenderLabelInner = function RenderLabelInner(node2, cascaderContext2) {
    var inputVal = cascaderContext2.inputVal;
    if (!inputVal && optionChild) {
      return optionChild;
    }
    var labelText = inputVal ? cascader_core_helper.getFullPathLabel(node2) : node2.label;
    if (inputVal) {
      var texts = labelText.split(inputVal);
      var doms = [];
      for (var index = 0; index < texts.length; index++) {
        doms.push(/* @__PURE__ */React__default["default"].createElement("span", {
          key: index
        }, texts[index]));
        if (index === texts.length - 1) break;
        doms.push(/* @__PURE__ */React__default["default"].createElement("span", {
          key: "".concat(index, "filter"),
          className: "".concat(COMPONENT_NAME, "-label--filter")
        }, inputVal));
      }
      return doms;
    }
    return labelText;
  };
  var RenderLabelContent = function RenderLabelContent(node2, cascaderContext2) {
    var label = RenderLabelInner(node2, cascaderContext2);
    var labelCont = /* @__PURE__ */React__default["default"].createElement("span", {
      title: cascaderContext2.inputVal ? cascader_core_helper.getFullPathLabel(node2) : node2.label,
      className: classNames__default["default"]("".concat(COMPONENT_NAME, "-label"), "".concat(COMPONENT_NAME, "-label--ellipsis")),
      role: "label"
    }, label);
    return labelCont;
  };
  var RenderCheckBox = function RenderCheckBox(node2, cascaderContext2) {
    var checkProps = cascaderContext2.checkProps,
      value = cascaderContext2.value,
      max = cascaderContext2.max,
      inputVal = cascaderContext2.inputVal,
      isParentFilterable = cascaderContext2.isParentFilterable;
    var label = RenderLabelInner(node2, cascaderContext2);
    return /* @__PURE__ */React__default["default"].createElement(checkbox_index.Checkbox, _objectSpread({
      checked: node2.checked,
      indeterminate: node2.indeterminate,
      disabled: node2.isDisabled() || value && value.length >= max && max !== 0,
      name: String(node2.value),
      stopLabelTrigger: !!node2.children && !isParentFilterable,
      title: inputVal ? cascader_core_helper.getFullPathLabel(node2) : node2.label,
      onChange: function onChange() {
        _onChange(node2);
      }
    }, checkProps), label);
  };
  var isFiltering = React.useMemo(function () {
    return Boolean(cascaderContext.filterable) || isFunction.isFunction(cascaderContext.filter);
  }, [cascaderContext.filterable, cascaderContext.filter]);
  return /* @__PURE__ */React__default["default"].createElement("li", {
    ref: ref,
    className: itemClass,
    onClick: function onClick(e) {
      var _e$nativeEvent, _e$nativeEvent$stopIm;
      e.stopPropagation();
      e === null || e === void 0 || (_e$nativeEvent = e.nativeEvent) === null || _e$nativeEvent === void 0 || (_e$nativeEvent$stopIm = _e$nativeEvent.stopImmediatePropagation) === null || _e$nativeEvent$stopIm === void 0 || _e$nativeEvent$stopIm.call(_e$nativeEvent);
      if (multiple && cascaderContext.inputVal && isFiltering) return;
      _onClick(node);
    },
    onMouseEnter: function onMouseEnter(e) {
      e.stopPropagation();
      if (cascaderContext.inputVal && isFiltering) return;
      _onMouseEnter(node);
    }
  }, multiple ? RenderCheckBox(node, cascaderContext) : RenderLabelContent(node, cascaderContext), node.children && !cascaderContext.isParentFilterable && (node.loading ? /* @__PURE__ */React__default["default"].createElement(loading_index.Loading, {
    className: iconClass,
    loading: true,
    size: "small"
  }) : /* @__PURE__ */React__default["default"].createElement(ChevronRightIcon, {
    className: iconClass
  })));
});

exports["default"] = Item;
