// tdesign-react/cjs/cascader/components/Panel.js
/**
 * tdesign v1.17.1
 * (c) 2026 tdesign
 * @license MIT
 */

'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var defineProperty = require('../../_chunks/dep-c446aa11.js');
var slicedToArray = require('../../_chunks/dep-c60a3e4c.js');
var React = require('react');
var classNames = require('classnames');
var _util_parseTNode = require('../../_util/parseTNode.js');
var hooks_useConfig = require('../../hooks/useConfig.js');
var locale_LocalReceiver = require('../../locale/LocalReceiver.js');
var cascader_core_effect = require('../core/effect.js');
var cascader_core_helper = require('../core/helper.js');
var cascader_components_Item = require('./Item.js');
require('../../_chunks/dep-c290284d.js');
require('../../_chunks/dep-98e6819c.js');
require('../../_chunks/dep-7f3fa935.js');
require('../../_chunks/dep-3c628d18.js');
require('../../_chunks/dep-11347ea1.js');
require('../../_chunks/dep-3d05de97.js');
require('../../_chunks/dep-a830ce1e.js');
require('../../config-provider/ConfigContext.js');
require('../../_chunks/dep-2ee09518.js');
require('../../_chunks/dep-a6fa90dc.js');
require('dayjs');
require('../../_chunks/dep-012d9a7c.js');
require('../../_chunks/dep-150a112c.js');
require('../../_chunks/dep-e360e17f.js');
require('../../_chunks/dep-4f1729f0.js');
require('../../_chunks/dep-a54f202c.js');
require('../../_chunks/dep-51284537.js');
require('../../_chunks/dep-8a493834.js');
require('../../_chunks/dep-9f5b2891.js');
require('../../_chunks/dep-338f013a.js');
require('../../_chunks/dep-c37c5e17.js');
require('../../_chunks/dep-eee1aab9.js');
require('../../_chunks/dep-770bc0ae.js');
require('../../_chunks/dep-aaf27e39.js');
require('../../_chunks/dep-b47d855f.js');
require('../../_chunks/dep-81bce4e6.js');
require('../../_chunks/dep-863f8f4d.js');
require('../../_chunks/dep-84e4d0da.js');
require('../../_chunks/dep-f78183d7.js');
require('../../_chunks/dep-2f45f6c4.js');
require('../../_chunks/dep-427f8bc0.js');
require('../../config-provider/ConfigProvider.js');
require('../../_chunks/dep-ba90997f.js');
require('../../_chunks/dep-22026fea.js');
require('../../_chunks/dep-551d5940.js');
require('../../_chunks/dep-8005b808.js');
require('../../_chunks/dep-2add8824.js');
require('../../_chunks/dep-77f065c8.js');
require('../../_chunks/dep-3784bbea.js');
require('../../_chunks/dep-d4a6a9c1.js');
require('../../_chunks/dep-36579bf6.js');
require('../../_chunks/dep-b199860d.js');
require('../../_chunks/dep-bc160f5b.js');
require('../../_chunks/dep-363864ae.js');
require('../../_chunks/dep-204188de.js');
require('../../_chunks/dep-7a4e0117.js');
require('../../_chunks/dep-fb263e64.js');
require('../../_chunks/dep-8de6ca9d.js');
require('../../_chunks/dep-6e5e0470.js');
require('../../_chunks/dep-cdef7593.js');
require('../../_chunks/dep-f5e1d11c.js');
require('../../_chunks/dep-bfa39549.js');
require('../../_chunks/dep-b8d0b96c.js');
require('../../_chunks/dep-2be6b7d2.js');
require('tdesign-icons-react');
require('../../checkbox/index.js');
require('../../_chunks/dep-ab5baf4d.js');
require('../../_util/forwardRefWithStatics.js');
require('hoist-non-react-statics');
require('../../common/Check.js');
require('../../_chunks/dep-e41137f7.js');
require('../../_util/helper.js');
require('../../_chunks/dep-b422817e.js');
require('../../_chunks/dep-45188d54.js');
require('../../_chunks/dep-eccb6a94.js');
require('../../hooks/useControlled.js');
require('../../_util/noop.js');
require('../../hooks/useDefaultProps.js');
require('../../_chunks/dep-05bc49e6.js');
require('../../_chunks/dep-8c880a65.js');
require('../../checkbox/defaultProps.js');
require('../../hooks/useCommonClassName.js');
require('../../hooks/useDomRefCallback.js');
require('../../hooks/useGlobalIcon.js');
require('../../loading/index.js');
require('../../loading/Loading.js');
require('../../_chunks/dep-ee71ddca.js');
require('../../common/Portal.js');
require('react-dom');
require('../../hooks/useLayoutEffect.js');
require('../../loading/defaultProps.js');
require('../../loading/gradient.js');
require('../../_chunks/dep-062bb1b1.js');
require('../../_chunks/dep-6e0da644.js');
require('../../loading/plugin.js');
require('../../_util/react-render.js');
require('../../_chunks/dep-7bf14076.js');
require('../../common/PluginContainer.js');
require('../core/className.js');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

var React__default = /*#__PURE__*/_interopDefaultLegacy(React);
var classNames__default = /*#__PURE__*/_interopDefaultLegacy(classNames);

var Panel = function Panel(props) {
  var cascaderContext = props.cascaderContext,
    option = props.option;
  var panels = React.useMemo(function () {
    return cascader_core_helper.getPanels(cascaderContext.treeNodes);
  }, [cascaderContext.treeNodes]);
  var handleExpand = function handleExpand(node, trigger) {
    var propsTrigger = props.trigger,
      cascaderContext2 = props.cascaderContext;
    cascader_core_effect.expendClickEffect(propsTrigger, trigger, node, cascaderContext2);
  };
  var _useConfig = hooks_useConfig["default"](),
    classPrefix = _useConfig.classPrefix;
  var _useLocaleReceiver = locale_LocalReceiver.useLocaleReceiver("cascader"),
    _useLocaleReceiver2 = slicedToArray._slicedToArray(_useLocaleReceiver, 1),
    global = _useLocaleReceiver2[0];
  var COMPONENT_NAME = "".concat(classPrefix, "-cascader");
  var renderItem = function renderItem(node, index) {
    return /* @__PURE__ */React__default["default"].createElement(cascader_components_Item["default"], {
      key: index,
      node: node,
      optionChild: node.data.content || _util_parseTNode["default"](option, {
        item: node.data,
        index: index,
        context: {
          node: node
        }
      }),
      cascaderContext: cascaderContext,
      onClick: function onClick() {
        handleExpand(node, "click");
      },
      onMouseEnter: function onMouseEnter() {
        handleExpand(node, "hover");
      },
      onChange: function onChange() {
        cascader_core_effect.valueChangeEffect(node, cascaderContext);
      }
    });
  };
  var renderList = function renderList(treeNodes) {
    var isFilter = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : false;
    var segment = arguments.length > 2 && arguments[2] !== undefined ? arguments[2] : true;
    var key = arguments.length > 3 && arguments[3] !== undefined ? arguments[3] : "1";
    return /* @__PURE__ */React__default["default"].createElement("ul", {
      className: classNames__default["default"]("".concat(COMPONENT_NAME, "__menu"), "narrow-scrollbar", defineProperty._defineProperty(defineProperty._defineProperty({}, "".concat(COMPONENT_NAME, "__menu--segment"), segment), "".concat(COMPONENT_NAME, "__menu--filter"), isFilter)),
      key: key
    }, treeNodes.map(function (node, index) {
      return renderItem(node, index);
    }));
  };
  var renderPanels = function renderPanels() {
    var _props$cascaderContex = props.cascaderContext,
      inputVal = _props$cascaderContex.inputVal,
      treeNodes = _props$cascaderContex.treeNodes;
    return inputVal ? renderList(treeNodes, true) : panels.map(function (treeNodes2, index) {
      return renderList(treeNodes2, false, index !== panels.length - 1, "".concat(COMPONENT_NAME, "__menu").concat(index));
    });
  };
  var content;
  if (props.loading) {
    var _props$loadingText;
    content = /* @__PURE__ */React__default["default"].createElement("div", {
      className: "".concat(COMPONENT_NAME, "__panel--empty")
    }, (_props$loadingText = props.loadingText) !== null && _props$loadingText !== void 0 ? _props$loadingText : global.loadingText);
  } else {
    var _props$empty;
    content = panels !== null && panels !== void 0 && panels.length ? renderPanels() : /* @__PURE__ */React__default["default"].createElement("div", {
      className: "".concat(COMPONENT_NAME, "__panel--empty")
    }, (_props$empty = props.empty) !== null && _props$empty !== void 0 ? _props$empty : global.empty);
  }
  return /* @__PURE__ */React__default["default"].createElement("div", {
    className: classNames__default["default"]("".concat(COMPONENT_NAME, "__panel"), defineProperty._defineProperty({}, "".concat(COMPONENT_NAME, "--normal"), panels.length && !props.loading), props.className),
    style: props.style
  }, content);
};

exports["default"] = Panel;
