// tdesign-react/lib/date-picker/panel/ExtraContent.js
/**
 * tdesign v1.17.1
 * (c) 2026 tdesign
 * @license MIT
 */

import React from 'react';
import DatePickerFooter from '../base/Footer.js';
import '../../_chunks/dep-55b77e26.js';
import '../../_chunks/dep-067cb22b.js';
import '../../_chunks/dep-d1eeb95c.js';
import '../../_chunks/dep-673e8854.js';
import '../../_chunks/dep-570a5adb.js';
import 'classnames';
import 'lodash-es';
import '../../button/index.js';
import '../../button/Button.js';
import '../../_chunks/dep-84926167.js';
import '../../_util/parseTNode.js';
import '../../_chunks/dep-65b369a5.js';
import '../../hooks/useConfig.js';
import '../../config-provider/ConfigContext.js';
import '../../_chunks/dep-c3e5e988.js';
import '../../_chunks/dep-65b05485.js';
import 'dayjs';
import '../../hooks/useDefaultProps.js';
import '../../hooks/useDomRefCallback.js';
import '../../hooks/useRipple.js';
import '../../_chunks/dep-e17ec8df.js';
import '../../_chunks/dep-68899e89.js';
import '../../hooks/useAnimation.js';
import '../../loading/index.js';
import '../../loading/Loading.js';
import '../../common/Portal.js';
import 'react-dom';
import '../../hooks/useLayoutEffect.js';
import '../../loading/defaultProps.js';
import '../../loading/gradient.js';
import '../../_chunks/dep-b0cbea79.js';
import '../../_chunks/dep-154fcb93.js';
import '../../_chunks/dep-e9c0bb63.js';
import '../../loading/plugin.js';
import '../../_util/react-render.js';
import '../../_chunks/dep-e29eee31.js';
import '../../common/PluginContainer.js';
import '../../config-provider/ConfigProvider.js';
import '../../button/defaultProps.js';
import '../../locale/LocalReceiver.js';

function ExtraContent(props) {
  var presets = props.presets,
    enableTimePicker = props.enableTimePicker,
    presetsPlacement = props.presetsPlacement,
    onPresetClick = props.onPresetClick,
    onConfirmClick = props.onConfirmClick,
    selectedValue = props.selectedValue,
    needConfirm = props.needConfirm,
    isSwitchTimeMode = props.isSwitchTimeMode;
  var showPanelFooter = enableTimePicker && needConfirm || presets;
  return showPanelFooter ? /* @__PURE__ */React.createElement(DatePickerFooter, {
    presets: presets,
    onPresetClick: onPresetClick,
    enableTimePicker: enableTimePicker,
    onConfirmClick: onConfirmClick,
    presetsPlacement: presetsPlacement,
    selectedValue: selectedValue,
    needConfirm: needConfirm,
    onTimePanelChange: props.onTimeModeChange,
    isDateRangeContent: props.isDateRangeContent,
    isSwitchTimeMode: isSwitchTimeMode
  }) : null;
}

export { ExtraContent as default };
