// tdesign-react/lib/cascader/components/Panel.js
/**
 * tdesign v1.17.1
 * (c) 2026 tdesign
 * @license MIT
 */

import { _ as _defineProperty } from '../../_chunks/dep-55b77e26.js';
import { _ as _slicedToArray } from '../../_chunks/dep-d1eeb95c.js';
import React, { useMemo } from 'react';
import classNames from 'classnames';
import parseTNode from '../../_util/parseTNode.js';
import useConfig from '../../hooks/useConfig.js';
import { useLocaleReceiver } from '../../locale/LocalReceiver.js';
import { valueChangeEffect, expendClickEffect } from '../core/effect.js';
import { getPanels } from '../core/helper.js';
import Item from './Item.js';
import '../../_chunks/dep-067cb22b.js';
import '../../_chunks/dep-673e8854.js';
import '../../_chunks/dep-570a5adb.js';
import 'lodash-es';
import '../../_chunks/dep-65b369a5.js';
import '../../config-provider/ConfigContext.js';
import '../../_chunks/dep-c3e5e988.js';
import '../../_chunks/dep-65b05485.js';
import 'dayjs';
import '../../config-provider/ConfigProvider.js';
import '../../_chunks/dep-5ed1b2fb.js';
import 'tdesign-icons-react';
import '../../checkbox/index.js';
import '../../_chunks/dep-954df161.js';
import '../../_util/forwardRefWithStatics.js';
import 'hoist-non-react-statics';
import '../../common/Check.js';
import '../../_chunks/dep-84926167.js';
import '../../_util/helper.js';
import '../../hooks/useControlled.js';
import '../../_util/noop.js';
import '../../hooks/useDefaultProps.js';
import '../../_chunks/dep-154fcb93.js';
import '../../_chunks/dep-e9c0bb63.js';
import '../../checkbox/defaultProps.js';
import '../../hooks/useCommonClassName.js';
import '../../hooks/useDomRefCallback.js';
import '../../hooks/useGlobalIcon.js';
import '../../loading/index.js';
import '../../loading/Loading.js';
import '../../_chunks/dep-68899e89.js';
import '../../common/Portal.js';
import 'react-dom';
import '../../hooks/useLayoutEffect.js';
import '../../loading/defaultProps.js';
import '../../loading/gradient.js';
import '../../_chunks/dep-e17ec8df.js';
import '../../_chunks/dep-b0cbea79.js';
import '../../loading/plugin.js';
import '../../_util/react-render.js';
import '../../_chunks/dep-e29eee31.js';
import '../../common/PluginContainer.js';
import '../core/className.js';

var Panel = function Panel(props) {
  var cascaderContext = props.cascaderContext,
    option = props.option;
  var panels = useMemo(function () {
    return getPanels(cascaderContext.treeNodes);
  }, [cascaderContext.treeNodes]);
  var handleExpand = function handleExpand(node, trigger) {
    var propsTrigger = props.trigger,
      cascaderContext2 = props.cascaderContext;
    expendClickEffect(propsTrigger, trigger, node, cascaderContext2);
  };
  var _useConfig = useConfig(),
    classPrefix = _useConfig.classPrefix;
  var _useLocaleReceiver = useLocaleReceiver("cascader"),
    _useLocaleReceiver2 = _slicedToArray(_useLocaleReceiver, 1),
    global = _useLocaleReceiver2[0];
  var COMPONENT_NAME = "".concat(classPrefix, "-cascader");
  var renderItem = function renderItem(node, index) {
    return /* @__PURE__ */React.createElement(Item, {
      key: index,
      node: node,
      optionChild: node.data.content || parseTNode(option, {
        item: node.data,
        index: index,
        context: {
          node: node
        }
      }),
      cascaderContext: cascaderContext,
      onClick: function onClick() {
        handleExpand(node, "click");
      },
      onMouseEnter: function onMouseEnter() {
        handleExpand(node, "hover");
      },
      onChange: function onChange() {
        valueChangeEffect(node, cascaderContext);
      }
    });
  };
  var renderList = function renderList(treeNodes) {
    var isFilter = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : false;
    var segment = arguments.length > 2 && arguments[2] !== undefined ? arguments[2] : true;
    var key = arguments.length > 3 && arguments[3] !== undefined ? arguments[3] : "1";
    return /* @__PURE__ */React.createElement("ul", {
      className: classNames("".concat(COMPONENT_NAME, "__menu"), "narrow-scrollbar", _defineProperty(_defineProperty({}, "".concat(COMPONENT_NAME, "__menu--segment"), segment), "".concat(COMPONENT_NAME, "__menu--filter"), isFilter)),
      key: key
    }, treeNodes.map(function (node, index) {
      return renderItem(node, index);
    }));
  };
  var renderPanels = function renderPanels() {
    var _props$cascaderContex = props.cascaderContext,
      inputVal = _props$cascaderContex.inputVal,
      treeNodes = _props$cascaderContex.treeNodes;
    return inputVal ? renderList(treeNodes, true) : panels.map(function (treeNodes2, index) {
      return renderList(treeNodes2, false, index !== panels.length - 1, "".concat(COMPONENT_NAME, "__menu").concat(index));
    });
  };
  var content;
  if (props.loading) {
    var _props$loadingText;
    content = /* @__PURE__ */React.createElement("div", {
      className: "".concat(COMPONENT_NAME, "__panel--empty")
    }, (_props$loadingText = props.loadingText) !== null && _props$loadingText !== void 0 ? _props$loadingText : global.loadingText);
  } else {
    var _props$empty;
    content = panels !== null && panels !== void 0 && panels.length ? renderPanels() : /* @__PURE__ */React.createElement("div", {
      className: "".concat(COMPONENT_NAME, "__panel--empty")
    }, (_props$empty = props.empty) !== null && _props$empty !== void 0 ? _props$empty : global.empty);
  }
  return /* @__PURE__ */React.createElement("div", {
    className: classNames("".concat(COMPONENT_NAME, "__panel"), _defineProperty({}, "".concat(COMPONENT_NAME, "--normal"), panels.length && !props.loading), props.className),
    style: props.style
  }, content);
};

export { Panel as default };
