// tdesign-react/lib/image-viewer/hooks/useScale.js
/**
 * tdesign v1.17.1
 * (c) 2026 tdesign
 * @license MIT
 */

import { _ as _slicedToArray } from '../../_chunks/dep-d1eeb95c.js';
import { _ as _defineProperty } from '../../_chunks/dep-55b77e26.js';
import { useCallback, useRef, useState, useEffect } from 'react';
import { D as DEFAULT_IMAGE_SCALE, a as clampScale, z as zoomIn, b as zoomOut } from '../../_chunks/dep-ce103a58.js';
import '../../_chunks/dep-673e8854.js';
import '../../_chunks/dep-570a5adb.js';
import '../../_chunks/dep-067cb22b.js';
import '../../_chunks/dep-f26ab88c.js';
import 'lodash-es';
import '../../_chunks/dep-65b369a5.js';

function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { _defineProperty(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
var useScale = function useScale(imageScale, visible, onWheel) {
  var _DEFAULT_IMAGE_SCALE$ = _objectSpread(_objectSpread({}, DEFAULT_IMAGE_SCALE), imageScale),
    max = _DEFAULT_IMAGE_SCALE$.max,
    min = _DEFAULT_IMAGE_SCALE$.min,
    step = _DEFAULT_IMAGE_SCALE$.step,
    defaultScale = _DEFAULT_IMAGE_SCALE$.defaultScale;
  var calcDefaultScale = useCallback(function () {
    return clampScale(defaultScale, min, max);
  }, [defaultScale, max, min]);
  var distance = useRef(0);
  var _useState = useState(calcDefaultScale()),
    _useState2 = _slicedToArray(_useState, 2),
    scale = _useState2[0],
    setScale = _useState2[1];
  var scaleRef = useRef(scale);
  var paramsRef = useRef({
    step: step,
    min: min,
    max: max
  });
  paramsRef.current = {
    step: step,
    min: min,
    max: max
  };
  var onZoomIn = useCallback(function (zoomOptions) {
    var _paramsRef$current = paramsRef.current,
      s = _paramsRef$current.step,
      mi = _paramsRef$current.min,
      ma = _paramsRef$current.max;
    var _zoomIn = zoomIn(scaleRef.current, s, mi, ma, zoomOptions),
      newScale = _zoomIn.newScale,
      zoomResult = _zoomIn.zoomResult;
    if (newScale === scaleRef.current) return {};
    scaleRef.current = newScale;
    setScale(newScale);
    return zoomResult;
  }, []);
  var onZoomOut = useCallback(function (zoomOptions) {
    var _paramsRef$current2 = paramsRef.current,
      s = _paramsRef$current2.step,
      mi = _paramsRef$current2.min,
      ma = _paramsRef$current2.max;
    var _zoomOut = zoomOut(scaleRef.current, s, mi, ma, zoomOptions),
      newScale = _zoomOut.newScale,
      zoomResult = _zoomOut.zoomResult;
    if (newScale === scaleRef.current) return {};
    scaleRef.current = newScale;
    setScale(newScale);
    return zoomResult;
  }, []);
  var onResetScale = useCallback(function () {
    var defaultVal = calcDefaultScale();
    scaleRef.current = defaultVal;
    setScale(defaultVal);
  }, [calcDefaultScale]);
  var onTouchStart = useCallback(function (e) {
    if (e.touches.length !== 2) return;
    e.preventDefault();
    var _Array$from = Array.from(e.touches),
      _Array$from2 = _slicedToArray(_Array$from, 2),
      touch1 = _Array$from2[0],
      touch2 = _Array$from2[1];
    distance.current = Math.hypot(touch2.pageX - touch1.pageX, touch2.pageY - touch1.pageY);
  }, []);
  var onTouchMove = useCallback(function (e) {
    if (e.touches.length !== 2) return;
    e.preventDefault();
    var _Array$from3 = Array.from(e.touches),
      _Array$from4 = _slicedToArray(_Array$from3, 2),
      touch1 = _Array$from4[0],
      touch2 = _Array$from4[1];
    var currentDistance = Math.hypot(touch2.pageX - touch1.pageX, touch2.pageY - touch1.pageY);
    if (currentDistance > distance.current) {
      onZoomIn();
    } else {
      onZoomOut();
    }
    distance.current = currentDistance;
  }, [onZoomIn, onZoomOut]);
  var onTouchEnd = useCallback(function () {
    distance.current = 0;
  }, []);
  var onWheelRef = useRef(onWheel);
  onWheelRef.current = onWheel;
  var stableOnWheel = useCallback(function (e) {
    var _onWheelRef$current;
    (_onWheelRef$current = onWheelRef.current) === null || _onWheelRef$current === void 0 || _onWheelRef$current.call(onWheelRef, e);
  }, []);
  useEffect(function () {
    if (!visible) return;
    document.addEventListener("wheel", stableOnWheel, {
      passive: false
    });
    document.addEventListener("touchstart", onTouchStart, {
      passive: false
    });
    document.addEventListener("touchmove", onTouchMove, {
      passive: false
    });
    document.addEventListener("touchend", onTouchEnd);
    return function () {
      document.removeEventListener("wheel", stableOnWheel);
      document.removeEventListener("touchstart", onTouchStart);
      document.removeEventListener("touchmove", onTouchMove);
      document.removeEventListener("touchend", onTouchEnd);
    };
  }, [visible, stableOnWheel, onTouchStart, onTouchMove, onTouchEnd]);
  return {
    scale: scale,
    onZoomIn: onZoomIn,
    onZoomOut: onZoomOut,
    onResetScale: onResetScale,
    onTouchStart: onTouchStart,
    onTouchMove: onTouchMove,
    onTouchEnd: onTouchEnd
  };
};

export { useScale as default };
