// tdesign-react/cjs/table/hooks/useDragSort.js
/**
 * tdesign v1.17.1
 * (c) 2026 tdesign
 * @license MIT
 */

'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var toConsumableArray = require('../../_chunks/dep-05bc49e6.js');
var defineProperty = require('../../_chunks/dep-c446aa11.js');
var slicedToArray = require('../../_chunks/dep-c60a3e4c.js');
var React = require('react');
var Sortable = require('sortablejs');
var log = require('../../_chunks/dep-3c628d18.js');
var utils = require('../../_chunks/dep-640fca41.js');
var _util_style = require('../../_chunks/dep-ee71ddca.js');
var hooks_useLatest = require('../../hooks/useLatest.js');
var table_hooks_useClassName = require('./useClassName.js');
var get = require('../../_chunks/dep-f8c20fd8.js');
require('../../_chunks/dep-7f3fa935.js');
require('../../_chunks/dep-8c880a65.js');
require('../../_chunks/dep-c290284d.js');
require('../../_chunks/dep-98e6819c.js');
require('../../_chunks/dep-11347ea1.js');
require('../../_chunks/dep-3d05de97.js');
require('../../_chunks/dep-a830ce1e.js');
require('../../_chunks/dep-427f8bc0.js');
require('../../_chunks/dep-a54f202c.js');
require('../../_chunks/dep-4f1729f0.js');
require('../../hooks/useConfig.js');
require('../../config-provider/ConfigContext.js');
require('../../_chunks/dep-2ee09518.js');
require('../../_chunks/dep-a6fa90dc.js');
require('dayjs');
require('../../_chunks/dep-012d9a7c.js');
require('../../_chunks/dep-150a112c.js');
require('../../_chunks/dep-e360e17f.js');
require('../../_chunks/dep-51284537.js');
require('../../_chunks/dep-8a493834.js');
require('../../_chunks/dep-9f5b2891.js');
require('../../_chunks/dep-338f013a.js');
require('../../_chunks/dep-c37c5e17.js');
require('../../_chunks/dep-eee1aab9.js');
require('../../_chunks/dep-770bc0ae.js');
require('../../_chunks/dep-aaf27e39.js');
require('../../_chunks/dep-b47d855f.js');
require('../../_chunks/dep-81bce4e6.js');
require('../../_chunks/dep-863f8f4d.js');
require('../../_chunks/dep-84e4d0da.js');
require('../../_chunks/dep-f78183d7.js');
require('../../_chunks/dep-2f45f6c4.js');
require('../../_chunks/dep-d4a6a9c1.js');
require('../../_chunks/dep-36579bf6.js');
require('../../_chunks/dep-b199860d.js');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

var Sortable__default = /*#__PURE__*/_interopDefaultLegacy(Sortable);

function swapDragArrayElement(data, currentIndex, targetIndex) {
  var newData = toConsumableArray._toConsumableArray(data);
  if (targetIndex - currentIndex > 0) {
    newData.splice(targetIndex + 1, 0, newData[currentIndex]);
    newData.splice(currentIndex, 1);
  } else {
    newData.splice(targetIndex, 0, newData[currentIndex]);
    newData.splice(currentIndex + 1, 1);
  }
  return newData;
}

function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { defineProperty._defineProperty(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
var EXPANDED_SUFFIX = "__expanded";
var DATA_ID_ATTR = "data-id";
var DATA_PARENT_ID_ATTR = "data-parent-id";
function useDragSort(props, options) {
  var sortOnRowDraggable = props.sortOnRowDraggable,
    dragSort = props.dragSort,
    data = props.data,
    onDragSort = props.onDragSort;
  var primaryTableRef = options.primaryTableRef,
    innerPagination = options.innerPagination;
  var _useClassName = table_hooks_useClassName["default"](),
    tableDraggableClasses = _useClassName.tableDraggableClasses,
    tableExpandClasses = _useClassName.tableExpandClasses,
    tableBaseClass = _useClassName.tableBaseClass,
    tableFullRowClasses = _useClassName.tableFullRowClasses;
  var _useState = React.useState(props.columns || []),
    _useState2 = slicedToArray._slicedToArray(_useState, 2),
    columns = _useState2[0],
    setDragSortColumns = _useState2[1];
  var dragCol = React.useMemo(function () {
    return columns.find(function (item) {
      return item.colKey === "drag";
    });
  }, [columns]);
  var isRowDraggable = React.useMemo(function () {
    return sortOnRowDraggable || dragSort === "row";
  }, [dragSort, sortOnRowDraggable]);
  var isRowHandlerDraggable = React.useMemo(function () {
    return ["row-handler", "row-handler-col"].includes(dragSort) && !!dragCol;
  }, [dragSort, dragCol]);
  var isColDraggable = React.useMemo(function () {
    return ["col", "row-handler-col"].includes(dragSort);
  }, [dragSort]);
  var tData = React.useRef(null);
  var dragRowInstance = React.useRef(null);
  var dragColInstance = React.useRef(null);
  var trIdList = React.useRef([]);
  var lastColIdList = React.useRef([]);
  var dragColumns = React.useRef([]);
  var originalColumns = React.useRef([]);
  var scrollRef = React.useRef(null);
  var onDragSortRef = hooks_useLatest["default"](onDragSort);
  if (props.sortOnRowDraggable) {
    log.log.warn("Table", "`sortOnRowDraggable` is going to be deprecated, use dragSort='row' instead.");
  }
  var updateLastRowList = function updateLastRowList() {
    var _dragRowInstance$curr;
    trIdList.current = ((_dragRowInstance$curr = dragRowInstance.current) === null || _dragRowInstance$curr === void 0 ? void 0 : _dragRowInstance$curr.toArray()) || [];
  };
  var lockScrollContainer = function lockScrollContainer() {
    var _tData$current, _primaryTableRef$curr;
    var isVirtual = (_tData$current = tData.current) === null || _tData$current === void 0 ? void 0 : _tData$current.some(function (d) {
      return d.__VIRTUAL_SCROLL_INDEX !== void 0;
    });
    if (!isVirtual) return;
    var el = (_primaryTableRef$curr = primaryTableRef.current) === null || _primaryTableRef$curr === void 0 ? void 0 : _primaryTableRef$curr.tableContentElement;
    if (!el || scrollRef.current) return;
    scrollRef.current = {
      el: el,
      prevOverflowY: el.style.overflowY
    };
    el.style.overflowY = "hidden";
  };
  var unlockScrollContainer = function unlockScrollContainer() {
    if (!scrollRef.current) return;
    var _scrollRef$current = scrollRef.current,
      el = _scrollRef$current.el,
      prevOverflowY = _scrollRef$current.prevOverflowY;
    el.style.overflowY = prevOverflowY;
    scrollRef.current = null;
  };
  var getDataPageIndex = function getDataPageIndex(index, pagination) {
    var _pagination$current, _pagination$pageSize;
    var current = (_pagination$current = pagination.current) !== null && _pagination$current !== void 0 ? _pagination$current : pagination.defaultCurrent;
    var pageSize = (_pagination$pageSize = pagination.pageSize) !== null && _pagination$pageSize !== void 0 ? _pagination$pageSize : pagination.defaultPageSize;
    if (!props.disableDataPage && pagination && data.length > pageSize) {
      return pageSize * (current - 1) + index;
    }
    return index;
  };
  var cloneNodeWithStyles = function cloneNodeWithStyles(sourceEl) {
    var clone = sourceEl.cloneNode(true);
    var sourceEls = sourceEl.querySelectorAll("*");
    var cloneEls = clone.querySelectorAll("*");
    var cloneStyles = function cloneStyles(src, dest) {
      if (!window) return dest;
      var computed = window.getComputedStyle(src);
      var cssText = Array.from(computed).map(function (name) {
        return "".concat(name, ":").concat(computed.getPropertyValue(name), ";");
      }).join("");
      dest.style.cssText = cssText;
    };
    cloneStyles(sourceEl, clone);
    sourceEls.forEach(function (src, i) {
      return cloneStyles(src, cloneEls[i]);
    });
    return clone;
  };
  var _getDescendantRows = function getDescendantRows(parentId) {
    var _primaryTableRef$curr2;
    var container = (_primaryTableRef$curr2 = primaryTableRef.current) === null || _primaryTableRef$curr2 === void 0 ? void 0 : _primaryTableRef$curr2.tableContentElement;
    var children = Array.from(container.querySelectorAll("tr[".concat(DATA_PARENT_ID_ATTR, "=\"").concat(parentId, "\"]")) || []);
    var allDescendants = [].concat(children);
    children.forEach(function (child) {
      var childId = child.getAttribute(DATA_ID_ATTR);
      if (childId) {
        allDescendants = allDescendants.concat(_getDescendantRows(childId));
      }
    });
    return allDescendants;
  };
  var getValidSiblingId = function getValidSiblingId(el, dir) {
    var node = el;
    while (node) {
      if (node.nodeType !== 1) {
        node = dir === "prev" ? node.previousElementSibling : node.nextElementSibling;
        continue;
      }
      var id = node.getAttribute(DATA_ID_ATTR);
      var isFullRow = _util_style.hasClass(node, tableFullRowClasses.base);
      var isExpandedRow = id && id.endsWith(EXPANDED_SUFFIX);
      if (id && !isFullRow && !isExpandedRow) return id;
      node = dir === "prev" ? node.previousElementSibling : node.nextElementSibling;
    }
    return null;
  };
  var registerRowDragEvent = function registerRowDragEvent(element) {
    var _element$children;
    if ((element === null || element === void 0 || (_element$children = element.children) === null || _element$children === void 0 ? void 0 : _element$children.length) === 0 || !isRowHandlerDraggable && !isRowDraggable) return;
    var dragContainer = element === null || element === void 0 ? void 0 : element.querySelector("tbody");
    if (!dragContainer) return null;
    var baseOptions = _objectSpread({
      animation: 150,
      dataIdAttr: DATA_ID_ATTR,
      ghostClass: tableDraggableClasses.ghost,
      chosenClass: tableDraggableClasses.chosen,
      dragClass: tableDraggableClasses.dragging,
      filter: ".".concat(tableFullRowClasses.base),
      setData: function setData(dataTransfer, dragEl) {
        var dragRowId = dragEl.getAttribute(DATA_ID_ATTR);
        var childRows = _getDescendantRows(dragRowId);
        if (!childRows || childRows.length === 0) return;
        var ghostNode = cloneNodeWithStyles(dragEl);
        var table = document.createElement("table");
        table.style.borderCollapse = "collapse";
        table.style.borderSpacing = "0";
        var tbody = document.createElement("tbody");
        tbody.appendChild(ghostNode);
        childRows.forEach(function (row) {
          tbody.appendChild(cloneNodeWithStyles(row));
        });
        table.appendChild(tbody);
        document.body.appendChild(table);
        dataTransfer.setDragImage(table, 10, 10);
        requestAnimationFrame(function () {
          if (document.body.contains(table)) {
            document.body.removeChild(table);
          }
        });
      },
      onStart: function onStart(evt) {
        lockScrollContainer();
        updateLastRowList();
        var dragRowId = evt.item.getAttribute(DATA_ID_ATTR);
        evt.to.style.overflow = "hidden";
        var childRows = _getDescendantRows(dragRowId);
        childRows.forEach(function (row) {
          row.style.display = "none";
        });
      },
      onMove: function onMove(evt) {
        var isFullRow = _util_style.hasClass(evt.related, tableFullRowClasses.base);
        if (isFullRow) return false;
        var related = evt.related,
          dragged = evt.dragged,
          willInsertAfter = evt.willInsertAfter;
        var isTargetExpandedParent = _util_style.hasClass(related, tableExpandClasses.expanded);
        var isTargetExpandedChild = _util_style.hasClass(related, tableExpandClasses.row);
        if (isTargetExpandedParent && willInsertAfter) return false;
        if (isTargetExpandedChild && !willInsertAfter) return false;
        if (!props.tree) return;
        var getDragLevel = function getDragLevel(el) {
          var levelClass = Array.from(el.classList).find(function (cls) {
            return cls.includes("-table-tr--level-");
          });
          return parseInt(levelClass === null || levelClass === void 0 ? void 0 : levelClass.split("level-")[1], 10);
        };
        var dragLevel = getDragLevel(dragged);
        var targetLevel = getDragLevel(related);
        return dragLevel === targetLevel;
      },
      onEnd: function onEnd(evt) {
        try {
          var dragId = evt.item.getAttribute(DATA_ID_ATTR);
          var prevId = getValidSiblingId(evt.item.previousElementSibling, "prev");
          var nextId = getValidSiblingId(evt.item.nextElementSibling, "next");
          var childRows = _getDescendantRows(dragId);
          childRows.forEach(function (row) {
            row.style.display = "";
          });
          var dataIdList = tData.current.map(function (item) {
            return String(get.get(item, props.rowKey));
          });
          var currentIndex = dataIdList.indexOf(dragId);
          if (currentIndex === -1) return;
          var targetIndex = -1;
          if (prevId) {
            var anchor = dataIdList.indexOf(prevId);
            if (anchor === -1) return;
            targetIndex = anchor < currentIndex ? anchor + 1 : anchor;
          } else if (nextId) {
            var _anchor = dataIdList.indexOf(nextId);
            if (_anchor === -1) return;
            targetIndex = _anchor > currentIndex ? _anchor - 1 : _anchor;
          } else {
            return;
          }
          if (targetIndex === currentIndex) return;
          var swapCurrent = currentIndex;
          var swapTarget = targetIndex;
          if (innerPagination.current) {
            swapCurrent = getDataPageIndex(swapCurrent, innerPagination.current);
            swapTarget = getDataPageIndex(swapTarget, innerPagination.current);
          }
          var newData = swapDragArrayElement(toConsumableArray._toConsumableArray(tData.current), swapCurrent, swapTarget);
          var params = {
            currentIndex: swapCurrent,
            current: tData.current[swapCurrent],
            targetIndex: swapTarget,
            target: tData.current[swapTarget],
            data: tData.current,
            newData: newData,
            e: evt,
            sort: "row"
          };
          params.currentData = params.newData;
          if (onDragSortRef.current) {
            onDragSortRef.current(params);
          } else {
            var _dragRowInstance$curr2;
            (_dragRowInstance$curr2 = dragRowInstance.current) === null || _dragRowInstance$curr2 === void 0 || _dragRowInstance$curr2.sort(trIdList.current);
          }
        } finally {
          unlockScrollContainer();
        }
      }
    }, props.dragSortOptions);
    if (!dragContainer) return;
    try {
      if (isRowDraggable) {
        dragRowInstance.current = new Sortable__default["default"](dragContainer, _objectSpread({}, baseOptions));
      } else if (isRowHandlerDraggable) {
        dragRowInstance.current = new Sortable__default["default"](dragContainer, _objectSpread(_objectSpread({}, baseOptions), {}, {
          handle: ".".concat(tableDraggableClasses.handle)
        }));
      }
    } catch (error) {}
    updateLastRowList();
  };
  var registerOneLevelColDragEvent = function registerOneLevelColDragEvent(container, recover) {
    var options2 = _objectSpread({
      animation: 150,
      dataIdAttr: "data-colkey",
      direction: "vertical",
      ghostClass: tableDraggableClasses.ghost,
      chosenClass: tableDraggableClasses.chosen,
      dragClass: tableDraggableClasses.dragging,
      handle: ".".concat(tableBaseClass.thCellInner),
      onEnd: function onEnd(evt) {
        var _onDragSortRef$curren;
        if (evt.newIndex === evt.oldIndex) return;
        if (recover) {
          var _dragColInstance$curr;
          (_dragColInstance$curr = dragColInstance.current) === null || _dragColInstance$curr === void 0 || _dragColInstance$curr.sort(toConsumableArray._toConsumableArray(lastColIdList.current));
        }
        var oldIndex = evt.oldIndex,
          newIndex = evt.newIndex,
          targetElement = evt.target;
        var currentIndex = recover ? oldIndex : newIndex;
        var targetIndex = recover ? newIndex : oldIndex;
        var oldElement = targetElement.children[currentIndex];
        var newElement = targetElement.children[targetIndex];
        var current = utils.getColumnDataByKey(originalColumns.current, oldElement.dataset.colkey);
        var target = utils.getColumnDataByKey(originalColumns.current, newElement.dataset.colkey);
        if (!current || !current.colKey) {
          log.log.error("Table", "colKey is missing in ".concat(JSON.stringify(current)));
        }
        if (!target || !target.colKey) {
          log.log.error("Table", "colKey is missing in ".concat(JSON.stringify(target)));
        }
        currentIndex = utils.getColumnIndexByKey(originalColumns.current, current.colKey);
        targetIndex = utils.getColumnIndexByKey(originalColumns.current, target.colKey);
        var params = {
          data: dragColumns.current,
          currentIndex: currentIndex,
          current: current,
          targetIndex: targetIndex,
          target: target,
          newData: swapDragArrayElement(toConsumableArray._toConsumableArray(originalColumns.current), currentIndex, targetIndex),
          e: evt,
          sort: "col"
        };
        params.currentData = params.newData;
        (_onDragSortRef$curren = onDragSortRef.current) === null || _onDragSortRef$curren === void 0 || _onDragSortRef$curren.call(onDragSortRef, params);
      }
    }, props.dragSortOptions);
    if (!container) return;
    dragColInstance.current = new Sortable__default["default"](container, options2);
    return dragColInstance.current;
  };
  var registerColDragEvent = function registerColDragEvent(tableElement) {
    if (!isColDraggable || !tableElement) return;
    var trList = tableElement.querySelectorAll("thead > tr");
    if (trList.length <= 1) {
      var container = trList[0];
      var dragInstanceTmp = registerOneLevelColDragEvent(container, true);
      lastColIdList.current = dragInstanceTmp === null || dragInstanceTmp === void 0 ? void 0 : dragInstanceTmp.toArray();
    } else {
      trList === null || trList === void 0 || trList.forEach(function (container) {
        registerOneLevelColDragEvent(container, false);
      });
    }
  };
  React.useEffect(function () {
    tData.current = data;
    updateLastRowList();
  }, [data]);
  React.useEffect(function () {
    lastColIdList.current = props.columns.map(function (t) {
      return t.colKey;
    });
    dragColumns.current = props.columns;
    originalColumns.current = props.columns;
  }, [props.columns]);
  React.useEffect(function () {
    if (!primaryTableRef.current) return;
    registerRowDragEvent(primaryTableRef.current.tableElement);
    registerColDragEvent(primaryTableRef.current.tableHtmlElement);
    primaryTableRef.current.onAffixHeaderMount(function (el) {
      registerColDragEvent(el);
    });
  }, [columns, dragSort, innerPagination]);
  return {
    isRowDraggable: isRowDraggable,
    isRowHandlerDraggable: isRowHandlerDraggable,
    isColDraggable: isColDraggable,
    updateLastRowList: updateLastRowList,
    setDragSortColumns: setDragSortColumns
  };
}

exports.DATA_ID_ATTR = DATA_ID_ATTR;
exports.DATA_PARENT_ID_ATTR = DATA_PARENT_ID_ATTR;
exports.EXPANDED_SUFFIX = EXPANDED_SUFFIX;
exports["default"] = useDragSort;
