// tdesign-react/cjs/table/hooks/useFilter.js
/**
 * tdesign v1.17.1
 * (c) 2026 tdesign
 * @license MIT
 */

'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var defineProperty = require('../../_chunks/dep-c446aa11.js');
var slicedToArray = require('../../_chunks/dep-c60a3e4c.js');
var _typeof = require('../../_chunks/dep-c290284d.js');
var React = require('react');
var utils = require('../../_chunks/dep-640fca41.js');
var button_index = require('../../button/index.js');
var hooks_useControlled = require('../../hooks/useControlled.js');
var locale_LocalReceiver = require('../../locale/LocalReceiver.js');
var table_FilterController = require('../FilterController.js');
var table_hooks_useClassName = require('./useClassName.js');
var isFunction = require('../../_chunks/dep-11347ea1.js');
require('../../_chunks/dep-98e6819c.js');
require('../../_chunks/dep-7f3fa935.js');
require('../../_chunks/dep-f8c20fd8.js');
require('../../_chunks/dep-d4a6a9c1.js');
require('../../_chunks/dep-a54f202c.js');
require('../../_chunks/dep-36579bf6.js');
require('../../_chunks/dep-3d05de97.js');
require('../../_chunks/dep-4f1729f0.js');
require('../../_chunks/dep-c37c5e17.js');
require('../../_chunks/dep-eee1aab9.js');
require('../../_chunks/dep-a830ce1e.js');
require('../../_chunks/dep-770bc0ae.js');
require('../../_chunks/dep-b199860d.js');
require('../../button/Button.js');
require('../../_chunks/dep-e41137f7.js');
require('classnames');
require('../../_util/parseTNode.js');
require('../../_chunks/dep-3c628d18.js');
require('../../hooks/useConfig.js');
require('../../config-provider/ConfigContext.js');
require('../../_chunks/dep-2ee09518.js');
require('../../_chunks/dep-a6fa90dc.js');
require('dayjs');
require('../../_chunks/dep-012d9a7c.js');
require('../../_chunks/dep-150a112c.js');
require('../../_chunks/dep-e360e17f.js');
require('../../_chunks/dep-51284537.js');
require('../../_chunks/dep-8a493834.js');
require('../../_chunks/dep-9f5b2891.js');
require('../../_chunks/dep-338f013a.js');
require('../../_chunks/dep-aaf27e39.js');
require('../../_chunks/dep-b47d855f.js');
require('../../_chunks/dep-81bce4e6.js');
require('../../_chunks/dep-863f8f4d.js');
require('../../_chunks/dep-84e4d0da.js');
require('../../_chunks/dep-f78183d7.js');
require('../../_chunks/dep-2f45f6c4.js');
require('../../hooks/useDefaultProps.js');
require('../../hooks/useDomRefCallback.js');
require('../../hooks/useRipple.js');
require('../../_chunks/dep-062bb1b1.js');
require('../../_chunks/dep-ee71ddca.js');
require('../../_chunks/dep-427f8bc0.js');
require('../../hooks/useAnimation.js');
require('../../loading/index.js');
require('../../loading/Loading.js');
require('../../common/Portal.js');
require('react-dom');
require('../../hooks/useLayoutEffect.js');
require('../../loading/defaultProps.js');
require('../../loading/gradient.js');
require('../../_chunks/dep-6e0da644.js');
require('../../_chunks/dep-05bc49e6.js');
require('../../_chunks/dep-8c880a65.js');
require('../../_chunks/dep-b8d0b96c.js');
require('../../loading/plugin.js');
require('../../_util/react-render.js');
require('../../_chunks/dep-7bf14076.js');
require('../../common/PluginContainer.js');
require('../../config-provider/ConfigProvider.js');
require('../../_chunks/dep-ba90997f.js');
require('../../_chunks/dep-22026fea.js');
require('../../_chunks/dep-551d5940.js');
require('../../_chunks/dep-8005b808.js');
require('../../_chunks/dep-2add8824.js');
require('../../button/defaultProps.js');
require('../../_util/noop.js');
require('../../_chunks/dep-45188d54.js');
require('../../_chunks/dep-eccb6a94.js');
require('../../_chunks/dep-f5e1d11c.js');
require('tdesign-icons-react');
require('../../checkbox/index.js');
require('../../_chunks/dep-ab5baf4d.js');
require('../../_util/forwardRefWithStatics.js');
require('hoist-non-react-statics');
require('../../common/Check.js');
require('../../_util/helper.js');
require('../../_chunks/dep-b422817e.js');
require('../../_chunks/dep-fb263e64.js');
require('../../checkbox/defaultProps.js');
require('../../hooks/useGlobalIcon.js');
require('../../input/index.js');
require('../../input/Input.js');
require('../../input/defaultProps.js');
require('../../input/InputGroup.js');
require('../../input/useLengthLimit.js');
require('../../popup/index.js');
require('../../popup/Popup.js');
require('react-transition-group');
require('../../hooks/useAttach.js');
require('../../hooks/useMutationObserver.js');
require('../../hooks/useLatest.js');
require('../../_chunks/dep-aa9a2c54.js');
require('../../_chunks/dep-77ef3011.js');
require('../../_chunks/dep-b36779a4.js');
require('../../_chunks/dep-f2809376.js');
require('../../_chunks/dep-3be56e1e.js');
require('../../hooks/usePopper.js');
require('react-fast-compare');
require('@popperjs/core');
require('../../hooks/useWindowSize.js');
require('../../popup/defaultProps.js');
require('../../popup/hooks/useTrigger.js');
require('../../_util/listener.js');
require('../../_util/ref.js');
require('react-is');
require('../../_util/isFragment.js');
require('../../popup/utils/transition.js');
require('../../popup/PopupPlugin.js');
require('../../radio/index.js');
require('../../_chunks/dep-3ddd7377.js');
require('../../radio/defaultProps.js');
require('../../_chunks/dep-2bc69864.js');
require('../../hooks/useCommonClassName.js');
require('../../hooks/useDeepEffect.js');
require('../../radio/useKeyboard.js');
require('../../_chunks/dep-29222115.js');
require('../../_chunks/dep-2be6b7d2.js');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

var React__default = /*#__PURE__*/_interopDefaultLegacy(React);

function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { defineProperty._defineProperty(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
function isFilterValueExist(value) {
  var isArrayTrue = value instanceof Array && value.length;
  var isObject = _typeof._typeof(value) === "object" && !(value instanceof Array);
  var isObjectTrue = isObject && Object.keys(value || {}).length;
  return isArrayTrue || isObjectTrue || ![null, "", void 0].includes(value);
}
function filterEmptyData(data) {
  var newFilterValue = {};
  Object.keys(data).forEach(function (key) {
    var item = data[key];
    if (isFilterValueExist(item)) {
      newFilterValue[key] = item;
    }
  });
  return newFilterValue;
}
function useFilter(props, primaryTableRef) {
  var columns = props.columns;
  var _useLocaleReceiver = locale_LocalReceiver.useLocaleReceiver("table"),
    _useLocaleReceiver2 = slicedToArray._slicedToArray(_useLocaleReceiver, 2),
    locale = _useLocaleReceiver2[0],
    t = _useLocaleReceiver2[1];
  var _useClassName = table_hooks_useClassName["default"](),
    tableFilterClasses = _useClassName.tableFilterClasses,
    isFocusClass = _useClassName.isFocusClass;
  var _useState = React.useState(),
    _useState2 = slicedToArray._slicedToArray(_useState, 2),
    isTableOverflowHidden = _useState2[0],
    setIsTableOverflowHidden = _useState2[1];
  var _useControlled = hooks_useControlled["default"](props, "filterValue", props.onFilterChange),
    _useControlled2 = slicedToArray._slicedToArray(_useControlled, 2),
    tFilterValue = _useControlled2[0],
    setTFilterValue = _useControlled2[1];
  var _useState3 = React.useState(tFilterValue),
    _useState4 = slicedToArray._slicedToArray(_useState3, 2),
    innerFilterValue = _useState4[0],
    setInnerFilterValue = _useState4[1];
  var hasEmptyCondition = function () {
    var filterEmpty = filterEmptyData(tFilterValue || {});
    return !tFilterValue || !Object.keys(filterEmpty).length;
  }();
  React.useEffect(function () {
    setInnerFilterValue(tFilterValue);
  }, [tFilterValue]);
  function renderFirstFilterRow() {
    var _props$pagination, _props$data;
    if (hasEmptyCondition) return null;
    var defaultNode = /* @__PURE__ */React__default["default"].createElement("div", {
      className: tableFilterClasses.result
    }, /* @__PURE__ */React__default["default"].createElement("span", null, t(locale.searchResultText, {
      result: getFilterResultContent(),
      count: ((_props$pagination = props.pagination) === null || _props$pagination === void 0 ? void 0 : _props$pagination.total) || ((_props$data = props.data) === null || _props$data === void 0 ? void 0 : _props$data.length)
    })), /* @__PURE__ */React__default["default"].createElement(button_index.Button, {
      theme: "primary",
      variant: "text",
      onClick: onResetAll
    }, locale.clearFilterResultButtonText));
    var filterContent = isFunction.isFunction(props.filterRow) ? props.filterRow() : props.filterRow;
    if (filterContent === null) return null;
    var r = filterContent || defaultNode;
    if (!r) return null;
    return /* @__PURE__ */React__default["default"].createElement("div", {
      className: tableFilterClasses.inner
    }, r);
  }
  function getFilterResultContent() {
    var arr = [];
    props.columns.filter(function (col) {
      return col.filter;
    }).forEach(function (col) {
      var value = tFilterValue[col.colKey];
      if (col.filter.list && !["null", "", "undefined"].includes(String(value))) {
        var formattedValue = value instanceof Array ? value : [value];
        var label = [];
        col.filter.list.forEach(function (option) {
          if (formattedValue.includes(option.value)) {
            label.push(option.label);
          }
        });
        value = label.join();
      }
      if (isFilterValueExist(value)) {
        arr.push("".concat(col.title, "\uFF1A").concat(value));
      }
    });
    return arr.join("\uFF1B");
  }
  function onInnerFilterChange(val, column) {
    var filterValue = _objectSpread(_objectSpread({}, innerFilterValue), {}, defineProperty._defineProperty({}, column.colKey, val));
    setInnerFilterValue(filterValue);
    if (!column.filter.showConfirmAndReset) {
      emitFilterChange(filterValue, "filter-change", column);
    }
  }
  function emitFilterChange(filterValue, trigger, column) {
    var _props$onChange;
    setTFilterValue(filterValue, {
      col: column,
      trigger: trigger
    });
    (_props$onChange = props.onChange) === null || _props$onChange === void 0 || _props$onChange.call(props, {
      filter: filterValue
    }, {
      trigger: "filter",
      currentData: props.data
    });
    requestAnimationFrame(function () {
      var _primaryTableRef$curr;
      (_primaryTableRef$curr = primaryTableRef.current) === null || _primaryTableRef$curr === void 0 || _primaryTableRef$curr.scrollToElement({
        index: -1
      });
    });
  }
  function onReset(column) {
    var filterValue = _objectSpread(_objectSpread({}, tFilterValue), {}, defineProperty._defineProperty({}, column.colKey, {
      single: "",
      multiple: [],
      input: ""
    }[column.filter.type] || column.filter.resetValue || ""));
    emitFilterChange(filterValue, "reset", column);
  }
  function onResetAll() {
    var resetValue = utils.getColumnsResetValue(columns);
    emitFilterChange(resetValue, "clear", void 0);
  }
  function onConfirm(column) {
    emitFilterChange(innerFilterValue, "confirm", column);
  }
  function renderFilterIcon(_ref) {
    var _primaryTableRef$curr2;
    var col = _ref.col,
      colIndex = _ref.colIndex;
    return /* @__PURE__ */React__default["default"].createElement(table_FilterController["default"], {
      column: col,
      colIndex: colIndex,
      filterIcon: props.filterIcon,
      tFilterValue: tFilterValue,
      innerFilterValue: innerFilterValue,
      tableFilterClasses: tableFilterClasses,
      isFocusClass: isFocusClass,
      popupProps: col.filter.popupProps,
      onReset: onReset,
      onConfirm: onConfirm,
      onInnerFilterChange: onInnerFilterChange,
      primaryTableElement: primaryTableRef === null || primaryTableRef === void 0 || (_primaryTableRef$curr2 = primaryTableRef.current) === null || _primaryTableRef$curr2 === void 0 ? void 0 : _primaryTableRef$curr2.tableElement,
      onVisibleChange: onPopupVisibleChange
    });
  }
  function onPopupVisibleChange(visible) {
    if (visible && !isTableOverflowHidden) {
      setIsTableOverflowHidden(!visible);
    }
  }
  return {
    hasEmptyCondition: hasEmptyCondition,
    isTableOverflowHidden: isTableOverflowHidden,
    renderFilterIcon: renderFilterIcon,
    renderFirstFilterRow: renderFirstFilterRow
  };
}

exports["default"] = useFilter;
