// tdesign-react/cjs/table/hooks/useRowExpand.js
/**
 * tdesign v1.17.1
 * (c) 2026 tdesign
 * @license MIT
 */

'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var defineProperty = require('../../_chunks/dep-c446aa11.js');
var toConsumableArray = require('../../_chunks/dep-05bc49e6.js');
var slicedToArray = require('../../_chunks/dep-c60a3e4c.js');
var React = require('react');
var classNames = require('classnames');
var tdesignIconsReact = require('tdesign-icons-react');
var _util_parseTNode = require('../../_util/parseTNode.js');
var hooks_useControlled = require('../../hooks/useControlled.js');
var hooks_useGlobalIcon = require('../../hooks/useGlobalIcon.js');
var locale_LocalReceiver = require('../../locale/LocalReceiver.js');
var table_utils = require('../utils.js');
var table_hooks_useClassName = require('./useClassName.js');
var table_hooks_useDragSort = require('./useDragSort.js');
var get = require('../../_chunks/dep-f8c20fd8.js');
var isFunction = require('../../_chunks/dep-11347ea1.js');
require('../../_chunks/dep-c290284d.js');
require('../../_chunks/dep-7f3fa935.js');
require('../../_chunks/dep-8c880a65.js');
require('../../_chunks/dep-98e6819c.js');
require('../../_chunks/dep-3c628d18.js');
require('../../_chunks/dep-3d05de97.js');
require('../../_chunks/dep-a830ce1e.js');
require('../../_util/noop.js');
require('../../_chunks/dep-45188d54.js');
require('../../_chunks/dep-eccb6a94.js');
require('../../_chunks/dep-f5e1d11c.js');
require('../../_chunks/dep-b199860d.js');
require('../../_chunks/dep-a54f202c.js');
require('../../_chunks/dep-36579bf6.js');
require('../../_chunks/dep-4f1729f0.js');
require('../../hooks/useConfig.js');
require('../../config-provider/ConfigContext.js');
require('../../_chunks/dep-2ee09518.js');
require('../../_chunks/dep-a6fa90dc.js');
require('dayjs');
require('../../_chunks/dep-012d9a7c.js');
require('../../_chunks/dep-150a112c.js');
require('../../_chunks/dep-e360e17f.js');
require('../../_chunks/dep-51284537.js');
require('../../_chunks/dep-8a493834.js');
require('../../_chunks/dep-9f5b2891.js');
require('../../_chunks/dep-338f013a.js');
require('../../_chunks/dep-c37c5e17.js');
require('../../_chunks/dep-eee1aab9.js');
require('../../_chunks/dep-770bc0ae.js');
require('../../_chunks/dep-aaf27e39.js');
require('../../_chunks/dep-b47d855f.js');
require('../../_chunks/dep-81bce4e6.js');
require('../../_chunks/dep-863f8f4d.js');
require('../../_chunks/dep-84e4d0da.js');
require('../../_chunks/dep-f78183d7.js');
require('../../_chunks/dep-2f45f6c4.js');
require('../../_chunks/dep-427f8bc0.js');
require('../../config-provider/ConfigProvider.js');
require('../../_chunks/dep-ba90997f.js');
require('../../_chunks/dep-22026fea.js');
require('../../_chunks/dep-551d5940.js');
require('../../_chunks/dep-8005b808.js');
require('../../_chunks/dep-2add8824.js');
require('../../_chunks/dep-6e0da644.js');
require('../../_chunks/dep-e41137f7.js');
require('../../_chunks/dep-b8d0b96c.js');
require('../../_chunks/dep-d4a6a9c1.js');
require('sortablejs');
require('../../_chunks/dep-640fca41.js');
require('../../_chunks/dep-ee71ddca.js');
require('../../hooks/useLatest.js');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

var React__default = /*#__PURE__*/_interopDefaultLegacy(React);
var classNames__default = /*#__PURE__*/_interopDefaultLegacy(classNames);

function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { defineProperty._defineProperty(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
function useRowExpand(props) {
  var _props$columns;
  var expandIcon = props.expandIcon,
    expandedRow = props.expandedRow;
  var _useGlobalIcon = hooks_useGlobalIcon["default"]({
      ChevronRightCircleIcon: tdesignIconsReact.ChevronRightCircleIcon
    }),
    ChevronRightCircleIcon = _useGlobalIcon.ChevronRightCircleIcon;
  var _useLocaleReceiver = locale_LocalReceiver.useLocaleReceiver("table"),
    _useLocaleReceiver2 = slicedToArray._slicedToArray(_useLocaleReceiver, 1),
    locale = _useLocaleReceiver2[0];
  var _useClassName = table_hooks_useClassName["default"](),
    tableExpandClasses = _useClassName.tableExpandClasses,
    positiveRotate90 = _useClassName.positiveRotate90,
    tableFullRowClasses = _useClassName.tableFullRowClasses;
  var _useControlled = hooks_useControlled["default"](props, "expandedRowKeys", props.onExpandChange, {
      defaultExpandedRowKeys: props.defaultExpandedRowKeys || []
    }),
    _useControlled2 = slicedToArray._slicedToArray(_useControlled, 2),
    tExpandedRowKeys = _useControlled2[0],
    setTExpandedRowKeys = _useControlled2[1];
  var showExpandedRow = Boolean(expandedRow);
  var getExpandedRowClass = React.useCallback(function (params) {
    if (!showExpandedRow) return null;
    var row = params.row,
      rowKey = params.rowKey;
    var currentRowKey = get.get(row, rowKey || "id");
    return tableExpandClasses[tExpandedRowKeys !== null && tExpandedRowKeys !== void 0 && tExpandedRowKeys.includes(currentRowKey) ? "rowExpanded" : "rowFolded"];
  }, [tExpandedRowKeys, tableExpandClasses, showExpandedRow]);
  var showExpandIconColumn = props.expandIcon !== false && showExpandedRow;
  var isFirstColumnFixed = ((_props$columns = props.columns) === null || _props$columns === void 0 || (_props$columns = _props$columns[0]) === null || _props$columns === void 0 ? void 0 : _props$columns.fixed) === "left";
  var onToggleExpand = function onToggleExpand(e, row) {
    props.expandOnRowClick && e.stopPropagation();
    var currentId = get.get(row, props.rowKey || "id");
    var index = tExpandedRowKeys.indexOf(currentId);
    var newKeys = toConsumableArray._toConsumableArray(tExpandedRowKeys);
    index !== -1 ? newKeys.splice(index, 1) : newKeys.push(currentId);
    setTExpandedRowKeys(newKeys, {
      expandedRowData: props.data.filter(function (t) {
        return newKeys.includes(get.get(t, props.rowKey || "id"));
      }),
      currentRowData: row
    });
  };
  var renderExpandIcon = function renderExpandIcon(p, expandIcon2) {
    var row = p.row,
      rowIndex = p.rowIndex;
    var currentId = get.get(row, props.rowKey || "id");
    var expanded = tExpandedRowKeys.includes(currentId);
    var defaultIcon = locale.expandIcon || /* @__PURE__ */React__default["default"].createElement(ChevronRightCircleIcon, null);
    var icon = defaultIcon;
    if (expandIcon2 === false || expandIcon2 === null) {
      icon = null;
    } else if (isFunction.isFunction(expandIcon2)) {
      icon = expandIcon2({
        row: row,
        index: rowIndex
      });
    }
    var classes = [tableExpandClasses.iconBox, tableExpandClasses[expanded ? "expanded" : "collapsed"], defineProperty._defineProperty({}, positiveRotate90, expanded)];
    return /* @__PURE__ */React__default["default"].createElement("span", {
      className: classNames__default["default"](classes),
      onClick: function onClick(e) {
        return onToggleExpand(e, row);
      }
    }, icon);
  };
  var getExpandColumn = function getExpandColumn() {
    var expandCol = {
      colKey: "__EXPAND_ROW_ICON_COLUMN__",
      width: 46,
      className: tableExpandClasses.iconCell,
      fixed: isFirstColumnFixed ? "left" : void 0,
      cell: function cell(p) {
        return renderExpandIcon(p, expandIcon);
      },
      stopPropagation: true
    };
    return expandCol;
  };
  var renderExpandedRow = function renderExpandedRow(p) {
    var rowId = get.get(p.row, props.rowKey || "id");
    if (!tExpandedRowKeys || !tExpandedRowKeys.includes(rowId)) return null;
    var isFixedLeft = p.isWidthOverflow && props.columns.find(function (item) {
      return item.fixed === "left";
    });
    var dragAttr = table_utils.enableRowDrag(props.dragSort) && defineProperty._defineProperty(defineProperty._defineProperty({}, table_hooks_useDragSort.DATA_ID_ATTR, "".concat(rowId).concat(table_hooks_useDragSort.EXPANDED_SUFFIX)), table_hooks_useDragSort.DATA_PARENT_ID_ATTR, rowId);
    return /* @__PURE__ */React__default["default"].createElement("tr", _objectSpread(_objectSpread({
      key: "".concat(rowId).concat(table_hooks_useDragSort.EXPANDED_SUFFIX)
    }, dragAttr), {}, {
      className: classNames__default["default"]([tableExpandClasses.row, defineProperty._defineProperty({}, tableFullRowClasses.base, isFixedLeft)])
    }), /* @__PURE__ */React__default["default"].createElement("td", {
      colSpan: p.columns.length
    }, /* @__PURE__ */React__default["default"].createElement("div", {
      className: classNames__default["default"]([tableExpandClasses.rowInner, defineProperty._defineProperty({}, tableFullRowClasses.innerFullRow, isFixedLeft)]),
      style: isFixedLeft ? {
        width: "".concat(p.tableWidth, "px")
      } : {}
    }, /* @__PURE__ */React__default["default"].createElement("div", {
      className: tableFullRowClasses.innerFullElement
    }, _util_parseTNode.parseContentTNode(expandedRow, p)))));
  };
  var onInnerExpandRowClick = function onInnerExpandRowClick(p) {
    onToggleExpand(p.e, p.row);
  };
  return {
    showExpandedRow: showExpandedRow,
    showExpandIconColumn: showExpandIconColumn,
    getExpandColumn: getExpandColumn,
    renderExpandedRow: renderExpandedRow,
    onInnerExpandRowClick: onInnerExpandRowClick,
    getExpandedRowClass: getExpandedRowClass
  };
}

exports["default"] = useRowExpand;
