// tdesign-react/cjs/table/hooks/useStyle.js
/**
 * tdesign v1.17.1
 * (c) 2026 tdesign
 * @license MIT
 */

'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var defineProperty = require('../../_chunks/dep-c446aa11.js');
var React = require('react');
var hooks_useCommonClassName = require('../../hooks/useCommonClassName.js');
var table_hooks_useClassName = require('./useClassName.js');
require('../../_chunks/dep-c290284d.js');
require('../../hooks/useConfig.js');
require('../../config-provider/ConfigContext.js');
require('../../_chunks/dep-2ee09518.js');
require('../../_chunks/dep-a6fa90dc.js');
require('dayjs');
require('../../_chunks/dep-012d9a7c.js');
require('../../_chunks/dep-150a112c.js');
require('../../_chunks/dep-e360e17f.js');
require('../../_chunks/dep-3d05de97.js');
require('../../_chunks/dep-4f1729f0.js');
require('../../_chunks/dep-a54f202c.js');
require('../../_chunks/dep-51284537.js');
require('../../_chunks/dep-8a493834.js');
require('../../_chunks/dep-9f5b2891.js');
require('../../_chunks/dep-338f013a.js');
require('../../_chunks/dep-c37c5e17.js');
require('../../_chunks/dep-eee1aab9.js');
require('../../_chunks/dep-11347ea1.js');
require('../../_chunks/dep-a830ce1e.js');
require('../../_chunks/dep-770bc0ae.js');
require('../../_chunks/dep-aaf27e39.js');
require('../../_chunks/dep-b47d855f.js');
require('../../_chunks/dep-81bce4e6.js');
require('../../_chunks/dep-863f8f4d.js');
require('../../_chunks/dep-84e4d0da.js');
require('../../_chunks/dep-f78183d7.js');
require('../../_chunks/dep-2f45f6c4.js');

function formatCSSUnit(unit) {
  if (!unit) return unit;
  return isNaN(Number(unit)) ? unit : "".concat(unit, "px");
}
function useStyle(props) {
  var className = props.className,
    size = props.size,
    bordered = props.bordered,
    stripe = props.stripe,
    hover = props.hover,
    verticalAlign = props.verticalAlign,
    height = props.height,
    maxHeight = props.maxHeight,
    tableContentWidth = props.tableContentWidth;
  var _useClassName = table_hooks_useClassName["default"](),
    tableBaseClass = _useClassName.tableBaseClass,
    tableAlignClasses = _useClassName.tableAlignClasses;
  var _useCommonClassName = hooks_useCommonClassName["default"](),
    sizeClassNames = _useCommonClassName.sizeClassNames;
  var tableClasses = React.useMemo(function () {
    return [tableBaseClass.table, size !== "medium" && sizeClassNames[size], verticalAlign !== "middle" && tableAlignClasses[verticalAlign], defineProperty._defineProperty(defineProperty._defineProperty(defineProperty._defineProperty(defineProperty._defineProperty(defineProperty._defineProperty(defineProperty._defineProperty({}, tableBaseClass.bordered, bordered), tableBaseClass.striped, stripe), tableBaseClass.hover, hover), tableBaseClass.loading, props.loading), tableBaseClass.affixedHeader, props.headerAffixedTop), tableBaseClass.rowspanAndColspan, props.rowspanAndColspan), className];
  }, [className, bordered, hover, props.loading, props.headerAffixedTop, props.rowspanAndColspan, size, sizeClassNames, stripe, tableAlignClasses, tableBaseClass, verticalAlign]);
  var tableContentStyles = React.useMemo(function () {
    return {
      height: formatCSSUnit(height),
      maxHeight: formatCSSUnit(maxHeight)
    };
  }, [height, maxHeight]);
  var tableElementStyles = React.useMemo(function () {
    return {
      width: formatCSSUnit(tableContentWidth)
    };
  }, [tableContentWidth]);
  return {
    tableClasses: tableClasses,
    sizeClassNames: sizeClassNames,
    tableElementStyles: tableElementStyles,
    tableContentStyles: tableContentStyles
  };
}

exports["default"] = useStyle;
exports.formatCSSUnit = formatCSSUnit;
