// tdesign-react/cjs/pagination/PaginationMini.js
/**
 * tdesign v1.17.1
 * (c) 2026 tdesign
 * @license MIT
 */

'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var defineProperty = require('../_chunks/dep-c446aa11.js');
var React = require('react');
var classNames = require('classnames');
var tdesignIconsReact = require('tdesign-icons-react');
var _util_noop = require('../_util/noop.js');
var button_index = require('../button/index.js');
var hooks_useConfig = require('../hooks/useConfig.js');
var hooks_useDefaultProps = require('../hooks/useDefaultProps.js');
var hooks_useGlobalIcon = require('../hooks/useGlobalIcon.js');
var pagination_defaultProps = require('./defaultProps.js');
var isObject = require('../_chunks/dep-a830ce1e.js');
require('../_chunks/dep-c290284d.js');
require('../button/Button.js');
require('../_chunks/dep-c60a3e4c.js');
require('../_chunks/dep-98e6819c.js');
require('../_chunks/dep-7f3fa935.js');
require('../_chunks/dep-e41137f7.js');
require('../_util/parseTNode.js');
require('../_chunks/dep-3c628d18.js');
require('../_chunks/dep-11347ea1.js');
require('../_chunks/dep-3d05de97.js');
require('../hooks/useDomRefCallback.js');
require('../hooks/useRipple.js');
require('../_chunks/dep-062bb1b1.js');
require('../_chunks/dep-ee71ddca.js');
require('../_chunks/dep-427f8bc0.js');
require('../_chunks/dep-a54f202c.js');
require('../_chunks/dep-4f1729f0.js');
require('../hooks/useAnimation.js');
require('../config-provider/ConfigContext.js');
require('../_chunks/dep-2ee09518.js');
require('../_chunks/dep-a6fa90dc.js');
require('dayjs');
require('../_chunks/dep-012d9a7c.js');
require('../_chunks/dep-150a112c.js');
require('../_chunks/dep-e360e17f.js');
require('../_chunks/dep-51284537.js');
require('../_chunks/dep-8a493834.js');
require('../_chunks/dep-9f5b2891.js');
require('../_chunks/dep-338f013a.js');
require('../_chunks/dep-c37c5e17.js');
require('../_chunks/dep-eee1aab9.js');
require('../_chunks/dep-770bc0ae.js');
require('../_chunks/dep-aaf27e39.js');
require('../_chunks/dep-b47d855f.js');
require('../_chunks/dep-81bce4e6.js');
require('../_chunks/dep-863f8f4d.js');
require('../_chunks/dep-84e4d0da.js');
require('../_chunks/dep-f78183d7.js');
require('../_chunks/dep-2f45f6c4.js');
require('../loading/index.js');
require('../loading/Loading.js');
require('../common/Portal.js');
require('react-dom');
require('../hooks/useLayoutEffect.js');
require('../loading/defaultProps.js');
require('../loading/gradient.js');
require('../_chunks/dep-6e0da644.js');
require('../_chunks/dep-05bc49e6.js');
require('../_chunks/dep-8c880a65.js');
require('../_chunks/dep-b8d0b96c.js');
require('../loading/plugin.js');
require('../_util/react-render.js');
require('../_chunks/dep-7bf14076.js');
require('../common/PluginContainer.js');
require('../config-provider/ConfigProvider.js');
require('../_chunks/dep-ba90997f.js');
require('../_chunks/dep-22026fea.js');
require('../_chunks/dep-551d5940.js');
require('../_chunks/dep-8005b808.js');
require('../_chunks/dep-2add8824.js');
require('../button/defaultProps.js');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

var React__default = /*#__PURE__*/_interopDefaultLegacy(React);
var classNames__default = /*#__PURE__*/_interopDefaultLegacy(classNames);

var PaginationMini = /*#__PURE__*/React.forwardRef(function (props, ref) {
  var _useConfig = hooks_useConfig["default"](),
    classPrefix = _useConfig.classPrefix;
  var _useGlobalIcon = hooks_useGlobalIcon["default"]({
      RoundIcon: tdesignIconsReact.RoundIcon,
      ChevronUpIcon: tdesignIconsReact.ChevronUpIcon,
      ChevronDownIcon: tdesignIconsReact.ChevronDownIcon,
      ChevronLeftIcon: tdesignIconsReact.ChevronLeftIcon,
      ChevronRightIcon: tdesignIconsReact.ChevronRightIcon
    }),
    RoundIcon = _useGlobalIcon.RoundIcon,
    ChevronUpIcon = _useGlobalIcon.ChevronUpIcon,
    ChevronDownIcon = _useGlobalIcon.ChevronDownIcon,
    ChevronLeftIcon = _useGlobalIcon.ChevronLeftIcon,
    ChevronRightIcon = _useGlobalIcon.ChevronRightIcon;
  var _useDefaultProps = hooks_useDefaultProps["default"](props, pagination_defaultProps.paginationMiniDefaultProps),
    variant = _useDefaultProps.variant,
    tips = _useDefaultProps.tips,
    showCurrent = _useDefaultProps.showCurrent,
    disabled = _useDefaultProps.disabled,
    layout = _useDefaultProps.layout,
    size = _useDefaultProps.size,
    _useDefaultProps$onCh = _useDefaultProps.onChange,
    onChange = _useDefaultProps$onCh === void 0 ? _util_noop["default"] : _useDefaultProps$onCh,
    className = _useDefaultProps.className,
    style = _useDefaultProps.style;
  var titleConfig = React.useMemo(function () {
    if (isObject.isObject(tips)) {
      return tips;
    }
    if (tips === true) {
      return {
        prev: "\u4E0A\u4E00\u9875",
        current: "\u5F53\u524D",
        next: "\u4E0B\u4E00\u9875"
      };
    }
    return {};
  }, [tips]);
  var disabledConfig = React.useMemo(function () {
    if (isObject.isObject(disabled)) {
      return disabled;
    }
    if (disabled === true) {
      return {
        prev: true,
        current: true,
        next: true
      };
    }
    return {
      prev: false,
      current: false,
      next: false
    };
  }, [disabled]);
  return /* @__PURE__ */React__default["default"].createElement("div", {
    className: classNames__default["default"]("".concat(classPrefix, "-pagination-mini"), className, defineProperty._defineProperty({}, "".concat(classPrefix, "-pagination-mini--outline"), variant === "outline")),
    ref: ref,
    style: style
  }, /* @__PURE__ */React__default["default"].createElement(button_index.Button, {
    title: titleConfig.prev,
    variant: variant,
    size: size,
    shape: "square",
    onClick: function onClick(e) {
      return onChange({
        e: e,
        trigger: "prev"
      });
    },
    icon: layout === "horizontal" ? /* @__PURE__ */React__default["default"].createElement(ChevronLeftIcon, null) : /* @__PURE__ */React__default["default"].createElement(ChevronUpIcon, null),
    className: "".concat(classPrefix, "-pagination-mini__prev"),
    disabled: disabledConfig.prev
  }), showCurrent && /* @__PURE__ */React__default["default"].createElement(button_index.Button, {
    title: titleConfig.current,
    variant: variant,
    size: size,
    shape: "square",
    onClick: function onClick(e) {
      return onChange({
        e: e,
        trigger: "current"
      });
    },
    icon: /* @__PURE__ */React__default["default"].createElement(RoundIcon, null),
    className: "".concat(classPrefix, "-pagination-mini__current"),
    disabled: disabledConfig.current
  }), /* @__PURE__ */React__default["default"].createElement(button_index.Button, {
    title: titleConfig.next,
    variant: variant,
    size: size,
    shape: "square",
    onClick: function onClick(e) {
      return onChange({
        e: e,
        trigger: "next"
      });
    },
    icon: layout === "horizontal" ? /* @__PURE__ */React__default["default"].createElement(ChevronRightIcon, null) : /* @__PURE__ */React__default["default"].createElement(ChevronDownIcon, null),
    className: "".concat(classPrefix, "-pagination-mini__next"),
    disabled: disabledConfig.next
  }));
});
PaginationMini.displayName = "PaginationMini";

exports["default"] = PaginationMini;
