// tdesign-react/cjs/rate/Rate.js
/**
 * tdesign v1.17.1
 * (c) 2026 tdesign
 * @license MIT
 */

'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var toConsumableArray = require('../_chunks/dep-05bc49e6.js');
var slicedToArray = require('../_chunks/dep-c60a3e4c.js');
var defineProperty = require('../_chunks/dep-c446aa11.js');
var objectWithoutProperties = require('../_chunks/dep-e41137f7.js');
var React = require('react');
var classNames = require('classnames');
var tdesignIconsReact = require('tdesign-icons-react');
var hooks_useConfig = require('../hooks/useConfig.js');
var hooks_useControlled = require('../hooks/useControlled.js');
var hooks_useDefaultProps = require('../hooks/useDefaultProps.js');
var hooks_useGlobalIcon = require('../hooks/useGlobalIcon.js');
var locale_LocalReceiver = require('../locale/LocalReceiver.js');
var tooltip_index = require('../tooltip/index.js');
var rate_defaultProps = require('./defaultProps.js');
require('../_chunks/dep-7f3fa935.js');
require('../_chunks/dep-8c880a65.js');
require('../_chunks/dep-98e6819c.js');
require('../_chunks/dep-c290284d.js');
require('../config-provider/ConfigContext.js');
require('../_chunks/dep-2ee09518.js');
require('../_chunks/dep-a6fa90dc.js');
require('dayjs');
require('../_chunks/dep-012d9a7c.js');
require('../_chunks/dep-150a112c.js');
require('../_chunks/dep-e360e17f.js');
require('../_chunks/dep-3d05de97.js');
require('../_chunks/dep-4f1729f0.js');
require('../_chunks/dep-a54f202c.js');
require('../_chunks/dep-51284537.js');
require('../_chunks/dep-8a493834.js');
require('../_chunks/dep-9f5b2891.js');
require('../_chunks/dep-338f013a.js');
require('../_chunks/dep-c37c5e17.js');
require('../_chunks/dep-eee1aab9.js');
require('../_chunks/dep-11347ea1.js');
require('../_chunks/dep-a830ce1e.js');
require('../_chunks/dep-770bc0ae.js');
require('../_chunks/dep-aaf27e39.js');
require('../_chunks/dep-b47d855f.js');
require('../_chunks/dep-81bce4e6.js');
require('../_chunks/dep-863f8f4d.js');
require('../_chunks/dep-84e4d0da.js');
require('../_chunks/dep-f78183d7.js');
require('../_chunks/dep-2f45f6c4.js');
require('../_util/noop.js');
require('../_chunks/dep-45188d54.js');
require('../_chunks/dep-eccb6a94.js');
require('../_chunks/dep-f5e1d11c.js');
require('../_chunks/dep-b199860d.js');
require('../_chunks/dep-36579bf6.js');
require('../_chunks/dep-427f8bc0.js');
require('../config-provider/ConfigProvider.js');
require('../_chunks/dep-ba90997f.js');
require('../_chunks/dep-22026fea.js');
require('../_chunks/dep-551d5940.js');
require('../_chunks/dep-8005b808.js');
require('../_chunks/dep-2add8824.js');
require('../tooltip/Tooltip.js');
require('../popup/index.js');
require('../popup/Popup.js');
require('react-transition-group');
require('../_chunks/dep-ee71ddca.js');
require('../common/Portal.js');
require('react-dom');
require('../hooks/useLayoutEffect.js');
require('../hooks/useAnimation.js');
require('../hooks/useAttach.js');
require('../hooks/useMutationObserver.js');
require('../hooks/useLatest.js');
require('../_chunks/dep-aa9a2c54.js');
require('../_chunks/dep-77ef3011.js');
require('../_chunks/dep-b36779a4.js');
require('../_chunks/dep-f2809376.js');
require('../_chunks/dep-3be56e1e.js');
require('../hooks/usePopper.js');
require('react-fast-compare');
require('@popperjs/core');
require('../hooks/useWindowSize.js');
require('../popup/defaultProps.js');
require('../popup/hooks/useTrigger.js');
require('../_util/listener.js');
require('../_util/ref.js');
require('react-is');
require('../_util/isFragment.js');
require('../popup/utils/transition.js');
require('../popup/PopupPlugin.js');
require('../_chunks/dep-7bf14076.js');
require('../_util/react-render.js');
require('../common/PluginContainer.js');
require('../tooltip/defaultProps.js');
require('../tooltip/TooltipLite.js');
require('../hooks/useSwitch.js');
require('../hooks/usePersistFn.js');
require('../_chunks/dep-d07eb641.js');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

var React__default = /*#__PURE__*/_interopDefaultLegacy(React);
var classNames__default = /*#__PURE__*/_interopDefaultLegacy(classNames);

var _excluded = ["icon"];
function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { defineProperty._defineProperty(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
var RateIcon = function RateIcon(_ref) {
  var icon = _ref.icon,
    props = objectWithoutProperties._objectWithoutProperties(_ref, _excluded);
  var _useGlobalIcon = hooks_useGlobalIcon["default"]({
      StarFilledIcon: tdesignIconsReact.StarFilledIcon
    }),
    StarFilledIcon = _useGlobalIcon.StarFilledIcon;
  if (/*#__PURE__*/React__default["default"].isValidElement(icon)) {
    return /*#__PURE__*/React__default["default"].cloneElement(icon, props);
  }
  return /* @__PURE__ */React__default["default"].createElement(StarFilledIcon, _objectSpread({}, props));
};
var Rate = /*#__PURE__*/React__default["default"].forwardRef(function (originalProps, ref) {
  var props = hooks_useDefaultProps["default"](originalProps, rate_defaultProps.rateDefaultProps);
  var allowHalf = props.allowHalf,
    color = props.color,
    count = props.count,
    disabled = props.disabled,
    gap = props.gap,
    showText = props.showText,
    size = props.size,
    icon = props.icon,
    className = props.className,
    style = props.style,
    onChange = props.onChange,
    texts = props.texts,
    clearable = props.clearable;
  var _useLocaleReceiver = locale_LocalReceiver.useLocaleReceiver("rate"),
    _useLocaleReceiver2 = slicedToArray._slicedToArray(_useLocaleReceiver, 2),
    locale = _useLocaleReceiver2[0],
    t = _useLocaleReceiver2[1];
  var displayTexts = texts.length ? texts : t(locale.rateText);
  var _useConfig = hooks_useConfig["default"](),
    classPrefix = _useConfig.classPrefix;
  var _useControlled = hooks_useControlled["default"](props, "value", onChange),
    _useControlled2 = slicedToArray._slicedToArray(_useControlled, 2),
    starValue = _useControlled2[0],
    setStarValue = _useControlled2[1];
  var _useState = React.useState(void 0),
    _useState2 = slicedToArray._slicedToArray(_useState, 2),
    hoverValue = _useState2[0],
    setHoverValue = _useState2[1];
  var displayValue = hoverValue || starValue;
  var rootRef = React__default["default"].useRef(null);
  var activeColor = Array.isArray(color) ? color[0] : color;
  var defaultColor = Array.isArray(color) ? color[1] : "var(--td-bg-color-component)";
  var getStarValue = function getStarValue(event, index) {
    if (allowHalf) {
      var rootNode = rootRef.current;
      var _rootNode$getBounding = rootNode.getBoundingClientRect(),
        left = _rootNode$getBounding.left;
      var firstStar = rootNode.firstChild;
      var _firstStar$getBoundin = firstStar.getBoundingClientRect(),
        width = _firstStar$getBoundin.width;
      var clientX = event.clientX;
      var starMiddle = width * (index - 0.5) + gap * (index - 1);
      if (clientX - left >= starMiddle) {
        return index;
      }
      if (clientX - left < starMiddle) {
        return index - 0.5;
      }
    }
    return index;
  };
  var mouseEnterHandler = function mouseEnterHandler(event, index) {
    if (disabled) {
      return;
    }
    setHoverValue(getStarValue(event, index));
  };
  var mouseLeaveHandler = function mouseLeaveHandler() {
    if (disabled) {
      return;
    }
    setHoverValue(void 0);
  };
  var clickHandler = function clickHandler(event, index) {
    if (disabled) {
      return;
    }
    var value = getStarValue(event, index);
    if (clearable && value === starValue) {
      value = 0;
      setHoverValue(void 0);
    }
    setStarValue(value);
  };
  var getStarCls = function getStarCls(index) {
    if (allowHalf && index + 0.5 === displayValue) {
      return "".concat(classPrefix, "-rate__item--half");
    }
    if (index >= displayValue) {
      return "";
    }
    if (index < displayValue) {
      return "".concat(classPrefix, "-rate__item--full");
    }
  };
  return /* @__PURE__ */React__default["default"].createElement("div", {
    ref: ref,
    style: style,
    className: classNames__default["default"]("".concat(classPrefix, "-rate"), className),
    onMouseLeave: mouseLeaveHandler
  }, /* @__PURE__ */React__default["default"].createElement("ul", {
    className: "".concat(classPrefix, "-rate__list"),
    style: {
      gap: gap
    },
    ref: rootRef
  }, toConsumableArray._toConsumableArray(Array(count)).map(function (_, index) {
    return /* @__PURE__ */React__default["default"].createElement("li", {
      key: index,
      className: classNames__default["default"]("".concat(classPrefix, "-rate__item"), getStarCls(index)),
      onClick: function onClick(event) {
        return clickHandler(event, index + 1);
      },
      onMouseMove: function onMouseMove(event) {
        return mouseEnterHandler(event, index + 1);
      }
    }, showText ? /* @__PURE__ */React__default["default"].createElement(tooltip_index.TooltipLite, {
      key: index,
      content: displayTexts[displayValue - 1]
    }, /* @__PURE__ */React__default["default"].createElement("div", {
      className: "".concat(classPrefix, "-rate__star-top")
    }, /* @__PURE__ */React__default["default"].createElement(RateIcon, {
      size: size,
      color: activeColor,
      icon: icon
    })), /* @__PURE__ */React__default["default"].createElement("div", {
      className: "".concat(classPrefix, "-rate__star-bottom")
    }, /* @__PURE__ */React__default["default"].createElement(RateIcon, {
      size: size,
      color: defaultColor,
      icon: icon
    }))) : /* @__PURE__ */React__default["default"].createElement(React__default["default"].Fragment, null, /* @__PURE__ */React__default["default"].createElement("div", {
      className: "".concat(classPrefix, "-rate__star-top")
    }, /* @__PURE__ */React__default["default"].createElement(RateIcon, {
      size: size,
      color: activeColor,
      icon: icon
    })), /* @__PURE__ */React__default["default"].createElement("div", {
      className: "".concat(classPrefix, "-rate__star-bottom")
    }, /* @__PURE__ */React__default["default"].createElement(RateIcon, {
      size: size,
      color: defaultColor,
      icon: icon
    }))));
  })), showText && /* @__PURE__ */React__default["default"].createElement("div", {
    className: "".concat(classPrefix, "-rate__text")
  }, displayTexts[displayValue - 1]));
});
Rate.displayName = "Rate";

exports["default"] = Rate;
