// tdesign-react/esm/image-viewer/defaultTrigger.js
/**
 * tdesign v1.17.1
 * (c) 2026 tdesign
 * @license MIT
 */

import { _ as _slicedToArray } from '../_chunks/dep-d1eeb95c.js';
import React from 'react';
import { BrowseIcon } from 'tdesign-icons-react';
import useConfig from '../hooks/useConfig.js';
import { Image } from '../image/index.js';
import { useLocaleReceiver } from '../locale/LocalReceiver.js';
import '../_chunks/dep-673e8854.js';
import '../_chunks/dep-570a5adb.js';
import '../config-provider/ConfigContext.js';
import '../_chunks/dep-55b77e26.js';
import '../_chunks/dep-067cb22b.js';
import 'lodash-es';
import '../locale/zh_CN.js';
import '../_chunks/dep-e56ffacc.js';
import 'dayjs';
import '../_chunks/dep-84036ea0.js';
import '../image/style/index.js';
import '../image/Image.js';
import '../_chunks/dep-84926167.js';
import 'classnames';
import '../_chunks/dep-5ca9c80d.js';
import '../hooks/useDefaultProps.js';
import '../hooks/useGlobalIcon.js';
import '../hooks/useImagePreviewUrl.js';
import '../_chunks/dep-0f787ecd.js';
import '../_chunks/dep-65b369a5.js';
import '../_chunks/dep-aec71025.js';
import '../space/index.js';
import '../space/style/index.js';
import '../space/Space.js';
import 'react-is';
import '../_chunks/dep-89bc2167.js';
import '../_chunks/dep-154fcb93.js';
import '../_chunks/dep-e9c0bb63.js';
import '../space/defaultProps.js';
import '../space/type.js';
import '../image/defaultProps.js';
import '../config-provider/index.js';
import '../config-provider/ConfigProvider.js';
import '../config-provider/type.js';

var DefaultTrigger = function DefaultTrigger(_ref) {
  var showImage = _ref.showImage,
    onClick = _ref.onClick;
  var _useConfig = useConfig(),
    classPrefix = _useConfig.classPrefix;
  var _useLocaleReceiver = useLocaleReceiver("imageViewer"),
    _useLocaleReceiver2 = _slicedToArray(_useLocaleReceiver, 1),
    local = _useLocaleReceiver2[0];
  return /* @__PURE__ */React.createElement("div", {
    className: "".concat(classPrefix, "-image-viewer__trigger"),
    onClick: onClick
  }, /* @__PURE__ */React.createElement(Image, {
    src: showImage,
    alt: "preview",
    fit: "contain",
    className: "".concat(classPrefix, "-image-viewer__trigger-img")
  }), /* @__PURE__ */React.createElement("div", {
    className: "".concat(classPrefix, "-image-viewer__trigger--hover")
  }, /* @__PURE__ */React.createElement("span", null, /* @__PURE__ */React.createElement(BrowseIcon, {
    size: "1.4em",
    className: "".concat(classPrefix, "-image-viewer__trigger-icon")
  }), local.previewText)));
};

export { DefaultTrigger };
