// tdesign-react/cjs/pagination/hooks/usePrevNextJumper.js
/**
 * tdesign v1.17.1
 * (c) 2026 tdesign
 * @license MIT
 */

'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var defineProperty = require('../../_chunks/dep-c446aa11.js');
var React = require('react');
var classNames = require('classnames');
var tdesignIconsReact = require('tdesign-icons-react');
var hooks_useConfig = require('../../hooks/useConfig.js');
var hooks_useGlobalIcon = require('../../hooks/useGlobalIcon.js');
require('../../_chunks/dep-c290284d.js');
require('../../config-provider/ConfigContext.js');
require('../../_chunks/dep-2ee09518.js');
require('../../_chunks/dep-a6fa90dc.js');
require('dayjs');
require('../../_chunks/dep-012d9a7c.js');
require('../../_chunks/dep-150a112c.js');
require('../../_chunks/dep-e360e17f.js');
require('../../_chunks/dep-3d05de97.js');
require('../../_chunks/dep-4f1729f0.js');
require('../../_chunks/dep-a54f202c.js');
require('../../_chunks/dep-51284537.js');
require('../../_chunks/dep-8a493834.js');
require('../../_chunks/dep-9f5b2891.js');
require('../../_chunks/dep-338f013a.js');
require('../../_chunks/dep-c37c5e17.js');
require('../../_chunks/dep-eee1aab9.js');
require('../../_chunks/dep-11347ea1.js');
require('../../_chunks/dep-a830ce1e.js');
require('../../_chunks/dep-770bc0ae.js');
require('../../_chunks/dep-aaf27e39.js');
require('../../_chunks/dep-b47d855f.js');
require('../../_chunks/dep-81bce4e6.js');
require('../../_chunks/dep-863f8f4d.js');
require('../../_chunks/dep-84e4d0da.js');
require('../../_chunks/dep-f78183d7.js');
require('../../_chunks/dep-2f45f6c4.js');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

var React__default = /*#__PURE__*/_interopDefaultLegacy(React);
var classNames__default = /*#__PURE__*/_interopDefaultLegacy(classNames);

function usePrevNextJumper(props) {
  var _useConfig = hooks_useConfig["default"](),
    classPrefix = _useConfig.classPrefix;
  var _useGlobalIcon = hooks_useGlobalIcon["default"]({
      ChevronLeftIcon: tdesignIconsReact.ChevronLeftIcon,
      ChevronRightIcon: tdesignIconsReact.ChevronRightIcon
    }),
    ChevronLeftIcon = _useGlobalIcon.ChevronLeftIcon,
    ChevronRightIcon = _useGlobalIcon.ChevronRightIcon;
  var name = "".concat(classPrefix, "-pagination");
  var showPreviousAndNextBtn = props.showPreviousAndNextBtn,
    disabled = props.disabled,
    current = props.current,
    pageCount = props.pageCount,
    changeCurrent = props.changeCurrent;
  var prevJumper = showPreviousAndNextBtn && /* @__PURE__ */React__default["default"].createElement("div", {
    className: classNames__default["default"]("".concat(name, "__btn"), "".concat(name, "__btn-prev"), defineProperty._defineProperty({}, "".concat(classPrefix, "-is-disabled"), disabled || current === 1)),
    onClick: function onClick() {
      if (disabled || current === 1) return;
      changeCurrent(current - 1);
    }
  }, /* @__PURE__ */React__default["default"].createElement(ChevronLeftIcon, null));
  var nextJumper = showPreviousAndNextBtn && /* @__PURE__ */React__default["default"].createElement("div", {
    className: classNames__default["default"]("".concat(name, "__btn"), "".concat(name, "__btn-next"), defineProperty._defineProperty({}, "".concat(classPrefix, "-is-disabled"), disabled || current === pageCount)),
    onClick: function onClick() {
      if (disabled || current === pageCount) return;
      changeCurrent(current + 1);
    }
  }, /* @__PURE__ */React__default["default"].createElement(ChevronRightIcon, null));
  return {
    prevJumper: prevJumper,
    nextJumper: nextJumper
  };
}

exports["default"] = usePrevNextJumper;
