// tdesign-react/esm/message/config.js
/**
 * tdesign v1.17.1
 * (c) 2026 tdesign
 * @license MIT
 */

import { _ as _defineProperty } from '../_chunks/dep-55b77e26.js';
import { D as DEFAULT_DURATION, b as DEFAULT_Z_INDEX } from '../_chunks/dep-a6da459b.js';
import { messageDefaultProps } from './defaultProps.js';
import '../_chunks/dep-067cb22b.js';

function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { _defineProperty(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
var globalConfig = {
  top: 32
};
var messageDefaultConfig = _objectSpread(_objectSpread({}, messageDefaultProps), {}, {
  duration: DEFAULT_DURATION,
  placement: "top",
  zIndex: DEFAULT_Z_INDEX
});
var getMessageConfig = function getMessageConfig(options) {
  var currentOptions = _objectSpread({}, options);
  for (var i in currentOptions) {
    if (typeof currentOptions[i] === "undefined") {
      delete currentOptions[i];
    }
  }
  if (typeof currentOptions.duration !== "number" || currentOptions.duration < 0) {
    delete currentOptions.duration;
  }
  return _objectSpread(_objectSpread({}, messageDefaultConfig), currentOptions);
};
var setGlobalConfig = function setGlobalConfig(options) {
  messageDefaultConfig = _objectSpread({}, getMessageConfig(options));
};
var getMessageDefaultConfig = function getMessageDefaultConfig() {
  return _objectSpread({}, messageDefaultConfig);
};

export { getMessageConfig, getMessageDefaultConfig, globalConfig, setGlobalConfig };
