// tdesign-react/esm/message/MessageComponent.js
/**
 * tdesign v1.17.1
 * (c) 2026 tdesign
 * @license MIT
 */

import { _ as _slicedToArray } from '../_chunks/dep-d1eeb95c.js';
import { _ as _objectWithoutProperties } from '../_chunks/dep-84926167.js';
import React, { forwardRef, useState, useEffect } from 'react';
import classNames from 'classnames';
import { a as fadeIn } from '../_chunks/dep-a6da459b.js';
import noop from '../_util/noop.js';
import { usePersistFn } from '../hooks/usePersistFn.js';
import MessageClose from './MessageClose.js';
import MessageIcon from './MessageIcon.js';
import { useMessageClass } from './useMessageClass.js';
import '../_chunks/dep-673e8854.js';
import '../_chunks/dep-570a5adb.js';
import 'tdesign-icons-react';
import '../hooks/useGlobalIcon.js';
import '../hooks/useConfig.js';
import '../config-provider/ConfigContext.js';
import '../_chunks/dep-55b77e26.js';
import '../_chunks/dep-067cb22b.js';
import 'lodash-es';
import '../locale/zh_CN.js';
import '../_chunks/dep-e56ffacc.js';
import 'dayjs';
import '../_chunks/dep-84036ea0.js';
import '../loading/index.js';
import '../loading/style/index.js';
import '../loading/Loading.js';
import '../_chunks/dep-8c226881.js';
import '../common/Portal.js';
import 'react-dom';
import '../hooks/useLayoutEffect.js';
import '../hooks/useDefaultProps.js';
import '../loading/defaultProps.js';
import '../loading/gradient.js';
import '../_chunks/dep-e17ec8df.js';
import '../_chunks/dep-89bc2167.js';
import '../_chunks/dep-154fcb93.js';
import '../_chunks/dep-e9c0bb63.js';
import '../hooks/useDomRefCallback.js';
import '../loading/plugin.js';
import '../_util/react-render.js';
import '../_chunks/dep-55f620a2.js';
import '../common/PluginContainer.js';
import '../config-provider/index.js';
import '../config-provider/ConfigProvider.js';
import '../config-provider/type.js';
import '../loading/type.js';

var _excluded = ["placement"];
var MessageComponent = /*#__PURE__*/forwardRef(function (props, ref) {
  var placement = props.placement,
    otherProps = _objectWithoutProperties(props, _excluded);
  var _useMessageClass = useMessageClass(),
    tdMessagePrefix = _useMessageClass.tdMessagePrefix,
    tdClassIsGenerator = _useMessageClass.tdClassIsGenerator;
  var _otherProps$theme = otherProps.theme,
    theme = _otherProps$theme === void 0 ? "info" : _otherProps$theme,
    className = otherProps.className,
    children = otherProps.children,
    style = otherProps.style,
    _otherProps$icon = otherProps.icon,
    icon = _otherProps$icon === void 0 ? true : _otherProps$icon,
    content = otherProps.content,
    closeBtn = otherProps.closeBtn,
    _otherProps$onCloseBt = otherProps.onCloseBtnClick,
    onCloseBtnClick = _otherProps$onCloseBt === void 0 ? noop : _otherProps$onCloseBt,
    _otherProps$onDuratio = otherProps.onDurationEnd,
    onDurationEnd = _otherProps$onDuratio === void 0 ? noop : _otherProps$onDuratio,
    _otherProps$onClose = otherProps.onClose,
    onClose = _otherProps$onClose === void 0 ? noop : _otherProps$onClose,
    duration = otherProps.duration;
  var _useState = useState(false),
    _useState2 = _slicedToArray(_useState, 2),
    isHovering = _useState2[0],
    setIsHovering = _useState2[1];
  var onCloseFn = usePersistFn(onClose);
  var onDurationEndFn = usePersistFn(onDurationEnd);
  var onCloseBtnClickFn = usePersistFn(onCloseBtnClick);
  function handleCloseBtnClick(e) {
    onCloseBtnClickFn(e);
    onCloseFn({
      trigger: "close-click"
    });
  }
  useEffect(function () {
    if (ref && "current" in ref && ref.current && placement) {
      fadeIn(ref.current, placement);
    }
  }, []);
  useEffect(function () {
    if (!isHovering && duration > 0) {
      var timer = setTimeout(function () {
        onDurationEndFn();
        onCloseFn({
          trigger: "duration-end"
        });
      }, duration);
      return function () {
        clearTimeout(timer);
      };
    }
  }, [duration, isHovering, onCloseFn, onDurationEndFn]);
  return /* @__PURE__ */React.createElement("div", {
    ref: ref,
    key: "message",
    style: style,
    className: classNames(className, "".concat(tdMessagePrefix), tdClassIsGenerator(theme), closeBtn ? tdClassIsGenerator("closable") : ""),
    onMouseEnter: function onMouseEnter() {
      return setIsHovering(true);
    },
    onMouseLeave: function onMouseLeave() {
      return setIsHovering(false);
    }
  }, icon === true ? /* @__PURE__ */React.createElement(MessageIcon, {
    theme: theme
  }) : icon, content ? content : children, /* @__PURE__ */React.createElement(MessageClose, {
    closeBtn: closeBtn,
    onCloseBtnClick: handleCloseBtnClick
  }));
});
MessageComponent.displayName = "Message";

export { MessageComponent as default };
