// tdesign-react/lib/message/useMessageClass.js
/**
 * tdesign v1.17.1
 * (c) 2026 tdesign
 * @license MIT
 */

import useConfig from '../hooks/useConfig.js';
import 'react';
import '../config-provider/ConfigContext.js';
import '../_chunks/dep-55b77e26.js';
import '../_chunks/dep-067cb22b.js';
import 'lodash-es';
import '../_chunks/dep-c3e5e988.js';
import '../_chunks/dep-65b05485.js';
import 'dayjs';

function useMessageClass() {
  var _useConfig = useConfig(),
    classPrefix = _useConfig.classPrefix;
  var tdMessagePrefix = "".concat(classPrefix, "-message");
  var tdMessageListClass = "".concat(tdMessagePrefix, "__list");
  var tdClassIsGenerator = function tdClassIsGenerator(append) {
    return "".concat(classPrefix, "-is-").concat(append);
  };
  var tdMessageClassGenerator = function tdMessageClassGenerator(append) {
    return "".concat(tdMessagePrefix, "__").concat(append);
  };
  var tdMessagePlacementClassGenerator = function tdMessagePlacementClassGenerator(placement) {
    return "".concat(tdMessagePrefix, "-placement--").concat(placement);
  };
  return {
    tdMessagePrefix: tdMessagePrefix,
    tdMessageListClass: tdMessageListClass,
    tdClassIsGenerator: tdClassIsGenerator,
    tdMessageClassGenerator: tdMessageClassGenerator,
    tdMessagePlacementClassGenerator: tdMessagePlacementClassGenerator
  };
}

export { useMessageClass };
