// tdesign-react/cjs/tabs/TabNavItem.js
/**
 * tdesign v1.17.1
 * (c) 2026 tdesign
 * @license MIT
 */

'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var defineProperty = require('../_chunks/dep-c446aa11.js');
var slicedToArray = require('../_chunks/dep-c60a3e4c.js');
var React = require('react');
var classNames = require('classnames');
var tdesignIconsReact = require('tdesign-icons-react');
var _util_noop = require('../_util/noop.js');
var hooks_useDomRefCallback = require('../hooks/useDomRefCallback.js');
var hooks_useGlobalIcon = require('../hooks/useGlobalIcon.js');
var hooks_useRipple = require('../hooks/useRipple.js');
var tabs_useTabClass = require('./useTabClass.js');
require('../_chunks/dep-c290284d.js');
require('../_chunks/dep-98e6819c.js');
require('../_chunks/dep-7f3fa935.js');
require('../hooks/useConfig.js');
require('../config-provider/ConfigContext.js');
require('../_chunks/dep-2ee09518.js');
require('../_chunks/dep-a6fa90dc.js');
require('dayjs');
require('../_chunks/dep-012d9a7c.js');
require('../_chunks/dep-150a112c.js');
require('../_chunks/dep-e360e17f.js');
require('../_chunks/dep-3d05de97.js');
require('../_chunks/dep-4f1729f0.js');
require('../_chunks/dep-a54f202c.js');
require('../_chunks/dep-51284537.js');
require('../_chunks/dep-8a493834.js');
require('../_chunks/dep-9f5b2891.js');
require('../_chunks/dep-338f013a.js');
require('../_chunks/dep-c37c5e17.js');
require('../_chunks/dep-eee1aab9.js');
require('../_chunks/dep-11347ea1.js');
require('../_chunks/dep-a830ce1e.js');
require('../_chunks/dep-770bc0ae.js');
require('../_chunks/dep-aaf27e39.js');
require('../_chunks/dep-b47d855f.js');
require('../_chunks/dep-81bce4e6.js');
require('../_chunks/dep-863f8f4d.js');
require('../_chunks/dep-84e4d0da.js');
require('../_chunks/dep-f78183d7.js');
require('../_chunks/dep-2f45f6c4.js');
require('../_chunks/dep-062bb1b1.js');
require('../_chunks/dep-ee71ddca.js');
require('../_chunks/dep-427f8bc0.js');
require('../hooks/useAnimation.js');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

var React__default = /*#__PURE__*/_interopDefaultLegacy(React);
var classNames__default = /*#__PURE__*/_interopDefaultLegacy(classNames);

function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { defineProperty._defineProperty(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
var TabNavItem = function TabNavItem(props) {
  var label = props.label,
    removable = props.removable,
    isActive = props.isActive,
    _props$onClick = props.onClick,
    onClick = _props$onClick === void 0 ? _util_noop["default"] : _props$onClick,
    theme = props.theme,
    placement = props.placement,
    _props$onRemove = props.onRemove,
    onRemove = _props$onRemove === void 0 ? _util_noop["default"] : _props$onRemove,
    value = props.value,
    _props$size = props.size,
    size = _props$size === void 0 ? "medium" : _props$size,
    _props$disabled = props.disabled,
    disabled = _props$disabled === void 0 ? false : _props$disabled,
    index = props.index,
    _props$onTabRemove = props.onTabRemove,
    onTabRemove = _props$onTabRemove === void 0 ? _util_noop["default"] : _props$onTabRemove,
    innerRef = props.innerRef,
    dragProps = props.dragProps;
  var _useGlobalIcon = hooks_useGlobalIcon["default"]({
      CloseIcon: tdesignIconsReact.CloseIcon
    }),
    CloseIcon = _useGlobalIcon.CloseIcon;
  var isCard = theme === "card";
  var _useTabClass = tabs_useTabClass.useTabClass(),
    tdTabsClassGenerator = _useTabClass.tdTabsClassGenerator,
    tdClassGenerator = _useTabClass.tdClassGenerator,
    tdSizeClassGenerator = _useTabClass.tdSizeClassGenerator;
  var _useDomRefCallback = hooks_useDomRefCallback["default"](),
    _useDomRefCallback2 = slicedToArray._slicedToArray(_useDomRefCallback, 2),
    navItemDom = _useDomRefCallback2[0],
    setRefCurrent = _useDomRefCallback2[1];
  hooks_useRipple["default"](disabled ? null : navItemDom);
  return /* @__PURE__ */React__default["default"].createElement("div", _objectSpread(_objectSpread({}, dragProps), {}, {
    ref: innerRef,
    onClick: disabled ? _util_noop["default"] : onClick,
    className: classNames__default["default"](tdTabsClassGenerator("nav-item"), isCard ? tdTabsClassGenerator("nav--card") : "", tdSizeClassGenerator(size), isActive ? tdClassGenerator("is-active") : "", tdClassGenerator("is-".concat(placement)), disabled ? tdClassGenerator("is-disabled") : "", props.className)
  }), isCard ? /* @__PURE__ */React__default["default"].createElement("span", {
    className: classNames__default["default"](tdTabsClassGenerator("nav-item-text-wrapper"))
  }, label) : /* @__PURE__ */React__default["default"].createElement("div", {
    ref: setRefCurrent,
    className: classNames__default["default"](tdTabsClassGenerator("nav-item-wrapper"))
  }, /* @__PURE__ */React__default["default"].createElement("span", {
    className: classNames__default["default"](tdTabsClassGenerator("nav-item-text-wrapper"))
  }, label)), removable && !disabled ? /* @__PURE__ */React__default["default"].createElement("span", {
    className: classNames__default["default"]("remove-btn"),
    onClick: function onClick(e) {
      e.stopPropagation();
      onRemove({
        value: value,
        e: e
      });
      onTabRemove({
        value: value,
        e: e,
        index: index
      });
    }
  }, /* @__PURE__ */React__default["default"].createElement(CloseIcon, null)) : null);
};

exports["default"] = TabNavItem;
