// tdesign-react/cjs/tabs/useTabClass.js
/**
 * tdesign v1.17.1
 * (c) 2026 tdesign
 * @license MIT
 */

'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var hooks_useConfig = require('../hooks/useConfig.js');
require('react');
require('../config-provider/ConfigContext.js');
require('../_chunks/dep-c446aa11.js');
require('../_chunks/dep-c290284d.js');
require('../_chunks/dep-2ee09518.js');
require('../_chunks/dep-a6fa90dc.js');
require('dayjs');
require('../_chunks/dep-012d9a7c.js');
require('../_chunks/dep-150a112c.js');
require('../_chunks/dep-e360e17f.js');
require('../_chunks/dep-3d05de97.js');
require('../_chunks/dep-4f1729f0.js');
require('../_chunks/dep-a54f202c.js');
require('../_chunks/dep-51284537.js');
require('../_chunks/dep-8a493834.js');
require('../_chunks/dep-9f5b2891.js');
require('../_chunks/dep-338f013a.js');
require('../_chunks/dep-c37c5e17.js');
require('../_chunks/dep-eee1aab9.js');
require('../_chunks/dep-11347ea1.js');
require('../_chunks/dep-a830ce1e.js');
require('../_chunks/dep-770bc0ae.js');
require('../_chunks/dep-aaf27e39.js');
require('../_chunks/dep-b47d855f.js');
require('../_chunks/dep-81bce4e6.js');
require('../_chunks/dep-863f8f4d.js');
require('../_chunks/dep-84e4d0da.js');
require('../_chunks/dep-f78183d7.js');
require('../_chunks/dep-2f45f6c4.js');

var useTabClass = function useTabClass() {
  var _useConfig = hooks_useConfig["default"](),
    classPrefix = _useConfig.classPrefix;
  var tdTabsClassPrefix = "".concat(classPrefix, "-tabs");
  var tdTabPanelClassPrefix = "".concat(classPrefix, "-tab-panel");
  var tdClassGenerator = function tdClassGenerator(append) {
    return "".concat(classPrefix, "-").concat(append);
  };
  var tdTabsClassGenerator = function tdTabsClassGenerator(append) {
    return "".concat(tdTabsClassPrefix, "__").concat(append);
  };
  var tdTabPanelClassGenerator = function tdTabPanelClassGenerator(append) {
    return "".concat(tdTabPanelClassPrefix, "__").concat(append);
  };
  var tdSizeClassGenerator = function tdSizeClassGenerator(size) {
    return "".concat(classPrefix, "-size-").concat(size === "large" ? "l" : "m");
  };
  return {
    tdTabsClassPrefix: tdTabsClassPrefix,
    tdTabPanelClassPrefix: tdTabPanelClassPrefix,
    tdClassGenerator: tdClassGenerator,
    tdTabsClassGenerator: tdTabsClassGenerator,
    tdTabPanelClassGenerator: tdTabPanelClassGenerator,
    tdSizeClassGenerator: tdSizeClassGenerator
  };
};

exports.useTabClass = useTabClass;
