// tdesign-react/lib/tree/TreeItem.js
/**
 * tdesign v1.17.1
 * (c) 2026 tdesign
 * @license MIT
 */

import { _ as _objectWithoutProperties } from '../_chunks/dep-84926167.js';
import { _ as _slicedToArray } from '../_chunks/dep-d1eeb95c.js';
import { _ as _defineProperty } from '../_chunks/dep-55b77e26.js';
import React, { forwardRef, useRef, useEffect, useState, isValidElement } from 'react';
import classNames from 'classnames';
import { isFunction } from 'lodash-es';
import { CaretRightSmallIcon } from 'tdesign-icons-react';
import { composeRefs } from '../_util/ref.js';
import { Checkbox } from '../checkbox/index.js';
import useConfig from '../hooks/useConfig.js';
import useDomRefCallback from '../hooks/useDomRefCallback.js';
import useGlobalIcon from '../hooks/useGlobalIcon.js';
import useRipple from '../hooks/useRipple.js';
import { Loading } from '../loading/index.js';
import useDraggable from './hooks/useDraggable.js';
import { useTreeConfig } from './hooks/useTreeConfig.js';
import '../_chunks/dep-673e8854.js';
import '../_chunks/dep-570a5adb.js';
import '../_chunks/dep-067cb22b.js';
import 'react-is';
import '../_util/isFragment.js';
import '../_chunks/dep-954df161.js';
import '../_util/forwardRefWithStatics.js';
import 'hoist-non-react-statics';
import '../common/Check.js';
import '../_util/helper.js';
import '../hooks/useControlled.js';
import '../_util/noop.js';
import '../config-provider/ConfigContext.js';
import '../_chunks/dep-c3e5e988.js';
import '../_chunks/dep-65b05485.js';
import 'dayjs';
import '../hooks/useDefaultProps.js';
import '../_chunks/dep-154fcb93.js';
import '../_chunks/dep-e9c0bb63.js';
import '../checkbox/defaultProps.js';
import '../_chunks/dep-e17ec8df.js';
import '../_chunks/dep-68899e89.js';
import '../hooks/useAnimation.js';
import '../loading/Loading.js';
import '../common/Portal.js';
import 'react-dom';
import '../hooks/useLayoutEffect.js';
import '../loading/defaultProps.js';
import '../loading/gradient.js';
import '../_chunks/dep-b0cbea79.js';
import '../loading/plugin.js';
import '../_util/react-render.js';
import '../_chunks/dep-e29eee31.js';
import '../common/PluginContainer.js';
import '../config-provider/ConfigProvider.js';
import '../hooks/usePersistFn.js';
import './hooks/TreeDraggableContext.js';
import '../_util/createHookContext.js';
import '../locale/LocalReceiver.js';

var _excluded = ["setData"];
function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { _defineProperty(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
var TreeItem = /*#__PURE__*/forwardRef(function (props, ref) {
  var node = props.node,
    icon = props.icon,
    label = props.label,
    line = props.line,
    expandOnClickNode = props.expandOnClickNode,
    activable = props.activable,
    checkProps = props.checkProps,
    operations = props.operations,
    onClick = props.onClick,
    _onChange = props.onChange,
    isVirtual = props.isVirtual,
    onTreeItemMounted = props.onTreeItemMounted,
    allowDrop = props.allowDrop;
  var _useGlobalIcon = useGlobalIcon({
      CaretRightSmallIcon: CaretRightSmallIcon
    }),
    CaretRightSmallIcon$1 = _useGlobalIcon.CaretRightSmallIcon;
  var level = node.level;
  var nodeRef = useRef(null);
  var _useTreeConfig = useTreeConfig(),
    treeClassNames = _useTreeConfig.treeClassNames,
    locale = _useTreeConfig.locale;
  var _useConfig = useConfig(),
    classPrefix = _useConfig.classPrefix;
  useEffect(function () {
    onTreeItemMounted === null || onTreeItemMounted === void 0 || onTreeItemMounted({
      ref: nodeRef.current,
      data: node
    });
  }, [isVirtual, nodeRef, node, onTreeItemMounted]);
  var handleClick = function handleClick(evt) {
    var _srcTarget$className, _srcTarget$className$;
    var srcTarget = evt.target;
    var isBranchTrigger = node.children && expandOnClickNode && (srcTarget.className === "".concat(classPrefix, "-checkbox__input") || srcTarget.tagName.toLowerCase() === "input");
    if (isBranchTrigger) return;
    if (expandOnClickNode && node.children && ((_srcTarget$className = srcTarget.className) === null || _srcTarget$className === void 0 || (_srcTarget$className$ = _srcTarget$className.indexOf) === null || _srcTarget$className$ === void 0 ? void 0 : _srcTarget$className$.call(_srcTarget$className, "".concat(classPrefix, "-tree__label"))) !== -1) evt.preventDefault();
    onClick === null || onClick === void 0 || onClick(node, {
      e: evt,
      expand: expandOnClickNode,
      active: activable,
      trigger: "node-click"
    });
  };
  var handleItemClick = function handleItemClick(evt) {
    if (node.loading) {
      return;
    }
    onClick === null || onClick === void 0 || onClick(node, {
      e: evt,
      expand: true,
      active: false,
      trigger: "icon-click"
    });
  };
  var handleIconClick = function handleIconClick(evt) {
    if (!icon) return;
    evt.stopPropagation();
    handleItemClick(evt);
  };
  var stopPropagation = function stopPropagation(e) {
    e.stopPropagation();
  };
  var renderIcon = function renderIcon() {
    var isDefaultIcon = false;
    var renderIconNode = function renderIconNode() {
      if (icon === false) {
        return null;
      }
      if (icon instanceof Function) {
        return icon(node.getModel());
      }
      if (/*#__PURE__*/React.isValidElement(icon)) {
        return icon;
      }
      if (icon && icon !== true) {
        throw new Error("invalid type of icon");
      }
      if (!node.isLeaf()) {
        isDefaultIcon = true;
        if (node.loading && node.expanded) {
          return /* @__PURE__ */React.createElement(Loading, {
            loading: true
          });
        }
        return /* @__PURE__ */React.createElement(CaretRightSmallIcon$1, {
          className: treeClassNames.treeIconRight
        });
      }
      return null;
    };
    var iconNode = renderIconNode();
    return /* @__PURE__ */React.createElement("span", {
      className: classNames(treeClassNames.treeIcon, treeClassNames.folderIcon, _defineProperty({}, treeClassNames.treeIconDefault, isDefaultIcon)),
      onClick: handleIconClick
    }, iconNode);
  };
  var renderLine = function renderLine() {
    var iconVisible = icon !== false;
    if (line === false) {
      return null;
    }
    if (isFunction(line)) {
      return line(node.getModel());
    }
    if (/*#__PURE__*/React.isValidElement(line)) {
      return line;
    }
    if (node.parent && node.tree) {
      var shadowStyles = [];
      var parents = node.getParents();
      parents.pop();
      parents.forEach(function (pnode, index) {
        if (!pnode.vmIsLast) {
          shadowStyles.push("calc(-".concat(index + 1, " * var(--space)) 0 var(--color)"));
        }
      });
      var styles = {
        "--level": level,
        boxShadow: shadowStyles.join(",")
      };
      return /* @__PURE__ */React.createElement("span", {
        className: classNames(treeClassNames.line, _defineProperty(_defineProperty({}, treeClassNames.lineIsLeaf, node.vmIsLeaf || !iconVisible), treeClassNames.lineIsFirst, node.vmIsFirst && iconVisible)),
        style: styles,
        onClick: stopPropagation
      });
    }
    return null;
  };
  var _useDomRefCallback = useDomRefCallback(),
    _useDomRefCallback2 = _slicedToArray(_useDomRefCallback, 2),
    labelDom = _useDomRefCallback2[0],
    setRefCurrent = _useDomRefCallback2[1];
  useRipple(labelDom);
  var _useState = useState({}),
    _useState2 = _slicedToArray(_useState, 2),
    updateRender = _useState2[1];
  var renderLabel = function renderLabel() {
    var _node$data;
    var emptyView = locale("empty");
    var labelText = "";
    if (label instanceof Function) {
      var _node$getModel = node.getModel(),
        nodeSetData = _node$getModel.setData,
        rest = _objectWithoutProperties(_node$getModel, _excluded);
      labelText = label(_objectSpread(_objectSpread({}, rest), {}, {
        setData: function setData(value) {
          nodeSetData(value);
          updateRender({});
        }
      })) || emptyView;
    } else {
      labelText = node.label || emptyView;
    }
    var labelClasses = classNames(treeClassNames.treeLabel, treeClassNames.treeLabelStrictly, _defineProperty({}, treeClassNames.actived, node.isActivable() ? node.actived : false));
    if (node.isCheckable()) {
      var checkboxProps;
      if (typeof checkProps === "function") {
        checkboxProps = checkProps(node.getModel());
      } else {
        checkboxProps = checkProps;
      }
      return /* @__PURE__ */React.createElement(Checkbox, _objectSpread({
        ref: setRefCurrent,
        checked: node.checked,
        indeterminate: node.indeterminate,
        disabled: node.disabled,
        name: String(node.value),
        onChange: function onChange(_, ctx) {
          return _onChange(node, ctx);
        },
        className: labelClasses,
        stopLabelTrigger: expandOnClickNode && !!node.children
      }, checkboxProps), /* @__PURE__ */React.createElement("span", {
        "data-target": "label"
      }, labelText));
    }
    var isCustomLabel = label instanceof Function || /*#__PURE__*/isValidElement(node.label);
    return /* @__PURE__ */React.createElement("span", {
      ref: setRefCurrent,
      "data-target": "label",
      className: labelClasses,
      title: isCustomLabel ? void 0 : String(((_node$data = node.data) === null || _node$data === void 0 ? void 0 : _node$data.text) || node.label || "")
    }, /* @__PURE__ */React.createElement("span", {
      style: {
        position: "relative"
      }
    }, labelText));
  };
  var renderOperations = function renderOperations() {
    var operationsView = null;
    if (operations) {
      if (/*#__PURE__*/React.isValidElement(operations)) {
        operationsView = operations;
      } else if (operations instanceof Function) {
        var treeNodeModel = node === null || node === void 0 ? void 0 : node.getModel();
        operationsView = operations(treeNodeModel);
      } else {
        throw new Error("invalid type of operations");
      }
    }
    if (operationsView) {
      return /* @__PURE__ */React.createElement("span", {
        className: treeClassNames.treeOperations,
        "data-target": "operations",
        onClick: stopPropagation
      }, operationsView);
    }
    return null;
  };
  var _useDraggable = useDraggable({
      node: node,
      nodeRef: nodeRef,
      allowDrop: allowDrop
    }),
    setDragStatus = _useDraggable.setDragStatus,
    isDragging = _useDraggable.isDragging,
    dropPosition = _useDraggable.dropPosition,
    isDragOver = _useDraggable.isDragOver;
  var handleDragStart = function handleDragStart(evt) {
    var node2 = props.node;
    if (!node2.isDraggable()) return;
    evt.stopPropagation();
    setDragStatus("dragStart", evt);
    try {
      var _evt$dataTransfer;
      (_evt$dataTransfer = evt.dataTransfer) === null || _evt$dataTransfer === void 0 || _evt$dataTransfer.setData("text/plain", "");
    } catch (e) {}
  };
  var handleDragEnd = function handleDragEnd(evt) {
    var node2 = props.node;
    if (!node2.isDraggable()) return;
    evt.stopPropagation();
    setDragStatus("dragEnd", evt);
  };
  var handleDragOver = function handleDragOver(evt) {
    var node2 = props.node;
    if (!node2.isDraggable()) return;
    evt.stopPropagation();
    evt.preventDefault();
    setDragStatus("dragOver", evt);
  };
  var handleDragLeave = function handleDragLeave(evt) {
    var node2 = props.node;
    if (!node2.isDraggable()) return;
    evt.stopPropagation();
    setDragStatus("dragLeave", evt);
  };
  var handleDrop = function handleDrop(evt) {
    var node2 = props.node;
    if (!node2.isDraggable()) return;
    evt.stopPropagation();
    evt.preventDefault();
    setDragStatus("drop", evt);
  };
  return /* @__PURE__ */React.createElement("div", {
    ref: composeRefs(ref, nodeRef),
    "data-value": node.value,
    "data-level": level,
    className: classNames(treeClassNames.treeNode, _defineProperty(_defineProperty(_defineProperty(_defineProperty(_defineProperty(_defineProperty(_defineProperty(_defineProperty({}, treeClassNames.treeNodeOpen, node.expanded), treeClassNames.actived, node.isActivable() ? node.actived : false), treeClassNames.disabled, node.isDisabled()), treeClassNames.treeNodeDraggable, !node.isDisabled() && node.isDraggable()), treeClassNames.treeNodeDragging, isDragging), treeClassNames.treeNodeDragTipTop, isDragOver && dropPosition < 0), treeClassNames.treeNodeDragTipBottom, isDragOver && dropPosition > 0), treeClassNames.treeNodeDragTipHighlight, !isDragging && isDragOver && dropPosition === 0)),
    style: {
      "--level": level,
      boxShadow: ""
    },
    onClick: handleClick,
    draggable: !node.isDisabled() && node.isDraggable(),
    onDragStart: handleDragStart,
    onDragEnd: handleDragEnd,
    onDragOver: handleDragOver,
    onDragLeave: handleDragLeave,
    onDrop: handleDrop
  }, renderLine(), renderIcon(), renderLabel(), renderOperations());
});
TreeItem.displayName = "TreeItem";

export { TreeItem as default };
