// tdesign-react/cjs/collapse/CollapsePanel.js
/**
 * tdesign v1.17.1
 * (c) 2026 tdesign
 * @license MIT
 */

'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var defineProperty = require('../_chunks/dep-c446aa11.js');
var React = require('react');
var reactTransitionGroup = require('react-transition-group');
var classNames = require('classnames');
var common_FakeArrow = require('../common/FakeArrow.js');
var hooks_useConfig = require('../hooks/useConfig.js');
var hooks_useDefaultProps = require('../hooks/useDefaultProps.js');
var collapse_CollapseContext = require('./CollapseContext.js');
var collapse_defaultProps = require('./defaultProps.js');
require('../_chunks/dep-c290284d.js');
require('../_chunks/dep-6e0da644.js');
require('../_chunks/dep-05bc49e6.js');
require('../_chunks/dep-7f3fa935.js');
require('../_chunks/dep-8c880a65.js');
require('../_chunks/dep-e41137f7.js');
require('../_chunks/dep-b8d0b96c.js');
require('../_chunks/dep-3d05de97.js');
require('../_chunks/dep-4f1729f0.js');
require('../_chunks/dep-427f8bc0.js');
require('../_chunks/dep-a54f202c.js');
require('../config-provider/ConfigContext.js');
require('../_chunks/dep-2ee09518.js');
require('../_chunks/dep-a6fa90dc.js');
require('dayjs');
require('../_chunks/dep-012d9a7c.js');
require('../_chunks/dep-150a112c.js');
require('../_chunks/dep-e360e17f.js');
require('../_chunks/dep-51284537.js');
require('../_chunks/dep-8a493834.js');
require('../_chunks/dep-9f5b2891.js');
require('../_chunks/dep-338f013a.js');
require('../_chunks/dep-c37c5e17.js');
require('../_chunks/dep-eee1aab9.js');
require('../_chunks/dep-11347ea1.js');
require('../_chunks/dep-a830ce1e.js');
require('../_chunks/dep-770bc0ae.js');
require('../_chunks/dep-aaf27e39.js');
require('../_chunks/dep-b47d855f.js');
require('../_chunks/dep-81bce4e6.js');
require('../_chunks/dep-863f8f4d.js');
require('../_chunks/dep-84e4d0da.js');
require('../_chunks/dep-f78183d7.js');
require('../_chunks/dep-2f45f6c4.js');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

var React__default = /*#__PURE__*/_interopDefaultLegacy(React);
var classNames__default = /*#__PURE__*/_interopDefaultLegacy(classNames);

function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { defineProperty._defineProperty(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
var CollapsePanel = function CollapsePanel(props) {
  var _useDefaultProps = hooks_useDefaultProps["default"](props, collapse_defaultProps.collapsePanelDefaultProps),
    value = _useDefaultProps.value,
    disabled = _useDefaultProps.disabled,
    destroyOnCollapse = _useDefaultProps.destroyOnCollapse,
    expandIcon = _useDefaultProps.expandIcon,
    className = _useDefaultProps.className,
    style = _useDefaultProps.style,
    header = _useDefaultProps.header,
    headerRightContent = _useDefaultProps.headerRightContent,
    children = _useDefaultProps.children,
    index = _useDefaultProps.index;
  var _useCollapseContext = collapse_CollapseContext.useCollapseContext(),
    disableAll = _useCollapseContext.disabled,
    defaultExpandAll = _useCollapseContext.defaultExpandAll,
    expandIconPlacement = _useCollapseContext.expandIconPlacement,
    expandOnRowClick = _useCollapseContext.expandOnRowClick,
    expandIconAll = _useCollapseContext.expandIcon,
    updateCollapseValue = _useCollapseContext.updateCollapseValue,
    collapseValue = _useCollapseContext.collapseValue;
  var _useConfig = hooks_useConfig["default"](),
    classPrefix = _useConfig.classPrefix;
  var componentName = "".concat(classPrefix, "-collapse-panel");
  var innerValue = value || index;
  var finalExpandIcon = (expandIcon === void 0 ? expandIconAll : expandIcon) || null;
  var headRef = React.useRef(null);
  var iconRef = React.useRef(null);
  var contentRef = React.useRef(null);
  var bodyRef = React.useRef(null);
  var isDisabled = disabled || !!disableAll;
  React.useEffect(function () {
    if (defaultExpandAll) {
      updateCollapseValue(innerValue);
    }
  }, []);
  var isActive = Array.isArray(collapseValue) ? collapseValue.includes(innerValue) : collapseValue === innerValue;
  var classes = classNames__default["default"](componentName, defineProperty._defineProperty({}, "".concat(classPrefix, "-is-disabled"), isDisabled), className);
  var handleClick = function handleClick(e) {
    var canExpand = expandOnRowClick && e.currentTarget === headRef.current || e.currentTarget === iconRef.current;
    if (canExpand && !isDisabled) {
      updateCollapseValue(innerValue, {
        e: e
      });
    }
    e.stopPropagation();
  };
  var renderIcon = function renderIcon() {
    var iconNode = null;
    if (/*#__PURE__*/React__default["default"].isValidElement(finalExpandIcon)) {
      iconNode = finalExpandIcon;
    } else if (finalExpandIcon) {
      iconNode = /* @__PURE__ */React__default["default"].createElement(common_FakeArrow["default"], {
        className: classNames__default["default"]("".concat(componentName, "__icon--default"))
      });
    }
    return iconNode && /* @__PURE__ */React__default["default"].createElement("div", {
      className: "".concat(componentName, "__icon ").concat(componentName, "__icon--").concat(expandIconPlacement, " ").concat(isActive ? "".concat(componentName, "__icon--active") : ""),
      ref: iconRef,
      onClick: handleClick
    }, iconNode);
  };
  var renderHeader = function renderHeader() {
    var cls = ["".concat(componentName, "__header"), defineProperty._defineProperty({}, "".concat(classPrefix, "-is-clickable"), expandOnRowClick && !isDisabled)];
    return /* @__PURE__ */React__default["default"].createElement("div", {
      ref: headRef,
      className: classNames__default["default"](cls),
      onClick: handleClick
    }, expandIconPlacement === "left" && renderIcon(), header, /* @__PURE__ */React__default["default"].createElement("div", {
      className: "".concat(componentName, "__header--blank")
    }), headerRightContent, expandIconPlacement === "right" && renderIcon());
  };
  var renderBody = function renderBody() {
    var transitionCallbacks = {
      onEnter: function onEnter() {
        bodyRef.current.style.height = "".concat(contentRef === null || contentRef === void 0 ? void 0 : contentRef.current.clientHeight, "px");
      },
      onEntered: function onEntered() {
        bodyRef.current.style.height = "auto";
      },
      onExit: function onExit() {
        bodyRef.current.style.height = "".concat(contentRef === null || contentRef === void 0 ? void 0 : contentRef.current.clientHeight, "px");
      },
      onExiting: function onExiting() {
        bodyRef.current.style.height = "0px";
      }
    };
    return /* @__PURE__ */React__default["default"].createElement(reactTransitionGroup.CSSTransition, _objectSpread({
      "in": isActive,
      appear: true,
      timeout: 300,
      nodeRef: bodyRef,
      unmountOnExit: destroyOnCollapse
    }, transitionCallbacks), /* @__PURE__ */React__default["default"].createElement("div", {
      style: {
        height: 0
      },
      className: classNames__default["default"]("".concat(componentName, "__body"), "".concat(classPrefix, "-slide-down-enter-active"), defineProperty._defineProperty({}, "".concat(componentName, "__body--collapsed"), !isActive)),
      ref: bodyRef
    }, /* @__PURE__ */React__default["default"].createElement("div", {
      className: "".concat(componentName, "__content"),
      ref: contentRef
    }, children)));
  };
  return /* @__PURE__ */React__default["default"].createElement("div", {
    className: classes,
    style: _objectSpread({}, style)
  }, /* @__PURE__ */React__default["default"].createElement("div", {
    className: "".concat(componentName, "__wrapper")
  }, renderHeader(), renderBody()));
};
CollapsePanel.displayName = "CollapsePanel";

exports["default"] = CollapsePanel;
