// tdesign-react/esm/tag-input/useTagList.js
/**
 * tdesign v1.17.1
 * (c) 2026 tdesign
 * @license MIT
 */

import { _ as _defineProperty } from '../_chunks/dep-55b77e26.js';
import { _ as _toConsumableArray } from '../_chunks/dep-154fcb93.js';
import { _ as _typeof } from '../_chunks/dep-067cb22b.js';
import { _ as _slicedToArray } from '../_chunks/dep-d1eeb95c.js';
import React, { useState, Fragment } from 'react';
import { isFunction } from 'lodash-es';
import '../_chunks/dep-0f787ecd.js';
import useConfig from '../hooks/useConfig.js';
import useControlled from '../hooks/useControlled.js';
import { Tag } from '../tag/index.js';
import { l as log } from '../_chunks/dep-65b369a5.js';
import '../_chunks/dep-570a5adb.js';
import '../_chunks/dep-e9c0bb63.js';
import '../_chunks/dep-673e8854.js';
import '../config-provider/ConfigContext.js';
import '../locale/zh_CN.js';
import '../_chunks/dep-e56ffacc.js';
import 'dayjs';
import '../_chunks/dep-84036ea0.js';
import '../_util/noop.js';
import '../tag/style/index.js';
import '../_util/forwardRefWithStatics.js';
import 'hoist-non-react-statics';
import '../tag/CheckTag.js';
import '../_chunks/dep-84926167.js';
import 'classnames';
import '../_chunks/dep-99c21f72.js';
import '../hooks/useDefaultProps.js';
import '../tag/defaultProps.js';
import '../tag/Tag.js';
import 'tdesign-icons-react';
import '../_chunks/dep-0c178c7f.js';
import '../hooks/useGlobalIcon.js';
import '../tag/CheckTagGroup.js';
import '../tag/type.js';

function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { _defineProperty(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
function useTagList(props) {
  var _useConfig = useConfig(),
    prefix = _useConfig.classPrefix;
  var onRemove = props.onRemove,
    max = props.max,
    minCollapsedNum = props.minCollapsedNum,
    size = props.size,
    disabled = props.disabled,
    tagProps = props.tagProps,
    tag = props.tag,
    collapsedItems = props.collapsedItems,
    getDragProps = props.getDragProps;
  var readOnly = props.readOnly || props.readonly;
  var _useControlled = useControlled(props, "value", props.onChange),
    _useControlled2 = _slicedToArray(_useControlled, 2),
    innerTagValue = _useControlled2[0],
    setTagValue = _useControlled2[1];
  var tagValue;
  if (Array.isArray(innerTagValue)) {
    tagValue = innerTagValue;
  } else {
    if (innerTagValue !== void 0 && innerTagValue !== null) {
      log.warnOnce("TagInput", "`value` is expected to be an array, but got ".concat(_typeof(innerTagValue), ". It will be treated as an empty array."));
    }
    tagValue = [];
  }
  var _useState = useState(),
    _useState2 = _slicedToArray(_useState, 2),
    oldInputValue = _useState2[0],
    setOldInputValue = _useState2[1];
  var onClose = function onClose(p) {
    var arr = _toConsumableArray(tagValue);
    var _arr$splice = arr.splice(p.index, 1),
      _arr$splice2 = _slicedToArray(_arr$splice, 1),
      item = _arr$splice2[0];
    setTagValue(arr, _objectSpread(_objectSpread({
      trigger: "tag-remove"
    }, p), {}, {
      item: item
    }));
    onRemove === null || onRemove === void 0 || onRemove(_objectSpread(_objectSpread({}, p), {}, {
      item: item,
      trigger: "tag-remove",
      value: arr
    }));
  };
  var clearAll = function clearAll(context) {
    setTagValue([], {
      trigger: "clear",
      e: context.e
    });
  };
  var onInnerEnter = function onInnerEnter(value, context) {
    var _tagValue, _props$onEnter;
    var valueStr = value ? String(value).trim() : "";
    var newValue = tagValue;
    var isLimitExceeded = max && ((_tagValue = tagValue) === null || _tagValue === void 0 ? void 0 : _tagValue.length) >= max;
    if (valueStr && !isLimitExceeded) {
      newValue = tagValue instanceof Array ? tagValue.concat(String(valueStr)) : [valueStr];
      setTagValue(newValue, {
        trigger: "enter",
        index: newValue.length - 1,
        item: valueStr,
        e: context.e
      });
    }
    props === null || props === void 0 || (_props$onEnter = props.onEnter) === null || _props$onEnter === void 0 || _props$onEnter.call(props, newValue, _objectSpread(_objectSpread({}, context), {}, {
      inputValue: value
    }));
  };
  var onInputBackspaceKeyUp = function onInputBackspaceKeyUp(value) {
    if (!tagValue || !tagValue.length) return;
    setOldInputValue(value);
  };
  var onInputBackspaceKeyDown = function onInputBackspaceKeyDown(value, context) {
    var e = context.e;
    if (!tagValue || !tagValue.length || readOnly) return;
    if (!value && ["Backspace", "NumpadDelete"].includes(e.key)) {
      var index = tagValue.length - 1;
      var item = tagValue[index];
      var trigger = "backspace";
      var newValue = tagValue.slice(0, -1);
      setTagValue(newValue, {
        e: e,
        index: index,
        item: item,
        trigger: trigger
      });
      onRemove === null || onRemove === void 0 || onRemove({
        e: e,
        index: index,
        item: item,
        trigger: trigger,
        value: newValue
      });
    }
  };
  var renderLabel = function renderLabel(_ref) {
    var displayNode = _ref.displayNode,
      label = _ref.label;
    var newList = minCollapsedNum ? tagValue.slice(0, minCollapsedNum) : tagValue;
    var list = displayNode ? [/* @__PURE__ */React.createElement(Fragment, {
      key: "display-node"
    }, displayNode)] : newList === null || newList === void 0 ? void 0 : newList.map(function (item, index) {
      var tagContent = isFunction(tag) ? tag({
        value: item
      }) : tag;
      var handleClose = function handleClose(context) {
        var _tagProps$onClose;
        tagProps === null || tagProps === void 0 || (_tagProps$onClose = tagProps.onClose) === null || _tagProps$onClose === void 0 || _tagProps$onClose.call(tagProps, context);
        onClose({
          e: context === null || context === void 0 ? void 0 : context.e,
          index: index
        });
      };
      return /* @__PURE__ */React.createElement(Tag, _objectSpread(_objectSpread(_objectSpread({
        key: index,
        size: size,
        disabled: disabled,
        closable: !readOnly && !disabled
      }, getDragProps === null || getDragProps === void 0 ? void 0 : getDragProps(index, item)), tagProps), {}, {
        onClose: handleClose
      }), tagContent !== null && tagContent !== void 0 ? tagContent : item);
    });
    if (label) {
      list === null || list === void 0 || list.unshift(/* @__PURE__ */React.createElement("div", {
        className: "".concat(prefix, "-tag-input__prefix"),
        key: "label"
      }, label));
    }
    if (newList.length !== tagValue.length) {
      var len = tagValue.length - newList.length;
      var options = Array.isArray(props === null || props === void 0 ? void 0 : props.options) ? props.options : tagValue;
      var params = {
        value: tagValue,
        count: tagValue.length - minCollapsedNum,
        collapsedTags: tagValue.slice(minCollapsedNum, tagValue.length),
        collapsedSelectedItems: options.slice(minCollapsedNum, tagValue.length),
        onClose: onClose
      };
      var more = isFunction(collapsedItems) ? collapsedItems(params) : collapsedItems;
      list.push(/* @__PURE__ */React.createElement(Fragment, {
        key: "more"
      }, more !== null && more !== void 0 ? more : /* @__PURE__ */React.createElement(Tag, _objectSpread({
        size: size
      }, tagProps), "+", len)));
    }
    return list;
  };
  return {
    tagValue: tagValue,
    clearAll: clearAll,
    onClose: onClose,
    onInnerEnter: onInnerEnter,
    onInputBackspaceKeyDown: onInputBackspaceKeyDown,
    onInputBackspaceKeyUp: onInputBackspaceKeyUp,
    renderLabel: renderLabel
  };
}

export { useTagList as default };
