// tdesign-react/lib/table/EnhancedTable.js
/**
 * tdesign v1.17.1
 * (c) 2026 tdesign
 * @license MIT
 */

import { _ as _defineProperty } from '../_chunks/dep-55b77e26.js';
import { _ as _objectWithoutProperties } from '../_chunks/dep-84926167.js';
import React, { forwardRef, useRef, useImperativeHandle } from 'react';
import { get } from 'lodash-es';
import useConfig from '../hooks/useConfig.js';
import useClassName from './hooks/useClassName.js';
import { DATA_PARENT_ID_ATTR } from './hooks/useDragSort.js';
import useTreeData from './hooks/useTreeData.js';
import useTreeSelect from './hooks/useTreeSelect.js';
import PrimaryTable from './PrimaryTable.js';
import { enableRowDrag } from './utils.js';
import '../_chunks/dep-067cb22b.js';
import '../config-provider/ConfigContext.js';
import '../_chunks/dep-c3e5e988.js';
import '../_chunks/dep-65b05485.js';
import 'dayjs';
import '../_chunks/dep-154fcb93.js';
import '../_chunks/dep-570a5adb.js';
import '../_chunks/dep-e9c0bb63.js';
import '../_chunks/dep-d1eeb95c.js';
import '../_chunks/dep-673e8854.js';
import 'sortablejs';
import '../_chunks/dep-65b369a5.js';
import '../_chunks/dep-69f8347e.js';
import '../_chunks/dep-68899e89.js';
import '../hooks/useLatest.js';
import 'classnames';
import 'tdesign-icons-react';
import '../_chunks/dep-e13fcc68.js';
import '../_chunks/dep-c4dcddfa.js';
import '../hooks/useControlled.js';
import '../_util/noop.js';
import '../hooks/usePrevious.js';
import '../_util/parseTNode.js';
import '../hooks/useGlobalIcon.js';
import '../locale/LocalReceiver.js';
import '../config-provider/ConfigProvider.js';
import './Cell.js';
import './Ellipsis.js';
import '../_util/scroll.js';
import 'raf';
import '../_util/easing.js';
import '../hooks/useDebounce.js';
import '../hooks/usePersistFn.js';
import '../tooltip/index.js';
import '../tooltip/Tooltip.js';
import '../hooks/useDefaultProps.js';
import '../popup/index.js';
import '../popup/Popup.js';
import 'react-transition-group';
import '../common/Portal.js';
import 'react-dom';
import '../hooks/useLayoutEffect.js';
import '../hooks/useAnimation.js';
import '../hooks/useAttach.js';
import '../hooks/useMutationObserver.js';
import '../hooks/usePopper.js';
import 'react-fast-compare';
import '@popperjs/core';
import '../hooks/useWindowSize.js';
import '../popup/defaultProps.js';
import '../popup/hooks/useTrigger.js';
import '../_util/listener.js';
import '../_util/ref.js';
import 'react-is';
import '../_util/isFragment.js';
import '../popup/utils/transition.js';
import '../popup/PopupPlugin.js';
import '../_chunks/dep-e29eee31.js';
import '../_util/react-render.js';
import '../common/PluginContainer.js';
import '../tooltip/defaultProps.js';
import '../tooltip/TooltipLite.js';
import '../hooks/useSwitch.js';
import './hooks/useFixed.js';
import '../_chunks/dep-8a383121.js';
import '../_chunks/dep-b0cbea79.js';
import '../hooks/useDeepEffect.js';
import './BaseTable.js';
import '../affix/index.js';
import '../affix/Affix.js';
import '../affix/defaultProps.js';
import '../hooks/useDomRefMount.js';
import '../hooks/useElementLazyRender.js';
import '../_chunks/dep-5ca9c80d.js';
import '../hooks/useVirtualScroll.js';
import '../loading/index.js';
import '../loading/Loading.js';
import '../loading/defaultProps.js';
import '../loading/gradient.js';
import '../_chunks/dep-e17ec8df.js';
import '../hooks/useDomRefCallback.js';
import '../loading/plugin.js';
import './defaultProps.js';
import './hooks/useAffix.js';
import './hooks/useColumnResize.js';
import './hooks/usePagination.js';
import '../pagination/index.js';
import '../pagination/Pagination.js';
import '../input-adornment/InputAdornment.js';
import '../input-number/index.js';
import '../input-number/InputNumber.js';
import '../button/index.js';
import '../button/Button.js';
import '../hooks/useRipple.js';
import '../button/defaultProps.js';
import '../input/index.js';
import '../input/Input.js';
import '../_util/forwardRefWithStatics.js';
import 'hoist-non-react-statics';
import '../input/defaultProps.js';
import '../input/InputGroup.js';
import '../input/useLengthLimit.js';
import '../input-number/defaultProps.js';
import '../input-number/useInputNumber.js';
import '../_chunks/dep-f26ab88c.js';
import '../hooks/useCommonClassName.js';
import '../select/index.js';
import '../select/base/Select.js';
import '../_util/helper.js';
import '../common/FakeArrow.js';
import '../select-input/index.js';
import '../select-input/SelectInput.js';
import '../select-input/defaultProps.js';
import '../select-input/useMultiple.js';
import '../tag-input/index.js';
import '../tag-input/TagInput.js';
import '../hooks/useDragSorter.js';
import '../hooks/useEventCallback.js';
import '../tag-input/defaultProps.js';
import '../tag-input/useHover.js';
import '../tag-input/useTagList.js';
import '../tag/index.js';
import '../tag/CheckTag.js';
import '../_chunks/dep-8bc822ae.js';
import '../tag/defaultProps.js';
import '../tag/Tag.js';
import '../_chunks/dep-0c178c7f.js';
import '../tag/CheckTagGroup.js';
import '../tag-input/useTagScroll.js';
import '../select-input/useOverlayInnerStyle.js';
import '../hooks/useInnerPopupVisible.js';
import '../select-input/useSingle.js';
import '../select/defaultProps.js';
import '../select/hooks/useKeyboardControl.js';
import '../_chunks/dep-0210405c.js';
import '../select/base/OptionGroup.js';
import '../select/base/PopupContent.js';
import '../select/hooks/usePanelVirtualScroll.js';
import '../pagination/defaultProps.js';
import '../pagination/hooks/useBoundaryJumper.js';
import '../pagination/hooks/usePageNumber.js';
import '../pagination/hooks/usePrevNextJumper.js';
import '../pagination/hooks/useTotal.js';
import '../pagination/validators.js';
import '../pagination/PaginationMini.js';
import './hooks/useStyle.js';
import './hooks/useTableHeader.js';
import './hooks/useMultiHeader.js';
import './TBody.js';
import './hooks/useRowspanAndColspan.js';
import './TR.js';
import './hooks/useLazyLoad.js';
import './TFoot.js';
import './THead.js';
import './EditableCell.js';
import '../_chunks/dep-c3dd7720.js';
import '../form/formModel.js';
import './hooks/useAsyncLoading.js';
import './hooks/useColumnController.js';
import '../checkbox/index.js';
import '../_chunks/dep-954df161.js';
import '../common/Check.js';
import '../checkbox/defaultProps.js';
import '../dialog/plugin.js';
import '../dialog/Dialog.js';
import '../hooks/useSetState.js';
import '../dialog/defaultProps.js';
import '../dialog/DialogCard.js';
import '../dialog/hooks/useDialogDrag.js';
import '../hooks/useMouseEvent.js';
import '../dialog/hooks/useDialogEsc.js';
import '../dialog/hooks/useDialogPosition.js';
import '../dialog/hooks/useLockStyle.js';
import './hooks/useEditableRow.js';
import './hooks/useFilter.js';
import './FilterController.js';
import '../radio/index.js';
import '../_chunks/dep-7d81069d.js';
import '../radio/defaultProps.js';
import '../radio/useKeyboard.js';
import './hooks/useRowExpand.js';
import './hooks/useRowSelect.js';
import './hooks/useSorter.js';
import './SorterButton.js';

var _excluded = ["store", "dataSource", "formatTreeColumn", "swapData", "onExpandFoldIconClick"];
function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { _defineProperty(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
var EnhancedTable = /*#__PURE__*/forwardRef(function (props, ref) {
  var _props$tree2;
  var tree = props.tree,
    columns = props.columns,
    style = props.style,
    className = props.className;
  var _useConfig = useConfig(),
    classPrefix = _useConfig.classPrefix;
  var _useClassName = useClassName(),
    tableExpandClasses = _useClassName.tableExpandClasses;
  var _useTreeData = useTreeData(props),
    store = _useTreeData.store,
    dataSource = _useTreeData.dataSource,
    formatTreeColumn = _useTreeData.formatTreeColumn,
    swapData = _useTreeData.swapData,
    onExpandFoldIconClick = _useTreeData.onExpandFoldIconClick,
    treeInstanceFunctions = _objectWithoutProperties(_useTreeData, _excluded);
  var treeDataMap = store === null || store === void 0 ? void 0 : store.treeDataMap;
  var _useTreeSelect = useTreeSelect(props, treeDataMap),
    tIndeterminateSelectedRowKeys = _useTreeSelect.tIndeterminateSelectedRowKeys,
    onInnerSelectChange = _useTreeSelect.onInnerSelectChange;
  var primaryTableRef = useRef(null);
  var _getColumns = function getColumns(columns2) {
    var arr = [];
    for (var i = 0, len = columns2.length; i < len; i++) {
      var _item$children, _item$children2;
      var item = _objectSpread({}, columns2[i]);
      item = formatTreeColumn(item);
      if ((_item$children = item.children) !== null && _item$children !== void 0 && _item$children.length) {
        item.children = _getColumns(item.children);
      }
      if (!item.children || (_item$children2 = item.children) !== null && _item$children2 !== void 0 && _item$children2.length) {
        arr.push(item);
      }
    }
    return arr;
  };
  var tColumns = function () {
    var isTreeData = !tree || !Object.keys(tree).length;
    return isTreeData ? columns : _getColumns(columns);
  }();
  var onEnhancedTableRowClick = function onEnhancedTableRowClick(p) {
    var _props$tree, _props$onRowClick;
    if ((_props$tree = props.tree) !== null && _props$tree !== void 0 && _props$tree.expandTreeNodeOnClick) {
      onExpandFoldIconClick({
        row: p.row,
        rowIndex: p.index
      }, "row-click");
    }
    (_props$onRowClick = props.onRowClick) === null || _props$onRowClick === void 0 || _props$onRowClick.call(props, p);
  };
  useImperativeHandle(ref, function () {
    return _objectSpread(_objectSpread({
      treeDataMap: treeDataMap
    }, treeInstanceFunctions), primaryTableRef.current);
  });
  var onDragSortChange = function onDragSortChange(params) {
    var _props$onDragSort;
    if (props.beforeDragSort && !props.beforeDragSort(params)) return;
    swapData({
      data: params.data,
      current: params.current,
      target: params.target,
      currentIndex: params.currentIndex,
      targetIndex: params.targetIndex
    });
    (_props$onDragSort = props.onDragSort) === null || _props$onDragSort === void 0 || _props$onDragSort.call(props, params);
  };
  var primaryTableProps = _objectSpread(_objectSpread(_objectSpread({}, props), {}, {
    data: dataSource,
    columns: tColumns,
    indeterminateSelectedRowKeys: tIndeterminateSelectedRowKeys,
    disableDataPage: Boolean(tree && Object.keys(tree).length) || props.disableDataPage,
    onSelectChange: onInnerSelectChange,
    onDragSort: onDragSortChange,
    rowClassName: function rowClassName(_ref) {
      var row = _ref.row;
      var rowValue = get(row, props.rowKey || "id");
      var rowState = treeDataMap.get(rowValue);
      if (!rowState) return [props.rowClassName];
      var classNames = [props.rowClassName, "".concat(classPrefix, "-table-tr--level-").concat(rowState.level)];
      if (rowState.expanded) {
        classNames.push(tableExpandClasses.expanded);
      } else {
        classNames.push(tableExpandClasses.collapsed);
      }
      return classNames;
    }
  }, enableRowDrag(props.dragSort) && {
    rowAttributes: function rowAttributes(params) {
      var row = params.row;
      var rowValue = get(row, props.rowKey);
      var rowState = treeDataMap.get(rowValue);
      var originalAttrs = typeof props.rowAttributes === "function" ? props.rowAttributes(params) : props.rowAttributes || {};
      var formatAttrs = !Array.isArray(originalAttrs) ? _objectSpread({}, originalAttrs) : {};
      if (rowState !== null && rowState !== void 0 && rowState.parent) {
        formatAttrs[DATA_PARENT_ID_ATTR] = String(rowState.parent.id);
      }
      return formatAttrs;
    }
  }), {}, {
    style: style,
    className: className
  });
  if ((_props$tree2 = props.tree) !== null && _props$tree2 !== void 0 && _props$tree2.expandTreeNodeOnClick) {
    primaryTableProps.onRowClick = onEnhancedTableRowClick;
  }
  return /* @__PURE__ */React.createElement(PrimaryTable, _objectSpread(_objectSpread({}, primaryTableProps), {}, {
    ref: primaryTableRef
  }));
});
EnhancedTable.displayName = "EnhancedTable";

export { EnhancedTable as default };
