// tdesign-react/esm/loading/plugin.js
/**
 * tdesign v1.17.1
 * (c) 2026 tdesign
 * @license MIT
 */

import { _ as _defineProperty } from '../_chunks/dep-55b77e26.js';
import React from 'react';
import { render, unmount } from '../_util/react-render.js';
import PluginContainer from '../common/PluginContainer.js';
import '../config-provider/index.js';
import Loading from './Loading.js';
import ConfigProvider from '../config-provider/ConfigProvider.js';
import '../_chunks/dep-067cb22b.js';
import '../_chunks/dep-55f620a2.js';
import '../_chunks/dep-e56ffacc.js';
import 'react-dom';
import '../config-provider/ConfigContext.js';
import 'lodash-es';
import '../locale/zh_CN.js';
import 'dayjs';
import '../_chunks/dep-84036ea0.js';
import '../config-provider/type.js';
import '../_chunks/dep-d1eeb95c.js';
import '../_chunks/dep-673e8854.js';
import '../_chunks/dep-570a5adb.js';
import 'classnames';
import '../_chunks/dep-8c226881.js';
import '../common/Portal.js';
import '../hooks/useConfig.js';
import '../hooks/useLayoutEffect.js';
import '../hooks/useDefaultProps.js';
import './defaultProps.js';
import './gradient.js';
import '../_chunks/dep-e17ec8df.js';
import '../_chunks/dep-89bc2167.js';
import '../_chunks/dep-154fcb93.js';
import '../_chunks/dep-e9c0bb63.js';
import '../_chunks/dep-84926167.js';
import '../hooks/useDomRefCallback.js';

function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { _defineProperty(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
function createContainer(attach) {
  if (typeof attach === "string") return document.querySelector(attach);
  if (typeof attach === "function") return attach();
  return document.body;
}
var LoadingPlugin = function LoadingPlugin(options) {
  if (options === false) return {
    hide: function hide() {
      return null;
    }
  };
  var props = typeof options === "boolean" ? {} : options;
  var attach = props.attach;
  var container = createContainer(attach);
  var div = document.createElement("div");
  div.setAttribute("style", "width: 100%; height: 100%; position: absolute; top: 0;");
  var defaultProps = {
    loading: true,
    attach: null,
    fullscreen: !attach,
    showOverlay: !!attach
  };
  var dGlobalConfig = ConfigProvider.getGlobalConfig();
  render(/* @__PURE__ */React.createElement(PluginContainer, {
    globalConfig: dGlobalConfig
  }, /* @__PURE__ */React.createElement(Loading, _objectSpread(_objectSpread(_objectSpread({}, defaultProps), props), {}, {
    attach: null
  }))), div);
  container.appendChild(div);
  return {
    hide: function hide() {
      unmount(div);
      div.remove();
    }
  };
};

export { LoadingPlugin };
