// tdesign-react/cjs/image-viewer/hooks/usePosition.js
/**
 * tdesign v1.17.1
 * (c) 2026 tdesign
 * @license MIT
 */

'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var slicedToArray = require('../../_chunks/dep-c60a3e4c.js');
var React = require('react');
var hooks_useMouseEvent = require('../../hooks/useMouseEvent.js');
require('../../_chunks/dep-98e6819c.js');
require('../../_chunks/dep-7f3fa935.js');
require('../../hooks/useLatest.js');

var usePosition = function usePosition(imgRef, options) {
  var _ref = options || {},
    _ref$initPosition = _ref.initPosition,
    initPosition = _ref$initPosition === void 0 ? [0, 0] : _ref$initPosition;
  var _useState = React.useState(initPosition),
    _useState2 = slicedToArray._slicedToArray(_useState, 2),
    position = _useState2[0],
    setPosition = _useState2[1];
  var _useState3 = React.useState(false),
    _useState4 = slicedToArray._slicedToArray(_useState3, 2),
    isDragging = _useState4[0],
    setIsDragging = _useState4[1];
  var lastScreenPositionRef = React.useRef(null);
  var positionRef = React.useRef(position);
  positionRef.current = position;
  var isDraggingRef = React.useRef(isDragging);
  isDraggingRef.current = isDragging;
  hooks_useMouseEvent["default"](imgRef, {
    onDown: function onDown(e) {
      var screenX = e.screenX,
        screenY = e.screenY;
      lastScreenPositionRef.current = {
        x: screenX,
        y: screenY
      };
      setIsDragging(true);
    },
    onMove: function onMove(e) {
      if (!lastScreenPositionRef.current) return;
      var screenX = e.screenX,
        screenY = e.screenY;
      var _lastScreenPositionRe = lastScreenPositionRef.current,
        lastX = _lastScreenPositionRe.x,
        lastY = _lastScreenPositionRe.y;
      setPosition(function (_ref2) {
        var _ref3 = slicedToArray._slicedToArray(_ref2, 2),
          x = _ref3[0],
          y = _ref3[1];
        return [x + screenX - lastX, y + screenY - lastY];
      });
      lastScreenPositionRef.current = {
        x: screenX,
        y: screenY
      };
    },
    onUp: function onUp() {
      lastScreenPositionRef.current = null;
      setIsDragging(false);
    }
  });
  var resetPosition = React.useCallback(function () {
    setPosition(initPosition);
  }, [initPosition]);
  return {
    position: position,
    positionRef: positionRef,
    setPosition: setPosition,
    resetPosition: resetPosition,
    isDragging: isDragging,
    isDraggingRef: isDraggingRef
  };
};

exports["default"] = usePosition;
