// tdesign-react/cjs/image-viewer/hooks/useScale.js
/**
 * tdesign v1.17.1
 * (c) 2026 tdesign
 * @license MIT
 */

'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var slicedToArray = require('../../_chunks/dep-c60a3e4c.js');
var defineProperty = require('../../_chunks/dep-c446aa11.js');
var React = require('react');
var transform = require('../../_chunks/dep-98b1e4fa.js');
require('../../_chunks/dep-98e6819c.js');
require('../../_chunks/dep-7f3fa935.js');
require('../../_chunks/dep-c290284d.js');
require('../../_chunks/dep-f7c108a2.js');
require('../../_chunks/dep-3c628d18.js');
require('../../_chunks/dep-a830ce1e.js');
require('../../_chunks/dep-427f8bc0.js');
require('../../_chunks/dep-3d05de97.js');
require('../../_chunks/dep-a54f202c.js');
require('../../_chunks/dep-4f1729f0.js');
require('../../_chunks/dep-b8d0b96c.js');
require('../../_chunks/dep-bfa39549.js');

function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { defineProperty._defineProperty(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
var useScale = function useScale(imageScale, visible, onWheel) {
  var _DEFAULT_IMAGE_SCALE$ = _objectSpread(_objectSpread({}, transform.DEFAULT_IMAGE_SCALE), imageScale),
    max = _DEFAULT_IMAGE_SCALE$.max,
    min = _DEFAULT_IMAGE_SCALE$.min,
    step = _DEFAULT_IMAGE_SCALE$.step,
    defaultScale = _DEFAULT_IMAGE_SCALE$.defaultScale;
  var calcDefaultScale = React.useCallback(function () {
    return transform.clampScale(defaultScale, min, max);
  }, [defaultScale, max, min]);
  var distance = React.useRef(0);
  var _useState = React.useState(calcDefaultScale()),
    _useState2 = slicedToArray._slicedToArray(_useState, 2),
    scale = _useState2[0],
    setScale = _useState2[1];
  var scaleRef = React.useRef(scale);
  var paramsRef = React.useRef({
    step: step,
    min: min,
    max: max
  });
  paramsRef.current = {
    step: step,
    min: min,
    max: max
  };
  var onZoomIn = React.useCallback(function (zoomOptions) {
    var _paramsRef$current = paramsRef.current,
      s = _paramsRef$current.step,
      mi = _paramsRef$current.min,
      ma = _paramsRef$current.max;
    var _zoomIn = transform.zoomIn(scaleRef.current, s, mi, ma, zoomOptions),
      newScale = _zoomIn.newScale,
      zoomResult = _zoomIn.zoomResult;
    if (newScale === scaleRef.current) return {};
    scaleRef.current = newScale;
    setScale(newScale);
    return zoomResult;
  }, []);
  var onZoomOut = React.useCallback(function (zoomOptions) {
    var _paramsRef$current2 = paramsRef.current,
      s = _paramsRef$current2.step,
      mi = _paramsRef$current2.min,
      ma = _paramsRef$current2.max;
    var _zoomOut = transform.zoomOut(scaleRef.current, s, mi, ma, zoomOptions),
      newScale = _zoomOut.newScale,
      zoomResult = _zoomOut.zoomResult;
    if (newScale === scaleRef.current) return {};
    scaleRef.current = newScale;
    setScale(newScale);
    return zoomResult;
  }, []);
  var onResetScale = React.useCallback(function () {
    var defaultVal = calcDefaultScale();
    scaleRef.current = defaultVal;
    setScale(defaultVal);
  }, [calcDefaultScale]);
  var onTouchStart = React.useCallback(function (e) {
    if (e.touches.length !== 2) return;
    e.preventDefault();
    var _Array$from = Array.from(e.touches),
      _Array$from2 = slicedToArray._slicedToArray(_Array$from, 2),
      touch1 = _Array$from2[0],
      touch2 = _Array$from2[1];
    distance.current = Math.hypot(touch2.pageX - touch1.pageX, touch2.pageY - touch1.pageY);
  }, []);
  var onTouchMove = React.useCallback(function (e) {
    if (e.touches.length !== 2) return;
    e.preventDefault();
    var _Array$from3 = Array.from(e.touches),
      _Array$from4 = slicedToArray._slicedToArray(_Array$from3, 2),
      touch1 = _Array$from4[0],
      touch2 = _Array$from4[1];
    var currentDistance = Math.hypot(touch2.pageX - touch1.pageX, touch2.pageY - touch1.pageY);
    if (currentDistance > distance.current) {
      onZoomIn();
    } else {
      onZoomOut();
    }
    distance.current = currentDistance;
  }, [onZoomIn, onZoomOut]);
  var onTouchEnd = React.useCallback(function () {
    distance.current = 0;
  }, []);
  var onWheelRef = React.useRef(onWheel);
  onWheelRef.current = onWheel;
  var stableOnWheel = React.useCallback(function (e) {
    var _onWheelRef$current;
    (_onWheelRef$current = onWheelRef.current) === null || _onWheelRef$current === void 0 || _onWheelRef$current.call(onWheelRef, e);
  }, []);
  React.useEffect(function () {
    if (!visible) return;
    document.addEventListener("wheel", stableOnWheel, {
      passive: false
    });
    document.addEventListener("touchstart", onTouchStart, {
      passive: false
    });
    document.addEventListener("touchmove", onTouchMove, {
      passive: false
    });
    document.addEventListener("touchend", onTouchEnd);
    return function () {
      document.removeEventListener("wheel", stableOnWheel);
      document.removeEventListener("touchstart", onTouchStart);
      document.removeEventListener("touchmove", onTouchMove);
      document.removeEventListener("touchend", onTouchEnd);
    };
  }, [visible, stableOnWheel, onTouchStart, onTouchMove, onTouchEnd]);
  return {
    scale: scale,
    onZoomIn: onZoomIn,
    onZoomOut: onZoomOut,
    onResetScale: onResetScale,
    onTouchStart: onTouchStart,
    onTouchMove: onTouchMove,
    onTouchEnd: onTouchEnd
  };
};

exports["default"] = useScale;
