// tdesign-react/cjs/layout/Layout.js
/**
 * tdesign v1.17.1
 * (c) 2026 tdesign
 * @license MIT
 */

'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var _typeof = require('../_chunks/dep-c290284d.js');
var defineProperty = require('../_chunks/dep-c446aa11.js');
var objectWithoutProperties = require('../_chunks/dep-e41137f7.js');
var React = require('react');
var classNames = require('classnames');
var _util_parseTNode = require('../_util/parseTNode.js');
var hooks_useConfig = require('../hooks/useConfig.js');
var layout_Aside = require('./Aside.js');
require('../_chunks/dep-3c628d18.js');
require('../_chunks/dep-11347ea1.js');
require('../_chunks/dep-3d05de97.js');
require('../_chunks/dep-a830ce1e.js');
require('../config-provider/ConfigContext.js');
require('../_chunks/dep-2ee09518.js');
require('../_chunks/dep-a6fa90dc.js');
require('dayjs');
require('../_chunks/dep-012d9a7c.js');
require('../_chunks/dep-150a112c.js');
require('../_chunks/dep-e360e17f.js');
require('../_chunks/dep-4f1729f0.js');
require('../_chunks/dep-a54f202c.js');
require('../_chunks/dep-51284537.js');
require('../_chunks/dep-8a493834.js');
require('../_chunks/dep-9f5b2891.js');
require('../_chunks/dep-338f013a.js');
require('../_chunks/dep-c37c5e17.js');
require('../_chunks/dep-eee1aab9.js');
require('../_chunks/dep-770bc0ae.js');
require('../_chunks/dep-aaf27e39.js');
require('../_chunks/dep-b47d855f.js');
require('../_chunks/dep-81bce4e6.js');
require('../_chunks/dep-863f8f4d.js');
require('../_chunks/dep-84e4d0da.js');
require('../_chunks/dep-f78183d7.js');
require('../_chunks/dep-2f45f6c4.js');
require('../hooks/useDefaultProps.js');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

var React__default = /*#__PURE__*/_interopDefaultLegacy(React);
var classNames__default = /*#__PURE__*/_interopDefaultLegacy(classNames);

var _excluded = ["className", "style", "children", "height"],
  _excluded2 = ["className", "style", "children", "height"],
  _excluded3 = ["className", "style", "children", "content"],
  _excluded4 = ["direction", "className", "style", "children"];
function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { defineProperty._defineProperty(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
var Header = function Header(props) {
  var _useConfig = hooks_useConfig["default"](),
    classPrefix = _useConfig.classPrefix;
  var className = props.className,
    _props$style = props.style,
    style = _props$style === void 0 ? {} : _props$style,
    children = props.children,
    height = props.height,
    others = objectWithoutProperties._objectWithoutProperties(props, _excluded);
  var renderHeight = isNaN(Number(height)) ? height : "".concat(height, "px");
  var headerClassNames = classNames__default["default"]("".concat(classPrefix, "-layout__header"), className);
  return /* @__PURE__ */React__default["default"].createElement("header", _objectSpread({
    className: headerClassNames,
    style: _objectSpread({
      height: renderHeight
    }, style)
  }, others), _util_parseTNode["default"](children));
};
var Footer = function Footer(props) {
  var _useConfig2 = hooks_useConfig["default"](),
    classPrefix = _useConfig2.classPrefix;
  var className = props.className,
    _props$style2 = props.style,
    style = _props$style2 === void 0 ? {} : _props$style2,
    children = props.children,
    height = props.height,
    others = objectWithoutProperties._objectWithoutProperties(props, _excluded2);
  var renderHeight = isNaN(Number(height)) ? height : "".concat(height, "px");
  var footerClassNames = classNames__default["default"]("".concat(classPrefix, "-layout__footer"), className);
  return /* @__PURE__ */React__default["default"].createElement("footer", _objectSpread({
    className: footerClassNames,
    style: _objectSpread({
      height: renderHeight
    }, style)
  }, others), _util_parseTNode["default"](children));
};
var Content = function Content(props) {
  var _useConfig3 = hooks_useConfig["default"](),
    classPrefix = _useConfig3.classPrefix;
  var className = props.className,
    style = props.style,
    children = props.children,
    content = props.content,
    others = objectWithoutProperties._objectWithoutProperties(props, _excluded3);
  var contentClassNames = classNames__default["default"]("".concat(classPrefix, "-layout__content"), className);
  return /* @__PURE__ */React__default["default"].createElement("main", _objectSpread({
    className: contentClassNames,
    style: style
  }, others), _util_parseTNode["default"](content) || _util_parseTNode["default"](children));
};
var Layout = function Layout(props) {
  var direction = props.direction,
    className = props.className,
    style = props.style,
    children = props.children,
    otherLayoutProps = objectWithoutProperties._objectWithoutProperties(props, _excluded4);
  var shouldAsides = React.useMemo(function () {
    var asides = [];
    React__default["default"].Children.forEach(children, function (child) {
      if (!child || _typeof._typeof(child) !== "object") {
        return;
      }
      if (child.type === layout_Aside["default"]) {
        asides.push(child);
      }
    });
    return !!asides.length;
  }, [children]);
  var _useConfig4 = hooks_useConfig["default"](),
    classPrefix = _useConfig4.classPrefix;
  var layoutClassNames = classNames__default["default"]("".concat(classPrefix, "-layout"), defineProperty._defineProperty(defineProperty._defineProperty({}, "".concat(classPrefix, "-layout--with-sider"), shouldAsides), "".concat(classPrefix, "-layout__direction-").concat(direction), direction), className);
  return /* @__PURE__ */React__default["default"].createElement("div", _objectSpread({
    className: layoutClassNames,
    style: style
  }, otherLayoutProps), _util_parseTNode["default"](children));
};
Layout.Header = Header;
Layout.Content = Content;
Layout.Footer = Footer;
Layout.Aside = layout_Aside["default"];
Header.displayName = "Header";
Content.displayName = "Content";
Footer.displayName = "Footer";
Layout.displayName = "Layout";

exports["default"] = Layout;
