// tdesign-react/cjs/empty/Empty.js
/**
 * tdesign v1.17.1
 * (c) 2026 tdesign
 * @license MIT
 */

'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var slicedToArray = require('../_chunks/dep-c60a3e4c.js');
var defineProperty = require('../_chunks/dep-c446aa11.js');
var React = require('react');
var classNames = require('classnames');
var hooks_useConfig = require('../hooks/useConfig.js');
var hooks_useDefaultProps = require('../hooks/useDefaultProps.js');
var image_index = require('../image/index.js');
var locale_LocalReceiver = require('../locale/LocalReceiver.js');
var empty_assets_EmptySvg = require('./assets/EmptySvg.js');
var empty_assets_FailSvg = require('./assets/FailSvg.js');
var empty_assets_MaintenanceSvg = require('./assets/MaintenanceSvg.js');
var empty_assets_NetworkErrorSvg = require('./assets/NetworkErrorSvg.js');
var empty_assets_SuccessSvg = require('./assets/SuccessSvg.js');
var empty_defaultProps = require('./defaultProps.js');
var isObject = require('../_chunks/dep-a830ce1e.js');
var isString = require('../_chunks/dep-427f8bc0.js');
require('../_chunks/dep-98e6819c.js');
require('../_chunks/dep-7f3fa935.js');
require('../_chunks/dep-c290284d.js');
require('../config-provider/ConfigContext.js');
require('../_chunks/dep-2ee09518.js');
require('../_chunks/dep-a6fa90dc.js');
require('dayjs');
require('../_chunks/dep-012d9a7c.js');
require('../_chunks/dep-150a112c.js');
require('../_chunks/dep-e360e17f.js');
require('../_chunks/dep-3d05de97.js');
require('../_chunks/dep-4f1729f0.js');
require('../_chunks/dep-a54f202c.js');
require('../_chunks/dep-51284537.js');
require('../_chunks/dep-8a493834.js');
require('../_chunks/dep-9f5b2891.js');
require('../_chunks/dep-338f013a.js');
require('../_chunks/dep-c37c5e17.js');
require('../_chunks/dep-eee1aab9.js');
require('../_chunks/dep-11347ea1.js');
require('../_chunks/dep-770bc0ae.js');
require('../_chunks/dep-aaf27e39.js');
require('../_chunks/dep-b47d855f.js');
require('../_chunks/dep-81bce4e6.js');
require('../_chunks/dep-863f8f4d.js');
require('../_chunks/dep-84e4d0da.js');
require('../_chunks/dep-f78183d7.js');
require('../_chunks/dep-2f45f6c4.js');
require('../image/Image.js');
require('../_chunks/dep-e41137f7.js');
require('tdesign-icons-react');
require('../_chunks/dep-2bc69864.js');
require('../hooks/useGlobalIcon.js');
require('../hooks/useImagePreviewUrl.js');
require('../_chunks/dep-3c628d18.js');
require('../_chunks/dep-24740dfb.js');
require('../space/index.js');
require('../space/Space.js');
require('react-is');
require('../_chunks/dep-6e0da644.js');
require('../_chunks/dep-05bc49e6.js');
require('../_chunks/dep-8c880a65.js');
require('../_chunks/dep-b8d0b96c.js');
require('../space/defaultProps.js');
require('../image/defaultProps.js');
require('../config-provider/ConfigProvider.js');
require('../_chunks/dep-ba90997f.js');
require('../_chunks/dep-22026fea.js');
require('../_chunks/dep-551d5940.js');
require('../_chunks/dep-8005b808.js');
require('../_chunks/dep-2add8824.js');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

var React__default = /*#__PURE__*/_interopDefaultLegacy(React);
var classNames__default = /*#__PURE__*/_interopDefaultLegacy(classNames);

function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { defineProperty._defineProperty(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
function getImageIns(data) {
  var result = data;
  if (/*#__PURE__*/React.isValidElement(data)) {
    result = data;
  } else if (isObject.isObject(data)) {
    result = /* @__PURE__ */React__default["default"].createElement(image_index.Image, _objectSpread({}, data));
  } else if (isString.isString(data)) {
    result = /* @__PURE__ */React__default["default"].createElement(image_index.Image, {
      src: data
    });
  }
  return data ? result : null;
}
var Empty = function Empty(props) {
  var _defaultMaps$type, _typeImageProps$image, _typeImageProps$title;
  var _useDefaultProps = hooks_useDefaultProps["default"](props, empty_defaultProps.emptyDefaultProps),
    propsImage = _useDefaultProps.image,
    imageStyle = _useDefaultProps.imageStyle,
    propsDescription = _useDefaultProps.description,
    propsTitle = _useDefaultProps.title,
    type = _useDefaultProps.type,
    action = _useDefaultProps.action,
    style = _useDefaultProps.style,
    className = _useDefaultProps.className,
    _useDefaultProps$size = _useDefaultProps.size,
    size = _useDefaultProps$size === void 0 ? "medium" : _useDefaultProps$size;
  var _useConfig = hooks_useConfig["default"](),
    classPrefix = _useConfig.classPrefix,
    empty = _useConfig.empty;
  var _useLocaleReceiver = locale_LocalReceiver.useLocaleReceiver("empty"),
    _useLocaleReceiver2 = slicedToArray._slicedToArray(_useLocaleReceiver, 2),
    local = _useLocaleReceiver2[0],
    t = _useLocaleReceiver2[1];
  var defaultMaps = {
    maintenance: {
      image: empty.image.maintenance || /* @__PURE__ */React__default["default"].createElement(empty_assets_MaintenanceSvg["default"], null),
      title: empty.titleText.maintenance || t(local.titleText.maintenance)
    },
    success: {
      image: empty.image.success || /* @__PURE__ */React__default["default"].createElement(empty_assets_SuccessSvg["default"], null),
      title: empty.titleText.success || t(local.titleText.success)
    },
    fail: {
      image: empty.image.fail || /* @__PURE__ */React__default["default"].createElement(empty_assets_FailSvg["default"], null),
      title: empty.titleText.fail || t(local.titleText.fail)
    },
    "network-error": {
      image: empty.image.networkError || /* @__PURE__ */React__default["default"].createElement(empty_assets_NetworkErrorSvg["default"], null),
      title: empty.titleText.networkError || t(local.titleText.networkError)
    },
    empty: {
      image: empty.image.empty || /* @__PURE__ */React__default["default"].createElement(empty_assets_EmptySvg["default"], null),
      title: empty.titleText.empty || t(local.titleText.empty)
    }
  };
  var defaultSize = {
    small: "".concat(classPrefix, "-size-s"),
    medium: "".concat(classPrefix, "-size"),
    large: "".concat(classPrefix, "-size-l")
  };
  var prefix = "".concat(classPrefix, "-empty");
  var emptyClasses = classNames__default["default"](prefix, className, defaultSize[size]);
  var titleClasses = classNames__default["default"]("".concat(prefix, "__title"));
  var imageClasses = classNames__default["default"]("".concat(prefix, "__image"));
  var descriptionClasses = classNames__default["default"]("".concat(prefix, "__description"));
  var actionCls = classNames__default["default"]("".concat(prefix, "__action"));
  var typeImageProps = (_defaultMaps$type = defaultMaps[type]) !== null && _defaultMaps$type !== void 0 ? _defaultMaps$type : null;
  var _image$title$descript = {
      image: propsImage ? propsImage : (_typeImageProps$image = typeImageProps === null || typeImageProps === void 0 ? void 0 : typeImageProps.image) !== null && _typeImageProps$image !== void 0 ? _typeImageProps$image : null,
      title: propsTitle ? propsTitle : (_typeImageProps$title = typeImageProps === null || typeImageProps === void 0 ? void 0 : typeImageProps.title) !== null && _typeImageProps$title !== void 0 ? _typeImageProps$title : null,
      description: propsDescription
    },
    image = _image$title$descript.image,
    description = _image$title$descript.description,
    title = _image$title$descript.title;
  function renderTitle() {
    if (!title) {
      return null;
    }
    return /* @__PURE__ */React__default["default"].createElement("div", {
      className: titleClasses
    }, title);
  }
  function renderDescription() {
    if (!description) {
      return null;
    }
    return /* @__PURE__ */React__default["default"].createElement("div", {
      className: descriptionClasses
    }, description);
  }
  var imageContent = getImageIns(image);
  return /* @__PURE__ */React__default["default"].createElement("div", {
    className: emptyClasses,
    style: style
  }, imageContent ? /* @__PURE__ */React__default["default"].createElement("div", {
    className: imageClasses,
    style: imageStyle
  }, imageContent) : null, renderTitle(), renderDescription(), action ? /* @__PURE__ */React__default["default"].createElement("div", {
    className: actionCls
  }, action) : null);
};

exports["default"] = Empty;
