// tdesign-react/cjs/date-picker/hooks/useDisableDate.js
/**
 * tdesign v1.17.1
 * (c) 2026 tdesign
 * @license MIT
 */

'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var dayjs = require('dayjs');
var utils = require('../../_chunks/dep-39cd4265.js');
var isObject = require('../../_chunks/dep-a830ce1e.js');
require('../../_chunks/dep-c60a3e4c.js');
require('../../_chunks/dep-98e6819c.js');
require('../../_chunks/dep-7f3fa935.js');
require('../../_chunks/dep-c446aa11.js');
require('../../_chunks/dep-c290284d.js');
require('../../_chunks/dep-a6fa90dc.js');
require('../../_chunks/dep-c1575286.js');
require('../../_chunks/dep-3be56e1e.js');
require('../../_chunks/dep-36579bf6.js');
require('../../_chunks/dep-3d05de97.js');
require('../../_chunks/dep-4f1729f0.js');
require('../../_chunks/dep-34b7c4f7.js');
require('../../_chunks/dep-b21e3f45.js');
require('../../_chunks/dep-3c628d18.js');
require('../../_chunks/dep-427f8bc0.js');
require('../../_chunks/dep-a54f202c.js');
require('../../_chunks/dep-11347ea1.js');
require('../../_chunks/dep-f5e1d11c.js');
require('../../_chunks/dep-2f45f6c4.js');
require('../../_chunks/dep-770bc0ae.js');
require('../../_chunks/dep-863f8f4d.js');
require('../../_chunks/dep-8a493834.js');
require('../../_chunks/dep-338f013a.js');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

var dayjs__default = /*#__PURE__*/_interopDefaultLegacy(dayjs);

function useDisableDate(props) {
  var _disableDate = props.disableDate,
    format = props.format,
    mode = props.mode,
    start = props.start,
    end = props.end;
  return {
    disableDate: function disableDate(value) {
      return !utils.isEnabledDate({
        disableDate: _disableDate,
        format: format,
        mode: mode,
        value: value
      });
    },
    minDate: isObject.isObject(_disableDate) && "before" in _disableDate ? new Date(dayjs__default["default"](_disableDate.before).startOf("day").format()) : start,
    maxDate: isObject.isObject(_disableDate) && "after" in _disableDate ? new Date(dayjs__default["default"](_disableDate.after).endOf("day").format()) : end
  };
}

exports["default"] = useDisableDate;
