// tdesign-react/cjs/date-picker/hooks/useRange.js
/**
 * tdesign v1.17.1
 * (c) 2026 tdesign
 * @license MIT
 */

'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var defineProperty = require('../../_chunks/dep-c446aa11.js');
var slicedToArray = require('../../_chunks/dep-c60a3e4c.js');
var React = require('react');
var classNames = require('classnames');
var tdesignIconsReact = require('tdesign-icons-react');
var format = require('../../_chunks/dep-b21e3f45.js');
var hooks_useConfig = require('../../hooks/useConfig.js');
var hooks_useGlobalIcon = require('../../hooks/useGlobalIcon.js');
var hooks_useInnerPopupVisible = require('../../hooks/useInnerPopupVisible.js');
var datePicker_hooks_useRangeValue = require('./useRangeValue.js');
var isObject = require('../../_chunks/dep-a830ce1e.js');
require('../../_chunks/dep-c290284d.js');
require('../../_chunks/dep-98e6819c.js');
require('../../_chunks/dep-7f3fa935.js');
require('dayjs');
require('../../_chunks/dep-a6fa90dc.js');
require('../../_chunks/dep-34b7c4f7.js');
require('../../_chunks/dep-3c628d18.js');
require('../../_chunks/dep-427f8bc0.js');
require('../../_chunks/dep-3d05de97.js');
require('../../_chunks/dep-a54f202c.js');
require('../../_chunks/dep-4f1729f0.js');
require('../../config-provider/ConfigContext.js');
require('../../_chunks/dep-2ee09518.js');
require('../../_chunks/dep-012d9a7c.js');
require('../../_chunks/dep-150a112c.js');
require('../../_chunks/dep-e360e17f.js');
require('../../_chunks/dep-51284537.js');
require('../../_chunks/dep-8a493834.js');
require('../../_chunks/dep-9f5b2891.js');
require('../../_chunks/dep-338f013a.js');
require('../../_chunks/dep-c37c5e17.js');
require('../../_chunks/dep-eee1aab9.js');
require('../../_chunks/dep-11347ea1.js');
require('../../_chunks/dep-770bc0ae.js');
require('../../_chunks/dep-aaf27e39.js');
require('../../_chunks/dep-b47d855f.js');
require('../../_chunks/dep-81bce4e6.js');
require('../../_chunks/dep-863f8f4d.js');
require('../../_chunks/dep-84e4d0da.js');
require('../../_chunks/dep-f78183d7.js');
require('../../_chunks/dep-2f45f6c4.js');
require('../../hooks/useControlled.js');
require('../../_util/noop.js');
require('../../_chunks/dep-45188d54.js');
require('../../_chunks/dep-eccb6a94.js');
require('../../_chunks/dep-f5e1d11c.js');
require('../../_chunks/dep-b199860d.js');
require('../../_chunks/dep-36579bf6.js');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

var React__default = /*#__PURE__*/_interopDefaultLegacy(React);
var classNames__default = /*#__PURE__*/_interopDefaultLegacy(classNames);

function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { defineProperty._defineProperty(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
var PARTIAL_MAP = {
  first: "start",
  second: "end"
};
function useRange(props) {
  var _props$separator, _props$placeholder, _props$suffixIcon, _props$popupProps$ove, _props$popupProps3, _props$popupProps4;
  var _useConfig = hooks_useConfig["default"](),
    classPrefix = _useConfig.classPrefix,
    globalDatePickerConfig = _useConfig.datePicker;
  var _useGlobalIcon = hooks_useGlobalIcon["default"]({
      CalendarIcon: tdesignIconsReact.CalendarIcon
    }),
    CalendarIcon = _useGlobalIcon.CalendarIcon;
  var name = "".concat(classPrefix, "-date-range-picker");
  var isMountedRef = React.useRef(false);
  var inputRef = React.useRef(null);
  var isSwitchTimeMode = React.useMemo(function () {
    return isObject.isObject(props.enableTimePicker) && props.enableTimePicker.mode === "switch";
  }, [props.enableTimePicker]);
  var _useRangeValue = datePicker_hooks_useRangeValue["default"](props),
    value = _useRangeValue.value,
    onChange = _useRangeValue.onChange,
    time = _useRangeValue.time,
    setTime = _useRangeValue.setTime,
    month = _useRangeValue.month,
    setMonth = _useRangeValue.setMonth,
    year = _useRangeValue.year,
    setYear = _useRangeValue.setYear,
    cacheValue = _useRangeValue.cacheValue,
    setCacheValue = _useRangeValue.setCacheValue,
    isFirstValueSelected = _useRangeValue.isFirstValueSelected,
    setIsFirstValueSelected = _useRangeValue.setIsFirstValueSelected;
  var _getDefaultFormat = format.getDefaultFormat({
      mode: props.mode,
      format: props.format,
      valueType: props.valueType,
      enableTimePicker: props.enableTimePicker
    }),
    format$1 = _getDefaultFormat.format,
    timeFormat = _getDefaultFormat.timeFormat,
    valueType = _getDefaultFormat.valueType;
  var _useState = React.useState(false),
    _useState2 = slicedToArray._slicedToArray(_useState, 2),
    popupVisible = _useState2[0],
    setPopupVisible = _useState2[1];
  var _useState3 = React.useState(false),
    _useState4 = slicedToArray._slicedToArray(_useState3, 2),
    isHoverCell = _useState4[0],
    setIsHoverCell = _useState4[1];
  var _useState5 = React.useState(0),
    _useState6 = slicedToArray._slicedToArray(_useState5, 2),
    activeIndex = _useState6[0],
    setActiveIndex = _useState6[1];
  var _useState7 = React.useState(function () {
      return format.formatDate(value, {
        format: format$1
      });
    }),
    _useState8 = slicedToArray._slicedToArray(_useState7, 2),
    inputValue = _useState8[0],
    setInputValue = _useState8[1];
  var handlePopupInvisible = function handlePopupInvisible() {
    var _props$popupProps, _props$popupProps$onV;
    setPopupVisible(false);
    (_props$popupProps = props.popupProps) === null || _props$popupProps === void 0 || (_props$popupProps$onV = _props$popupProps.onVisibleChange) === null || _props$popupProps$onV === void 0 || _props$popupProps$onV.call(_props$popupProps, false, {});
  };
  var rangeInputProps = _objectSpread(_objectSpread({}, props.rangeInputProps), {}, {
    ref: inputRef,
    borderless: props.borderless,
    size: props.size,
    separator: (_props$separator = props.separator) !== null && _props$separator !== void 0 ? _props$separator : globalDatePickerConfig.rangeSeparator,
    clearable: props.clearable,
    prefixIcon: props.prefixIcon,
    readOnly: !props.allowInput,
    placeholder: (_props$placeholder = props.placeholder) !== null && _props$placeholder !== void 0 ? _props$placeholder : globalDatePickerConfig.placeholder[props.mode],
    activeIndex: popupVisible ? activeIndex : void 0,
    suffixIcon: (_props$suffixIcon = props.suffixIcon) !== null && _props$suffixIcon !== void 0 ? _props$suffixIcon : /* @__PURE__ */React__default["default"].createElement(CalendarIcon, null),
    label: props.label,
    className: classNames__default["default"](defineProperty._defineProperty({}, "".concat(name, "__input--placeholder"), isHoverCell)),
    onClick: function onClick(_ref) {
      var position = _ref.position;
      setActiveIndex(position === "first" ? 0 : 1);
    },
    onClear: function onClear(_ref2) {
      var _props$onClear;
      var e = _ref2.e;
      e.stopPropagation();
      handlePopupInvisible();
      onChange([], {
        dayjsValue: [],
        trigger: "clear"
      });
      (_props$onClear = props.onClear) === null || _props$onClear === void 0 || _props$onClear.call(props, {
        e: e
      });
    },
    onBlur: function onBlur(newVal, _ref3) {
      var _props$onBlur;
      var e = _ref3.e,
        position = _ref3.position;
      (_props$onBlur = props.onBlur) === null || _props$onBlur === void 0 || _props$onBlur.call(props, {
        value: newVal,
        partial: PARTIAL_MAP[position],
        e: e
      });
    },
    onFocus: function onFocus(newVal, _ref4) {
      var _props$onFocus;
      var e = _ref4.e,
        position = _ref4.position;
      (_props$onFocus = props.onFocus) === null || _props$onFocus === void 0 || _props$onFocus.call(props, {
        value: newVal,
        partial: PARTIAL_MAP[position],
        e: e
      });
      setActiveIndex(position === "first" ? 0 : 1);
    },
    onChange: function onChange(newVal, _ref5) {
      var _props$onInput;
      var e = _ref5.e,
        position = _ref5.position;
      var index = position === "first" ? 0 : 1;
      (_props$onInput = props.onInput) === null || _props$onInput === void 0 || _props$onInput.call(props, {
        input: newVal[index],
        value: value,
        partial: PARTIAL_MAP[position],
        e: e
      });
      setInputValue(newVal);
      if (!format.isValidDate(newVal, format$1)) return;
      setCacheValue(newVal);
      var newYear = [];
      var newMonth = [];
      var newTime = [];
      newVal.forEach(function (v) {
        newYear.push(format.parseToDayjs(v, format$1).year());
        newMonth.push(format.parseToDayjs(v, format$1).month());
        newTime.push(format.parseToDayjs(v, format$1).format(timeFormat));
      });
      setYear(newYear);
      setMonth(newMonth);
      setTime(newTime);
    },
    onEnter: function onEnter(newVal) {
      if (!format.isValidDate(newVal, format$1) && !format.isValidDate(value, format$1)) return;
      handlePopupInvisible();
      if (format.isValidDate(newVal, format$1)) {
        onChange(format.formatDate(newVal, {
          format: format$1,
          targetFormat: valueType,
          autoSwap: true
        }), {
          dayjsValue: newVal.map(function (v) {
            return format.parseToDayjs(v, format$1);
          }),
          trigger: "enter"
        });
      } else if (format.isValidDate(value, format$1)) {
        setInputValue(format.formatDate(value, {
          format: format$1
        }));
      } else {
        setInputValue([]);
      }
    }
  });
  var handleInnerVisibleChange = hooks_useInnerPopupVisible["default"](function (visible, context) {
    var _props$popupProps2, _props$popupProps2$on;
    if (props.disabled) return;
    (_props$popupProps2 = props.popupProps) === null || _props$popupProps2 === void 0 || (_props$popupProps2$on = _props$popupProps2.onVisibleChange) === null || _props$popupProps2$on === void 0 || _props$popupProps2$on.call(_props$popupProps2, visible, context);
    if (context.trigger === "trigger-element-mousedown") {
      var indexMap = {
        0: "first",
        1: "second"
      };
      inputRef.current.focus({
        position: indexMap[activeIndex]
      });
      return setPopupVisible(true);
    }
    setPopupVisible(visible);
  });
  var popupProps = _objectSpread(_objectSpread({
    expandAnimation: true
  }, props.popupProps), {}, {
    trigger: "mousedown",
    overlayInnerStyle: (_props$popupProps$ove = (_props$popupProps3 = props.popupProps) === null || _props$popupProps3 === void 0 ? void 0 : _props$popupProps3.overlayInnerStyle) !== null && _props$popupProps$ove !== void 0 ? _props$popupProps$ove : {
      width: "auto"
    },
    overlayClassName: classNames__default["default"]((_props$popupProps4 = props.popupProps) === null || _props$popupProps4 === void 0 ? void 0 : _props$popupProps4.overlayClassName, "".concat(name, "__panel-container")),
    onVisibleChange: handleInnerVisibleChange
  });
  React.useEffect(function () {
    if (!value) {
      setInputValue([]);
      return;
    }
    if (!format.isValidDate(value, format$1)) return;
    setInputValue(format.formatDate(value, {
      format: format$1
    }));
  }, [value]);
  React.useEffect(function () {
    if (!popupVisible) return;
    if (!isMountedRef.current) {
      isMountedRef.current = true;
      return;
    }
    var indexMap = {
      0: "first",
      1: "second"
    };
    inputRef.current.focus({
      position: indexMap[activeIndex]
    });
  }, [activeIndex]);
  return {
    year: year,
    month: month,
    value: value,
    time: time,
    inputValue: inputValue,
    popupVisible: popupVisible,
    rangeInputProps: rangeInputProps,
    popupProps: popupProps,
    isHoverCell: isHoverCell,
    onChange: onChange,
    setYear: setYear,
    setMonth: setMonth,
    setTime: setTime,
    setIsHoverCell: setIsHoverCell,
    setInputValue: setInputValue,
    setPopupVisible: setPopupVisible,
    activeIndex: activeIndex,
    setActiveIndex: setActiveIndex,
    isFirstValueSelected: isFirstValueSelected,
    setIsFirstValueSelected: setIsFirstValueSelected,
    cacheValue: cacheValue,
    setCacheValue: setCacheValue,
    isSwitchTimeMode: isSwitchTimeMode
  };
}

exports.PARTIAL_MAP = PARTIAL_MAP;
exports["default"] = useRange;
