// tdesign-react/cjs/date-picker/hooks/useRangeValue.js
/**
 * tdesign v1.17.1
 * (c) 2026 tdesign
 * @license MIT
 */

'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var slicedToArray = require('../../_chunks/dep-c60a3e4c.js');
var React = require('react');
var format = require('../../_chunks/dep-b21e3f45.js');
var log = require('../../_chunks/dep-3c628d18.js');
var hooks_useControlled = require('../../hooks/useControlled.js');
var isObject = require('../../_chunks/dep-a830ce1e.js');
require('../../_chunks/dep-98e6819c.js');
require('../../_chunks/dep-7f3fa935.js');
require('dayjs');
require('../../_chunks/dep-a6fa90dc.js');
require('../../_chunks/dep-34b7c4f7.js');
require('../../_chunks/dep-427f8bc0.js');
require('../../_chunks/dep-3d05de97.js');
require('../../_chunks/dep-c290284d.js');
require('../../_chunks/dep-a54f202c.js');
require('../../_chunks/dep-4f1729f0.js');
require('../../_util/noop.js');
require('../../_chunks/dep-45188d54.js');
require('../../_chunks/dep-eccb6a94.js');
require('../../_chunks/dep-f5e1d11c.js');
require('../../_chunks/dep-b199860d.js');
require('../../_chunks/dep-36579bf6.js');

var PARTIAL_MAP = {
  first: "start",
  second: "end"
};
function useRange(props) {
  var _useControlled = hooks_useControlled["default"](props, "value", props.onChange),
    _useControlled2 = slicedToArray._slicedToArray(_useControlled, 2),
    value = _useControlled2[0],
    onChange = _useControlled2[1];
  var _getDefaultFormat = format.getDefaultFormat({
      mode: props.mode,
      format: props.format,
      enableTimePicker: props.enableTimePicker
    }),
    format$1 = _getDefaultFormat.format,
    timeFormat = _getDefaultFormat.timeFormat;
  if (props.enableTimePicker) {
    if (!format.extractTimeFormat(format$1)) log.log.error("DatePicker", "format: ".concat(format$1, " is invalid\uFF0Ctime selection must include time formatting HH:mm:ss"));
  }
  if (!Array.isArray(value)) {
    log.log.error("DatePicker", "typeof value: ".concat(value, " must be Array!"));
  } else if (!format.isValidDate(value, format$1)) {
    log.log.error("DatePicker", "value: ".concat(value, " is invalid dateTime! Check whether the value is consistent with format: ").concat(format$1));
  }
  var _useState = React.useState(false),
    _useState2 = slicedToArray._slicedToArray(_useState, 2),
    isFirstValueSelected = _useState2[0],
    setIsFirstValueSelected = _useState2[1];
  var _useState3 = React.useState(function () {
      return format.initYearMonthTime({
        value: value,
        mode: props.mode,
        format: format$1,
        enableTimePicker: props.enableTimePicker
      }).time;
    }),
    _useState4 = slicedToArray._slicedToArray(_useState3, 2),
    time = _useState4[0],
    setTime = _useState4[1];
  var _useState5 = React.useState(function () {
      return format.initYearMonthTime({
        value: value,
        mode: props.mode,
        format: format$1,
        enableTimePicker: props.enableTimePicker
      }).month;
    }),
    _useState6 = slicedToArray._slicedToArray(_useState5, 2),
    month = _useState6[0],
    setMonth = _useState6[1];
  var _useState7 = React.useState(function () {
      return format.initYearMonthTime({
        value: value,
        mode: props.mode,
        format: format$1
      }).year;
    }),
    _useState8 = slicedToArray._slicedToArray(_useState7, 2),
    year = _useState8[0],
    setYear = _useState8[1];
  var _useState9 = React.useState(function () {
      return format.formatDate(value, {
        format: format$1
      });
    }),
    _useState0 = slicedToArray._slicedToArray(_useState9, 2),
    cacheValue = _useState0[0],
    setCacheValue = _useState0[1];
  var isSwitchTimeMode = React.useMemo(function () {
    return isObject.isObject(props.enableTimePicker) && props.enableTimePicker.mode === "switch";
  }, [props.enableTimePicker]);
  React.useEffect(function () {
    if (!value) {
      setCacheValue([]);
      return;
    }
    if (!format.isValidDate(value, format$1)) return;
    setCacheValue(format.formatDate(value, {
      format: format$1
    }));
    setTime(format.formatTime(value, format$1, timeFormat, props.defaultTime));
  }, [value]);
  return {
    value: value,
    onChange: onChange,
    year: year,
    setYear: setYear,
    month: month,
    setMonth: setMonth,
    time: time,
    setTime: setTime,
    isFirstValueSelected: isFirstValueSelected,
    setIsFirstValueSelected: setIsFirstValueSelected,
    cacheValue: cacheValue,
    setCacheValue: setCacheValue,
    isSwitchTimeMode: isSwitchTimeMode
  };
}

exports.PARTIAL_MAP = PARTIAL_MAP;
exports["default"] = useRange;
