// tdesign-react/cjs/date-picker/hooks/useSelectRange.js
/**
 * tdesign v1.17.1
 * (c) 2026 tdesign
 * @license MIT
 */

'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var React = require('react');
var utils = require('../../_chunks/dep-39cd4265.js');
var isArray = require('../../_chunks/dep-a54f202c.js');
var isFunction = require('../../_chunks/dep-11347ea1.js');
require('../../_chunks/dep-c60a3e4c.js');
require('../../_chunks/dep-98e6819c.js');
require('../../_chunks/dep-7f3fa935.js');
require('../../_chunks/dep-c446aa11.js');
require('../../_chunks/dep-c290284d.js');
require('dayjs');
require('../../_chunks/dep-a6fa90dc.js');
require('../../_chunks/dep-c1575286.js');
require('../../_chunks/dep-3be56e1e.js');
require('../../_chunks/dep-a830ce1e.js');
require('../../_chunks/dep-36579bf6.js');
require('../../_chunks/dep-3d05de97.js');
require('../../_chunks/dep-4f1729f0.js');
require('../../_chunks/dep-34b7c4f7.js');
require('../../_chunks/dep-b21e3f45.js');
require('../../_chunks/dep-3c628d18.js');
require('../../_chunks/dep-427f8bc0.js');
require('../../_chunks/dep-f5e1d11c.js');
require('../../_chunks/dep-2f45f6c4.js');
require('../../_chunks/dep-770bc0ae.js');
require('../../_chunks/dep-863f8f4d.js');
require('../../_chunks/dep-8a493834.js');
require('../../_chunks/dep-338f013a.js');

function useSelectRange(props) {
  var rangeBounds = React.useMemo(function () {
    return utils.getRangeBounds(props.range);
  }, [props.range]);
  var monthAllowed = function monthAllowed(year, month) {
    return utils.monthHasAnyAllowed(props.range, year, month);
  };
  var yearAllowed = function yearAllowed(year) {
    return utils.yearHasAnyAllowed(props.range, year);
  };
  var decadeAllowed = function decadeAllowed(decadeEndYear) {
    return utils.decadeHasAnyAllowed(props.range, decadeEndYear);
  };
  var paginationDisabled = React.useMemo(function () {
    return utils.computePaginationDisabled(props.range, props.mode, props.year, props.month);
  }, [props]);
  var canLoadMoreTop = function canLoadMoreTop(firstValue) {
    var min = rangeBounds.min;
    if (isArray.isArray(props.range)) {
      if (!min) return true;
      var minYear = min.getFullYear();
      if (props.mode === "year") {
        var minDecadeEnd = Math.floor(minYear / 10) * 10 + 9;
        return firstValue > minDecadeEnd;
      }
      return firstValue > minYear;
    }
    if (isFunction.isFunction(props.range)) {
      if (props.mode === "year") {
        for (var i = firstValue - 10; i >= firstValue - 50; i -= 10) {
          if (decadeAllowed(i)) return true;
        }
        return false;
      }
      for (var _i = firstValue - 1; _i > firstValue - 10; _i--) {
        if (yearAllowed(_i)) return true;
      }
      return false;
    }
    return true;
  };
  var canLoadMoreBottom = function canLoadMoreBottom(lastValue) {
    var max = rangeBounds.max;
    if (isArray.isArray(props.range)) {
      if (!max) return true;
      var maxYear = max.getFullYear();
      if (props.mode === "year") {
        var maxDecadeEnd = Math.floor(maxYear / 10) * 10 + 9;
        return lastValue < maxDecadeEnd;
      }
      return lastValue < maxYear;
    }
    if (isFunction.isFunction(props.range)) {
      if (props.mode === "year") {
        for (var i = lastValue + 10; i <= lastValue + 50; i += 10) {
          if (decadeAllowed(i)) return true;
        }
        return false;
      }
      for (var _i2 = lastValue + 1; _i2 <= lastValue + 10; _i2++) {
        if (yearAllowed(_i2)) return true;
      }
      return false;
    }
    return true;
  };
  return {
    rangeBounds: rangeBounds,
    monthHasAnyAllowed: monthAllowed,
    yearHasAnyAllowed: yearAllowed,
    decadeHasAnyAllowed: decadeAllowed,
    paginationDisabled: paginationDisabled,
    canLoadMoreTop: canLoadMoreTop,
    canLoadMoreBottom: canLoadMoreBottom
  };
}

exports.useSelectRange = useSelectRange;
