// tdesign-react/cjs/date-picker/hooks/useTableData.js
/**
 * tdesign v1.17.1
 * (c) 2026 tdesign
 * @license MIT
 */

'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var slicedToArray = require('../../_chunks/dep-c60a3e4c.js');
var dayjs = require('dayjs');
var utils = require('../../_chunks/dep-39cd4265.js');
var locale_LocalReceiver = require('../../locale/LocalReceiver.js');
var isArray = require('../../_chunks/dep-a54f202c.js');
var isFunction = require('../../_chunks/dep-11347ea1.js');
require('../../_chunks/dep-98e6819c.js');
require('../../_chunks/dep-7f3fa935.js');
require('../../_chunks/dep-c446aa11.js');
require('../../_chunks/dep-c290284d.js');
require('../../_chunks/dep-a6fa90dc.js');
require('../../_chunks/dep-c1575286.js');
require('../../_chunks/dep-3be56e1e.js');
require('../../_chunks/dep-a830ce1e.js');
require('../../_chunks/dep-36579bf6.js');
require('../../_chunks/dep-3d05de97.js');
require('../../_chunks/dep-4f1729f0.js');
require('../../_chunks/dep-34b7c4f7.js');
require('../../_chunks/dep-b21e3f45.js');
require('../../_chunks/dep-3c628d18.js');
require('../../_chunks/dep-427f8bc0.js');
require('../../_chunks/dep-f5e1d11c.js');
require('../../_chunks/dep-2f45f6c4.js');
require('../../_chunks/dep-770bc0ae.js');
require('../../_chunks/dep-863f8f4d.js');
require('../../_chunks/dep-8a493834.js');
require('../../_chunks/dep-338f013a.js');
require('react');
require('../../config-provider/ConfigContext.js');
require('../../_chunks/dep-2ee09518.js');
require('../../_chunks/dep-012d9a7c.js');
require('../../_chunks/dep-150a112c.js');
require('../../_chunks/dep-e360e17f.js');
require('../../_chunks/dep-51284537.js');
require('../../_chunks/dep-9f5b2891.js');
require('../../_chunks/dep-c37c5e17.js');
require('../../_chunks/dep-eee1aab9.js');
require('../../_chunks/dep-aaf27e39.js');
require('../../_chunks/dep-b47d855f.js');
require('../../_chunks/dep-81bce4e6.js');
require('../../_chunks/dep-84e4d0da.js');
require('../../_chunks/dep-f78183d7.js');
require('../../config-provider/ConfigProvider.js');
require('../../_chunks/dep-ba90997f.js');
require('../../_chunks/dep-22026fea.js');
require('../../_chunks/dep-551d5940.js');
require('../../_chunks/dep-8005b808.js');
require('../../_chunks/dep-2add8824.js');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

var dayjs__default = /*#__PURE__*/_interopDefaultLegacy(dayjs);

function useTableData(props) {
  var _useLocaleReceiver = locale_LocalReceiver.useLocaleReceiver("datePicker"),
    _useLocaleReceiver2 = slicedToArray._slicedToArray(_useLocaleReceiver, 2),
    local = _useLocaleReceiver2[0],
    t = _useLocaleReceiver2[1];
  var monthLocal = t(local.months);
  var quarterLocal = t(local.quarters);
  var start = props.start,
    end = props.end,
    hoverStart = props.hoverStart,
    hoverEnd = props.hoverEnd,
    year = props.year,
    month = props.month,
    mode = props.mode,
    firstDayOfWeek = props.firstDayOfWeek,
    disableDate = props.disableDate,
    minDate = props.minDate,
    maxDate = props.maxDate,
    isRange = props.isRange,
    value = props.value,
    multiple = props.multiple;
  var options = {
    minDate: minDate,
    maxDate: maxDate,
    disableDate: disableDate,
    firstDayOfWeek: firstDayOfWeek,
    monthLocal: monthLocal,
    quarterLocal: quarterLocal,
    showWeekOfYear: mode === "week",
    dayjsLocale: local.dayjsLocale,
    cancelRangeSelectLimit: props.cancelRangeSelectLimit
  };
  var baseDisableDate = options.disableDate;
  var finalDisableDate = function finalDisableDate(date) {
    return baseDisableDate ? !!baseDisableDate(date) : false;
  };
  if (isArray.isArray(props.range)) {
    var _getRangeBounds = utils.getRangeBounds(props.range),
      rawMin = _getRangeBounds.min,
      rawMax = _getRangeBounds.max;
    var rangeMin = rawMin ? new Date(dayjs__default["default"](rawMin).startOf("day").format()) : null;
    var rangeMax = rawMax ? new Date(dayjs__default["default"](rawMax).endOf("day").format()) : null;
    var min = rangeMin;
    var max = rangeMax;
    if (["date", "week"].includes(props.mode)) {
      if (min) {
        options.minDate = options.minDate ? new Date(Math.max(options.minDate.getTime(), min.getTime())) : min;
      }
      if (max) {
        options.maxDate = options.maxDate ? new Date(Math.min(options.maxDate.getTime(), max.getTime())) : max;
      }
    }
    var intervalOverlap = function intervalOverlap(aStart, aEnd, bStart, bEnd) {
      var s1 = aStart.getTime();
      var e1 = aEnd.getTime();
      var s2 = bStart ? bStart.getTime() : -Infinity;
      var e2 = bEnd ? bEnd.getTime() : Infinity;
      return !(e1 < s2 || s1 > e2);
    };
    finalDisableDate = function finalDisableDate(date) {
      var base = baseDisableDate ? !!baseDisableDate(date) : false;
      if (props.mode === "date" || props.mode === "week") {
        var t2 = date.getTime();
        var aboveMin = min ? t2 >= min.getTime() : true;
        var belowMax = max ? t2 <= max.getTime() : true;
        var allowed = aboveMin && belowMax;
        return base || !allowed;
      }
      if (props.mode === "month") {
        var start2 = new Date(date.getFullYear(), date.getMonth(), 1);
        var end2 = new Date(date.getFullYear(), date.getMonth() + 1, 0);
        end2.setHours(23, 59, 59, 999);
        var _allowed = intervalOverlap(start2, end2, min, max);
        return base || !_allowed;
      }
      if (props.mode === "quarter") {
        var startMonth = Math.floor(date.getMonth() / 3) * 3;
        var _start = new Date(date.getFullYear(), startMonth, 1);
        var _end = new Date(date.getFullYear(), startMonth + 3, 0);
        _end.setHours(23, 59, 59, 999);
        var _allowed2 = intervalOverlap(_start, _end, min, max);
        return base || !_allowed2;
      }
      if (props.mode === "year") {
        var _start2 = new Date(date.getFullYear(), 0, 1);
        var _end2 = new Date(date.getFullYear(), 11, 31);
        _end2.setHours(23, 59, 59, 999);
        var _allowed3 = intervalOverlap(_start2, _end2, min, max);
        return base || !_allowed3;
      }
      return base;
    };
  } else if (isFunction.isFunction(props.range)) {
    var allow = props.range;
    finalDisableDate = function finalDisableDate(date) {
      var base = baseDisableDate ? !!baseDisableDate(date) : false;
      if (props.mode === "date" || props.mode === "week") {
        var disallowedByRange = !allow(date);
        return base || disallowedByRange;
      }
      if (props.mode === "month") {
        var allowed = utils.monthHasAnyAllowed(props.range, date.getFullYear(), date.getMonth());
        return base || !allowed;
      }
      if (props.mode === "quarter") {
        var startMonth = Math.floor(date.getMonth() / 3) * 3;
        var _allowed4 = false;
        for (var m = startMonth; m < startMonth + 3; m++) {
          if (utils.monthHasAnyAllowed(props.range, date.getFullYear(), m)) {
            _allowed4 = true;
            break;
          }
        }
        return base || !_allowed4;
      }
      if (props.mode === "year") {
        var _allowed5 = utils.yearHasAnyAllowed(props.range, date.getFullYear());
        return base || !_allowed5;
      }
      return base;
    };
  }
  options.disableDate = finalDisableDate;
  var data = [];
  if (mode === "date") {
    data = utils.getWeeks({
      year: year,
      month: month
    }, options);
  } else if (mode === "week") {
    data = utils.getWeeks({
      year: year,
      month: month
    }, options);
  } else if (mode === "quarter") {
    data = utils.getQuarters(year, options);
  } else if (mode === "month") {
    data = utils.getMonths(year, options);
  } else if (mode === "year") {
    data = utils.getYears(year, options);
  }
  return utils.flagActive(data, {
    start: start,
    end: end,
    hoverStart: hoverStart,
    hoverEnd: hoverEnd,
    type: mode,
    isRange: isRange,
    value: value,
    multiple: multiple
  });
}

exports["default"] = useTableData;
