// tdesign-react/cjs/select-input/useMultiple.js
/**
 * tdesign v1.17.1
 * (c) 2026 tdesign
 * @license MIT
 */

'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var defineProperty = require('../_chunks/dep-c446aa11.js');
var slicedToArray = require('../_chunks/dep-c60a3e4c.js');
var React = require('react');
var classNames = require('classnames');
var hooks_useConfig = require('../hooks/useConfig.js');
var hooks_useControlled = require('../hooks/useControlled.js');
var tagInput_index = require('../tag-input/index.js');
var isObject = require('../_chunks/dep-a830ce1e.js');
require('../_chunks/dep-c290284d.js');
require('../_chunks/dep-98e6819c.js');
require('../_chunks/dep-7f3fa935.js');
require('../config-provider/ConfigContext.js');
require('../_chunks/dep-2ee09518.js');
require('../_chunks/dep-a6fa90dc.js');
require('dayjs');
require('../_chunks/dep-012d9a7c.js');
require('../_chunks/dep-150a112c.js');
require('../_chunks/dep-e360e17f.js');
require('../_chunks/dep-3d05de97.js');
require('../_chunks/dep-4f1729f0.js');
require('../_chunks/dep-a54f202c.js');
require('../_chunks/dep-51284537.js');
require('../_chunks/dep-8a493834.js');
require('../_chunks/dep-9f5b2891.js');
require('../_chunks/dep-338f013a.js');
require('../_chunks/dep-c37c5e17.js');
require('../_chunks/dep-eee1aab9.js');
require('../_chunks/dep-11347ea1.js');
require('../_chunks/dep-770bc0ae.js');
require('../_chunks/dep-aaf27e39.js');
require('../_chunks/dep-b47d855f.js');
require('../_chunks/dep-81bce4e6.js');
require('../_chunks/dep-863f8f4d.js');
require('../_chunks/dep-84e4d0da.js');
require('../_chunks/dep-f78183d7.js');
require('../_chunks/dep-2f45f6c4.js');
require('../_util/noop.js');
require('../_chunks/dep-45188d54.js');
require('../_chunks/dep-eccb6a94.js');
require('../_chunks/dep-f5e1d11c.js');
require('../_chunks/dep-b199860d.js');
require('../_chunks/dep-36579bf6.js');
require('../tag-input/TagInput.js');
require('tdesign-icons-react');
require('../hooks/useDefaultProps.js');
require('../hooks/useDragSorter.js');
require('../hooks/useEventCallback.js');
require('../hooks/useGlobalIcon.js');
require('../input/index.js');
require('../input/Input.js');
require('../_chunks/dep-e41137f7.js');
require('../_util/forwardRefWithStatics.js');
require('hoist-non-react-statics');
require('../_util/parseTNode.js');
require('../_chunks/dep-3c628d18.js');
require('../hooks/useLayoutEffect.js');
require('../_chunks/dep-ee71ddca.js');
require('../_chunks/dep-427f8bc0.js');
require('../locale/LocalReceiver.js');
require('../config-provider/ConfigProvider.js');
require('../_chunks/dep-ba90997f.js');
require('../_chunks/dep-22026fea.js');
require('../_chunks/dep-551d5940.js');
require('../_chunks/dep-8005b808.js');
require('../_chunks/dep-2add8824.js');
require('../input/defaultProps.js');
require('../input/InputGroup.js');
require('../input/useLengthLimit.js');
require('../_chunks/dep-6e0da644.js');
require('../_chunks/dep-05bc49e6.js');
require('../_chunks/dep-8c880a65.js');
require('../_chunks/dep-b8d0b96c.js');
require('../tag-input/defaultProps.js');
require('../tag-input/useHover.js');
require('../tag-input/useTagList.js');
require('../tag/index.js');
require('../tag/CheckTag.js');
require('../_chunks/dep-29222115.js');
require('../tag/defaultProps.js');
require('../tag/Tag.js');
require('../_chunks/dep-183643e1.js');
require('../tag/CheckTagGroup.js');
require('../tag-input/useTagScroll.js');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

var React__default = /*#__PURE__*/_interopDefaultLegacy(React);
var classNames__default = /*#__PURE__*/_interopDefaultLegacy(classNames);

function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { defineProperty._defineProperty(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
var DEFAULT_KEYS = {
  label: "label",
  key: "key",
  children: "children"
};
function useMultiple(props) {
  var value = props.value;
  var readOnly = props.readOnly || props.readonly;
  var _useConfig = hooks_useConfig["default"](),
    classPrefix = _useConfig.classPrefix;
  var _useControlled = hooks_useControlled["default"](props, "inputValue", props.onInputChange),
    _useControlled2 = slicedToArray._slicedToArray(_useControlled, 2),
    tInputValue = _useControlled2[0],
    setTInputValue = _useControlled2[1];
  var tagInputRef = React.useRef(null);
  var blurTimeoutRef = React.useRef(null);
  var iKeys = _objectSpread(_objectSpread({}, DEFAULT_KEYS), props.keys);
  var getTags = function getTags() {
    if (!(value instanceof Array)) {
      if (["", null, void 0].includes(value)) return [];
      return isObject.isObject(value) ? [value[iKeys.label]] : [value];
    }
    return value.map(function (item) {
      return isObject.isObject(item) ? item[iKeys.label] : item;
    });
  };
  var tags = getTags();
  var tPlaceholder = !tags || !tags.length ? props.placeholder : "";
  var onTagInputChange = function onTagInputChange(val, context) {
    var _props$onTagChange;
    if (context.trigger === "tag-remove") {
      var _context$e;
      (_context$e = context.e) === null || _context$e === void 0 || _context$e.stopPropagation();
    }
    (_props$onTagChange = props.onTagChange) === null || _props$onTagChange === void 0 || _props$onTagChange.call(props, val, context);
  };
  var renderSelectMultiple = function renderSelectMultiple(p) {
    var _props$tagInputProps, _props$tagInputProps2;
    var handleBlur = function handleBlur(value2, context) {
      if (blurTimeoutRef.current) {
        clearTimeout(blurTimeoutRef.current);
      }
      blurTimeoutRef.current = setTimeout(function () {
        if (blurTimeoutRef.current) {
          if (!p.popupVisible) {
            p.onInnerBlur(context);
          } else if (!props.panel) {
            var _props$onBlur;
            (_props$onBlur = props.onBlur) === null || _props$onBlur === void 0 || _props$onBlur.call(props, value2, {
              e: context.e,
              inputValue: tInputValue,
              tagInputValue: tags
            });
          }
        }
        blurTimeoutRef.current = null;
      }, 150);
    };
    var handleFocus = function handleFocus(val, context) {
      var _props$onFocus;
      if (blurTimeoutRef.current) {
        clearTimeout(blurTimeoutRef.current);
        blurTimeoutRef.current = null;
      }
      (_props$onFocus = props.onFocus) === null || _props$onFocus === void 0 || _props$onFocus.call(props, props.value, _objectSpread(_objectSpread({}, context), {}, {
        tagInputValue: val
      }));
    };
    return /* @__PURE__ */React__default["default"].createElement(tagInput_index.TagInput, _objectSpread(_objectSpread(_objectSpread({
      ref: tagInputRef
    }, p.commonInputProps), {}, {
      autoWidth: props.autoWidth,
      readOnly: readOnly,
      minCollapsedNum: props.minCollapsedNum,
      collapsedItems: props.collapsedItems,
      tag: props.tag,
      valueDisplay: props.valueDisplay,
      placeholder: tPlaceholder,
      options: props.options,
      value: tags,
      inputValue: p.popupVisible && p.allowInput ? tInputValue : "",
      onChange: onTagInputChange,
      onInputChange: function onInputChange(val, context) {
        if ((context === null || context === void 0 ? void 0 : context.trigger) === "enter" || (context === null || context === void 0 ? void 0 : context.trigger) === "blur") return;
        setTInputValue(val, {
          trigger: context.trigger,
          e: context.e
        });
      },
      onClear: p.onInnerClear,
      onFocus: handleFocus,
      onBlur: handleBlur
    }, props.tagInputProps), {}, {
      tagProps: _objectSpread(_objectSpread({}, props.tagProps), (_props$tagInputProps = props.tagInputProps) === null || _props$tagInputProps === void 0 ? void 0 : _props$tagInputProps.tagProps),
      inputProps: _objectSpread(_objectSpread({}, props.inputProps), {}, {
        autocomplete: "off",
        readOnly: !props.allowInput || readOnly,
        readonly: !props.allowInput || readOnly,
        inputClass: classNames__default["default"]((_props$tagInputProps2 = props.tagInputProps) === null || _props$tagInputProps2 === void 0 ? void 0 : _props$tagInputProps2.className, defineProperty._defineProperty(defineProperty._defineProperty({}, "".concat(classPrefix, "-input--focused"), p.popupVisible), "".concat(classPrefix, "-is-focused"), p.popupVisible))
      })
    }));
  };
  return {
    tags: tags,
    tPlaceholder: tPlaceholder,
    tagInputRef: tagInputRef,
    multipleInputValue: tInputValue,
    renderSelectMultiple: renderSelectMultiple
  };
}

exports["default"] = useMultiple;
