// tdesign-react/cjs/select-input/useOverlayInnerStyle.js
/**
 * tdesign v1.17.1
 * (c) 2026 tdesign
 * @license MIT
 */

'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var defineProperty = require('../_chunks/dep-c446aa11.js');
var _typeof = require('../_chunks/dep-c290284d.js');
var slicedToArray = require('../_chunks/dep-c60a3e4c.js');
var React = require('react');
var hooks_useControlled = require('../hooks/useControlled.js');
var hooks_useInnerPopupVisible = require('../hooks/useInnerPopupVisible.js');
var isFunction = require('../_chunks/dep-11347ea1.js');
var isObject = require('../_chunks/dep-a830ce1e.js');
require('../_chunks/dep-98e6819c.js');
require('../_chunks/dep-7f3fa935.js');
require('../_util/noop.js');
require('../_chunks/dep-45188d54.js');
require('../_chunks/dep-eccb6a94.js');
require('../_chunks/dep-f5e1d11c.js');
require('../_chunks/dep-b199860d.js');
require('../_chunks/dep-3d05de97.js');
require('../_chunks/dep-a54f202c.js');
require('../_chunks/dep-36579bf6.js');
require('../_chunks/dep-4f1729f0.js');
require('../hooks/useConfig.js');
require('../config-provider/ConfigContext.js');
require('../_chunks/dep-2ee09518.js');
require('../_chunks/dep-a6fa90dc.js');
require('dayjs');
require('../_chunks/dep-012d9a7c.js');
require('../_chunks/dep-150a112c.js');
require('../_chunks/dep-e360e17f.js');
require('../_chunks/dep-51284537.js');
require('../_chunks/dep-8a493834.js');
require('../_chunks/dep-9f5b2891.js');
require('../_chunks/dep-338f013a.js');
require('../_chunks/dep-c37c5e17.js');
require('../_chunks/dep-eee1aab9.js');
require('../_chunks/dep-770bc0ae.js');
require('../_chunks/dep-aaf27e39.js');
require('../_chunks/dep-b47d855f.js');
require('../_chunks/dep-81bce4e6.js');
require('../_chunks/dep-863f8f4d.js');
require('../_chunks/dep-84e4d0da.js');
require('../_chunks/dep-f78183d7.js');
require('../_chunks/dep-2f45f6c4.js');

function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { defineProperty._defineProperty(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
var MAX_POPUP_WIDTH = 1e3;
function useOverlayInnerStyle(props, extra) {
  var popupProps = props.popupProps,
    autoWidth = props.autoWidth,
    disabled = props.disabled,
    onPopupVisibleChange = props.onPopupVisibleChange,
    allowInput = props.allowInput;
  var readonly = props.readOnly || props.readonly;
  var _useControlled = hooks_useControlled["default"](props, "popupVisible", onPopupVisibleChange),
    _useControlled2 = slicedToArray._slicedToArray(_useControlled, 2),
    innerPopupVisible = _useControlled2[0],
    setInnerPopupVisible = _useControlled2[1];
  var skipNextBlur = React.useRef(false);
  var matchWidthFunc = function matchWidthFunc(triggerElement, popupElement) {
    if (!triggerElement || !popupElement) return;
    var prevDisplay = popupElement.style.display;
    popupElement.style.display = "";
    var overlayScrollWidth = popupElement.offsetWidth - popupElement.scrollWidth;
    var width = popupElement.offsetWidth - overlayScrollWidth > triggerElement.offsetWidth ? popupElement.scrollWidth : triggerElement.offsetWidth - overlayScrollWidth;
    if (prevDisplay === "none") {
      popupElement.style.display = "none";
    }
    var otherOverlayInnerStyle = {};
    if (popupProps && _typeof._typeof(popupProps.overlayInnerStyle) === "object" && !popupProps.overlayInnerStyle.width) {
      otherOverlayInnerStyle = popupProps.overlayInnerStyle;
    }
    return _objectSpread({
      width: "".concat(Math.min(width, MAX_POPUP_WIDTH), "px")
    }, otherOverlayInnerStyle);
  };
  var onInnerPopupVisibleChange = hooks_useInnerPopupVisible["default"](function (visible, context) {
    skipNextBlur.current = false;
    if (disabled || readonly) return;
    var newVisible = context.trigger === "trigger-element-click" && allowInput ? true : visible;
    if (props.popupVisible !== newVisible) {
      setInnerPopupVisible(newVisible, context);
      if (!newVisible) {
        var _extra$afterHidePopup;
        extra === null || extra === void 0 || (_extra$afterHidePopup = extra.afterHidePopup) === null || _extra$afterHidePopup === void 0 || _extra$afterHidePopup.call(extra, context);
        skipNextBlur.current = true;
      }
    }
  });
  var tOverlayInnerStyle = React.useMemo(function () {
    var result = {};
    var overlayInnerStyle = (popupProps === null || popupProps === void 0 ? void 0 : popupProps.overlayInnerStyle) || {};
    if (isFunction.isFunction(overlayInnerStyle) || isObject.isObject(overlayInnerStyle) && overlayInnerStyle.width) {
      result = overlayInnerStyle;
    } else if (!autoWidth) {
      result = matchWidthFunc;
    }
    return result;
  }, [autoWidth, popupProps === null || popupProps === void 0 ? void 0 : popupProps.overlayInnerStyle]);
  return {
    tOverlayInnerStyle: tOverlayInnerStyle,
    innerPopupVisible: innerPopupVisible,
    skipNextBlur: skipNextBlur,
    onInnerPopupVisibleChange: onInnerPopupVisibleChange
  };
}

exports["default"] = useOverlayInnerStyle;
